// velocityjs/dist/cjs/compile/set.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const utils = require("../utils.cjs");
const baseCompile = require("./base-compile.cjs");
const PROTO_KEY = "__proto__";
const PROTOTYPE_CHAIN_KEYS = /* @__PURE__ */ new Set(["constructor", "prototype"]);
function hasOwnProperty(baseRef, key) {
  return (typeof baseRef === "object" || typeof baseRef === "function") && baseRef !== null && Object.prototype.hasOwnProperty.call(baseRef, key);
}
class SetValue extends baseCompile.Compile {
  /**
   * get variable from context, if run in block, return local context, else return global context
   */
  getContext(idName) {
    const local = this.local;
    for (const condition of this.conditions) {
      if (local[condition].hasOwnProperty(idName)) {
        return local[condition];
      }
    }
    return this.context;
  }
  getPathKey(exp) {
    if (exp.type === "property") {
      return exp.id;
    }
    if (exp.type === "index" && exp.id) {
      return String(this.getLiteral(exp.id));
    }
    return "";
  }
  getPathKeys(ref) {
    return (ref.path || []).map((exp) => this.getPathKey(exp));
  }
  isBlockedPathKey(baseRef, key, isEnd) {
    if (key === PROTO_KEY) {
      return true;
    }
    if (key === "prototype" && typeof baseRef === "function") {
      return true;
    }
    return !isEnd && PROTOTYPE_CHAIN_KEYS.has(key) && !hasOwnProperty(baseRef, key);
  }
  hasBlockedUnknownPath(keys, startIndex) {
    return keys.slice(startIndex).some((key, i) => {
      const isEnd = startIndex + i === keys.length - 1;
      return key === PROTO_KEY || !isEnd && PROTOTYPE_CHAIN_KEYS.has(key);
    });
  }
  resolveSetPath(context, ref, pathKeys) {
    if (pathKeys.length === 0) {
      return { blocked: ref.id === PROTO_KEY };
    }
    if (ref.id === PROTO_KEY) {
      return { blocked: true };
    }
    let rootRef = context[ref.id];
    let shouldSetRoot = false;
    if (typeof rootRef !== "object") {
      rootRef = {};
      shouldSetRoot = true;
    }
    let baseRef = rootRef;
    for (let i = 0; i < pathKeys.length; i++) {
      if (baseRef === void 0 || baseRef === null) {
        return { blocked: this.hasBlockedUnknownPath(pathKeys, i), rootRef, shouldSetRoot };
      }
      const key = pathKeys[i];
      const isEnd = pathKeys.length === i + 1;
      if (this.isBlockedPathKey(baseRef, key, isEnd)) {
        return { blocked: true };
      }
      if (isEnd) {
        return { blocked: false, rootRef, shouldSetRoot, baseRef, key };
      }
      baseRef = baseRef[key];
    }
    return { blocked: false, rootRef, shouldSetRoot };
  }
  /**
   * parse #set
   */
  setValue(ast) {
    var _a, _b;
    const ref = ast.equal[0];
    let context = this.getContext(ref.id);
    if (this.contextId && this.contextId.indexOf("macro:") === 0) {
      context = this.context;
    }
    const pathKeys = this.getPathKeys(ref);
    let setPath = this.resolveSetPath(context, ref, pathKeys);
    if (setPath.blocked) {
      return;
    }
    const valAst = ast.equal[1];
    let val;
    if (valAst.type === "math") {
      val = this.getExpression(valAst);
    } else {
      val = (_b = (_a = this.config).valueMapper) == null ? void 0 : _b.call(_a, this.getLiteral(ast.equal[1]));
    }
    if (!ref.path) {
      context[ref.id] = val;
      return;
    }
    setPath = this.resolveSetPath(context, ref, pathKeys);
    if (setPath.blocked) {
      return;
    }
    if (setPath.shouldSetRoot) {
      context[ref.id] = setPath.rootRef;
    }
    if (setPath.key !== void 0) {
      setPath.baseRef[setPath.key] = val;
    }
  }
}
utils.applyMixins(baseCompile.Compile, [SetValue]);
exports.SetValue = SetValue;
