// @aws-sdk/client-kinesis/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var eventStreams = require('@smithy/core/event-streams');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var KinesisServiceException = require('./models/KinesisServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "kinesis",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class KinesisClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = eventStreams.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = resolveRuntimeExtensions(_config_8, configuration?.extensions || []);
        this.config = _config_9;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultKinesisHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AddTagsToStreamCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "AddTagsToStream", {})
    .n("KinesisClient", "AddTagsToStreamCommand")
    .sc(schemas_0.AddTagsToStream$)
    .build() {
}

class CreateStreamCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "CreateStream", {})
    .n("KinesisClient", "CreateStreamCommand")
    .sc(schemas_0.CreateStream$)
    .build() {
}

class DecreaseStreamRetentionPeriodCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DecreaseStreamRetentionPeriod", {})
    .n("KinesisClient", "DecreaseStreamRetentionPeriodCommand")
    .sc(schemas_0.DecreaseStreamRetentionPeriod$)
    .build() {
}

class DeleteResourcePolicyCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DeleteResourcePolicy", {})
    .n("KinesisClient", "DeleteResourcePolicyCommand")
    .sc(schemas_0.DeleteResourcePolicy$)
    .build() {
}

class DeleteStreamCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DeleteStream", {})
    .n("KinesisClient", "DeleteStreamCommand")
    .sc(schemas_0.DeleteStream$)
    .build() {
}

class DeregisterStreamConsumerCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    ConsumerARN: { type: "contextParams", name: "ConsumerARN" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DeregisterStreamConsumer", {})
    .n("KinesisClient", "DeregisterStreamConsumerCommand")
    .sc(schemas_0.DeregisterStreamConsumer$)
    .build() {
}

class DescribeAccountSettingsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DescribeAccountSettings", {})
    .n("KinesisClient", "DescribeAccountSettingsCommand")
    .sc(schemas_0.DescribeAccountSettings$)
    .build() {
}

class DescribeLimitsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DescribeLimits", {})
    .n("KinesisClient", "DescribeLimitsCommand")
    .sc(schemas_0.DescribeLimits$)
    .build() {
}

class DescribeStreamCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DescribeStream", {})
    .n("KinesisClient", "DescribeStreamCommand")
    .sc(schemas_0.DescribeStream$)
    .build() {
}

class DescribeStreamConsumerCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    ConsumerARN: { type: "contextParams", name: "ConsumerARN" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DescribeStreamConsumer", {})
    .n("KinesisClient", "DescribeStreamConsumerCommand")
    .sc(schemas_0.DescribeStreamConsumer$)
    .build() {
}

class DescribeStreamSummaryCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DescribeStreamSummary", {})
    .n("KinesisClient", "DescribeStreamSummaryCommand")
    .sc(schemas_0.DescribeStreamSummary$)
    .build() {
}

class DisableEnhancedMonitoringCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "DisableEnhancedMonitoring", {})
    .n("KinesisClient", "DisableEnhancedMonitoringCommand")
    .sc(schemas_0.DisableEnhancedMonitoring$)
    .build() {
}

class EnableEnhancedMonitoringCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "EnableEnhancedMonitoring", {})
    .n("KinesisClient", "EnableEnhancedMonitoringCommand")
    .sc(schemas_0.EnableEnhancedMonitoring$)
    .build() {
}

class GetRecordsCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `data` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "GetRecords", {})
    .n("KinesisClient", "GetRecordsCommand")
    .sc(schemas_0.GetRecords$)
    .build() {
}

class GetResourcePolicyCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "GetResourcePolicy", {})
    .n("KinesisClient", "GetResourcePolicyCommand")
    .sc(schemas_0.GetResourcePolicy$)
    .build() {
}

class GetShardIteratorCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `data` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "GetShardIterator", {})
    .n("KinesisClient", "GetShardIteratorCommand")
    .sc(schemas_0.GetShardIterator$)
    .build() {
}

class IncreaseStreamRetentionPeriodCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "IncreaseStreamRetentionPeriod", {})
    .n("KinesisClient", "IncreaseStreamRetentionPeriodCommand")
    .sc(schemas_0.IncreaseStreamRetentionPeriod$)
    .build() {
}

class ListShardsCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "ListShards", {})
    .n("KinesisClient", "ListShardsCommand")
    .sc(schemas_0.ListShards$)
    .build() {
}

class ListStreamConsumersCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "ListStreamConsumers", {})
    .n("KinesisClient", "ListStreamConsumersCommand")
    .sc(schemas_0.ListStreamConsumers$)
    .build() {
}

class ListStreamsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "ListStreams", {})
    .n("KinesisClient", "ListStreamsCommand")
    .sc(schemas_0.ListStreams$)
    .build() {
}

class ListTagsForResourceCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "ListTagsForResource", {})
    .n("KinesisClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTagsForStreamCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "ListTagsForStream", {})
    .n("KinesisClient", "ListTagsForStreamCommand")
    .sc(schemas_0.ListTagsForStream$)
    .build() {
}

class MergeShardsCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "MergeShards", {})
    .n("KinesisClient", "MergeShardsCommand")
    .sc(schemas_0.MergeShards$)
    .build() {
}

class PutRecordCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `data` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "PutRecord", {})
    .n("KinesisClient", "PutRecordCommand")
    .sc(schemas_0.PutRecord$)
    .build() {
}

class PutRecordsCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `data` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "PutRecords", {})
    .n("KinesisClient", "PutRecordsCommand")
    .sc(schemas_0.PutRecords$)
    .build() {
}

class PutResourcePolicyCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "PutResourcePolicy", {})
    .n("KinesisClient", "PutResourcePolicyCommand")
    .sc(schemas_0.PutResourcePolicy$)
    .build() {
}

class RegisterStreamConsumerCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "RegisterStreamConsumer", {})
    .n("KinesisClient", "RegisterStreamConsumerCommand")
    .sc(schemas_0.RegisterStreamConsumer$)
    .build() {
}

class RemoveTagsFromStreamCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "RemoveTagsFromStream", {})
    .n("KinesisClient", "RemoveTagsFromStreamCommand")
    .sc(schemas_0.RemoveTagsFromStream$)
    .build() {
}

class SplitShardCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "SplitShard", {})
    .n("KinesisClient", "SplitShardCommand")
    .sc(schemas_0.SplitShard$)
    .build() {
}

class StartStreamEncryptionCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "StartStreamEncryption", {})
    .n("KinesisClient", "StartStreamEncryptionCommand")
    .sc(schemas_0.StartStreamEncryption$)
    .build() {
}

class StopStreamEncryptionCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "StopStreamEncryption", {})
    .n("KinesisClient", "StopStreamEncryptionCommand")
    .sc(schemas_0.StopStreamEncryption$)
    .build() {
}

class SubscribeToShardCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `data` },
    StreamId: { type: "contextParams", name: "StreamId" },
    ConsumerARN: { type: "contextParams", name: "ConsumerARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "SubscribeToShard", {
    eventStream: {
        output: true,
    },
})
    .n("KinesisClient", "SubscribeToShardCommand")
    .sc(schemas_0.SubscribeToShard$)
    .build() {
}

class TagResourceCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "TagResource", {})
    .n("KinesisClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    ResourceARN: { type: "contextParams", name: "ResourceARN" },
    StreamId: { type: "contextParams", name: "StreamId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UntagResource", {})
    .n("KinesisClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateAccountSettingsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UpdateAccountSettings", {})
    .n("KinesisClient", "UpdateAccountSettingsCommand")
    .sc(schemas_0.UpdateAccountSettings$)
    .build() {
}

class UpdateMaxRecordSizeCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UpdateMaxRecordSize", {})
    .n("KinesisClient", "UpdateMaxRecordSizeCommand")
    .sc(schemas_0.UpdateMaxRecordSize$)
    .build() {
}

class UpdateShardCountCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UpdateShardCount", {})
    .n("KinesisClient", "UpdateShardCountCommand")
    .sc(schemas_0.UpdateShardCount$)
    .build() {
}

class UpdateStreamModeCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UpdateStreamMode", {})
    .n("KinesisClient", "UpdateStreamModeCommand")
    .sc(schemas_0.UpdateStreamMode$)
    .build() {
}

class UpdateStreamWarmThroughputCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    OperationType: { type: "staticContextParams", value: `control` },
    StreamId: { type: "contextParams", name: "StreamId" },
    StreamARN: { type: "contextParams", name: "StreamARN" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Kinesis_20131202", "UpdateStreamWarmThroughput", {})
    .n("KinesisClient", "UpdateStreamWarmThroughputCommand")
    .sc(schemas_0.UpdateStreamWarmThroughput$)
    .build() {
}

const paginateListStreamConsumers = core.createPaginator(KinesisClient, ListStreamConsumersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStreams = core.createPaginator(KinesisClient, ListStreamsCommand, "NextToken", "NextToken", "Limit");

const checkState$1 = async (client$1, input) => {
    let reason;
    try {
        let result = await client$1.send(new DescribeStreamCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.StreamDescription.StreamStatus;
            };
            if (returnComparator() === "ACTIVE") {
                return { state: client.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: client.WaiterState.RETRY, reason };
};
const waitForStreamExists = async (params, input) => {
    const serviceDefaults = { minDelay: 10, maxDelay: 120 };
    return client.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilStreamExists = async (params, input) => {
    const serviceDefaults = { minDelay: 10, maxDelay: 120 };
    const result = await client.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return client.checkExceptions(result);
};

const checkState = async (client$1, input) => {
    let reason;
    try {
        let result = await client$1.send(new DescribeStreamCommand(input));
        reason = result;
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ResourceNotFoundException") {
            return { state: client.WaiterState.SUCCESS, reason };
        }
    }
    return { state: client.WaiterState.RETRY, reason };
};
const waitForStreamNotExists = async (params, input) => {
    const serviceDefaults = { minDelay: 10, maxDelay: 120 };
    return client.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilStreamNotExists = async (params, input) => {
    const serviceDefaults = { minDelay: 10, maxDelay: 120 };
    const result = await client.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return client.checkExceptions(result);
};

const commands = {
    AddTagsToStreamCommand,
    CreateStreamCommand,
    DecreaseStreamRetentionPeriodCommand,
    DeleteResourcePolicyCommand,
    DeleteStreamCommand,
    DeregisterStreamConsumerCommand,
    DescribeAccountSettingsCommand,
    DescribeLimitsCommand,
    DescribeStreamCommand,
    DescribeStreamConsumerCommand,
    DescribeStreamSummaryCommand,
    DisableEnhancedMonitoringCommand,
    EnableEnhancedMonitoringCommand,
    GetRecordsCommand,
    GetResourcePolicyCommand,
    GetShardIteratorCommand,
    IncreaseStreamRetentionPeriodCommand,
    ListShardsCommand,
    ListStreamConsumersCommand,
    ListStreamsCommand,
    ListTagsForResourceCommand,
    ListTagsForStreamCommand,
    MergeShardsCommand,
    PutRecordCommand,
    PutRecordsCommand,
    PutResourcePolicyCommand,
    RegisterStreamConsumerCommand,
    RemoveTagsFromStreamCommand,
    SplitShardCommand,
    StartStreamEncryptionCommand,
    StopStreamEncryptionCommand,
    SubscribeToShardCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAccountSettingsCommand,
    UpdateMaxRecordSizeCommand,
    UpdateShardCountCommand,
    UpdateStreamModeCommand,
    UpdateStreamWarmThroughputCommand,
};
const paginators = {
    paginateListStreamConsumers,
    paginateListStreams,
};
const waiters = {
    waitUntilStreamExists,
    waitUntilStreamNotExists,
};
class Kinesis extends KinesisClient {
}
client.createAggregatedClient(commands, Kinesis, { paginators, waiters });

const ConsumerStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
};
const StreamMode = {
    ON_DEMAND: "ON_DEMAND",
    PROVISIONED: "PROVISIONED",
};
const MinimumThroughputBillingCommitmentOutputStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    ENABLED_UNTIL_EARLIEST_ALLOWED_END: "ENABLED_UNTIL_EARLIEST_ALLOWED_END",
};
const EncryptionType = {
    KMS: "KMS",
    NONE: "NONE",
};
const MetricsName = {
    ALL: "ALL",
    INCOMING_BYTES: "IncomingBytes",
    INCOMING_RECORDS: "IncomingRecords",
    ITERATOR_AGE_MILLISECONDS: "IteratorAgeMilliseconds",
    OUTGOING_BYTES: "OutgoingBytes",
    OUTGOING_RECORDS: "OutgoingRecords",
    READ_PROVISIONED_THROUGHPUT_EXCEEDED: "ReadProvisionedThroughputExceeded",
    WRITE_PROVISIONED_THROUGHPUT_EXCEEDED: "WriteProvisionedThroughputExceeded",
};
const StreamStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    UPDATING: "UPDATING",
};
const ShardIteratorType = {
    AFTER_SEQUENCE_NUMBER: "AFTER_SEQUENCE_NUMBER",
    AT_SEQUENCE_NUMBER: "AT_SEQUENCE_NUMBER",
    AT_TIMESTAMP: "AT_TIMESTAMP",
    LATEST: "LATEST",
    TRIM_HORIZON: "TRIM_HORIZON",
};
const ShardFilterType = {
    AFTER_SHARD_ID: "AFTER_SHARD_ID",
    AT_LATEST: "AT_LATEST",
    AT_TIMESTAMP: "AT_TIMESTAMP",
    AT_TRIM_HORIZON: "AT_TRIM_HORIZON",
    FROM_TIMESTAMP: "FROM_TIMESTAMP",
    FROM_TRIM_HORIZON: "FROM_TRIM_HORIZON",
};
const MinimumThroughputBillingCommitmentInputStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ScalingType = {
    UNIFORM_SCALING: "UNIFORM_SCALING",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.KinesisServiceException = KinesisServiceException.KinesisServiceException;
exports.AddTagsToStreamCommand = AddTagsToStreamCommand;
exports.ConsumerStatus = ConsumerStatus;
exports.CreateStreamCommand = CreateStreamCommand;
exports.DecreaseStreamRetentionPeriodCommand = DecreaseStreamRetentionPeriodCommand;
exports.DeleteResourcePolicyCommand = DeleteResourcePolicyCommand;
exports.DeleteStreamCommand = DeleteStreamCommand;
exports.DeregisterStreamConsumerCommand = DeregisterStreamConsumerCommand;
exports.DescribeAccountSettingsCommand = DescribeAccountSettingsCommand;
exports.DescribeLimitsCommand = DescribeLimitsCommand;
exports.DescribeStreamCommand = DescribeStreamCommand;
exports.DescribeStreamConsumerCommand = DescribeStreamConsumerCommand;
exports.DescribeStreamSummaryCommand = DescribeStreamSummaryCommand;
exports.DisableEnhancedMonitoringCommand = DisableEnhancedMonitoringCommand;
exports.EnableEnhancedMonitoringCommand = EnableEnhancedMonitoringCommand;
exports.EncryptionType = EncryptionType;
exports.GetRecordsCommand = GetRecordsCommand;
exports.GetResourcePolicyCommand = GetResourcePolicyCommand;
exports.GetShardIteratorCommand = GetShardIteratorCommand;
exports.IncreaseStreamRetentionPeriodCommand = IncreaseStreamRetentionPeriodCommand;
exports.Kinesis = Kinesis;
exports.KinesisClient = KinesisClient;
exports.ListShardsCommand = ListShardsCommand;
exports.ListStreamConsumersCommand = ListStreamConsumersCommand;
exports.ListStreamsCommand = ListStreamsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForStreamCommand = ListTagsForStreamCommand;
exports.MergeShardsCommand = MergeShardsCommand;
exports.MetricsName = MetricsName;
exports.MinimumThroughputBillingCommitmentInputStatus = MinimumThroughputBillingCommitmentInputStatus;
exports.MinimumThroughputBillingCommitmentOutputStatus = MinimumThroughputBillingCommitmentOutputStatus;
exports.PutRecordCommand = PutRecordCommand;
exports.PutRecordsCommand = PutRecordsCommand;
exports.PutResourcePolicyCommand = PutResourcePolicyCommand;
exports.RegisterStreamConsumerCommand = RegisterStreamConsumerCommand;
exports.RemoveTagsFromStreamCommand = RemoveTagsFromStreamCommand;
exports.ScalingType = ScalingType;
exports.ShardFilterType = ShardFilterType;
exports.ShardIteratorType = ShardIteratorType;
exports.SplitShardCommand = SplitShardCommand;
exports.StartStreamEncryptionCommand = StartStreamEncryptionCommand;
exports.StopStreamEncryptionCommand = StopStreamEncryptionCommand;
exports.StreamMode = StreamMode;
exports.StreamStatus = StreamStatus;
exports.SubscribeToShardCommand = SubscribeToShardCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateAccountSettingsCommand = UpdateAccountSettingsCommand;
exports.UpdateMaxRecordSizeCommand = UpdateMaxRecordSizeCommand;
exports.UpdateShardCountCommand = UpdateShardCountCommand;
exports.UpdateStreamModeCommand = UpdateStreamModeCommand;
exports.UpdateStreamWarmThroughputCommand = UpdateStreamWarmThroughputCommand;
exports.paginateListStreamConsumers = paginateListStreamConsumers;
exports.paginateListStreams = paginateListStreams;
exports.waitForStreamExists = waitForStreamExists;
exports.waitForStreamNotExists = waitForStreamNotExists;
exports.waitUntilStreamExists = waitUntilStreamExists;
exports.waitUntilStreamNotExists = waitUntilStreamNotExists;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
