// @aws-sdk/client-eventbridge/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var EventBridgeServiceException = require('./models/EventBridgeServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "events",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class EventBridgeClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultEventBridgeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
                "aws.auth#sigv4a": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class ActivateEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ActivateEventSource", {})
    .n("EventBridgeClient", "ActivateEventSourceCommand")
    .sc(schemas_0.ActivateEventSource$)
    .build() {
}

class CancelReplayCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CancelReplay", {})
    .n("EventBridgeClient", "CancelReplayCommand")
    .sc(schemas_0.CancelReplay$)
    .build() {
}

class CreateApiDestinationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreateApiDestination", {})
    .n("EventBridgeClient", "CreateApiDestinationCommand")
    .sc(schemas_0.CreateApiDestination$)
    .build() {
}

class CreateArchiveCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreateArchive", {})
    .n("EventBridgeClient", "CreateArchiveCommand")
    .sc(schemas_0.CreateArchive$)
    .build() {
}

class CreateConnectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreateConnection", {})
    .n("EventBridgeClient", "CreateConnectionCommand")
    .sc(schemas_0.CreateConnection$)
    .build() {
}

class CreateEndpointCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreateEndpoint", {})
    .n("EventBridgeClient", "CreateEndpointCommand")
    .sc(schemas_0.CreateEndpoint$)
    .build() {
}

class CreateEventBusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreateEventBus", {})
    .n("EventBridgeClient", "CreateEventBusCommand")
    .sc(schemas_0.CreateEventBus$)
    .build() {
}

class CreatePartnerEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "CreatePartnerEventSource", {})
    .n("EventBridgeClient", "CreatePartnerEventSourceCommand")
    .sc(schemas_0.CreatePartnerEventSource$)
    .build() {
}

class DeactivateEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeactivateEventSource", {})
    .n("EventBridgeClient", "DeactivateEventSourceCommand")
    .sc(schemas_0.DeactivateEventSource$)
    .build() {
}

class DeauthorizeConnectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeauthorizeConnection", {})
    .n("EventBridgeClient", "DeauthorizeConnectionCommand")
    .sc(schemas_0.DeauthorizeConnection$)
    .build() {
}

class DeleteApiDestinationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteApiDestination", {})
    .n("EventBridgeClient", "DeleteApiDestinationCommand")
    .sc(schemas_0.DeleteApiDestination$)
    .build() {
}

class DeleteArchiveCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteArchive", {})
    .n("EventBridgeClient", "DeleteArchiveCommand")
    .sc(schemas_0.DeleteArchive$)
    .build() {
}

class DeleteConnectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteConnection", {})
    .n("EventBridgeClient", "DeleteConnectionCommand")
    .sc(schemas_0.DeleteConnection$)
    .build() {
}

class DeleteEndpointCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteEndpoint", {})
    .n("EventBridgeClient", "DeleteEndpointCommand")
    .sc(schemas_0.DeleteEndpoint$)
    .build() {
}

class DeleteEventBusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteEventBus", {})
    .n("EventBridgeClient", "DeleteEventBusCommand")
    .sc(schemas_0.DeleteEventBus$)
    .build() {
}

class DeletePartnerEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeletePartnerEventSource", {})
    .n("EventBridgeClient", "DeletePartnerEventSourceCommand")
    .sc(schemas_0.DeletePartnerEventSource$)
    .build() {
}

class DeleteRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DeleteRule", {})
    .n("EventBridgeClient", "DeleteRuleCommand")
    .sc(schemas_0.DeleteRule$)
    .build() {
}

class DescribeApiDestinationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeApiDestination", {})
    .n("EventBridgeClient", "DescribeApiDestinationCommand")
    .sc(schemas_0.DescribeApiDestination$)
    .build() {
}

class DescribeArchiveCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeArchive", {})
    .n("EventBridgeClient", "DescribeArchiveCommand")
    .sc(schemas_0.DescribeArchive$)
    .build() {
}

class DescribeConnectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeConnection", {})
    .n("EventBridgeClient", "DescribeConnectionCommand")
    .sc(schemas_0.DescribeConnection$)
    .build() {
}

class DescribeEndpointCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeEndpoint", {})
    .n("EventBridgeClient", "DescribeEndpointCommand")
    .sc(schemas_0.DescribeEndpoint$)
    .build() {
}

class DescribeEventBusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeEventBus", {})
    .n("EventBridgeClient", "DescribeEventBusCommand")
    .sc(schemas_0.DescribeEventBus$)
    .build() {
}

class DescribeEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeEventSource", {})
    .n("EventBridgeClient", "DescribeEventSourceCommand")
    .sc(schemas_0.DescribeEventSource$)
    .build() {
}

class DescribePartnerEventSourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribePartnerEventSource", {})
    .n("EventBridgeClient", "DescribePartnerEventSourceCommand")
    .sc(schemas_0.DescribePartnerEventSource$)
    .build() {
}

class DescribeReplayCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeReplay", {})
    .n("EventBridgeClient", "DescribeReplayCommand")
    .sc(schemas_0.DescribeReplay$)
    .build() {
}

class DescribeRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DescribeRule", {})
    .n("EventBridgeClient", "DescribeRuleCommand")
    .sc(schemas_0.DescribeRule$)
    .build() {
}

class DisableRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "DisableRule", {})
    .n("EventBridgeClient", "DisableRuleCommand")
    .sc(schemas_0.DisableRule$)
    .build() {
}

class EnableRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "EnableRule", {})
    .n("EventBridgeClient", "EnableRuleCommand")
    .sc(schemas_0.EnableRule$)
    .build() {
}

class ListApiDestinationsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListApiDestinations", {})
    .n("EventBridgeClient", "ListApiDestinationsCommand")
    .sc(schemas_0.ListApiDestinations$)
    .build() {
}

class ListArchivesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListArchives", {})
    .n("EventBridgeClient", "ListArchivesCommand")
    .sc(schemas_0.ListArchives$)
    .build() {
}

class ListConnectionsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListConnections", {})
    .n("EventBridgeClient", "ListConnectionsCommand")
    .sc(schemas_0.ListConnections$)
    .build() {
}

class ListEndpointsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListEndpoints", {})
    .n("EventBridgeClient", "ListEndpointsCommand")
    .sc(schemas_0.ListEndpoints$)
    .build() {
}

class ListEventBusesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListEventBuses", {})
    .n("EventBridgeClient", "ListEventBusesCommand")
    .sc(schemas_0.ListEventBuses$)
    .build() {
}

class ListEventSourcesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListEventSources", {})
    .n("EventBridgeClient", "ListEventSourcesCommand")
    .sc(schemas_0.ListEventSources$)
    .build() {
}

class ListPartnerEventSourceAccountsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListPartnerEventSourceAccounts", {})
    .n("EventBridgeClient", "ListPartnerEventSourceAccountsCommand")
    .sc(schemas_0.ListPartnerEventSourceAccounts$)
    .build() {
}

class ListPartnerEventSourcesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListPartnerEventSources", {})
    .n("EventBridgeClient", "ListPartnerEventSourcesCommand")
    .sc(schemas_0.ListPartnerEventSources$)
    .build() {
}

class ListReplaysCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListReplays", {})
    .n("EventBridgeClient", "ListReplaysCommand")
    .sc(schemas_0.ListReplays$)
    .build() {
}

class ListRuleNamesByTargetCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListRuleNamesByTarget", {})
    .n("EventBridgeClient", "ListRuleNamesByTargetCommand")
    .sc(schemas_0.ListRuleNamesByTarget$)
    .build() {
}

class ListRulesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListRules", {})
    .n("EventBridgeClient", "ListRulesCommand")
    .sc(schemas_0.ListRules$)
    .build() {
}

class ListTagsForResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListTagsForResource", {})
    .n("EventBridgeClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTargetsByRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "ListTargetsByRule", {})
    .n("EventBridgeClient", "ListTargetsByRuleCommand")
    .sc(schemas_0.ListTargetsByRule$)
    .build() {
}

class PutEventsCommand extends client.Command
    .classBuilder()
    .ep({
    ...commonParams,
    EndpointId: { type: "contextParams", name: "EndpointId" },
})
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "PutEvents", {})
    .n("EventBridgeClient", "PutEventsCommand")
    .sc(schemas_0.PutEvents$)
    .build() {
}

class PutPartnerEventsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "PutPartnerEvents", {})
    .n("EventBridgeClient", "PutPartnerEventsCommand")
    .sc(schemas_0.PutPartnerEvents$)
    .build() {
}

class PutPermissionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "PutPermission", {})
    .n("EventBridgeClient", "PutPermissionCommand")
    .sc(schemas_0.PutPermission$)
    .build() {
}

class PutRuleCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "PutRule", {})
    .n("EventBridgeClient", "PutRuleCommand")
    .sc(schemas_0.PutRule$)
    .build() {
}

class PutTargetsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "PutTargets", {})
    .n("EventBridgeClient", "PutTargetsCommand")
    .sc(schemas_0.PutTargets$)
    .build() {
}

class RemovePermissionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "RemovePermission", {})
    .n("EventBridgeClient", "RemovePermissionCommand")
    .sc(schemas_0.RemovePermission$)
    .build() {
}

class RemoveTargetsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "RemoveTargets", {})
    .n("EventBridgeClient", "RemoveTargetsCommand")
    .sc(schemas_0.RemoveTargets$)
    .build() {
}

class StartReplayCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "StartReplay", {})
    .n("EventBridgeClient", "StartReplayCommand")
    .sc(schemas_0.StartReplay$)
    .build() {
}

class TagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "TagResource", {})
    .n("EventBridgeClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class TestEventPatternCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "TestEventPattern", {})
    .n("EventBridgeClient", "TestEventPatternCommand")
    .sc(schemas_0.TestEventPattern$)
    .build() {
}

class UntagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UntagResource", {})
    .n("EventBridgeClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateApiDestinationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UpdateApiDestination", {})
    .n("EventBridgeClient", "UpdateApiDestinationCommand")
    .sc(schemas_0.UpdateApiDestination$)
    .build() {
}

class UpdateArchiveCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UpdateArchive", {})
    .n("EventBridgeClient", "UpdateArchiveCommand")
    .sc(schemas_0.UpdateArchive$)
    .build() {
}

class UpdateConnectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UpdateConnection", {})
    .n("EventBridgeClient", "UpdateConnectionCommand")
    .sc(schemas_0.UpdateConnection$)
    .build() {
}

class UpdateEndpointCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UpdateEndpoint", {})
    .n("EventBridgeClient", "UpdateEndpointCommand")
    .sc(schemas_0.UpdateEndpoint$)
    .build() {
}

class UpdateEventBusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSEvents", "UpdateEventBus", {})
    .n("EventBridgeClient", "UpdateEventBusCommand")
    .sc(schemas_0.UpdateEventBus$)
    .build() {
}

const commands = {
    ActivateEventSourceCommand,
    CancelReplayCommand,
    CreateApiDestinationCommand,
    CreateArchiveCommand,
    CreateConnectionCommand,
    CreateEndpointCommand,
    CreateEventBusCommand,
    CreatePartnerEventSourceCommand,
    DeactivateEventSourceCommand,
    DeauthorizeConnectionCommand,
    DeleteApiDestinationCommand,
    DeleteArchiveCommand,
    DeleteConnectionCommand,
    DeleteEndpointCommand,
    DeleteEventBusCommand,
    DeletePartnerEventSourceCommand,
    DeleteRuleCommand,
    DescribeApiDestinationCommand,
    DescribeArchiveCommand,
    DescribeConnectionCommand,
    DescribeEndpointCommand,
    DescribeEventBusCommand,
    DescribeEventSourceCommand,
    DescribePartnerEventSourceCommand,
    DescribeReplayCommand,
    DescribeRuleCommand,
    DisableRuleCommand,
    EnableRuleCommand,
    ListApiDestinationsCommand,
    ListArchivesCommand,
    ListConnectionsCommand,
    ListEndpointsCommand,
    ListEventBusesCommand,
    ListEventSourcesCommand,
    ListPartnerEventSourceAccountsCommand,
    ListPartnerEventSourcesCommand,
    ListReplaysCommand,
    ListRuleNamesByTargetCommand,
    ListRulesCommand,
    ListTagsForResourceCommand,
    ListTargetsByRuleCommand,
    PutEventsCommand,
    PutPartnerEventsCommand,
    PutPermissionCommand,
    PutRuleCommand,
    PutTargetsCommand,
    RemovePermissionCommand,
    RemoveTargetsCommand,
    StartReplayCommand,
    TagResourceCommand,
    TestEventPatternCommand,
    UntagResourceCommand,
    UpdateApiDestinationCommand,
    UpdateArchiveCommand,
    UpdateConnectionCommand,
    UpdateEndpointCommand,
    UpdateEventBusCommand,
};
class EventBridge extends EventBridgeClient {
}
client.createAggregatedClient(commands, EventBridge);

const ApiDestinationState = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const ApiDestinationHttpMethod = {
    DELETE: "DELETE",
    GET: "GET",
    HEAD: "HEAD",
    OPTIONS: "OPTIONS",
    PATCH: "PATCH",
    POST: "POST",
    PUT: "PUT",
};
const ArchiveState = {
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
};
const AssignPublicIp = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ReplayState = {
    CANCELLED: "CANCELLED",
    CANCELLING: "CANCELLING",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    STARTING: "STARTING",
};
const ConnectionAuthorizationType = {
    API_KEY: "API_KEY",
    BASIC: "BASIC",
    OAUTH_CLIENT_CREDENTIALS: "OAUTH_CLIENT_CREDENTIALS",
};
const ConnectionOAuthHttpMethod = {
    GET: "GET",
    POST: "POST",
    PUT: "PUT",
};
const ConnectionState = {
    ACTIVE: "ACTIVE",
    AUTHORIZED: "AUTHORIZED",
    AUTHORIZING: "AUTHORIZING",
    CREATING: "CREATING",
    DEAUTHORIZED: "DEAUTHORIZED",
    DEAUTHORIZING: "DEAUTHORIZING",
    DELETING: "DELETING",
    FAILED_CONNECTIVITY: "FAILED_CONNECTIVITY",
    UPDATING: "UPDATING",
};
const ReplicationState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const EndpointState = {
    ACTIVE: "ACTIVE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETE_FAILED: "DELETE_FAILED",
    DELETING: "DELETING",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
};
const IncludeDetail = {
    FULL: "FULL",
    NONE: "NONE",
};
const Level = {
    ERROR: "ERROR",
    INFO: "INFO",
    OFF: "OFF",
    TRACE: "TRACE",
};
const EventSourceState = {
    ACTIVE: "ACTIVE",
    DELETED: "DELETED",
    PENDING: "PENDING",
};
const RuleState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    ENABLED_WITH_ALL_CLOUDTRAIL_MANAGEMENT_EVENTS: "ENABLED_WITH_ALL_CLOUDTRAIL_MANAGEMENT_EVENTS",
};
const LaunchType = {
    EC2: "EC2",
    EXTERNAL: "EXTERNAL",
    FARGATE: "FARGATE",
};
const PlacementConstraintType = {
    DISTINCT_INSTANCE: "distinctInstance",
    MEMBER_OF: "memberOf",
};
const PlacementStrategyType = {
    BINPACK: "binpack",
    RANDOM: "random",
    SPREAD: "spread",
};
const PropagateTags = {
    TASK_DEFINITION: "TASK_DEFINITION",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.EventBridgeServiceException = EventBridgeServiceException.EventBridgeServiceException;
exports.ActivateEventSourceCommand = ActivateEventSourceCommand;
exports.ApiDestinationHttpMethod = ApiDestinationHttpMethod;
exports.ApiDestinationState = ApiDestinationState;
exports.ArchiveState = ArchiveState;
exports.AssignPublicIp = AssignPublicIp;
exports.CancelReplayCommand = CancelReplayCommand;
exports.ConnectionAuthorizationType = ConnectionAuthorizationType;
exports.ConnectionOAuthHttpMethod = ConnectionOAuthHttpMethod;
exports.ConnectionState = ConnectionState;
exports.CreateApiDestinationCommand = CreateApiDestinationCommand;
exports.CreateArchiveCommand = CreateArchiveCommand;
exports.CreateConnectionCommand = CreateConnectionCommand;
exports.CreateEndpointCommand = CreateEndpointCommand;
exports.CreateEventBusCommand = CreateEventBusCommand;
exports.CreatePartnerEventSourceCommand = CreatePartnerEventSourceCommand;
exports.DeactivateEventSourceCommand = DeactivateEventSourceCommand;
exports.DeauthorizeConnectionCommand = DeauthorizeConnectionCommand;
exports.DeleteApiDestinationCommand = DeleteApiDestinationCommand;
exports.DeleteArchiveCommand = DeleteArchiveCommand;
exports.DeleteConnectionCommand = DeleteConnectionCommand;
exports.DeleteEndpointCommand = DeleteEndpointCommand;
exports.DeleteEventBusCommand = DeleteEventBusCommand;
exports.DeletePartnerEventSourceCommand = DeletePartnerEventSourceCommand;
exports.DeleteRuleCommand = DeleteRuleCommand;
exports.DescribeApiDestinationCommand = DescribeApiDestinationCommand;
exports.DescribeArchiveCommand = DescribeArchiveCommand;
exports.DescribeConnectionCommand = DescribeConnectionCommand;
exports.DescribeEndpointCommand = DescribeEndpointCommand;
exports.DescribeEventBusCommand = DescribeEventBusCommand;
exports.DescribeEventSourceCommand = DescribeEventSourceCommand;
exports.DescribePartnerEventSourceCommand = DescribePartnerEventSourceCommand;
exports.DescribeReplayCommand = DescribeReplayCommand;
exports.DescribeRuleCommand = DescribeRuleCommand;
exports.DisableRuleCommand = DisableRuleCommand;
exports.EnableRuleCommand = EnableRuleCommand;
exports.EndpointState = EndpointState;
exports.EventBridge = EventBridge;
exports.EventBridgeClient = EventBridgeClient;
exports.EventSourceState = EventSourceState;
exports.IncludeDetail = IncludeDetail;
exports.LaunchType = LaunchType;
exports.Level = Level;
exports.ListApiDestinationsCommand = ListApiDestinationsCommand;
exports.ListArchivesCommand = ListArchivesCommand;
exports.ListConnectionsCommand = ListConnectionsCommand;
exports.ListEndpointsCommand = ListEndpointsCommand;
exports.ListEventBusesCommand = ListEventBusesCommand;
exports.ListEventSourcesCommand = ListEventSourcesCommand;
exports.ListPartnerEventSourceAccountsCommand = ListPartnerEventSourceAccountsCommand;
exports.ListPartnerEventSourcesCommand = ListPartnerEventSourcesCommand;
exports.ListReplaysCommand = ListReplaysCommand;
exports.ListRuleNamesByTargetCommand = ListRuleNamesByTargetCommand;
exports.ListRulesCommand = ListRulesCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTargetsByRuleCommand = ListTargetsByRuleCommand;
exports.PlacementConstraintType = PlacementConstraintType;
exports.PlacementStrategyType = PlacementStrategyType;
exports.PropagateTags = PropagateTags;
exports.PutEventsCommand = PutEventsCommand;
exports.PutPartnerEventsCommand = PutPartnerEventsCommand;
exports.PutPermissionCommand = PutPermissionCommand;
exports.PutRuleCommand = PutRuleCommand;
exports.PutTargetsCommand = PutTargetsCommand;
exports.RemovePermissionCommand = RemovePermissionCommand;
exports.RemoveTargetsCommand = RemoveTargetsCommand;
exports.ReplayState = ReplayState;
exports.ReplicationState = ReplicationState;
exports.RuleState = RuleState;
exports.StartReplayCommand = StartReplayCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TestEventPatternCommand = TestEventPatternCommand;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateApiDestinationCommand = UpdateApiDestinationCommand;
exports.UpdateArchiveCommand = UpdateArchiveCommand;
exports.UpdateConnectionCommand = UpdateConnectionCommand;
exports.UpdateEndpointCommand = UpdateEndpointCommand;
exports.UpdateEventBusCommand = UpdateEventBusCommand;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
