// @aws-sdk/client-api-gateway/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteIntegrationResponseRequest$ = exports.DeleteIntegrationRequest$ = exports.DeleteGatewayResponseRequest$ = exports.DeleteDomainNameRequest$ = exports.DeleteDomainNameAccessAssociationRequest$ = exports.DeleteDocumentationVersionRequest$ = exports.DeleteDocumentationPartRequest$ = exports.DeleteDeploymentRequest$ = exports.DeleteClientCertificateRequest$ = exports.DeleteBasePathMappingRequest$ = exports.DeleteAuthorizerRequest$ = exports.DeleteApiKeyRequest$ = exports.CreateVpcLinkRequest$ = exports.CreateUsagePlanRequest$ = exports.CreateUsagePlanKeyRequest$ = exports.CreateStageRequest$ = exports.CreateRestApiRequest$ = exports.CreateResourceRequest$ = exports.CreateRequestValidatorRequest$ = exports.CreateModelRequest$ = exports.CreateDomainNameRequest$ = exports.CreateDomainNameAccessAssociationRequest$ = exports.CreateDocumentationVersionRequest$ = exports.CreateDocumentationPartRequest$ = exports.CreateDeploymentRequest$ = exports.CreateBasePathMappingRequest$ = exports.CreateAuthorizerRequest$ = exports.CreateApiKeyRequest$ = exports.ClientCertificates$ = exports.ClientCertificate$ = exports.CanarySettings$ = exports.BasePathMappings$ = exports.BasePathMapping$ = exports.Authorizers$ = exports.Authorizer$ = exports.ApiStage$ = exports.ApiKeys$ = exports.ApiKeyIds$ = exports.ApiKey$ = exports.Account$ = exports.AccessLogSettings$ = exports.errorTypeRegistries = exports.UnauthorizedException$ = exports.TooManyRequestsException$ = exports.ServiceUnavailableException$ = exports.NotFoundException$ = exports.LimitExceededException$ = exports.ConflictException$ = exports.BadRequestException$ = exports.APIGatewayServiceException$ = void 0;
exports.GetGatewayResponseRequest$ = exports.GetExportRequest$ = exports.GetDomainNamesRequest$ = exports.GetDomainNameRequest$ = exports.GetDomainNameAccessAssociationsRequest$ = exports.GetDocumentationVersionsRequest$ = exports.GetDocumentationVersionRequest$ = exports.GetDocumentationPartsRequest$ = exports.GetDocumentationPartRequest$ = exports.GetDeploymentsRequest$ = exports.GetDeploymentRequest$ = exports.GetClientCertificatesRequest$ = exports.GetClientCertificateRequest$ = exports.GetBasePathMappingsRequest$ = exports.GetBasePathMappingRequest$ = exports.GetAuthorizersRequest$ = exports.GetAuthorizerRequest$ = exports.GetApiKeysRequest$ = exports.GetApiKeyRequest$ = exports.GetAccountRequest$ = exports.GenerateClientCertificateRequest$ = exports.GatewayResponses$ = exports.GatewayResponse$ = exports.FlushStageCacheRequest$ = exports.FlushStageAuthorizersCacheRequest$ = exports.ExportResponse$ = exports.EndpointConfiguration$ = exports.DomainNames$ = exports.DomainNameAccessAssociations$ = exports.DomainNameAccessAssociation$ = exports.DomainName$ = exports.DocumentationVersions$ = exports.DocumentationVersion$ = exports.DocumentationParts$ = exports.DocumentationPartLocation$ = exports.DocumentationPartIds$ = exports.DocumentationPart$ = exports.Deployments$ = exports.DeploymentCanarySettings$ = exports.Deployment$ = exports.DeleteVpcLinkRequest$ = exports.DeleteUsagePlanRequest$ = exports.DeleteUsagePlanKeyRequest$ = exports.DeleteStageRequest$ = exports.DeleteRestApiRequest$ = exports.DeleteResourceRequest$ = exports.DeleteRequestValidatorRequest$ = exports.DeleteModelRequest$ = exports.DeleteMethodResponseRequest$ = exports.DeleteMethodRequest$ = void 0;
exports.RequestValidator$ = exports.RejectDomainNameAccessAssociationRequest$ = exports.QuotaSettings$ = exports.PutRestApiRequest$ = exports.PutMethodResponseRequest$ = exports.PutMethodRequest$ = exports.PutIntegrationResponseRequest$ = exports.PutIntegrationRequest$ = exports.PutGatewayResponseRequest$ = exports.PatchOperation$ = exports.MutualTlsAuthenticationInput$ = exports.MutualTlsAuthentication$ = exports.Models$ = exports.Model$ = exports.MethodSnapshot$ = exports.MethodSetting$ = exports.MethodResponse$ = exports.Method$ = exports.IntegrationResponse$ = exports.Integration$ = exports.ImportRestApiRequest$ = exports.ImportDocumentationPartsRequest$ = exports.ImportApiKeysRequest$ = exports.GetVpcLinksRequest$ = exports.GetVpcLinkRequest$ = exports.GetUsageRequest$ = exports.GetUsagePlansRequest$ = exports.GetUsagePlanRequest$ = exports.GetUsagePlanKeysRequest$ = exports.GetUsagePlanKeyRequest$ = exports.GetTagsRequest$ = exports.GetStagesRequest$ = exports.GetStageRequest$ = exports.GetSdkTypesRequest$ = exports.GetSdkTypeRequest$ = exports.GetSdkRequest$ = exports.GetRestApisRequest$ = exports.GetRestApiRequest$ = exports.GetResourcesRequest$ = exports.GetResourceRequest$ = exports.GetRequestValidatorsRequest$ = exports.GetRequestValidatorRequest$ = exports.GetModelTemplateRequest$ = exports.GetModelsRequest$ = exports.GetModelRequest$ = exports.GetMethodResponseRequest$ = exports.GetMethodRequest$ = exports.GetIntegrationResponseRequest$ = exports.GetIntegrationRequest$ = exports.GetGatewayResponsesRequest$ = void 0;
exports.VpcLink$ = exports.UsagePlans$ = exports.UsagePlanKeys$ = exports.UsagePlanKey$ = exports.UsagePlan$ = exports.Usage$ = exports.UpdateVpcLinkRequest$ = exports.UpdateUsageRequest$ = exports.UpdateUsagePlanRequest$ = exports.UpdateStageRequest$ = exports.UpdateRestApiRequest$ = exports.UpdateResourceRequest$ = exports.UpdateRequestValidatorRequest$ = exports.UpdateModelRequest$ = exports.UpdateMethodResponseRequest$ = exports.UpdateMethodRequest$ = exports.UpdateIntegrationResponseRequest$ = exports.UpdateIntegrationRequest$ = exports.UpdateGatewayResponseRequest$ = exports.UpdateDomainNameRequest$ = exports.UpdateDocumentationVersionRequest$ = exports.UpdateDocumentationPartRequest$ = exports.UpdateDeploymentRequest$ = exports.UpdateClientCertificateRequest$ = exports.UpdateBasePathMappingRequest$ = exports.UpdateAuthorizerRequest$ = exports.UpdateApiKeyRequest$ = exports.UpdateAccountRequest$ = exports.UntagResourceRequest$ = exports.TlsConfig$ = exports.ThrottleSettings$ = exports.TestInvokeMethodResponse$ = exports.TestInvokeMethodRequest$ = exports.TestInvokeAuthorizerResponse$ = exports.TestInvokeAuthorizerRequest$ = exports.Template$ = exports.Tags$ = exports.TagResourceRequest$ = exports.Stages$ = exports.StageKey$ = exports.Stage$ = exports.SdkTypes$ = exports.SdkType$ = exports.SdkResponse$ = exports.SdkConfigurationProperty$ = exports.RestApis$ = exports.RestApi$ = exports.Resources$ = exports.Resource$ = exports.RequestValidators$ = void 0;
exports.GetClientCertificate$ = exports.GetBasePathMappings$ = exports.GetBasePathMapping$ = exports.GetAuthorizers$ = exports.GetAuthorizer$ = exports.GetApiKeys$ = exports.GetApiKey$ = exports.GetAccount$ = exports.GenerateClientCertificate$ = exports.FlushStageCache$ = exports.FlushStageAuthorizersCache$ = exports.DeleteVpcLink$ = exports.DeleteUsagePlanKey$ = exports.DeleteUsagePlan$ = exports.DeleteStage$ = exports.DeleteRestApi$ = exports.DeleteResource$ = exports.DeleteRequestValidator$ = exports.DeleteModel$ = exports.DeleteMethodResponse$ = exports.DeleteMethod$ = exports.DeleteIntegrationResponse$ = exports.DeleteIntegration$ = exports.DeleteGatewayResponse$ = exports.DeleteDomainNameAccessAssociation$ = exports.DeleteDomainName$ = exports.DeleteDocumentationVersion$ = exports.DeleteDocumentationPart$ = exports.DeleteDeployment$ = exports.DeleteClientCertificate$ = exports.DeleteBasePathMapping$ = exports.DeleteAuthorizer$ = exports.DeleteApiKey$ = exports.CreateVpcLink$ = exports.CreateUsagePlanKey$ = exports.CreateUsagePlan$ = exports.CreateStage$ = exports.CreateRestApi$ = exports.CreateResource$ = exports.CreateRequestValidator$ = exports.CreateModel$ = exports.CreateDomainNameAccessAssociation$ = exports.CreateDomainName$ = exports.CreateDocumentationVersion$ = exports.CreateDocumentationPart$ = exports.CreateDeployment$ = exports.CreateBasePathMapping$ = exports.CreateAuthorizer$ = exports.CreateApiKey$ = exports.VpcLinks$ = void 0;
exports.TagResource$ = exports.RejectDomainNameAccessAssociation$ = exports.PutRestApi$ = exports.PutMethodResponse$ = exports.PutMethod$ = exports.PutIntegrationResponse$ = exports.PutIntegration$ = exports.PutGatewayResponse$ = exports.ImportRestApi$ = exports.ImportDocumentationParts$ = exports.ImportApiKeys$ = exports.GetVpcLinks$ = exports.GetVpcLink$ = exports.GetUsagePlans$ = exports.GetUsagePlanKeys$ = exports.GetUsagePlanKey$ = exports.GetUsagePlan$ = exports.GetUsage$ = exports.GetTags$ = exports.GetStages$ = exports.GetStage$ = exports.GetSdkTypes$ = exports.GetSdkType$ = exports.GetSdk$ = exports.GetRestApis$ = exports.GetRestApi$ = exports.GetResources$ = exports.GetResource$ = exports.GetRequestValidators$ = exports.GetRequestValidator$ = exports.GetModelTemplate$ = exports.GetModels$ = exports.GetModel$ = exports.GetMethodResponse$ = exports.GetMethod$ = exports.GetIntegrationResponse$ = exports.GetIntegration$ = exports.GetGatewayResponses$ = exports.GetGatewayResponse$ = exports.GetExport$ = exports.GetDomainNames$ = exports.GetDomainNameAccessAssociations$ = exports.GetDomainName$ = exports.GetDocumentationVersions$ = exports.GetDocumentationVersion$ = exports.GetDocumentationParts$ = exports.GetDocumentationPart$ = exports.GetDeployments$ = exports.GetDeployment$ = exports.GetClientCertificates$ = void 0;
exports.UpdateVpcLink$ = exports.UpdateUsagePlan$ = exports.UpdateUsage$ = exports.UpdateStage$ = exports.UpdateRestApi$ = exports.UpdateResource$ = exports.UpdateRequestValidator$ = exports.UpdateModel$ = exports.UpdateMethodResponse$ = exports.UpdateMethod$ = exports.UpdateIntegrationResponse$ = exports.UpdateIntegration$ = exports.UpdateGatewayResponse$ = exports.UpdateDomainName$ = exports.UpdateDocumentationVersion$ = exports.UpdateDocumentationPart$ = exports.UpdateDeployment$ = exports.UpdateClientCertificate$ = exports.UpdateBasePathMapping$ = exports.UpdateAuthorizer$ = exports.UpdateApiKey$ = exports.UpdateAccount$ = exports.UntagResource$ = exports.TestInvokeMethod$ = exports.TestInvokeAuthorizer$ = void 0;
const _A = "Account";
const _AK = "ApiKey";
const _AKI = "ApiKeyIds";
const _AKp = "ApiKeys";
const _ALS = "AccessLogSettings";
const _AS = "ApiStage";
const _Ac = "Accept";
const _Au = "Authorizer";
const _Aut = "Authorizers";
const _BPM = "BasePathMapping";
const _BPMa = "BasePathMappings";
const _BRE = "BadRequestException";
const _CA = "CreateAuthorizer";
const _CAK = "CreateApiKey";
const _CAKR = "CreateApiKeyRequest";
const _CAR = "CreateAuthorizerRequest";
const _CBPM = "CreateBasePathMapping";
const _CBPMR = "CreateBasePathMappingRequest";
const _CC = "ClientCertificate";
const _CCl = "ClientCertificates";
const _CD = "Content-Disposition";
const _CDN = "CreateDomainName";
const _CDNAA = "CreateDomainNameAccessAssociation";
const _CDNAAR = "CreateDomainNameAccessAssociationRequest";
const _CDNR = "CreateDomainNameRequest";
const _CDP = "CreateDocumentationPart";
const _CDPR = "CreateDocumentationPartRequest";
const _CDR = "CreateDeploymentRequest";
const _CDV = "CreateDocumentationVersion";
const _CDVR = "CreateDocumentationVersionRequest";
const _CDr = "CreateDeployment";
const _CE = "ConflictException";
const _CM = "CreateModel";
const _CMR = "CreateModelRequest";
const _CR = "CreateResource";
const _CRA = "CreateRestApi";
const _CRAR = "CreateRestApiRequest";
const _CRR = "CreateResourceRequest";
const _CRV = "CreateRequestValidator";
const _CRVR = "CreateRequestValidatorRequest";
const _CS = "CanarySettings";
const _CSR = "CreateStageRequest";
const _CSr = "CreateStage";
const _CT = "Content-Type";
const _CUP = "CreateUsagePlan";
const _CUPK = "CreateUsagePlanKey";
const _CUPKR = "CreateUsagePlanKeyRequest";
const _CUPR = "CreateUsagePlanRequest";
const _CVL = "CreateVpcLink";
const _CVLR = "CreateVpcLinkRequest";
const _D = "Deployment";
const _DA = "DeleteAuthorizer";
const _DAK = "DeleteApiKey";
const _DAKR = "DeleteApiKeyRequest";
const _DAR = "DeleteAuthorizerRequest";
const _DBPM = "DeleteBasePathMapping";
const _DBPMR = "DeleteBasePathMappingRequest";
const _DCC = "DeleteClientCertificate";
const _DCCR = "DeleteClientCertificateRequest";
const _DCS = "DeploymentCanarySettings";
const _DD = "DeleteDeployment";
const _DDN = "DeleteDomainName";
const _DDNAA = "DeleteDomainNameAccessAssociation";
const _DDNAAR = "DeleteDomainNameAccessAssociationRequest";
const _DDNR = "DeleteDomainNameRequest";
const _DDP = "DeleteDocumentationPart";
const _DDPR = "DeleteDocumentationPartRequest";
const _DDR = "DeleteDeploymentRequest";
const _DDV = "DeleteDocumentationVersion";
const _DDVR = "DeleteDocumentationVersionRequest";
const _DGR = "DeleteGatewayResponse";
const _DGRR = "DeleteGatewayResponseRequest";
const _DI = "DeleteIntegration";
const _DIR = "DeleteIntegrationRequest";
const _DIRR = "DeleteIntegrationResponseRequest";
const _DIRe = "DeleteIntegrationResponse";
const _DM = "DeleteMethod";
const _DMR = "DeleteMethodRequest";
const _DMRR = "DeleteMethodResponseRequest";
const _DMRe = "DeleteModelRequest";
const _DMRel = "DeleteMethodResponse";
const _DMe = "DeleteModel";
const _DN = "DomainName";
const _DNAA = "DomainNameAccessAssociation";
const _DNAAo = "DomainNameAccessAssociations";
const _DNo = "DomainNames";
const _DP = "DocumentationPart";
const _DPI = "DocumentationPartIds";
const _DPL = "DocumentationPartLocation";
const _DPo = "DocumentationParts";
const _DR = "DeleteResource";
const _DRA = "DeleteRestApi";
const _DRAR = "DeleteRestApiRequest";
const _DRR = "DeleteResourceRequest";
const _DRV = "DeleteRequestValidator";
const _DRVR = "DeleteRequestValidatorRequest";
const _DS = "DeleteStage";
const _DSR = "DeleteStageRequest";
const _DUP = "DeleteUsagePlan";
const _DUPK = "DeleteUsagePlanKey";
const _DUPKR = "DeleteUsagePlanKeyRequest";
const _DUPR = "DeleteUsagePlanRequest";
const _DV = "DocumentationVersion";
const _DVL = "DeleteVpcLink";
const _DVLR = "DeleteVpcLinkRequest";
const _DVo = "DocumentationVersions";
const _De = "Deployments";
const _EC = "EndpointConfiguration";
const _ER = "ExportResponse";
const _FSAC = "FlushStageAuthorizersCache";
const _FSACR = "FlushStageAuthorizersCacheRequest";
const _FSC = "FlushStageCache";
const _FSCR = "FlushStageCacheRequest";
const _GA = "GetAccount";
const _GAK = "GetApiKey";
const _GAKR = "GetApiKeyRequest";
const _GAKRe = "GetApiKeysRequest";
const _GAKe = "GetApiKeys";
const _GAR = "GetAccountRequest";
const _GARe = "GetAuthorizerRequest";
const _GARet = "GetAuthorizersRequest";
const _GAe = "GetAuthorizer";
const _GAet = "GetAuthorizers";
const _GBPM = "GetBasePathMapping";
const _GBPMR = "GetBasePathMappingRequest";
const _GBPMRe = "GetBasePathMappingsRequest";
const _GBPMe = "GetBasePathMappings";
const _GCC = "GenerateClientCertificate";
const _GCCR = "GenerateClientCertificateRequest";
const _GCCRe = "GetClientCertificateRequest";
const _GCCRet = "GetClientCertificatesRequest";
const _GCCe = "GetClientCertificate";
const _GCCet = "GetClientCertificates";
const _GD = "GetDeployment";
const _GDN = "GetDomainName";
const _GDNAA = "GetDomainNameAccessAssociations";
const _GDNAAR = "GetDomainNameAccessAssociationsRequest";
const _GDNR = "GetDomainNameRequest";
const _GDNRe = "GetDomainNamesRequest";
const _GDNe = "GetDomainNames";
const _GDP = "GetDocumentationPart";
const _GDPR = "GetDocumentationPartRequest";
const _GDPRe = "GetDocumentationPartsRequest";
const _GDPe = "GetDocumentationParts";
const _GDR = "GetDeploymentRequest";
const _GDRe = "GetDeploymentsRequest";
const _GDV = "GetDocumentationVersion";
const _GDVR = "GetDocumentationVersionRequest";
const _GDVRe = "GetDocumentationVersionsRequest";
const _GDVe = "GetDocumentationVersions";
const _GDe = "GetDeployments";
const _GE = "GetExport";
const _GER = "GetExportRequest";
const _GGR = "GetGatewayResponse";
const _GGRR = "GetGatewayResponseRequest";
const _GGRRe = "GetGatewayResponsesRequest";
const _GGRe = "GetGatewayResponses";
const _GI = "GetIntegration";
const _GIR = "GetIntegrationRequest";
const _GIRR = "GetIntegrationResponseRequest";
const _GIRe = "GetIntegrationResponse";
const _GM = "GetMethod";
const _GMR = "GetMethodRequest";
const _GMRR = "GetMethodResponseRequest";
const _GMRe = "GetModelRequest";
const _GMRet = "GetModelsRequest";
const _GMRete = "GetMethodResponse";
const _GMT = "GetModelTemplate";
const _GMTR = "GetModelTemplateRequest";
const _GMe = "GetModel";
const _GMet = "GetModels";
const _GR = "GatewayResponse";
const _GRA = "GetRestApi";
const _GRAR = "GetRestApiRequest";
const _GRARe = "GetRestApisRequest";
const _GRAe = "GetRestApis";
const _GRR = "GetResourceRequest";
const _GRRe = "GetResourcesRequest";
const _GRV = "GetRequestValidator";
const _GRVR = "GetRequestValidatorRequest";
const _GRVRe = "GetRequestValidatorsRequest";
const _GRVe = "GetRequestValidators";
const _GRa = "GatewayResponses";
const _GRe = "GetResource";
const _GRet = "GetResources";
const _GS = "GetSdk";
const _GSR = "GetSdkRequest";
const _GSRe = "GetStageRequest";
const _GSRet = "GetStagesRequest";
const _GST = "GetSdkType";
const _GSTR = "GetSdkTypeRequest";
const _GSTRe = "GetSdkTypesRequest";
const _GSTe = "GetSdkTypes";
const _GSe = "GetStage";
const _GSet = "GetStages";
const _GT = "GetTags";
const _GTR = "GetTagsRequest";
const _GU = "GetUsage";
const _GUP = "GetUsagePlan";
const _GUPK = "GetUsagePlanKey";
const _GUPKR = "GetUsagePlanKeyRequest";
const _GUPKRe = "GetUsagePlanKeysRequest";
const _GUPKe = "GetUsagePlanKeys";
const _GUPR = "GetUsagePlanRequest";
const _GUPRe = "GetUsagePlansRequest";
const _GUPe = "GetUsagePlans";
const _GUR = "GetUsageRequest";
const _GVL = "GetVpcLink";
const _GVLR = "GetVpcLinkRequest";
const _GVLRe = "GetVpcLinksRequest";
const _GVLe = "GetVpcLinks";
const _I = "Integration";
const _IAK = "ImportApiKeys";
const _IAKR = "ImportApiKeysRequest";
const _IDP = "ImportDocumentationParts";
const _IDPR = "ImportDocumentationPartsRequest";
const _IR = "IntegrationResponse";
const _IRA = "ImportRestApi";
const _IRAR = "ImportRestApiRequest";
const _LEE = "LimitExceededException";
const _LOA = "ListOfAuthorizer";
const _LOAK = "ListOfApiKey";
const _LOAS = "ListOfApiStage";
const _LOBPM = "ListOfBasePathMapping";
const _LOCC = "ListOfClientCertificate";
const _LOD = "ListOfDeployment";
const _LODN = "ListOfDomainName";
const _LODNAA = "ListOfDomainNameAccessAssociation";
const _LODP = "ListOfDocumentationPart";
const _LODV = "ListOfDocumentationVersion";
const _LOGR = "ListOfGatewayResponse";
const _LOM = "ListOfModel";
const _LOPO = "ListOfPatchOperation";
const _LOR = "ListOfResource";
const _LORA = "ListOfRestApi";
const _LORV = "ListOfRequestValidator";
const _LOS = "ListOfStage";
const _LOSCP = "ListOfSdkConfigurationProperty";
const _LOSK = "ListOfStageKeys";
const _LOST = "ListOfSdkType";
const _LOU = "ListOfUsage";
const _LOUP = "ListOfUsagePlan";
const _LOUPK = "ListOfUsagePlanKey";
const _LOVL = "ListOfVpcLink";
const _M = "Method";
const _MOASTS = "MapOfApiStageThrottleSettings";
const _MOIR = "MapOfIntegrationResponse";
const _MOKU = "MapOfKeyUsages";
const _MOM = "MapOfMethod";
const _MOMR = "MapOfMethodResponse";
const _MOMS = "MapOfMethodSettings";
const _MOMSa = "MapOfMethodSnapshot";
const _MOSTL = "MapOfStringToList";
const _MR = "MethodResponse";
const _MS = "MethodSetting";
const _MSe = "MethodSnapshot";
const _MTA = "MutualTlsAuthentication";
const _MTAI = "MutualTlsAuthenticationInput";
const _Mo = "Model";
const _Mod = "Models";
const _NFE = "NotFoundException";
const _PGR = "PutGatewayResponse";
const _PGRR = "PutGatewayResponseRequest";
const _PI = "PutIntegration";
const _PIR = "PutIntegrationRequest";
const _PIRR = "PutIntegrationResponseRequest";
const _PIRu = "PutIntegrationResponse";
const _PM = "PutMethod";
const _PMR = "PutMethodRequest";
const _PMRR = "PutMethodResponseRequest";
const _PMRu = "PutMethodResponse";
const _PO = "PatchOperation";
const _PRA = "PutRestApi";
const _PRAR = "PutRestApiRequest";
const _PTMOMS = "PathToMapOfMethodSnapshot";
const _QS = "QuotaSettings";
const _R = "Resource";
const _RA = "Retry-After";
const _RAe = "RestApi";
const _RAes = "RestApis";
const _RDNAA = "RejectDomainNameAccessAssociation";
const _RDNAAR = "RejectDomainNameAccessAssociationRequest";
const _RV = "RequestValidator";
const _RVe = "RequestValidators";
const _Re = "Resources";
const _S = "Stage";
const _SCP = "SdkConfigurationProperty";
const _SK = "StageKey";
const _SR = "SdkResponse";
const _ST = "SdkType";
const _STd = "SdkTypes";
const _SUE = "ServiceUnavailableException";
const _St = "Stages";
const _T = "Tags";
const _TC = "TlsConfig";
const _TIA = "TestInvokeAuthorizer";
const _TIAR = "TestInvokeAuthorizerRequest";
const _TIARe = "TestInvokeAuthorizerResponse";
const _TIM = "TestInvokeMethod";
const _TIMR = "TestInvokeMethodRequest";
const _TIMRe = "TestInvokeMethodResponse";
const _TMRE = "TooManyRequestsException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TS = "ThrottleSettings";
const _Te = "Template";
const _U = "Usage";
const _UA = "UpdateAccount";
const _UAK = "UpdateApiKey";
const _UAKR = "UpdateApiKeyRequest";
const _UAR = "UpdateAccountRequest";
const _UARp = "UpdateAuthorizerRequest";
const _UAp = "UpdateAuthorizer";
const _UBPM = "UpdateBasePathMapping";
const _UBPMR = "UpdateBasePathMappingRequest";
const _UCC = "UpdateClientCertificate";
const _UCCR = "UpdateClientCertificateRequest";
const _UD = "UpdateDeployment";
const _UDN = "UpdateDomainName";
const _UDNR = "UpdateDomainNameRequest";
const _UDP = "UpdateDocumentationPart";
const _UDPR = "UpdateDocumentationPartRequest";
const _UDR = "UpdateDeploymentRequest";
const _UDV = "UpdateDocumentationVersion";
const _UDVR = "UpdateDocumentationVersionRequest";
const _UE = "UnauthorizedException";
const _UGR = "UpdateGatewayResponse";
const _UGRR = "UpdateGatewayResponseRequest";
const _UI = "UpdateIntegration";
const _UIR = "UpdateIntegrationRequest";
const _UIRR = "UpdateIntegrationResponseRequest";
const _UIRp = "UpdateIntegrationResponse";
const _UM = "UpdateMethod";
const _UMR = "UpdateMethodRequest";
const _UMRR = "UpdateMethodResponseRequest";
const _UMRp = "UpdateModelRequest";
const _UMRpd = "UpdateMethodResponse";
const _UMp = "UpdateModel";
const _UP = "UsagePlan";
const _UPK = "UsagePlanKey";
const _UPKs = "UsagePlanKeys";
const _UPs = "UsagePlans";
const _UR = "UntagResource";
const _URA = "UpdateRestApi";
const _URAR = "UpdateRestApiRequest";
const _URR = "UntagResourceRequest";
const _URRp = "UpdateResourceRequest";
const _URV = "UpdateRequestValidator";
const _URVR = "UpdateRequestValidatorRequest";
const _URp = "UpdateResource";
const _US = "UpdateStage";
const _USR = "UpdateStageRequest";
const _UU = "UpdateUsage";
const _UUP = "UpdateUsagePlan";
const _UUPR = "UpdateUsagePlanRequest";
const _UUR = "UpdateUsageRequest";
const _UVL = "UpdateVpcLink";
const _UVLR = "UpdateVpcLinkRequest";
const _VL = "VpcLink";
const _VLp = "VpcLinks";
const _a = "accepts";
const _aAS = "accessAssociationSource";
const _aAST = "accessAssociationSourceType";
const _aC = "authorizerCredentials";
const _aCd = "additionalContext";
const _aI = "apiId";
const _aIu = "authorizerId";
const _aK = "apiKey";
const _aKR = "apiKeyRequired";
const _aKS = "apiKeySource";
const _aKV = "apiKeyVersion";
const _aLS = "accessLogSettings";
const _aRTIS = "authorizerResultTtlInSeconds";
const _aS = "apiStages";
const _aSM = "apiStatusMessage";
const _aSp = "apiSummary";
const _aSpi = "apiStatus";
const _aSu = "authorizationScopes";
const _aT = "authType";
const _aTu = "authorizationType";
const _aU = "authorizerUri";
const _au = "authorization";
const _b = "body";
const _bL = "burstLimit";
const _bMT = "binaryMediaTypes";
const _bP = "basePath";
const _c = "client";
const _cA = "certificateArn";
const _cB = "certificateBody";
const _cC = "certificateChain";
const _cCE = "cacheClusterEnabled";
const _cCI = "clientCertificateId";
const _cCS = "cacheClusterSize";
const _cCSa = "cacheClusterStatus";
const _cD = "createdDate";
const _cDE = "cacheDataEncrypted";
const _cDo = "contentDisposition";
const _cE = "cachingEnabled";
const _cF = "cloneFrom";
const _cH = "contentHandling";
const _cI = "customerId";
const _cIo = "connectionId";
const _cKP = "cacheKeyParameters";
const _cN = "certificateName";
const _cNa = "cacheNamespace";
const _cP = "configurationProperties";
const _cPK = "certificatePrivateKey";
const _cRA = "cloudwatchRoleArn";
const _cS = "canarySettings";
const _cSl = "clientStatus";
const _cT = "contentType";
const _cTIS = "cacheTtlInSeconds";
const _cTo = "connectionType";
const _cUD = "certificateUploadDate";
const _cl = "claims";
const _cr = "credentials";
const _d = "description";
const _dA = "destinationArn";
const _dDN = "distributionDomainName";
const _dEAE = "disableExecuteApiEndpoint";
const _dHZI = "distributionHostedZoneId";
const _dI = "deploymentId";
const _dN = "domainName";
const _dNA = "domainNameArn";
const _dNAAA = "domainNameAccessAssociationArn";
const _dNI = "domainNameId";
const _dNS = "domainNameStatus";
const _dNSM = "domainNameStatusMessage";
const _dPI = "documentationPartId";
const _dR = "defaultResponse";
const _dTE = "dataTraceEnabled";
const _dV = "documentationVersion";
const _dVe = "defaultValue";
const _e = "error";
const _eAM = "endpointAccessMode";
const _eC = "endpointConfiguration";
const _eD = "expirationDate";
const _eDn = "endDate";
const _eT = "exportType";
const _em = "embed";
const _en = "enabled";
const _f = "format";
const _fN = "friendlyName";
const _fOW = "failOnWarnings";
const _fa = "failonwarnings";
const _fe = "features";
const _fl = "flatten";
const _fr = "from";
const _gDI = "generateDistinctId";
const _h = "headers";
const _hE = "httpError";
const _hH = "httpHeader";
const _hL = "httpLabel";
const _hM = "httpMethod";
const _hQ = "httpQuery";
const _ht = "http";
const _i = "id";
const _iAT = "ipAddressType";
const _iHM = "integrationHttpMethod";
const _iR = "integrationResponses";
const _iS = "identitySource";
const _iSV = "insecureSkipVerification";
const _iT = "integrationTarget";
const _iV = "includeValue";
const _iVE = "identityValidationExpression";
const _iVn = "includeValues";
const _id = "ids";
const _it = "items";
const _ite = "item";
const _jN = "jsonName";
const _kI = "keyId";
const _kT = "keyType";
const _l = "location";
const _lL = "loggingLevel";
const _lS = "locationStatus";
const _lUD = "lastUpdatedDate";
const _la = "latency";
const _li = "limit";
const _lo = "log";
const _m = "message";
const _mCS = "minimumCompressionSize";
const _mE = "metricsEnabled";
const _mI = "methodIntegration";
const _mN = "modelName";
const _mP = "managementPolicy";
const _mR = "methodResponses";
const _mS = "methodSettings";
const _mTA = "mutualTlsAuthentication";
const _mVH = "multiValueHeaders";
const _me = "method";
const _mo = "mode";
const _n = "name";
const _nQ = "nameQuery";
const _o = "op";
const _oN = "operationName";
const _oVCA = "ownershipVerificationCertificateArn";
const _of = "offset";
const _p = "position";
const _pARN = "providerARNs";
const _pB = "passthroughBehavior";
const _pC = "productCode";
const _pEC = "pemEncodedCertificate";
const _pI = "parentId";
const _pIr = "principalId";
const _pO = "patchOperations";
const _pP = "pathPart";
const _pT = "percentTraffic";
const _pWQS = "pathWithQueryString";
const _pa = "path";
const _par = "parameters";
const _pe = "period";
const _po = "policy";
const _pr = "properties";
const _q = "quota";
const _r = "required";
const _rA = "resourceArn";
const _rAFCC = "requireAuthorizationForCacheControl";
const _rAI = "restApiId";
const _rAS = "retryAfterSeconds";
const _rCA = "regionalCertificateArn";
const _rCN = "regionalCertificateName";
const _rDN = "regionalDomainName";
const _rHM = "requestHttpMethod";
const _rHZI = "regionalHostedZoneId";
const _rI = "resourceId";
const _rL = "rateLimit";
const _rM = "routingMode";
const _rMe = "requestModels";
const _rMes = "responseModels";
const _rMeso = "resourceMethods";
const _rO = "resourceOwner";
const _rP = "responseParameters";
const _rPe = "requestParameters";
const _rRI = "rootResourceId";
const _rT = "responseType";
const _rTM = "responseTransferMode";
const _rTe = "responseTemplates";
const _rTeq = "requestTemplates";
const _rVI = "requestValidatorId";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.apigateway";
const _sC = "statusCode";
const _sD = "stageDescription";
const _sDt = "startDate";
const _sK = "stageKeys";
const _sM = "statusMessage";
const _sN = "stageName";
const _sP = "securityPolicy";
const _sPe = "selectionPattern";
const _sT = "sdkType";
const _sV = "stageVariables";
const _sVO = "stageVariableOverrides";
const _sc = "schema";
const _se = "server";
const _st = "stage";
const _sta = "status";
const _t = "tags";
const _tA = "targetArns";
const _tBL = "throttlingBurstLimit";
const _tC = "tlsConfig";
const _tE = "tracingEnabled";
const _tIM = "timeoutInMillis";
const _tK = "tagKeys";
const _tRL = "throttlingRateLimit";
const _tS = "throttleSettings";
const _tU = "truststoreUri";
const _tV = "truststoreVersion";
const _tW = "truststoreWarnings";
const _th = "throttle";
const _ty = "type";
const _typ = "types";
const _u = "uri";
const _uCCHS = "unauthorizedCacheControlHeaderStrategy";
const _uPI = "usagePlanId";
const _uSC = "useStageCache";
const _v = "value";
const _vEI = "vpcEndpointIds";
const _vLI = "vpcLinkId";
const _vRB = "validateRequestBody";
const _vRP = "validateRequestParameters";
const _va = "variables";
const _val = "values";
const _ve = "version";
const _w = "warnings";
const _wAA = "webAclArn";
const n0 = "com.amazonaws.apigateway";
const schema_1 = require("@smithy/core/schema");
const APIGatewayServiceException_1 = require("../models/APIGatewayServiceException");
const errors_1 = require("../models/errors");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.APIGatewayServiceException$ = [-3, _s, "APIGatewayServiceException", 0, [], []];
_s_registry.registerError(exports.APIGatewayServiceException$, APIGatewayServiceException_1.APIGatewayServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.BadRequestException$ = [-3, n0, _BRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.BadRequestException$, errors_1.BadRequestException);
exports.ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ConflictException$, errors_1.ConflictException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 429 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.NotFoundException$ = [-3, n0, _NFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(exports.NotFoundException$, errors_1.NotFoundException);
exports.ServiceUnavailableException$ = [-3, n0, _SUE,
    { [_e]: _se, [_hE]: 503 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
n0_registry.registerError(exports.ServiceUnavailableException$, errors_1.ServiceUnavailableException);
exports.TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c, [_hE]: 429 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
n0_registry.registerError(exports.TooManyRequestsException$, errors_1.TooManyRequestsException);
exports.UnauthorizedException$ = [-3, n0, _UE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnauthorizedException$, errors_1.UnauthorizedException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
exports.AccessLogSettings$ = [3, n0, _ALS,
    0,
    [_f, _dA],
    [0, 0]
];
exports.Account$ = [3, n0, _A,
    0,
    [_cRA, _tS, _fe, _aKV],
    [0, () => exports.ThrottleSettings$, 64 | 0, 0]
];
exports.ApiKey$ = [3, n0, _AK,
    0,
    [_i, _v, _n, _cI, _d, _en, _cD, _lUD, _sK, _t],
    [0, 0, 0, 0, 0, 2, 4, 4, 64 | 0, 128 | 0]
];
exports.ApiKeyIds$ = [3, n0, _AKI,
    0,
    [_id, _w],
    [64 | 0, 64 | 0]
];
exports.ApiKeys$ = [3, n0, _AKp,
    0,
    [_w, _it, _p],
    [64 | 0, [() => ListOfApiKey, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.ApiStage$ = [3, n0, _AS,
    0,
    [_aI, _st, _th],
    [0, 0, () => MapOfApiStageThrottleSettings]
];
exports.Authorizer$ = [3, n0, _Au,
    0,
    [_i, _n, _ty, _pARN, _aT, _aU, _aC, _iS, _iVE, _aRTIS],
    [0, 0, 0, 64 | 0, 0, 0, 0, 0, 0, 1]
];
exports.Authorizers$ = [3, n0, _Aut,
    0,
    [_it, _p],
    [[() => ListOfAuthorizer, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.BasePathMapping$ = [3, n0, _BPM,
    0,
    [_bP, _rAI, _st],
    [0, 0, 0]
];
exports.BasePathMappings$ = [3, n0, _BPMa,
    0,
    [_it, _p],
    [[() => ListOfBasePathMapping, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.CanarySettings$ = [3, n0, _CS,
    0,
    [_pT, _dI, _sVO, _uSC],
    [1, 0, 128 | 0, 2]
];
exports.ClientCertificate$ = [3, n0, _CC,
    0,
    [_cCI, _d, _pEC, _cD, _eD, _t],
    [0, 0, 0, 4, 4, 128 | 0]
];
exports.ClientCertificates$ = [3, n0, _CCl,
    0,
    [_it, _p],
    [[() => ListOfClientCertificate, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.CreateApiKeyRequest$ = [3, n0, _CAKR,
    0,
    [_n, _d, _en, _gDI, _v, _sK, _cI, _t],
    [0, 0, 2, 2, 0, () => ListOfStageKeys, 0, 128 | 0]
];
exports.CreateAuthorizerRequest$ = [3, n0, _CAR,
    0,
    [_rAI, _n, _ty, _pARN, _aT, _aU, _aC, _iS, _iVE, _aRTIS],
    [[0, 1], 0, 0, 64 | 0, 0, 0, 0, 0, 0, 1], 3
];
exports.CreateBasePathMappingRequest$ = [3, n0, _CBPMR,
    0,
    [_dN, _rAI, _dNI, _bP, _st],
    [[0, 1], 0, [0, { [_hQ]: _dNI }], 0, 0], 2
];
exports.CreateDeploymentRequest$ = [3, n0, _CDR,
    0,
    [_rAI, _sN, _sD, _d, _cCE, _cCS, _va, _cS, _tE],
    [[0, 1], 0, 0, 0, 2, 0, 128 | 0, () => exports.DeploymentCanarySettings$, 2], 1
];
exports.CreateDocumentationPartRequest$ = [3, n0, _CDPR,
    0,
    [_rAI, _l, _pr],
    [[0, 1], () => exports.DocumentationPartLocation$, 0], 3
];
exports.CreateDocumentationVersionRequest$ = [3, n0, _CDVR,
    0,
    [_rAI, _dV, _sN, _d],
    [[0, 1], 0, 0, 0], 2
];
exports.CreateDomainNameAccessAssociationRequest$ = [3, n0, _CDNAAR,
    0,
    [_dNA, _aAST, _aAS, _t],
    [0, 0, 0, 128 | 0], 3
];
exports.CreateDomainNameRequest$ = [3, n0, _CDNR,
    0,
    [_dN, _cN, _cB, _cPK, _cC, _cA, _rCN, _rCA, _eC, _t, _sP, _eAM, _mTA, _oVCA, _po, _rM],
    [0, 0, 0, 0, 0, 0, 0, 0, () => exports.EndpointConfiguration$, 128 | 0, 0, 0, () => exports.MutualTlsAuthenticationInput$, 0, 0, 0], 1
];
exports.CreateModelRequest$ = [3, n0, _CMR,
    0,
    [_rAI, _n, _cT, _d, _sc],
    [[0, 1], 0, 0, 0, 0], 3
];
exports.CreateRequestValidatorRequest$ = [3, n0, _CRVR,
    0,
    [_rAI, _n, _vRB, _vRP],
    [[0, 1], 0, 2, 2], 1
];
exports.CreateResourceRequest$ = [3, n0, _CRR,
    0,
    [_rAI, _pI, _pP],
    [[0, 1], [0, 1], 0], 3
];
exports.CreateRestApiRequest$ = [3, n0, _CRAR,
    0,
    [_n, _d, _ve, _cF, _bMT, _mCS, _aKS, _eC, _po, _t, _dEAE, _sP, _eAM],
    [0, 0, 0, 0, 64 | 0, 1, 0, () => exports.EndpointConfiguration$, 0, 128 | 0, 2, 0, 0], 1
];
exports.CreateStageRequest$ = [3, n0, _CSR,
    0,
    [_rAI, _sN, _dI, _d, _cCE, _cCS, _va, _dV, _cS, _tE, _t],
    [[0, 1], 0, 0, 0, 2, 0, 128 | 0, 0, () => exports.CanarySettings$, 2, 128 | 0], 3
];
exports.CreateUsagePlanKeyRequest$ = [3, n0, _CUPKR,
    0,
    [_uPI, _kI, _kT],
    [[0, 1], 0, 0], 3
];
exports.CreateUsagePlanRequest$ = [3, n0, _CUPR,
    0,
    [_n, _d, _aS, _th, _q, _t],
    [0, 0, () => ListOfApiStage, () => exports.ThrottleSettings$, () => exports.QuotaSettings$, 128 | 0], 1
];
exports.CreateVpcLinkRequest$ = [3, n0, _CVLR,
    0,
    [_n, _tA, _d, _t],
    [0, 64 | 0, 0, 128 | 0], 2
];
exports.DeleteApiKeyRequest$ = [3, n0, _DAKR,
    0,
    [_aK],
    [[0, 1]], 1
];
exports.DeleteAuthorizerRequest$ = [3, n0, _DAR,
    0,
    [_rAI, _aIu],
    [[0, 1], [0, 1]], 2
];
exports.DeleteBasePathMappingRequest$ = [3, n0, _DBPMR,
    0,
    [_dN, _bP, _dNI],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }]], 2
];
exports.DeleteClientCertificateRequest$ = [3, n0, _DCCR,
    0,
    [_cCI],
    [[0, 1]], 1
];
exports.DeleteDeploymentRequest$ = [3, n0, _DDR,
    0,
    [_rAI, _dI],
    [[0, 1], [0, 1]], 2
];
exports.DeleteDocumentationPartRequest$ = [3, n0, _DDPR,
    0,
    [_rAI, _dPI],
    [[0, 1], [0, 1]], 2
];
exports.DeleteDocumentationVersionRequest$ = [3, n0, _DDVR,
    0,
    [_rAI, _dV],
    [[0, 1], [0, 1]], 2
];
exports.DeleteDomainNameAccessAssociationRequest$ = [3, n0, _DDNAAR,
    0,
    [_dNAAA],
    [[0, 1]], 1
];
exports.DeleteDomainNameRequest$ = [3, n0, _DDNR,
    0,
    [_dN, _dNI],
    [[0, 1], [0, { [_hQ]: _dNI }]], 1
];
exports.DeleteGatewayResponseRequest$ = [3, n0, _DGRR,
    0,
    [_rAI, _rT],
    [[0, 1], [0, 1]], 2
];
exports.DeleteIntegrationRequest$ = [3, n0, _DIR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
exports.DeleteIntegrationResponseRequest$ = [3, n0, _DIRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.DeleteMethodRequest$ = [3, n0, _DMR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
exports.DeleteMethodResponseRequest$ = [3, n0, _DMRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.DeleteModelRequest$ = [3, n0, _DMRe,
    0,
    [_rAI, _mN],
    [[0, 1], [0, 1]], 2
];
exports.DeleteRequestValidatorRequest$ = [3, n0, _DRVR,
    0,
    [_rAI, _rVI],
    [[0, 1], [0, 1]], 2
];
exports.DeleteResourceRequest$ = [3, n0, _DRR,
    0,
    [_rAI, _rI],
    [[0, 1], [0, 1]], 2
];
exports.DeleteRestApiRequest$ = [3, n0, _DRAR,
    0,
    [_rAI],
    [[0, 1]], 1
];
exports.DeleteStageRequest$ = [3, n0, _DSR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
exports.DeleteUsagePlanKeyRequest$ = [3, n0, _DUPKR,
    0,
    [_uPI, _kI],
    [[0, 1], [0, 1]], 2
];
exports.DeleteUsagePlanRequest$ = [3, n0, _DUPR,
    0,
    [_uPI],
    [[0, 1]], 1
];
exports.DeleteVpcLinkRequest$ = [3, n0, _DVLR,
    0,
    [_vLI],
    [[0, 1]], 1
];
exports.Deployment$ = [3, n0, _D,
    0,
    [_i, _d, _cD, _aSp],
    [0, 0, 4, () => PathToMapOfMethodSnapshot]
];
exports.DeploymentCanarySettings$ = [3, n0, _DCS,
    0,
    [_pT, _sVO, _uSC],
    [1, 128 | 0, 2]
];
exports.Deployments$ = [3, n0, _De,
    0,
    [_it, _p],
    [[() => ListOfDeployment, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.DocumentationPart$ = [3, n0, _DP,
    0,
    [_i, _l, _pr],
    [0, () => exports.DocumentationPartLocation$, 0]
];
exports.DocumentationPartIds$ = [3, n0, _DPI,
    0,
    [_id, _w],
    [64 | 0, 64 | 0]
];
exports.DocumentationPartLocation$ = [3, n0, _DPL,
    0,
    [_ty, _pa, _me, _sC, _n],
    [0, 0, 0, 0, 0], 1
];
exports.DocumentationParts$ = [3, n0, _DPo,
    0,
    [_it, _p],
    [[() => ListOfDocumentationPart, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.DocumentationVersion$ = [3, n0, _DV,
    0,
    [_ve, _cD, _d],
    [0, 4, 0]
];
exports.DocumentationVersions$ = [3, n0, _DVo,
    0,
    [_it, _p],
    [[() => ListOfDocumentationVersion, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.DomainName$ = [3, n0, _DN,
    0,
    [_dN, _dNI, _dNA, _cN, _cA, _cUD, _rDN, _rHZI, _rCN, _rCA, _dDN, _dHZI, _eC, _dNS, _dNSM, _sP, _eAM, _t, _mTA, _oVCA, _mP, _po, _rM],
    [0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, () => exports.EndpointConfiguration$, 0, 0, 0, 0, 128 | 0, () => exports.MutualTlsAuthentication$, 0, 0, 0, 0]
];
exports.DomainNameAccessAssociation$ = [3, n0, _DNAA,
    0,
    [_dNAAA, _dNA, _aAST, _aAS, _t],
    [0, 0, 0, 0, 128 | 0]
];
exports.DomainNameAccessAssociations$ = [3, n0, _DNAAo,
    0,
    [_it, _p],
    [[() => ListOfDomainNameAccessAssociation, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.DomainNames$ = [3, n0, _DNo,
    0,
    [_it, _p],
    [[() => ListOfDomainName, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.EndpointConfiguration$ = [3, n0, _EC,
    0,
    [_typ, _iAT, _vEI],
    [64 | 0, 0, 64 | 0]
];
exports.ExportResponse$ = [3, n0, _ER,
    0,
    [_cT, _cDo, _b],
    [[0, { [_hH]: _CT }], [0, { [_hH]: _CD }], [21, 16]]
];
exports.FlushStageAuthorizersCacheRequest$ = [3, n0, _FSACR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
exports.FlushStageCacheRequest$ = [3, n0, _FSCR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
exports.GatewayResponse$ = [3, n0, _GR,
    0,
    [_rT, _sC, _rP, _rTe, _dR],
    [0, 0, 128 | 0, 128 | 0, 2]
];
exports.GatewayResponses$ = [3, n0, _GRa,
    0,
    [_it, _p],
    [[() => ListOfGatewayResponse, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.GenerateClientCertificateRequest$ = [3, n0, _GCCR,
    0,
    [_d, _t],
    [0, 128 | 0]
];
exports.GetAccountRequest$ = [3, n0, _GAR,
    0,
    [],
    []
];
exports.GetApiKeyRequest$ = [3, n0, _GAKR,
    0,
    [_aK, _iV],
    [[0, 1], [2, { [_hQ]: _iV }]], 1
];
exports.GetApiKeysRequest$ = [3, n0, _GAKRe,
    0,
    [_p, _li, _nQ, _cI, _iVn],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _n }], [0, { [_hQ]: _cI }], [2, { [_hQ]: _iVn }]]
];
exports.GetAuthorizerRequest$ = [3, n0, _GARe,
    0,
    [_rAI, _aIu],
    [[0, 1], [0, 1]], 2
];
exports.GetAuthorizersRequest$ = [3, n0, _GARet,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetBasePathMappingRequest$ = [3, n0, _GBPMR,
    0,
    [_dN, _bP, _dNI],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }]], 2
];
exports.GetBasePathMappingsRequest$ = [3, n0, _GBPMRe,
    0,
    [_dN, _dNI, _p, _li],
    [[0, 1], [0, { [_hQ]: _dNI }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetClientCertificateRequest$ = [3, n0, _GCCRe,
    0,
    [_cCI],
    [[0, 1]], 1
];
exports.GetClientCertificatesRequest$ = [3, n0, _GCCRet,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
exports.GetDeploymentRequest$ = [3, n0, _GDR,
    0,
    [_rAI, _dI, _em],
    [[0, 1], [0, 1], [64 | 0, { [_hQ]: _em }]], 2
];
exports.GetDeploymentsRequest$ = [3, n0, _GDRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetDocumentationPartRequest$ = [3, n0, _GDPR,
    0,
    [_rAI, _dPI],
    [[0, 1], [0, 1]], 2
];
exports.GetDocumentationPartsRequest$ = [3, n0, _GDPRe,
    0,
    [_rAI, _ty, _nQ, _pa, _p, _li, _lS],
    [[0, 1], [0, { [_hQ]: _ty }], [0, { [_hQ]: _n }], [0, { [_hQ]: _pa }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _lS }]], 1
];
exports.GetDocumentationVersionRequest$ = [3, n0, _GDVR,
    0,
    [_rAI, _dV],
    [[0, 1], [0, 1]], 2
];
exports.GetDocumentationVersionsRequest$ = [3, n0, _GDVRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetDomainNameAccessAssociationsRequest$ = [3, n0, _GDNAAR,
    0,
    [_p, _li, _rO],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _rO }]]
];
exports.GetDomainNameRequest$ = [3, n0, _GDNR,
    0,
    [_dN, _dNI],
    [[0, 1], [0, { [_hQ]: _dNI }]], 1
];
exports.GetDomainNamesRequest$ = [3, n0, _GDNRe,
    0,
    [_p, _li, _rO],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _rO }]]
];
exports.GetExportRequest$ = [3, n0, _GER,
    0,
    [_rAI, _sN, _eT, _par, _a],
    [[0, 1], [0, 1], [0, 1], [128 | 0, 64], [0, { [_hH]: _Ac }]], 3
];
exports.GetGatewayResponseRequest$ = [3, n0, _GGRR,
    0,
    [_rAI, _rT],
    [[0, 1], [0, 1]], 2
];
exports.GetGatewayResponsesRequest$ = [3, n0, _GGRRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetIntegrationRequest$ = [3, n0, _GIR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
exports.GetIntegrationResponseRequest$ = [3, n0, _GIRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.GetMethodRequest$ = [3, n0, _GMR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
exports.GetMethodResponseRequest$ = [3, n0, _GMRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.GetModelRequest$ = [3, n0, _GMRe,
    0,
    [_rAI, _mN, _fl],
    [[0, 1], [0, 1], [2, { [_hQ]: _fl }]], 2
];
exports.GetModelsRequest$ = [3, n0, _GMRet,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetModelTemplateRequest$ = [3, n0, _GMTR,
    0,
    [_rAI, _mN],
    [[0, 1], [0, 1]], 2
];
exports.GetRequestValidatorRequest$ = [3, n0, _GRVR,
    0,
    [_rAI, _rVI],
    [[0, 1], [0, 1]], 2
];
exports.GetRequestValidatorsRequest$ = [3, n0, _GRVRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetResourceRequest$ = [3, n0, _GRR,
    0,
    [_rAI, _rI, _em],
    [[0, 1], [0, 1], [64 | 0, { [_hQ]: _em }]], 2
];
exports.GetResourcesRequest$ = [3, n0, _GRRe,
    0,
    [_rAI, _p, _li, _em],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [64 | 0, { [_hQ]: _em }]], 1
];
exports.GetRestApiRequest$ = [3, n0, _GRAR,
    0,
    [_rAI],
    [[0, 1]], 1
];
exports.GetRestApisRequest$ = [3, n0, _GRARe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
exports.GetSdkRequest$ = [3, n0, _GSR,
    0,
    [_rAI, _sN, _sT, _par],
    [[0, 1], [0, 1], [0, 1], [128 | 0, 64]], 3
];
exports.GetSdkTypeRequest$ = [3, n0, _GSTR,
    0,
    [_i],
    [[0, 1]], 1
];
exports.GetSdkTypesRequest$ = [3, n0, _GSTRe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
exports.GetStageRequest$ = [3, n0, _GSRe,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
exports.GetStagesRequest$ = [3, n0, _GSRet,
    0,
    [_rAI, _dI],
    [[0, 1], [0, { [_hQ]: _dI }]], 1
];
exports.GetTagsRequest$ = [3, n0, _GTR,
    0,
    [_rA, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
exports.GetUsagePlanKeyRequest$ = [3, n0, _GUPKR,
    0,
    [_uPI, _kI],
    [[0, 1], [0, 1]], 2
];
exports.GetUsagePlanKeysRequest$ = [3, n0, _GUPKRe,
    0,
    [_uPI, _p, _li, _nQ],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _n }]], 1
];
exports.GetUsagePlanRequest$ = [3, n0, _GUPR,
    0,
    [_uPI],
    [[0, 1]], 1
];
exports.GetUsagePlansRequest$ = [3, n0, _GUPRe,
    0,
    [_p, _kI, _li],
    [[0, { [_hQ]: _p }], [0, { [_hQ]: _kI }], [1, { [_hQ]: _li }]]
];
exports.GetUsageRequest$ = [3, n0, _GUR,
    0,
    [_uPI, _sDt, _eDn, _kI, _p, _li],
    [[0, 1], [0, { [_hQ]: _sDt }], [0, { [_hQ]: _eDn }], [0, { [_hQ]: _kI }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 3
];
exports.GetVpcLinkRequest$ = [3, n0, _GVLR,
    0,
    [_vLI],
    [[0, 1]], 1
];
exports.GetVpcLinksRequest$ = [3, n0, _GVLRe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
exports.ImportApiKeysRequest$ = [3, n0, _IAKR,
    0,
    [_b, _f, _fOW],
    [[21, 16], [0, { [_hQ]: _f }], [2, { [_hQ]: _fa }]], 2
];
exports.ImportDocumentationPartsRequest$ = [3, n0, _IDPR,
    0,
    [_rAI, _b, _mo, _fOW],
    [[0, 1], [21, 16], [0, { [_hQ]: _mo }], [2, { [_hQ]: _fa }]], 2
];
exports.ImportRestApiRequest$ = [3, n0, _IRAR,
    0,
    [_b, _fOW, _par],
    [[21, 16], [2, { [_hQ]: _fa }], [128 | 0, 64]], 1
];
exports.Integration$ = [3, n0, _I,
    0,
    [_ty, _hM, _u, _cTo, _cIo, _cr, _rPe, _rTeq, _pB, _cH, _tIM, _cNa, _cKP, _iR, _tC, _rTM, _iT],
    [0, 0, 0, 0, 0, 0, 128 | 0, 128 | 0, 0, 0, 1, 0, 64 | 0, () => MapOfIntegrationResponse, () => exports.TlsConfig$, 0, 0]
];
exports.IntegrationResponse$ = [3, n0, _IR,
    0,
    [_sC, _sPe, _rP, _rTe, _cH],
    [0, 0, 128 | 0, 128 | 0, 0]
];
exports.Method$ = [3, n0, _M,
    0,
    [_hM, _aTu, _aIu, _aKR, _rVI, _oN, _rPe, _rMe, _mR, _mI, _aSu],
    [0, 0, 0, 2, 0, 0, 128 | 2, 128 | 0, () => MapOfMethodResponse, () => exports.Integration$, 64 | 0]
];
exports.MethodResponse$ = [3, n0, _MR,
    0,
    [_sC, _rP, _rMes],
    [0, 128 | 2, 128 | 0]
];
exports.MethodSetting$ = [3, n0, _MS,
    0,
    [_mE, _lL, _dTE, _tBL, _tRL, _cE, _cTIS, _cDE, _rAFCC, _uCCHS],
    [2, 0, 2, 1, 1, 2, 1, 2, 2, 0]
];
exports.MethodSnapshot$ = [3, n0, _MSe,
    0,
    [_aTu, _aKR],
    [0, 2]
];
exports.Model$ = [3, n0, _Mo,
    0,
    [_i, _n, _d, _sc, _cT],
    [0, 0, 0, 0, 0]
];
exports.Models$ = [3, n0, _Mod,
    0,
    [_it, _p],
    [[() => ListOfModel, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.MutualTlsAuthentication$ = [3, n0, _MTA,
    0,
    [_tU, _tV, _tW],
    [0, 0, 64 | 0]
];
exports.MutualTlsAuthenticationInput$ = [3, n0, _MTAI,
    0,
    [_tU, _tV],
    [0, 0]
];
exports.PatchOperation$ = [3, n0, _PO,
    0,
    [_o, _pa, _v, _fr],
    [0, 0, 0, 0]
];
exports.PutGatewayResponseRequest$ = [3, n0, _PGRR,
    0,
    [_rAI, _rT, _sC, _rP, _rTe],
    [[0, 1], [0, 1], 0, 128 | 0, 128 | 0], 2
];
exports.PutIntegrationRequest$ = [3, n0, _PIR,
    0,
    [_rAI, _rI, _hM, _ty, _iHM, _u, _cTo, _cIo, _cr, _rPe, _rTeq, _pB, _cNa, _cKP, _cH, _tIM, _tC, _rTM, _iT],
    [[0, 1], [0, 1], [0, { [_hL]: 1, [_jN]: _rHM }], 0, [0, { [_jN]: _hM }], 0, 0, 0, 0, 128 | 0, 128 | 0, 0, 0, 64 | 0, 0, 1, () => exports.TlsConfig$, 0, 0], 4
];
exports.PutIntegrationResponseRequest$ = [3, n0, _PIRR,
    0,
    [_rAI, _rI, _hM, _sC, _sPe, _rP, _rTe, _cH],
    [[0, 1], [0, 1], [0, 1], [0, 1], 0, 128 | 0, 128 | 0, 0], 4
];
exports.PutMethodRequest$ = [3, n0, _PMR,
    0,
    [_rAI, _rI, _hM, _aTu, _aIu, _aKR, _oN, _rPe, _rMe, _rVI, _aSu],
    [[0, 1], [0, 1], [0, 1], 0, 0, 2, 0, 128 | 2, 128 | 0, 0, 64 | 0], 4
];
exports.PutMethodResponseRequest$ = [3, n0, _PMRR,
    0,
    [_rAI, _rI, _hM, _sC, _rP, _rMes],
    [[0, 1], [0, 1], [0, 1], [0, 1], 128 | 2, 128 | 0], 4
];
exports.PutRestApiRequest$ = [3, n0, _PRAR,
    0,
    [_rAI, _b, _mo, _fOW, _par],
    [[0, 1], [21, 16], [0, { [_hQ]: _mo }], [2, { [_hQ]: _fa }], [128 | 0, 64]], 2
];
exports.QuotaSettings$ = [3, n0, _QS,
    0,
    [_li, _of, _pe],
    [1, 1, 0]
];
exports.RejectDomainNameAccessAssociationRequest$ = [3, n0, _RDNAAR,
    0,
    [_dNAAA, _dNA],
    [[0, { [_hQ]: _dNAAA }], [0, { [_hQ]: _dNA }]], 2
];
exports.RequestValidator$ = [3, n0, _RV,
    0,
    [_i, _n, _vRB, _vRP],
    [0, 0, 2, 2]
];
exports.RequestValidators$ = [3, n0, _RVe,
    0,
    [_it, _p],
    [[() => ListOfRequestValidator, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.Resource$ = [3, n0, _R,
    0,
    [_i, _pI, _pP, _pa, _rMeso],
    [0, 0, 0, 0, () => MapOfMethod]
];
exports.Resources$ = [3, n0, _Re,
    0,
    [_it, _p],
    [[() => ListOfResource, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.RestApi$ = [3, n0, _RAe,
    0,
    [_i, _n, _d, _cD, _ve, _w, _bMT, _mCS, _aKS, _eC, _po, _t, _dEAE, _rRI, _sP, _eAM, _aSpi, _aSM],
    [0, 0, 0, 4, 0, 64 | 0, 64 | 0, 1, 0, () => exports.EndpointConfiguration$, 0, 128 | 0, 2, 0, 0, 0, 0, 0]
];
exports.RestApis$ = [3, n0, _RAes,
    0,
    [_it, _p],
    [[() => ListOfRestApi, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.SdkConfigurationProperty$ = [3, n0, _SCP,
    0,
    [_n, _fN, _d, _r, _dVe],
    [0, 0, 0, 2, 0]
];
exports.SdkResponse$ = [3, n0, _SR,
    0,
    [_cT, _cDo, _b],
    [[0, { [_hH]: _CT }], [0, { [_hH]: _CD }], [21, 16]]
];
exports.SdkType$ = [3, n0, _ST,
    0,
    [_i, _fN, _d, _cP],
    [0, 0, 0, () => ListOfSdkConfigurationProperty]
];
exports.SdkTypes$ = [3, n0, _STd,
    0,
    [_it],
    [[() => ListOfSdkType, { [_jN]: _ite }]]
];
exports.Stage$ = [3, n0, _S,
    0,
    [_dI, _cCI, _sN, _d, _cCE, _cCS, _cCSa, _mS, _va, _dV, _aLS, _cS, _tE, _wAA, _t, _cD, _lUD],
    [0, 0, 0, 0, 2, 0, 0, () => MapOfMethodSettings, 128 | 0, 0, () => exports.AccessLogSettings$, () => exports.CanarySettings$, 2, 0, 128 | 0, 4, 4]
];
exports.StageKey$ = [3, n0, _SK,
    0,
    [_rAI, _sN],
    [0, 0]
];
exports.Stages$ = [3, n0, _St,
    0,
    [_ite],
    [() => ListOfStage]
];
exports.TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rA, _t],
    [[0, 1], 128 | 0], 2
];
exports.Tags$ = [3, n0, _T,
    0,
    [_t],
    [128 | 0]
];
exports.Template$ = [3, n0, _Te,
    0,
    [_v],
    [0]
];
exports.TestInvokeAuthorizerRequest$ = [3, n0, _TIAR,
    0,
    [_rAI, _aIu, _h, _mVH, _pWQS, _b, _sV, _aCd],
    [[0, 1], [0, 1], 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 0, 128 | 0, 128 | 0], 2
];
exports.TestInvokeAuthorizerResponse$ = [3, n0, _TIARe,
    0,
    [_cSl, _lo, _la, _pIr, _po, _au, _cl],
    [1, 0, 1, 0, 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 128 | 0]
];
exports.TestInvokeMethodRequest$ = [3, n0, _TIMR,
    0,
    [_rAI, _rI, _hM, _pWQS, _b, _h, _mVH, _cCI, _sV],
    [[0, 1], [0, 1], [0, 1], 0, 0, 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 128 | 0], 3
];
exports.TestInvokeMethodResponse$ = [3, n0, _TIMRe,
    0,
    [_sta, _b, _h, _mVH, _lo, _la],
    [1, 0, 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 1]
];
exports.ThrottleSettings$ = [3, n0, _TS,
    0,
    [_bL, _rL],
    [1, 1]
];
exports.TlsConfig$ = [3, n0, _TC,
    0,
    [_iSV],
    [2]
];
exports.UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rA, _tK],
    [[0, 1], [64 | 0, { [_hQ]: _tK }]], 2
];
exports.UpdateAccountRequest$ = [3, n0, _UAR,
    0,
    [_pO],
    [() => ListOfPatchOperation]
];
exports.UpdateApiKeyRequest$ = [3, n0, _UAKR,
    0,
    [_aK, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
exports.UpdateAuthorizerRequest$ = [3, n0, _UARp,
    0,
    [_rAI, _aIu, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateBasePathMappingRequest$ = [3, n0, _UBPMR,
    0,
    [_dN, _bP, _dNI, _pO],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }], () => ListOfPatchOperation], 2
];
exports.UpdateClientCertificateRequest$ = [3, n0, _UCCR,
    0,
    [_cCI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
exports.UpdateDeploymentRequest$ = [3, n0, _UDR,
    0,
    [_rAI, _dI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateDocumentationPartRequest$ = [3, n0, _UDPR,
    0,
    [_rAI, _dPI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateDocumentationVersionRequest$ = [3, n0, _UDVR,
    0,
    [_rAI, _dV, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateDomainNameRequest$ = [3, n0, _UDNR,
    0,
    [_dN, _dNI, _pO],
    [[0, 1], [0, { [_hQ]: _dNI }], () => ListOfPatchOperation], 1
];
exports.UpdateGatewayResponseRequest$ = [3, n0, _UGRR,
    0,
    [_rAI, _rT, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateIntegrationRequest$ = [3, n0, _UIR,
    0,
    [_rAI, _rI, _hM, _pO],
    [[0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 3
];
exports.UpdateIntegrationResponseRequest$ = [3, n0, _UIRR,
    0,
    [_rAI, _rI, _hM, _sC, _pO],
    [[0, 1], [0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 4
];
exports.UpdateMethodRequest$ = [3, n0, _UMR,
    0,
    [_rAI, _rI, _hM, _pO],
    [[0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 3
];
exports.UpdateMethodResponseRequest$ = [3, n0, _UMRR,
    0,
    [_rAI, _rI, _hM, _sC, _pO],
    [[0, 1], [0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 4
];
exports.UpdateModelRequest$ = [3, n0, _UMRp,
    0,
    [_rAI, _mN, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateRequestValidatorRequest$ = [3, n0, _URVR,
    0,
    [_rAI, _rVI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateResourceRequest$ = [3, n0, _URRp,
    0,
    [_rAI, _rI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateRestApiRequest$ = [3, n0, _URAR,
    0,
    [_rAI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
exports.UpdateStageRequest$ = [3, n0, _USR,
    0,
    [_rAI, _sN, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateUsagePlanRequest$ = [3, n0, _UUPR,
    0,
    [_uPI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
exports.UpdateUsageRequest$ = [3, n0, _UUR,
    0,
    [_uPI, _kI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
exports.UpdateVpcLinkRequest$ = [3, n0, _UVLR,
    0,
    [_vLI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
exports.Usage$ = [3, n0, _U,
    0,
    [_uPI, _sDt, _eDn, _it, _p],
    [0, 0, 0, [[2, n0, _MOKU, 0, 0, [1, n0, _LOU, 0, 64 | 1]], { [_jN]: _val }], [0, { [_hQ]: _p }]]
];
exports.UsagePlan$ = [3, n0, _UP,
    0,
    [_i, _n, _d, _aS, _th, _q, _pC, _t],
    [0, 0, 0, () => ListOfApiStage, () => exports.ThrottleSettings$, () => exports.QuotaSettings$, 0, 128 | 0]
];
exports.UsagePlanKey$ = [3, n0, _UPK,
    0,
    [_i, _ty, _v, _n],
    [0, 0, 0, 0]
];
exports.UsagePlanKeys$ = [3, n0, _UPKs,
    0,
    [_it, _p],
    [[() => ListOfUsagePlanKey, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.UsagePlans$ = [3, n0, _UPs,
    0,
    [_it, _p],
    [[() => ListOfUsagePlan, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
exports.VpcLink$ = [3, n0, _VL,
    0,
    [_i, _n, _d, _tA, _sta, _sM, _t],
    [0, 0, 0, 64 | 0, 0, 0, 128 | 0]
];
exports.VpcLinks$ = [3, n0, _VLp,
    0,
    [_it, _p],
    [[() => ListOfVpcLink, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var __Unit = "unit";
var ListOfApiKey = [1, n0, _LOAK,
    0, () => exports.ApiKey$
];
var ListOfApiStage = [1, n0, _LOAS,
    0, () => exports.ApiStage$
];
var ListOfARNs = 64 | 0;
var ListOfAuthorizer = [1, n0, _LOA,
    0, () => exports.Authorizer$
];
var ListOfBasePathMapping = [1, n0, _LOBPM,
    0, () => exports.BasePathMapping$
];
var ListOfClientCertificate = [1, n0, _LOCC,
    0, () => exports.ClientCertificate$
];
var ListOfDeployment = [1, n0, _LOD,
    0, () => exports.Deployment$
];
var ListOfDocumentationPart = [1, n0, _LODP,
    0, () => exports.DocumentationPart$
];
var ListOfDocumentationVersion = [1, n0, _LODV,
    0, () => exports.DocumentationVersion$
];
var ListOfDomainName = [1, n0, _LODN,
    0, () => exports.DomainName$
];
var ListOfDomainNameAccessAssociation = [1, n0, _LODNAA,
    0, () => exports.DomainNameAccessAssociation$
];
var ListOfEndpointType = 64 | 0;
var ListOfGatewayResponse = [1, n0, _LOGR,
    0, () => exports.GatewayResponse$
];
var ListOfLong = 64 | 1;
var ListOfModel = [1, n0, _LOM,
    0, () => exports.Model$
];
var ListOfPatchOperation = [1, n0, _LOPO,
    0, () => exports.PatchOperation$
];
var ListOfRequestValidator = [1, n0, _LORV,
    0, () => exports.RequestValidator$
];
var ListOfResource = [1, n0, _LOR,
    0, () => exports.Resource$
];
var ListOfRestApi = [1, n0, _LORA,
    0, () => exports.RestApi$
];
var ListOfSdkConfigurationProperty = [1, n0, _LOSCP,
    0, () => exports.SdkConfigurationProperty$
];
var ListOfSdkType = [1, n0, _LOST,
    0, () => exports.SdkType$
];
var ListOfStage = [1, n0, _LOS,
    0, () => exports.Stage$
];
var ListOfStageKeys = [1, n0, _LOSK,
    0, () => exports.StageKey$
];
var ListOfString = 64 | 0;
var ListOfUsage = [1, n0, _LOU,
    0, 64 | 1
];
var ListOfUsagePlan = [1, n0, _LOUP,
    0, () => exports.UsagePlan$
];
var ListOfUsagePlanKey = [1, n0, _LOUPK,
    0, () => exports.UsagePlanKey$
];
var ListOfVpcLink = [1, n0, _LOVL,
    0, () => exports.VpcLink$
];
var MapOfApiStageThrottleSettings = [2, n0, _MOASTS,
    0, 0, () => exports.ThrottleSettings$
];
var MapOfIntegrationResponse = [2, n0, _MOIR,
    0, 0, () => exports.IntegrationResponse$
];
var MapOfKeyUsages = [2, n0, _MOKU,
    0, 0, [1, n0, _LOU, 0, 64 | 1]
];
var MapOfMethod = [2, n0, _MOM,
    0, 0, () => exports.Method$
];
var MapOfMethodResponse = [2, n0, _MOMR,
    0, 0, () => exports.MethodResponse$
];
var MapOfMethodSettings = [2, n0, _MOMS,
    0, 0, () => exports.MethodSetting$
];
var MapOfMethodSnapshot = [2, n0, _MOMSa,
    0, 0, () => exports.MethodSnapshot$
];
var MapOfStringToBoolean = 128 | 2;
var MapOfStringToList = [2, n0, _MOSTL,
    0, 0, 64 | 0
];
var MapOfStringToString = 128 | 0;
var PathToMapOfMethodSnapshot = [2, n0, _PTMOMS,
    0, 0, () => MapOfMethodSnapshot
];
exports.CreateApiKey$ = [9, n0, _CAK,
    { [_ht]: ["POST", "/apikeys", 201] }, () => exports.CreateApiKeyRequest$, () => exports.ApiKey$
];
exports.CreateAuthorizer$ = [9, n0, _CA,
    { [_ht]: ["POST", "/restapis/{restApiId}/authorizers", 201] }, () => exports.CreateAuthorizerRequest$, () => exports.Authorizer$
];
exports.CreateBasePathMapping$ = [9, n0, _CBPM,
    { [_ht]: ["POST", "/domainnames/{domainName}/basepathmappings", 201] }, () => exports.CreateBasePathMappingRequest$, () => exports.BasePathMapping$
];
exports.CreateDeployment$ = [9, n0, _CDr,
    { [_ht]: ["POST", "/restapis/{restApiId}/deployments", 201] }, () => exports.CreateDeploymentRequest$, () => exports.Deployment$
];
exports.CreateDocumentationPart$ = [9, n0, _CDP,
    { [_ht]: ["POST", "/restapis/{restApiId}/documentation/parts", 201] }, () => exports.CreateDocumentationPartRequest$, () => exports.DocumentationPart$
];
exports.CreateDocumentationVersion$ = [9, n0, _CDV,
    { [_ht]: ["POST", "/restapis/{restApiId}/documentation/versions", 201] }, () => exports.CreateDocumentationVersionRequest$, () => exports.DocumentationVersion$
];
exports.CreateDomainName$ = [9, n0, _CDN,
    { [_ht]: ["POST", "/domainnames", 201] }, () => exports.CreateDomainNameRequest$, () => exports.DomainName$
];
exports.CreateDomainNameAccessAssociation$ = [9, n0, _CDNAA,
    { [_ht]: ["POST", "/domainnameaccessassociations", 201] }, () => exports.CreateDomainNameAccessAssociationRequest$, () => exports.DomainNameAccessAssociation$
];
exports.CreateModel$ = [9, n0, _CM,
    { [_ht]: ["POST", "/restapis/{restApiId}/models", 201] }, () => exports.CreateModelRequest$, () => exports.Model$
];
exports.CreateRequestValidator$ = [9, n0, _CRV,
    { [_ht]: ["POST", "/restapis/{restApiId}/requestvalidators", 201] }, () => exports.CreateRequestValidatorRequest$, () => exports.RequestValidator$
];
exports.CreateResource$ = [9, n0, _CR,
    { [_ht]: ["POST", "/restapis/{restApiId}/resources/{parentId}", 201] }, () => exports.CreateResourceRequest$, () => exports.Resource$
];
exports.CreateRestApi$ = [9, n0, _CRA,
    { [_ht]: ["POST", "/restapis", 201] }, () => exports.CreateRestApiRequest$, () => exports.RestApi$
];
exports.CreateStage$ = [9, n0, _CSr,
    { [_ht]: ["POST", "/restapis/{restApiId}/stages", 201] }, () => exports.CreateStageRequest$, () => exports.Stage$
];
exports.CreateUsagePlan$ = [9, n0, _CUP,
    { [_ht]: ["POST", "/usageplans", 201] }, () => exports.CreateUsagePlanRequest$, () => exports.UsagePlan$
];
exports.CreateUsagePlanKey$ = [9, n0, _CUPK,
    { [_ht]: ["POST", "/usageplans/{usagePlanId}/keys", 201] }, () => exports.CreateUsagePlanKeyRequest$, () => exports.UsagePlanKey$
];
exports.CreateVpcLink$ = [9, n0, _CVL,
    { [_ht]: ["POST", "/vpclinks", 202] }, () => exports.CreateVpcLinkRequest$, () => exports.VpcLink$
];
exports.DeleteApiKey$ = [9, n0, _DAK,
    { [_ht]: ["DELETE", "/apikeys/{apiKey}", 202] }, () => exports.DeleteApiKeyRequest$, () => __Unit
];
exports.DeleteAuthorizer$ = [9, n0, _DA,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/authorizers/{authorizerId}", 202] }, () => exports.DeleteAuthorizerRequest$, () => __Unit
];
exports.DeleteBasePathMapping$ = [9, n0, _DBPM,
    { [_ht]: ["DELETE", "/domainnames/{domainName}/basepathmappings/{basePath}", 202] }, () => exports.DeleteBasePathMappingRequest$, () => __Unit
];
exports.DeleteClientCertificate$ = [9, n0, _DCC,
    { [_ht]: ["DELETE", "/clientcertificates/{clientCertificateId}", 202] }, () => exports.DeleteClientCertificateRequest$, () => __Unit
];
exports.DeleteDeployment$ = [9, n0, _DD,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/deployments/{deploymentId}", 202] }, () => exports.DeleteDeploymentRequest$, () => __Unit
];
exports.DeleteDocumentationPart$ = [9, n0, _DDP,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 202] }, () => exports.DeleteDocumentationPartRequest$, () => __Unit
];
exports.DeleteDocumentationVersion$ = [9, n0, _DDV,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 202] }, () => exports.DeleteDocumentationVersionRequest$, () => __Unit
];
exports.DeleteDomainName$ = [9, n0, _DDN,
    { [_ht]: ["DELETE", "/domainnames/{domainName}", 202] }, () => exports.DeleteDomainNameRequest$, () => __Unit
];
exports.DeleteDomainNameAccessAssociation$ = [9, n0, _DDNAA,
    { [_ht]: ["DELETE", "/domainnameaccessassociations/{domainNameAccessAssociationArn}", 202] }, () => exports.DeleteDomainNameAccessAssociationRequest$, () => __Unit
];
exports.DeleteGatewayResponse$ = [9, n0, _DGR,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/gatewayresponses/{responseType}", 202] }, () => exports.DeleteGatewayResponseRequest$, () => __Unit
];
exports.DeleteIntegration$ = [9, n0, _DI,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 204] }, () => exports.DeleteIntegrationRequest$, () => __Unit
];
exports.DeleteIntegrationResponse$ = [9, n0, _DIRe,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 204] }, () => exports.DeleteIntegrationResponseRequest$, () => __Unit
];
exports.DeleteMethod$ = [9, n0, _DM,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 204] }, () => exports.DeleteMethodRequest$, () => __Unit
];
exports.DeleteMethodResponse$ = [9, n0, _DMRel,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 204] }, () => exports.DeleteMethodResponseRequest$, () => __Unit
];
exports.DeleteModel$ = [9, n0, _DMe,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/models/{modelName}", 202] }, () => exports.DeleteModelRequest$, () => __Unit
];
exports.DeleteRequestValidator$ = [9, n0, _DRV,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 202] }, () => exports.DeleteRequestValidatorRequest$, () => __Unit
];
exports.DeleteResource$ = [9, n0, _DR,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}", 202] }, () => exports.DeleteResourceRequest$, () => __Unit
];
exports.DeleteRestApi$ = [9, n0, _DRA,
    { [_ht]: ["DELETE", "/restapis/{restApiId}", 202] }, () => exports.DeleteRestApiRequest$, () => __Unit
];
exports.DeleteStage$ = [9, n0, _DS,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}", 202] }, () => exports.DeleteStageRequest$, () => __Unit
];
exports.DeleteUsagePlan$ = [9, n0, _DUP,
    { [_ht]: ["DELETE", "/usageplans/{usagePlanId}", 202] }, () => exports.DeleteUsagePlanRequest$, () => __Unit
];
exports.DeleteUsagePlanKey$ = [9, n0, _DUPK,
    { [_ht]: ["DELETE", "/usageplans/{usagePlanId}/keys/{keyId}", 202] }, () => exports.DeleteUsagePlanKeyRequest$, () => __Unit
];
exports.DeleteVpcLink$ = [9, n0, _DVL,
    { [_ht]: ["DELETE", "/vpclinks/{vpcLinkId}", 202] }, () => exports.DeleteVpcLinkRequest$, () => __Unit
];
exports.FlushStageAuthorizersCache$ = [9, n0, _FSAC,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}/cache/authorizers", 202] }, () => exports.FlushStageAuthorizersCacheRequest$, () => __Unit
];
exports.FlushStageCache$ = [9, n0, _FSC,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}/cache/data", 202] }, () => exports.FlushStageCacheRequest$, () => __Unit
];
exports.GenerateClientCertificate$ = [9, n0, _GCC,
    { [_ht]: ["POST", "/clientcertificates", 201] }, () => exports.GenerateClientCertificateRequest$, () => exports.ClientCertificate$
];
exports.GetAccount$ = [9, n0, _GA,
    { [_ht]: ["GET", "/account", 200] }, () => exports.GetAccountRequest$, () => exports.Account$
];
exports.GetApiKey$ = [9, n0, _GAK,
    { [_ht]: ["GET", "/apikeys/{apiKey}", 200] }, () => exports.GetApiKeyRequest$, () => exports.ApiKey$
];
exports.GetApiKeys$ = [9, n0, _GAKe,
    { [_ht]: ["GET", "/apikeys", 200] }, () => exports.GetApiKeysRequest$, () => exports.ApiKeys$
];
exports.GetAuthorizer$ = [9, n0, _GAe,
    { [_ht]: ["GET", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => exports.GetAuthorizerRequest$, () => exports.Authorizer$
];
exports.GetAuthorizers$ = [9, n0, _GAet,
    { [_ht]: ["GET", "/restapis/{restApiId}/authorizers", 200] }, () => exports.GetAuthorizersRequest$, () => exports.Authorizers$
];
exports.GetBasePathMapping$ = [9, n0, _GBPM,
    { [_ht]: ["GET", "/domainnames/{domainName}/basepathmappings/{basePath}", 200] }, () => exports.GetBasePathMappingRequest$, () => exports.BasePathMapping$
];
exports.GetBasePathMappings$ = [9, n0, _GBPMe,
    { [_ht]: ["GET", "/domainnames/{domainName}/basepathmappings", 200] }, () => exports.GetBasePathMappingsRequest$, () => exports.BasePathMappings$
];
exports.GetClientCertificate$ = [9, n0, _GCCe,
    { [_ht]: ["GET", "/clientcertificates/{clientCertificateId}", 200] }, () => exports.GetClientCertificateRequest$, () => exports.ClientCertificate$
];
exports.GetClientCertificates$ = [9, n0, _GCCet,
    { [_ht]: ["GET", "/clientcertificates", 200] }, () => exports.GetClientCertificatesRequest$, () => exports.ClientCertificates$
];
exports.GetDeployment$ = [9, n0, _GD,
    { [_ht]: ["GET", "/restapis/{restApiId}/deployments/{deploymentId}", 200] }, () => exports.GetDeploymentRequest$, () => exports.Deployment$
];
exports.GetDeployments$ = [9, n0, _GDe,
    { [_ht]: ["GET", "/restapis/{restApiId}/deployments", 200] }, () => exports.GetDeploymentsRequest$, () => exports.Deployments$
];
exports.GetDocumentationPart$ = [9, n0, _GDP,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 200] }, () => exports.GetDocumentationPartRequest$, () => exports.DocumentationPart$
];
exports.GetDocumentationParts$ = [9, n0, _GDPe,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/parts", 200] }, () => exports.GetDocumentationPartsRequest$, () => exports.DocumentationParts$
];
exports.GetDocumentationVersion$ = [9, n0, _GDV,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 200] }, () => exports.GetDocumentationVersionRequest$, () => exports.DocumentationVersion$
];
exports.GetDocumentationVersions$ = [9, n0, _GDVe,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/versions", 200] }, () => exports.GetDocumentationVersionsRequest$, () => exports.DocumentationVersions$
];
exports.GetDomainName$ = [9, n0, _GDN,
    { [_ht]: ["GET", "/domainnames/{domainName}", 200] }, () => exports.GetDomainNameRequest$, () => exports.DomainName$
];
exports.GetDomainNameAccessAssociations$ = [9, n0, _GDNAA,
    { [_ht]: ["GET", "/domainnameaccessassociations", 200] }, () => exports.GetDomainNameAccessAssociationsRequest$, () => exports.DomainNameAccessAssociations$
];
exports.GetDomainNames$ = [9, n0, _GDNe,
    { [_ht]: ["GET", "/domainnames", 200] }, () => exports.GetDomainNamesRequest$, () => exports.DomainNames$
];
exports.GetExport$ = [9, n0, _GE,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}/exports/{exportType}", 200] }, () => exports.GetExportRequest$, () => exports.ExportResponse$
];
exports.GetGatewayResponse$ = [9, n0, _GGR,
    { [_ht]: ["GET", "/restapis/{restApiId}/gatewayresponses/{responseType}", 200] }, () => exports.GetGatewayResponseRequest$, () => exports.GatewayResponse$
];
exports.GetGatewayResponses$ = [9, n0, _GGRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/gatewayresponses", 200] }, () => exports.GetGatewayResponsesRequest$, () => exports.GatewayResponses$
];
exports.GetIntegration$ = [9, n0, _GI,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 200] }, () => exports.GetIntegrationRequest$, () => exports.Integration$
];
exports.GetIntegrationResponse$ = [9, n0, _GIRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 200] }, () => exports.GetIntegrationResponseRequest$, () => exports.IntegrationResponse$
];
exports.GetMethod$ = [9, n0, _GM,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => exports.GetMethodRequest$, () => exports.Method$
];
exports.GetMethodResponse$ = [9, n0, _GMRete,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 200] }, () => exports.GetMethodResponseRequest$, () => exports.MethodResponse$
];
exports.GetModel$ = [9, n0, _GMe,
    { [_ht]: ["GET", "/restapis/{restApiId}/models/{modelName}", 200] }, () => exports.GetModelRequest$, () => exports.Model$
];
exports.GetModels$ = [9, n0, _GMet,
    { [_ht]: ["GET", "/restapis/{restApiId}/models", 200] }, () => exports.GetModelsRequest$, () => exports.Models$
];
exports.GetModelTemplate$ = [9, n0, _GMT,
    { [_ht]: ["GET", "/restapis/{restApiId}/models/{modelName}/default_template", 200] }, () => exports.GetModelTemplateRequest$, () => exports.Template$
];
exports.GetRequestValidator$ = [9, n0, _GRV,
    { [_ht]: ["GET", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 200] }, () => exports.GetRequestValidatorRequest$, () => exports.RequestValidator$
];
exports.GetRequestValidators$ = [9, n0, _GRVe,
    { [_ht]: ["GET", "/restapis/{restApiId}/requestvalidators", 200] }, () => exports.GetRequestValidatorsRequest$, () => exports.RequestValidators$
];
exports.GetResource$ = [9, n0, _GRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}", 200] }, () => exports.GetResourceRequest$, () => exports.Resource$
];
exports.GetResources$ = [9, n0, _GRet,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources", 200] }, () => exports.GetResourcesRequest$, () => exports.Resources$
];
exports.GetRestApi$ = [9, n0, _GRA,
    { [_ht]: ["GET", "/restapis/{restApiId}", 200] }, () => exports.GetRestApiRequest$, () => exports.RestApi$
];
exports.GetRestApis$ = [9, n0, _GRAe,
    { [_ht]: ["GET", "/restapis", 200] }, () => exports.GetRestApisRequest$, () => exports.RestApis$
];
exports.GetSdk$ = [9, n0, _GS,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}/sdks/{sdkType}", 200] }, () => exports.GetSdkRequest$, () => exports.SdkResponse$
];
exports.GetSdkType$ = [9, n0, _GST,
    { [_ht]: ["GET", "/sdktypes/{id}", 200] }, () => exports.GetSdkTypeRequest$, () => exports.SdkType$
];
exports.GetSdkTypes$ = [9, n0, _GSTe,
    { [_ht]: ["GET", "/sdktypes", 200] }, () => exports.GetSdkTypesRequest$, () => exports.SdkTypes$
];
exports.GetStage$ = [9, n0, _GSe,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}", 200] }, () => exports.GetStageRequest$, () => exports.Stage$
];
exports.GetStages$ = [9, n0, _GSet,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages", 200] }, () => exports.GetStagesRequest$, () => exports.Stages$
];
exports.GetTags$ = [9, n0, _GT,
    { [_ht]: ["GET", "/tags/{resourceArn}", 200] }, () => exports.GetTagsRequest$, () => exports.Tags$
];
exports.GetUsage$ = [9, n0, _GU,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/usage", 200] }, () => exports.GetUsageRequest$, () => exports.Usage$
];
exports.GetUsagePlan$ = [9, n0, _GUP,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}", 200] }, () => exports.GetUsagePlanRequest$, () => exports.UsagePlan$
];
exports.GetUsagePlanKey$ = [9, n0, _GUPK,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/keys/{keyId}", 200] }, () => exports.GetUsagePlanKeyRequest$, () => exports.UsagePlanKey$
];
exports.GetUsagePlanKeys$ = [9, n0, _GUPKe,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/keys", 200] }, () => exports.GetUsagePlanKeysRequest$, () => exports.UsagePlanKeys$
];
exports.GetUsagePlans$ = [9, n0, _GUPe,
    { [_ht]: ["GET", "/usageplans", 200] }, () => exports.GetUsagePlansRequest$, () => exports.UsagePlans$
];
exports.GetVpcLink$ = [9, n0, _GVL,
    { [_ht]: ["GET", "/vpclinks/{vpcLinkId}", 200] }, () => exports.GetVpcLinkRequest$, () => exports.VpcLink$
];
exports.GetVpcLinks$ = [9, n0, _GVLe,
    { [_ht]: ["GET", "/vpclinks", 200] }, () => exports.GetVpcLinksRequest$, () => exports.VpcLinks$
];
exports.ImportApiKeys$ = [9, n0, _IAK,
    { [_ht]: ["POST", "/apikeys?mode=import", 201] }, () => exports.ImportApiKeysRequest$, () => exports.ApiKeyIds$
];
exports.ImportDocumentationParts$ = [9, n0, _IDP,
    { [_ht]: ["PUT", "/restapis/{restApiId}/documentation/parts", 200] }, () => exports.ImportDocumentationPartsRequest$, () => exports.DocumentationPartIds$
];
exports.ImportRestApi$ = [9, n0, _IRA,
    { [_ht]: ["POST", "/restapis?mode=import", 201] }, () => exports.ImportRestApiRequest$, () => exports.RestApi$
];
exports.PutGatewayResponse$ = [9, n0, _PGR,
    { [_ht]: ["PUT", "/restapis/{restApiId}/gatewayresponses/{responseType}", 201] }, () => exports.PutGatewayResponseRequest$, () => exports.GatewayResponse$
];
exports.PutIntegration$ = [9, n0, _PI,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 201] }, () => exports.PutIntegrationRequest$, () => exports.Integration$
];
exports.PutIntegrationResponse$ = [9, n0, _PIRu,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 201] }, () => exports.PutIntegrationResponseRequest$, () => exports.IntegrationResponse$
];
exports.PutMethod$ = [9, n0, _PM,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 201] }, () => exports.PutMethodRequest$, () => exports.Method$
];
exports.PutMethodResponse$ = [9, n0, _PMRu,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 201] }, () => exports.PutMethodResponseRequest$, () => exports.MethodResponse$
];
exports.PutRestApi$ = [9, n0, _PRA,
    { [_ht]: ["PUT", "/restapis/{restApiId}", 200] }, () => exports.PutRestApiRequest$, () => exports.RestApi$
];
exports.RejectDomainNameAccessAssociation$ = [9, n0, _RDNAA,
    { [_ht]: ["POST", "/rejectdomainnameaccessassociations", 202] }, () => exports.RejectDomainNameAccessAssociationRequest$, () => __Unit
];
exports.TagResource$ = [9, n0, _TR,
    { [_ht]: ["PUT", "/tags/{resourceArn}", 204] }, () => exports.TagResourceRequest$, () => __Unit
];
exports.TestInvokeAuthorizer$ = [9, n0, _TIA,
    { [_ht]: ["POST", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => exports.TestInvokeAuthorizerRequest$, () => exports.TestInvokeAuthorizerResponse$
];
exports.TestInvokeMethod$ = [9, n0, _TIM,
    { [_ht]: ["POST", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => exports.TestInvokeMethodRequest$, () => exports.TestInvokeMethodResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    { [_ht]: ["DELETE", "/tags/{resourceArn}", 204] }, () => exports.UntagResourceRequest$, () => __Unit
];
exports.UpdateAccount$ = [9, n0, _UA,
    { [_ht]: ["PATCH", "/account", 200] }, () => exports.UpdateAccountRequest$, () => exports.Account$
];
exports.UpdateApiKey$ = [9, n0, _UAK,
    { [_ht]: ["PATCH", "/apikeys/{apiKey}", 200] }, () => exports.UpdateApiKeyRequest$, () => exports.ApiKey$
];
exports.UpdateAuthorizer$ = [9, n0, _UAp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => exports.UpdateAuthorizerRequest$, () => exports.Authorizer$
];
exports.UpdateBasePathMapping$ = [9, n0, _UBPM,
    { [_ht]: ["PATCH", "/domainnames/{domainName}/basepathmappings/{basePath}", 200] }, () => exports.UpdateBasePathMappingRequest$, () => exports.BasePathMapping$
];
exports.UpdateClientCertificate$ = [9, n0, _UCC,
    { [_ht]: ["PATCH", "/clientcertificates/{clientCertificateId}", 200] }, () => exports.UpdateClientCertificateRequest$, () => exports.ClientCertificate$
];
exports.UpdateDeployment$ = [9, n0, _UD,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/deployments/{deploymentId}", 200] }, () => exports.UpdateDeploymentRequest$, () => exports.Deployment$
];
exports.UpdateDocumentationPart$ = [9, n0, _UDP,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 200] }, () => exports.UpdateDocumentationPartRequest$, () => exports.DocumentationPart$
];
exports.UpdateDocumentationVersion$ = [9, n0, _UDV,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 200] }, () => exports.UpdateDocumentationVersionRequest$, () => exports.DocumentationVersion$
];
exports.UpdateDomainName$ = [9, n0, _UDN,
    { [_ht]: ["PATCH", "/domainnames/{domainName}", 200] }, () => exports.UpdateDomainNameRequest$, () => exports.DomainName$
];
exports.UpdateGatewayResponse$ = [9, n0, _UGR,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/gatewayresponses/{responseType}", 200] }, () => exports.UpdateGatewayResponseRequest$, () => exports.GatewayResponse$
];
exports.UpdateIntegration$ = [9, n0, _UI,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 200] }, () => exports.UpdateIntegrationRequest$, () => exports.Integration$
];
exports.UpdateIntegrationResponse$ = [9, n0, _UIRp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 200] }, () => exports.UpdateIntegrationResponseRequest$, () => exports.IntegrationResponse$
];
exports.UpdateMethod$ = [9, n0, _UM,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => exports.UpdateMethodRequest$, () => exports.Method$
];
exports.UpdateMethodResponse$ = [9, n0, _UMRpd,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 201] }, () => exports.UpdateMethodResponseRequest$, () => exports.MethodResponse$
];
exports.UpdateModel$ = [9, n0, _UMp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/models/{modelName}", 200] }, () => exports.UpdateModelRequest$, () => exports.Model$
];
exports.UpdateRequestValidator$ = [9, n0, _URV,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 200] }, () => exports.UpdateRequestValidatorRequest$, () => exports.RequestValidator$
];
exports.UpdateResource$ = [9, n0, _URp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}", 200] }, () => exports.UpdateResourceRequest$, () => exports.Resource$
];
exports.UpdateRestApi$ = [9, n0, _URA,
    { [_ht]: ["PATCH", "/restapis/{restApiId}", 200] }, () => exports.UpdateRestApiRequest$, () => exports.RestApi$
];
exports.UpdateStage$ = [9, n0, _US,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/stages/{stageName}", 200] }, () => exports.UpdateStageRequest$, () => exports.Stage$
];
exports.UpdateUsage$ = [9, n0, _UU,
    { [_ht]: ["PATCH", "/usageplans/{usagePlanId}/keys/{keyId}/usage", 200] }, () => exports.UpdateUsageRequest$, () => exports.Usage$
];
exports.UpdateUsagePlan$ = [9, n0, _UUP,
    { [_ht]: ["PATCH", "/usageplans/{usagePlanId}", 200] }, () => exports.UpdateUsagePlanRequest$, () => exports.UsagePlan$
];
exports.UpdateVpcLink$ = [9, n0, _UVL,
    { [_ht]: ["PATCH", "/vpclinks/{vpcLinkId}", 200] }, () => exports.UpdateVpcLinkRequest$, () => exports.VpcLink$
];
