// @aws-sdk/middleware-flexible-checksums/dist-cjs/index.js
'use strict';

var checksums = require('@aws-sdk/checksums');



exports.CONFIG_REQUEST_CHECKSUM_CALCULATION = checksums.CONFIG_REQUEST_CHECKSUM_CALCULATION;
exports.CONFIG_RESPONSE_CHECKSUM_VALIDATION = checksums.CONFIG_RESPONSE_CHECKSUM_VALIDATION;
exports.ChecksumAlgorithm = checksums.ChecksumAlgorithm;
exports.ChecksumLocation = checksums.ChecksumLocation;
exports.DEFAULT_CHECKSUM_ALGORITHM = checksums.DEFAULT_CHECKSUM_ALGORITHM;
exports.DEFAULT_REQUEST_CHECKSUM_CALCULATION = checksums.DEFAULT_REQUEST_CHECKSUM_CALCULATION;
exports.DEFAULT_RESPONSE_CHECKSUM_VALIDATION = checksums.DEFAULT_RESPONSE_CHECKSUM_VALIDATION;
exports.ENV_REQUEST_CHECKSUM_CALCULATION = checksums.ENV_REQUEST_CHECKSUM_CALCULATION;
exports.ENV_RESPONSE_CHECKSUM_VALIDATION = checksums.ENV_RESPONSE_CHECKSUM_VALIDATION;
exports.NODE_REQUEST_CHECKSUM_CALCULATION_CONFIG_OPTIONS = checksums.NODE_REQUEST_CHECKSUM_CALCULATION_CONFIG_OPTIONS;
exports.NODE_RESPONSE_CHECKSUM_VALIDATION_CONFIG_OPTIONS = checksums.NODE_RESPONSE_CHECKSUM_VALIDATION_CONFIG_OPTIONS;
exports.RequestChecksumCalculation = checksums.RequestChecksumCalculation;
exports.ResponseChecksumValidation = checksums.ResponseChecksumValidation;
exports.flexibleChecksumsMiddleware = checksums.flexibleChecksumsMiddleware;
exports.flexibleChecksumsMiddlewareOptions = checksums.flexibleChecksumsMiddlewareOptions;
exports.getFlexibleChecksumsPlugin = checksums.getFlexibleChecksumsPlugin;
exports.resolveFlexibleChecksumsConfig = checksums.resolveFlexibleChecksumsConfig;
