// @aws-sdk/client-cloudformation/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContinueUpdateRollbackOutput$ = exports.ContinueUpdateRollbackInput$ = exports.ChangeSetSummary$ = exports.ChangeSetHookTargetDetails$ = exports.ChangeSetHookResourceTargetDetails$ = exports.ChangeSetHook$ = exports.Change$ = exports.CancelUpdateStackInput$ = exports.BatchDescribeTypeConfigurationsOutput$ = exports.BatchDescribeTypeConfigurationsInput$ = exports.BatchDescribeTypeConfigurationsError$ = exports.AutoDeployment$ = exports.Annotation$ = exports.ActivateTypeOutput$ = exports.ActivateTypeInput$ = exports.ActivateOrganizationsAccessOutput$ = exports.ActivateOrganizationsAccessInput$ = exports.AccountLimit$ = exports.AccountGateResult$ = exports.errorTypeRegistries = exports.TypeNotFoundException$ = exports.TypeConfigurationNotFoundException$ = exports.TokenAlreadyExistsException$ = exports.StaleRequestException$ = exports.StackSetNotFoundException$ = exports.StackSetNotEmptyException$ = exports.StackRefactorNotFoundException$ = exports.StackNotFoundException$ = exports.StackInstanceNotFoundException$ = exports.ResourceScanNotFoundException$ = exports.ResourceScanLimitExceededException$ = exports.ResourceScanInProgressException$ = exports.OperationStatusCheckFailedException$ = exports.OperationNotFoundException$ = exports.OperationInProgressException$ = exports.OperationIdAlreadyExistsException$ = exports.NameAlreadyExistsException$ = exports.LimitExceededException$ = exports.InvalidStateTransitionException$ = exports.InvalidOperationException$ = exports.InvalidChangeSetStatusException$ = exports.InsufficientCapabilitiesException$ = exports.HookResultNotFoundException$ = exports.GeneratedTemplateNotFoundException$ = exports.CreatedButModifiedException$ = exports.ConcurrentResourcesLimitExceededException$ = exports.ChangeSetNotFoundException$ = exports.CFNRegistryException$ = exports.AlreadyExistsException$ = exports.CloudFormationServiceException$ = void 0;
exports.DescribeStackRefactorInput$ = exports.DescribeStackInstanceOutput$ = exports.DescribeStackInstanceInput$ = exports.DescribeStackEventsOutput$ = exports.DescribeStackEventsInput$ = exports.DescribeStackDriftDetectionStatusOutput$ = exports.DescribeStackDriftDetectionStatusInput$ = exports.DescribeResourceScanOutput$ = exports.DescribeResourceScanInput$ = exports.DescribePublisherOutput$ = exports.DescribePublisherInput$ = exports.DescribeOrganizationsAccessOutput$ = exports.DescribeOrganizationsAccessInput$ = exports.DescribeGeneratedTemplateOutput$ = exports.DescribeGeneratedTemplateInput$ = exports.DescribeEventsOutput$ = exports.DescribeEventsInput$ = exports.DescribeChangeSetOutput$ = exports.DescribeChangeSetInput$ = exports.DescribeChangeSetHooksOutput$ = exports.DescribeChangeSetHooksInput$ = exports.DescribeAccountLimitsOutput$ = exports.DescribeAccountLimitsInput$ = exports.DeregisterTypeOutput$ = exports.DeregisterTypeInput$ = exports.DeploymentTargets$ = exports.DeleteStackSetOutput$ = exports.DeleteStackSetInput$ = exports.DeleteStackInstancesOutput$ = exports.DeleteStackInstancesInput$ = exports.DeleteStackInput$ = exports.DeleteGeneratedTemplateInput$ = exports.DeleteChangeSetOutput$ = exports.DeleteChangeSetInput$ = exports.DeactivateTypeOutput$ = exports.DeactivateTypeInput$ = exports.DeactivateOrganizationsAccessOutput$ = exports.DeactivateOrganizationsAccessInput$ = exports.CreateStackSetOutput$ = exports.CreateStackSetInput$ = exports.CreateStackRefactorOutput$ = exports.CreateStackRefactorInput$ = exports.CreateStackOutput$ = exports.CreateStackInstancesOutput$ = exports.CreateStackInstancesInput$ = exports.CreateStackInput$ = exports.CreateGeneratedTemplateOutput$ = exports.CreateGeneratedTemplateInput$ = exports.CreateChangeSetOutput$ = exports.CreateChangeSetInput$ = void 0;
exports.ListGeneratedTemplatesOutput$ = exports.ListGeneratedTemplatesInput$ = exports.ListExportsOutput$ = exports.ListExportsInput$ = exports.ListChangeSetsOutput$ = exports.ListChangeSetsInput$ = exports.ImportStacksToStackSetOutput$ = exports.ImportStacksToStackSetInput$ = exports.HookTarget$ = exports.HookResultSummary$ = exports.GetTemplateSummaryOutput$ = exports.GetTemplateSummaryInput$ = exports.GetTemplateOutput$ = exports.GetTemplateInput$ = exports.GetStackPolicyOutput$ = exports.GetStackPolicyInput$ = exports.GetHookResultOutput$ = exports.GetHookResultInput$ = exports.GetGeneratedTemplateOutput$ = exports.GetGeneratedTemplateInput$ = exports.Export$ = exports.ExecuteStackRefactorInput$ = exports.ExecuteChangeSetOutput$ = exports.ExecuteChangeSetInput$ = exports.EventFilter$ = exports.EstimateTemplateCostOutput$ = exports.EstimateTemplateCostInput$ = exports.DetectStackSetDriftOutput$ = exports.DetectStackSetDriftInput$ = exports.DetectStackResourceDriftOutput$ = exports.DetectStackResourceDriftInput$ = exports.DetectStackDriftOutput$ = exports.DetectStackDriftInput$ = exports.DescribeTypeRegistrationOutput$ = exports.DescribeTypeRegistrationInput$ = exports.DescribeTypeOutput$ = exports.DescribeTypeInput$ = exports.DescribeStacksOutput$ = exports.DescribeStacksInput$ = exports.DescribeStackSetOutput$ = exports.DescribeStackSetOperationOutput$ = exports.DescribeStackSetOperationInput$ = exports.DescribeStackSetInput$ = exports.DescribeStackResourcesOutput$ = exports.DescribeStackResourcesInput$ = exports.DescribeStackResourceOutput$ = exports.DescribeStackResourceInput$ = exports.DescribeStackResourceDriftsOutput$ = exports.DescribeStackResourceDriftsInput$ = exports.DescribeStackRefactorOutput$ = void 0;
exports.PublishTypeInput$ = exports.PropertyDifference$ = exports.PhysicalResourceIdContextKeyValuePair$ = exports.ParameterDeclaration$ = exports.ParameterConstraints$ = exports.Parameter$ = exports.Output$ = exports.OperationResultFilter$ = exports.OperationEvent$ = exports.OperationEntry$ = exports.ModuleInfo$ = exports.ManagedExecution$ = exports.LoggingConfig$ = exports.LiveResourceDrift$ = exports.ListTypeVersionsOutput$ = exports.ListTypeVersionsInput$ = exports.ListTypesOutput$ = exports.ListTypesInput$ = exports.ListTypeRegistrationsOutput$ = exports.ListTypeRegistrationsInput$ = exports.ListStacksOutput$ = exports.ListStacksInput$ = exports.ListStackSetsOutput$ = exports.ListStackSetsInput$ = exports.ListStackSetOperationsOutput$ = exports.ListStackSetOperationsInput$ = exports.ListStackSetOperationResultsOutput$ = exports.ListStackSetOperationResultsInput$ = exports.ListStackSetAutoDeploymentTargetsOutput$ = exports.ListStackSetAutoDeploymentTargetsInput$ = exports.ListStackResourcesOutput$ = exports.ListStackResourcesInput$ = exports.ListStackRefactorsOutput$ = exports.ListStackRefactorsInput$ = exports.ListStackRefactorActionsOutput$ = exports.ListStackRefactorActionsInput$ = exports.ListStackInstancesOutput$ = exports.ListStackInstancesInput$ = exports.ListStackInstanceResourceDriftsOutput$ = exports.ListStackInstanceResourceDriftsInput$ = exports.ListResourceScansOutput$ = exports.ListResourceScansInput$ = exports.ListResourceScanResourcesOutput$ = exports.ListResourceScanResourcesInput$ = exports.ListResourceScanRelatedResourcesOutput$ = exports.ListResourceScanRelatedResourcesInput$ = exports.ListImportsOutput$ = exports.ListImportsInput$ = exports.ListHookResultsOutput$ = exports.ListHookResultsInput$ = void 0;
exports.StackResourceSummary$ = exports.StackResourceDriftInformationSummary$ = exports.StackResourceDriftInformation$ = exports.StackResourceDrift$ = exports.StackResourceDetail$ = exports.StackResource$ = exports.StackRefactorSummary$ = exports.StackRefactorAction$ = exports.StackInstanceSummary$ = exports.StackInstanceResourceDriftsSummary$ = exports.StackInstanceFilter$ = exports.StackInstanceComprehensiveStatus$ = exports.StackInstance$ = exports.StackEvent$ = exports.StackDriftInformationSummary$ = exports.StackDriftInformation$ = exports.StackDefinition$ = exports.Stack$ = exports.SignalResourceInput$ = exports.SetTypeDefaultVersionOutput$ = exports.SetTypeDefaultVersionInput$ = exports.SetTypeConfigurationOutput$ = exports.SetTypeConfigurationInput$ = exports.SetStackPolicyInput$ = exports.ScannedResourceIdentifier$ = exports.ScannedResource$ = exports.ScanFilter$ = exports.RollbackTrigger$ = exports.RollbackStackOutput$ = exports.RollbackStackInput$ = exports.RollbackConfiguration$ = exports.ResourceToImport$ = exports.ResourceTargetDefinition$ = exports.ResourceScanSummary$ = exports.ResourceMapping$ = exports.ResourceLocation$ = exports.ResourceIdentifierSummary$ = exports.ResourceDriftIgnoredAttribute$ = exports.ResourceDetail$ = exports.ResourceDefinition$ = exports.ResourceChangeDetail$ = exports.ResourceChange$ = exports.RequiredActivatedType$ = exports.RegisterTypeOutput$ = exports.RegisterTypeInput$ = exports.RegisterPublisherOutput$ = exports.RegisterPublisherInput$ = exports.RecordHandlerProgressOutput$ = exports.RecordHandlerProgressInput$ = exports.PublishTypeOutput$ = void 0;
exports.CreateStack$ = exports.CreateGeneratedTemplate$ = exports.CreateChangeSet$ = exports.ContinueUpdateRollback$ = exports.CancelUpdateStack$ = exports.BatchDescribeTypeConfigurations$ = exports.ActivateType$ = exports.ActivateOrganizationsAccess$ = exports.Warnings$ = exports.WarningProperty$ = exports.WarningDetail$ = exports.ValidateTemplateOutput$ = exports.ValidateTemplateInput$ = exports.UpdateTerminationProtectionOutput$ = exports.UpdateTerminationProtectionInput$ = exports.UpdateStackSetOutput$ = exports.UpdateStackSetInput$ = exports.UpdateStackOutput$ = exports.UpdateStackInstancesOutput$ = exports.UpdateStackInstancesInput$ = exports.UpdateStackInput$ = exports.UpdateGeneratedTemplateOutput$ = exports.UpdateGeneratedTemplateInput$ = exports.TypeVersionSummary$ = exports.TypeSummary$ = exports.TypeFilters$ = exports.TypeConfigurationIdentifier$ = exports.TypeConfigurationDetails$ = exports.TestTypeOutput$ = exports.TestTypeInput$ = exports.TemplateSummaryConfig$ = exports.TemplateSummary$ = exports.TemplateProgress$ = exports.TemplateParameter$ = exports.TemplateConfiguration$ = exports.Tag$ = exports.StopStackSetOperationOutput$ = exports.StopStackSetOperationInput$ = exports.StartResourceScanOutput$ = exports.StartResourceScanInput$ = exports.StackSummary$ = exports.StackSetSummary$ = exports.StackSetOperationSummary$ = exports.StackSetOperationStatusDetails$ = exports.StackSetOperationResultSummary$ = exports.StackSetOperationPreferences$ = exports.StackSetOperation$ = exports.StackSetDriftDetectionDetails$ = exports.StackSetAutoDeploymentTargetSummary$ = exports.StackSet$ = void 0;
exports.ListResourceScanResources$ = exports.ListResourceScanRelatedResources$ = exports.ListImports$ = exports.ListHookResults$ = exports.ListGeneratedTemplates$ = exports.ListExports$ = exports.ListChangeSets$ = exports.ImportStacksToStackSet$ = exports.GetTemplateSummary$ = exports.GetTemplate$ = exports.GetStackPolicy$ = exports.GetHookResult$ = exports.GetGeneratedTemplate$ = exports.ExecuteStackRefactor$ = exports.ExecuteChangeSet$ = exports.EstimateTemplateCost$ = exports.DetectStackSetDrift$ = exports.DetectStackResourceDrift$ = exports.DetectStackDrift$ = exports.DescribeTypeRegistration$ = exports.DescribeType$ = exports.DescribeStackSetOperation$ = exports.DescribeStackSet$ = exports.DescribeStacks$ = exports.DescribeStackResources$ = exports.DescribeStackResourceDrifts$ = exports.DescribeStackResource$ = exports.DescribeStackRefactor$ = exports.DescribeStackInstance$ = exports.DescribeStackEvents$ = exports.DescribeStackDriftDetectionStatus$ = exports.DescribeResourceScan$ = exports.DescribePublisher$ = exports.DescribeOrganizationsAccess$ = exports.DescribeGeneratedTemplate$ = exports.DescribeEvents$ = exports.DescribeChangeSetHooks$ = exports.DescribeChangeSet$ = exports.DescribeAccountLimits$ = exports.DeregisterType$ = exports.DeleteStackSet$ = exports.DeleteStackInstances$ = exports.DeleteStack$ = exports.DeleteGeneratedTemplate$ = exports.DeleteChangeSet$ = exports.DeactivateType$ = exports.DeactivateOrganizationsAccess$ = exports.CreateStackSet$ = exports.CreateStackRefactor$ = exports.CreateStackInstances$ = void 0;
exports.ValidateTemplate$ = exports.UpdateTerminationProtection$ = exports.UpdateStackSet$ = exports.UpdateStackInstances$ = exports.UpdateStack$ = exports.UpdateGeneratedTemplate$ = exports.TestType$ = exports.StopStackSetOperation$ = exports.StartResourceScan$ = exports.SignalResource$ = exports.SetTypeDefaultVersion$ = exports.SetTypeConfiguration$ = exports.SetStackPolicy$ = exports.RollbackStack$ = exports.RegisterType$ = exports.RegisterPublisher$ = exports.RecordHandlerProgress$ = exports.PublishType$ = exports.ListTypeVersions$ = exports.ListTypes$ = exports.ListTypeRegistrations$ = exports.ListStackSets$ = exports.ListStackSetOperations$ = exports.ListStackSetOperationResults$ = exports.ListStackSetAutoDeploymentTargets$ = exports.ListStacks$ = exports.ListStackResources$ = exports.ListStackRefactors$ = exports.ListStackRefactorActions$ = exports.ListStackInstances$ = exports.ListStackInstanceResourceDrifts$ = exports.ListResourceScans$ = void 0;
const _A = "Arn";
const _AC = "AfterContext";
const _ACT = "AttributeChangeType";
const _AD = "AutoDeployment";
const _AEE = "AlreadyExistsException";
const _AFT = "AccountFilterType";
const _AGR = "AccountGateResult";
const _AL = "AccountLimit";
const _ALL = "AccountLimitList";
const _ALc = "AccountLimits";
const _ALn = "AnnotationList";
const _AN = "AnnotationName";
const _AOA = "ActivateOrganizationsAccess";
const _AOAI = "ActivateOrganizationsAccessInput";
const _AOAO = "ActivateOrganizationsAccessOutput";
const _AP = "ActualProperties";
const _AR = "AddResources";
const _ARARN = "AdministrationRoleARN";
const _AT = "ActivateType";
const _ATAC = "AcceptTermsAndConditions";
const _ATI = "ActivateTypeInput";
const _ATO = "ActivateTypeOutput";
const _AU = "AutoUpdate";
const _AUc = "AccountsUrl";
const _AV = "ActualValue";
const _AVF = "AfterValueFrom";
const _AVf = "AfterValue";
const _AVl = "AllowedValues";
const _Ac = "Accounts";
const _Acc = "Account";
const _Act = "Action";
const _Acti = "Active";
const _Al = "Alias";
const _An = "Annotation";
const _Ann = "Annotations";
const _At = "Attribute";
const _BC = "BeforeContext";
const _BDTC = "BatchDescribeTypeConfigurations";
const _BDTCE = "BatchDescribeTypeConfigurationsError";
const _BDTCEa = "BatchDescribeTypeConfigurationsErrors";
const _BDTCI = "BatchDescribeTypeConfigurationsInput";
const _BDTCO = "BatchDescribeTypeConfigurationsOutput";
const _BT = "BearerToken";
const _BV = "BeforeValue";
const _BVF = "BeforeValueFrom";
const _C = "Change";
const _CA = "CallAs";
const _CAo = "ConnectionArn";
const _CAon = "ConfigurationAlias";
const _CAonf = "ConfigurationArn";
const _CBME = "CreatedButModifiedException";
const _CCS = "CreateChangeSet";
const _CCSI = "CreateChangeSetInput";
const _CCSO = "CreateChangeSetOutput";
const _CE = "CausingEntity";
const _CFNRE = "CFNRegistryException";
const _CGT = "CreateGeneratedTemplate";
const _CGTI = "CreateGeneratedTemplateInput";
const _CGTO = "CreateGeneratedTemplateOutput";
const _CM = "ConcurrencyMode";
const _COS = "CurrentOperationStatus";
const _CR = "CapabilitiesReason";
const _CRLEE = "ConcurrentResourcesLimitExceededException";
const _CRT = "ClientRequestToken";
const _CS = "ConfigurationSchema";
const _CSH = "ChangeSetHook";
const _CSHRTD = "ChangeSetHookResourceTargetDetails";
const _CSHTD = "ChangeSetHookTargetDetails";
const _CSHh = "ChangeSetHooks";
const _CSI = "ChangeSetId";
const _CSII = "CreateStackInstancesInput";
const _CSIO = "CreateStackInstancesOutput";
const _CSIr = "CreateStackInput";
const _CSIre = "CreateStackInstances";
const _CSN = "ChangeSetName";
const _CSNFE = "ChangeSetNotFoundException";
const _CSO = "CreateStackOutput";
const _CSR = "CreateStackRefactor";
const _CSRI = "CreateStackRefactorInput";
const _CSRO = "CreateStackRefactorOutput";
const _CSS = "ChangeSetSummary";
const _CSSI = "CreateStackSetInput";
const _CSSO = "CreateStackSetOutput";
const _CSSh = "ChangeSetSummaries";
const _CSSr = "CreateStackSet";
const _CST = "ChangeSetType";
const _CSh = "ChangeSource";
const _CSr = "CreateStack";
const _CT = "CreationTime";
const _CTl = "ClientToken";
const _CTr = "CreationTimestamp";
const _CUR = "ContinueUpdateRollback";
const _CURI = "ContinueUpdateRollbackInput";
const _CURO = "ContinueUpdateRollbackOutput";
const _CUS = "CancelUpdateStack";
const _CUSI = "CancelUpdateStackInput";
const _Ca = "Capabilities";
const _Cat = "Category";
const _Ch = "Changes";
const _Co = "Configuration";
const _D = "Description";
const _DAL = "DescribeAccountLimits";
const _DALI = "DescribeAccountLimitsInput";
const _DALO = "DescribeAccountLimitsOutput";
const _DCS = "DeleteChangeSet";
const _DCSH = "DescribeChangeSetHooks";
const _DCSHI = "DescribeChangeSetHooksInput";
const _DCSHO = "DescribeChangeSetHooksOutput";
const _DCSI = "DeleteChangeSetInput";
const _DCSIe = "DescribeChangeSetInput";
const _DCSO = "DeleteChangeSetOutput";
const _DCSOe = "DescribeChangeSetOutput";
const _DCSe = "DescribeChangeSet";
const _DDS = "DriftDetectionStatus";
const _DDT = "DriftDetectionTimestamp";
const _DE = "DescribeEvents";
const _DEI = "DescribeEventsInput";
const _DEO = "DescribeEventsOutput";
const _DGT = "DeleteGeneratedTemplate";
const _DGTI = "DeleteGeneratedTemplateInput";
const _DGTIe = "DescribeGeneratedTemplateInput";
const _DGTO = "DescribeGeneratedTemplateOutput";
const _DGTe = "DescribeGeneratedTemplate";
const _DI = "DriftInformation";
const _DM = "DeploymentMode";
const _DMe = "DeletionMode";
const _DO = "DependsOn";
const _DOA = "DeactivateOrganizationsAccess";
const _DOAI = "DeactivateOrganizationsAccessInput";
const _DOAIe = "DescribeOrganizationsAccessInput";
const _DOAO = "DeactivateOrganizationsAccessOutput";
const _DOAOe = "DescribeOrganizationsAccessOutput";
const _DOAe = "DescribeOrganizationsAccess";
const _DP = "DeletionPolicy";
const _DPI = "DescribePublisherInput";
const _DPO = "DescribePublisherOutput";
const _DPe = "DescribePublisher";
const _DR = "DisableRollback";
const _DRS = "DescribeResourceScan";
const _DRSI = "DescribeResourceScanInput";
const _DRSO = "DescribeResourceScanOutput";
const _DRe = "DetectionReason";
const _DS = "DetectionStatus";
const _DSD = "DetectStackDrift";
const _DSDDS = "DescribeStackDriftDetectionStatus";
const _DSDDSI = "DescribeStackDriftDetectionStatusInput";
const _DSDDSO = "DescribeStackDriftDetectionStatusOutput";
const _DSDI = "DetectStackDriftInput";
const _DSDO = "DetectStackDriftOutput";
const _DSE = "DescribeStackEvents";
const _DSEI = "DescribeStackEventsInput";
const _DSEO = "DescribeStackEventsOutput";
const _DSI = "DeleteStackInput";
const _DSIC = "DriftedStackInstancesCount";
const _DSII = "DeleteStackInstancesInput";
const _DSIIe = "DescribeStackInstanceInput";
const _DSIO = "DeleteStackInstancesOutput";
const _DSIOe = "DescribeStackInstanceOutput";
const _DSIe = "DescribeStacksInput";
const _DSIel = "DeleteStackInstances";
const _DSIes = "DescribeStackInstance";
const _DSO = "DescribeStacksOutput";
const _DSR = "DetectionStatusReason";
const _DSRC = "DriftedStackResourceCount";
const _DSRD = "DescribeStackResourceDrifts";
const _DSRDI = "DescribeStackResourceDriftsInput";
const _DSRDIe = "DetectStackResourceDriftInput";
const _DSRDO = "DescribeStackResourceDriftsOutput";
const _DSRDOe = "DetectStackResourceDriftOutput";
const _DSRDe = "DetectStackResourceDrift";
const _DSRI = "DescribeStackRefactorInput";
const _DSRIe = "DescribeStackResourceInput";
const _DSRIes = "DescribeStackResourcesInput";
const _DSRO = "DescribeStackRefactorOutput";
const _DSROe = "DescribeStackResourceOutput";
const _DSROes = "DescribeStackResourcesOutput";
const _DSRe = "DescribeStackRefactor";
const _DSRes = "DescribeStackResource";
const _DSResc = "DescribeStackResources";
const _DSRr = "DriftStatusReason";
const _DSS = "DeleteStackSet";
const _DSSD = "DetectStackSetDrift";
const _DSSDI = "DetectStackSetDriftInput";
const _DSSDO = "DetectStackSetDriftOutput";
const _DSSI = "DeleteStackSetInput";
const _DSSIe = "DescribeStackSetInput";
const _DSSO = "DeleteStackSetOutput";
const _DSSOI = "DescribeStackSetOperationInput";
const _DSSOO = "DescribeStackSetOperationOutput";
const _DSSOe = "DescribeStackSetOutput";
const _DSSOes = "DescribeStackSetOperation";
const _DSSe = "DescribeStackSet";
const _DSe = "DeprecatedStatus";
const _DSel = "DeleteStack";
const _DSes = "DescribeStacks";
const _DSet = "DetailedStatus";
const _DSr = "DriftStatus";
const _DT = "DeploymentTargets";
const _DTI = "DeactivateTypeInput";
const _DTIe = "DeregisterTypeInput";
const _DTIes = "DescribeTypeInput";
const _DTO = "DeactivateTypeOutput";
const _DTOe = "DeregisterTypeOutput";
const _DTOes = "DescribeTypeOutput";
const _DTR = "DescribeTypeRegistration";
const _DTRI = "DescribeTypeRegistrationInput";
const _DTRO = "DescribeTypeRegistrationOutput";
const _DTe = "DeclaredTransforms";
const _DTea = "DeactivateType";
const _DTel = "DeletionTime";
const _DTer = "DeregisterType";
const _DTes = "DescribeType";
const _DTi = "DifferenceType";
const _DU = "DocumentationUrl";
const _DV = "DefaultValue";
const _DVI = "DefaultVersionId";
const _De = "Details";
const _Des = "Destination";
const _Det = "Detection";
const _Dr = "Drift";
const _E = "Enabled";
const _EC = "ErrorCode";
const _ECS = "ExecuteChangeSet";
const _ECSI = "ExecuteChangeSetInput";
const _ECSO = "ExecuteChangeSetOutput";
const _EF = "EventFilter";
const _EI = "EventId";
const _EM = "ErrorMessage";
const _EN = "ExportName";
const _EP = "ExpectedProperties";
const _ERA = "ExecutionRoleArn";
const _ERN = "ExecutionRoleName";
const _ES = "ExecutionStatus";
const _ESC = "EnableStackCreation";
const _ESF = "ExecutionStatusFilter";
const _ESI = "ExportingStackId";
const _ESR = "ExecutionStatusReason";
const _ESRI = "ExecuteStackRefactorInput";
const _ESRx = "ExecuteStackRefactor";
const _ET = "EndTime";
const _ETC = "EstimateTemplateCost";
const _ETCI = "EstimateTemplateCostInput";
const _ETCO = "EstimateTemplateCostOutput";
const _ETP = "EnableTerminationProtection";
const _ETn = "EndTimestamp";
const _ETv = "EventType";
const _EV = "ExpectedValue";
const _En = "Entity";
const _Er = "Errors";
const _Ev = "Evaluation";
const _Ex = "Export";
const _Exp = "Exports";
const _F = "Filters";
const _FE = "FailedEvents";
const _FM = "FailureMode";
const _FSIC = "FailedStackInstancesCount";
const _FTC = "FailureToleranceCount";
const _FTP = "FailureTolerancePercentage";
const _Fo = "Format";
const _GGT = "GetGeneratedTemplate";
const _GGTI = "GetGeneratedTemplateInput";
const _GGTO = "GetGeneratedTemplateOutput";
const _GHR = "GetHookResult";
const _GHRI = "GetHookResultInput";
const _GHRO = "GetHookResultOutput";
const _GSP = "GetStackPolicy";
const _GSPI = "GetStackPolicyInput";
const _GSPO = "GetStackPolicyOutput";
const _GT = "GetTemplate";
const _GTI = "GeneratedTemplateId";
const _GTIe = "GetTemplateInput";
const _GTN = "GeneratedTemplateName";
const _GTNFE = "GeneratedTemplateNotFoundException";
const _GTO = "GetTemplateOutput";
const _GTS = "GetTemplateSummary";
const _GTSI = "GetTemplateSummaryInput";
const _GTSO = "GetTemplateSummaryOutput";
const _H = "Hooks";
const _HET = "HookExecutionTarget";
const _HFM = "HookFailureMode";
const _HIC = "HookInvocationCount";
const _HII = "HookInvocationId";
const _HIP = "HookInvocationPoint";
const _HR = "HookResults";
const _HRI = "HookResultId";
const _HRNFE = "HookResultNotFoundException";
const _HRS = "HookResultSummary";
const _HRSo = "HookResultSummaries";
const _HS = "HookStatus";
const _HSR = "HookStatusReason";
const _HT = "HookTarget";
const _HTo = "HookType";
const _I = "Id";
const _IA = "IsActivated";
const _IAn = "InvokedAt";
const _ICE = "InsufficientCapabilitiesException";
const _ICSSE = "InvalidChangeSetStatusException";
const _IDC = "IsDefaultConfiguration";
const _IDV = "IsDefaultVersion";
const _IER = "ImportExistingResources";
const _INS = "IncludeNestedStacks";
const _IOE = "InvalidOperationException";
const _IP = "InvocationPoint";
const _IPSIC = "InProgressStackInstancesCount";
const _IPV = "IncludePropertyValues";
const _IPd = "IdentityProvider";
const _ISSIC = "InSyncStackInstancesCount";
const _ISTE = "InvalidStateTransitionException";
const _ISTSS = "ImportStacksToStackSet";
const _ISTSSI = "ImportStacksToStackSetInput";
const _ISTSSO = "ImportStacksToStackSetOutput";
const _Im = "Imports";
const _K = "Key";
const _LC = "LoggingConfig";
const _LCS = "ListChangeSets";
const _LCSI = "ListChangeSetsInput";
const _LCSO = "ListChangeSetsOutput";
const _LCT = "LastCheckTimestamp";
const _LDB = "LogDeliveryBucket";
const _LDCT = "LastDriftCheckTimestamp";
const _LE = "ListExports";
const _LEE = "LimitExceededException";
const _LEI = "ListExportsInput";
const _LEO = "ListExportsOutput";
const _LGN = "LogGroupName";
const _LGT = "ListGeneratedTemplates";
const _LGTI = "ListGeneratedTemplatesInput";
const _LGTO = "ListGeneratedTemplatesOutput";
const _LHR = "ListHookResults";
const _LHRI = "ListHookResultsInput";
const _LHRO = "ListHookResultsOutput";
const _LI = "ListImports";
const _LIH = "LogicalIdHierarchy";
const _LII = "ListImportsInput";
const _LIO = "ListImportsOutput";
const _LO = "LastOperations";
const _LOI = "LastOperationId";
const _LPV = "LatestPublicVersion";
const _LRA = "LogRoleArn";
const _LRD = "LiveResourceDrift";
const _LRI = "LogicalResourceId";
const _LRIo = "LogicalResourceIds";
const _LRS = "ListResourceScans";
const _LRSI = "ListResourceScansInput";
const _LRSO = "ListResourceScansOutput";
const _LRSR = "ListResourceScanResources";
const _LRSRI = "ListResourceScanResourcesInput";
const _LRSRO = "ListResourceScanResourcesOutput";
const _LRSRR = "ListResourceScanRelatedResources";
const _LRSRRI = "ListResourceScanRelatedResourcesInput";
const _LRSRRO = "ListResourceScanRelatedResourcesOutput";
const _LS = "ListStacks";
const _LSI = "ListStacksInput";
const _LSII = "ListStackInstancesInput";
const _LSIO = "ListStackInstancesOutput";
const _LSIRD = "ListStackInstanceResourceDrifts";
const _LSIRDI = "ListStackInstanceResourceDriftsInput";
const _LSIRDO = "ListStackInstanceResourceDriftsOutput";
const _LSIi = "ListStackInstances";
const _LSO = "ListStacksOutput";
const _LSR = "ListStackRefactors";
const _LSRA = "ListStackRefactorActions";
const _LSRAI = "ListStackRefactorActionsInput";
const _LSRAO = "ListStackRefactorActionsOutput";
const _LSRI = "ListStackRefactorsInput";
const _LSRIi = "ListStackResourcesInput";
const _LSRO = "ListStackRefactorsOutput";
const _LSROi = "ListStackResourcesOutput";
const _LSRi = "ListStackResources";
const _LSS = "ListStackSets";
const _LSSADT = "ListStackSetAutoDeploymentTargets";
const _LSSADTI = "ListStackSetAutoDeploymentTargetsInput";
const _LSSADTO = "ListStackSetAutoDeploymentTargetsOutput";
const _LSSI = "ListStackSetsInput";
const _LSSO = "ListStackSetsOutput";
const _LSSOI = "ListStackSetOperationsInput";
const _LSSOO = "ListStackSetOperationsOutput";
const _LSSOR = "ListStackSetOperationResults";
const _LSSORI = "ListStackSetOperationResultsInput";
const _LSSORO = "ListStackSetOperationResultsOutput";
const _LSSOi = "ListStackSetOperations";
const _LT = "ListTypes";
const _LTI = "ListTypesInput";
const _LTO = "ListTypesOutput";
const _LTR = "ListTypeRegistrations";
const _LTRI = "ListTypeRegistrationsInput";
const _LTRO = "ListTypeRegistrationsOutput";
const _LTV = "ListTypeVersions";
const _LTVI = "ListTypeVersionsInput";
const _LTVO = "ListTypeVersionsOutput";
const _LU = "LastUpdated";
const _LUT = "LastUpdatedTime";
const _LUTa = "LastUpdatedTimestamp";
const _M = "Message";
const _MBS = "ManagedByStack";
const _MCC = "MaxConcurrentCount";
const _MCP = "MaxConcurrentPercentage";
const _ME = "ManagedExecution";
const _MI = "ModuleInfo";
const _MR = "MaxResults";
const _MTIM = "MonitoringTimeInMinutes";
const _MV = "MajorVersion";
const _Me = "Metadata";
const _N = "Name";
const _NAEE = "NameAlreadyExistsException";
const _NARN = "NotificationARNs";
const _NE = "NoEcho";
const _NGTN = "NewGeneratedTemplateName";
const _NOR = "NumberOfResources";
const _NT = "NextToken";
const _O = "Output";
const _OE = "OperationEvents";
const _OEp = "OperationEntry";
const _OEpe = "OperationEvent";
const _OF = "OnFailure";
const _OI = "OperationId";
const _OIAEE = "OperationIdAlreadyExistsException";
const _OIPE = "OperationInProgressException";
const _OK = "OutputKey";
const _ONFE = "OperationNotFoundException";
const _OP = "OperationPreferences";
const _ORF = "OperationResultFilter";
const _ORFp = "OperationResultFilters";
const _OS = "OperationStatus";
const _OSCFE = "OperationStatusCheckFailedException";
const _OSF = "OnStackFailure";
const _OT = "OperationType";
const _OTA = "OriginalTypeArn";
const _OTN = "OriginalTypeName";
const _OUI = "OrganizationalUnitIds";
const _OUIr = "OrganizationalUnitId";
const _OV = "OutputValue";
const _Ou = "Outputs";
const _P = "Parameters";
const _PA = "PolicyAction";
const _PC = "PercentageCompleted";
const _PCSI = "ParentChangeSetId";
const _PCa = "ParameterConstraints";
const _PD = "ParameterDeclaration";
const _PDC = "PreviousDeploymentContext";
const _PDa = "ParameterDeclarations";
const _PDr = "PropertyDifference";
const _PDro = "PropertyDifferences";
const _PI = "PublisherId";
const _PIa = "ParentId";
const _PIu = "PublisherIdentity";
const _PK = "ParameterKey";
const _PM = "PermissionModel";
const _PN = "PublisherName";
const _PO = "ParameterOverrides";
const _PP = "PublisherProfile";
const _PPr = "PropertyPath";
const _PRI = "PhysicalResourceId";
const _PRIC = "PhysicalResourceIdContext";
const _PRICKVP = "PhysicalResourceIdContextKeyValuePair";
const _PS = "PublisherStatus";
const _PSr = "ProgressStatus";
const _PT = "ProvisioningType";
const _PTA = "PublicTypeArn";
const _PTI = "PublishTypeInput";
const _PTO = "PublishTypeOutput";
const _PTa = "ParameterType";
const _PTu = "PublishType";
const _PV = "PreviousValue";
const _PVN = "PublicVersionNumber";
const _PVa = "ParameterValue";
const _Pa = "Parameter";
const _Pat = "Path";
const _Pr = "Progress";
const _Pro = "Properties";
const _R = "Resources";
const _RA = "ResourceAction";
const _RAR = "RefreshAllResources";
const _RARN = "RoleARN";
const _RAT = "RequiredActivatedTypes";
const _RATe = "RequiredActivatedType";
const _RC = "ResourceChange";
const _RCD = "ResourceChangeDetail";
const _RCDe = "ResourceChangeDetails";
const _RCSI = "RootChangeSetId";
const _RCT = "RegionConcurrencyType";
const _RCo = "RollbackConfiguration";
const _RD = "ResourceDefinition";
const _RDIA = "ResourceDriftIgnoredAttributes";
const _RDIAe = "ResourceDriftIgnoredAttribute";
const _RDS = "ResourceDriftStatus";
const _RDe = "ResourceDetail";
const _RDes = "ResourceDefinitions";
const _RDeso = "ResourceDetails";
const _REOC = "RetainExceptOnCreate";
const _RF = "ResourcesFailed";
const _RHP = "RecordHandlerProgress";
const _RHPI = "RecordHandlerProgressInput";
const _RHPO = "RecordHandlerProgressOutput";
const _RI = "ResourceIdentifier";
const _RIS = "ResourceIdentifierSummaries";
const _RISe = "ResourceIdentifierSummary";
const _RIe = "ResourceIdentifiers";
const _RIo = "RootId";
const _RL = "RemediationLink";
const _RLe = "ResourceLocation";
const _RM = "RemediationMessage";
const _RMe = "ResourceMappings";
const _RMes = "ResourceModel";
const _RMeso = "ResourceMapping";
const _RO = "RegionOrder";
const _RP = "ResourceProperties";
const _RPI = "RegisterPublisherInput";
const _RPO = "RegisterPublisherOutput";
const _RPe = "ResourcesProcessing";
const _RPeg = "RegisterPublisher";
const _RPes = "ResourcesPending";
const _RR = "RetainResources";
const _RRe = "ResourcesRead";
const _RRel = "RelatedResources";
const _RRem = "RemoveResources";
const _RReq = "RequiresRecreation";
const _RS = "RetainStacks";
const _RSF = "RegistrationStatusFilter";
const _RSI = "ResourceScanId";
const _RSIPE = "ResourceScanInProgressException";
const _RSIo = "RollbackStackInput";
const _RSLEE = "ResourceScanLimitExceededException";
const _RSNFE = "ResourceScanNotFoundException";
const _RSO = "RollbackStackOutput";
const _RSOAR = "RetainStacksOnAccountRemoval";
const _RSR = "ResourceStatusReason";
const _RSS = "ResourceScanSummaries";
const _RSSe = "ResourceScanSummary";
const _RSe = "ResourcesScanned";
const _RSes = "ResourceStatus";
const _RSeso = "ResourcesSucceeded";
const _RSo = "RollbackStack";
const _RT = "ResourceType";
const _RTD = "ResourceTargetDetails";
const _RTDe = "ResourceTargetDefinition";
const _RTI = "ResourcesToImport";
const _RTIe = "RegisterTypeInput";
const _RTIes = "ResourceToImport";
const _RTL = "RegistrationTokenList";
const _RTO = "RegisterTypeOutput";
const _RTP = "ResourceTypePrefix";
const _RTS = "ResourcesToSkip";
const _RTe = "ResourceTypes";
const _RTeg = "RegistrationToken";
const _RTegi = "RegisterType";
const _RTo = "RollbackTriggers";
const _RTol = "RollbackTrigger";
const _RV = "ResolvedValue";
const _Re = "Regions";
const _Rea = "Reason";
const _Reg = "Region";
const _Rep = "Replacement";
const _Req = "Required";
const _S = "Status";
const _SA = "StagesAvailable";
const _SD = "StackDefinitions";
const _SDDI = "StackDriftDetectionId";
const _SDI = "StackDriftInformation";
const _SDIS = "StackDriftInformationSummary";
const _SDS = "StackDriftStatus";
const _SDt = "StackDefinition";
const _SDta = "StatusDetails";
const _SE = "StackEvents";
const _SEt = "StackEvent";
const _SF = "ScanFilters";
const _SFc = "ScanFilter";
const _SHP = "SchemaHandlerPackage";
const _SI = "StackId";
const _SIA = "StackInstanceAccount";
const _SICS = "StackInstanceComprehensiveStatus";
const _SIF = "StackInstanceFilter";
const _SIFt = "StackInstanceFilters";
const _SINFE = "StackInstanceNotFoundException";
const _SIR = "StackInstanceRegion";
const _SIRDS = "StackInstanceResourceDriftStatuses";
const _SIRDSt = "StackInstanceResourceDriftsSummary";
const _SIRDSta = "StackInstanceResourceDriftsSummaries";
const _SIS = "StackInstanceStatus";
const _SISt = "StackInstanceSummary";
const _SISta = "StackInstanceSummaries";
const _SIU = "StackIdsUrl";
const _SIt = "StackInstance";
const _SIta = "StackIds";
const _SL = "SeverityLevel";
const _SM = "StatusMessage";
const _SMV = "SupportedMajorVersions";
const _SN = "StackName";
const _SNFE = "StackNotFoundException";
const _SPB = "StackPolicyBody";
const _SPDUB = "StackPolicyDuringUpdateBody";
const _SPDUURL = "StackPolicyDuringUpdateURL";
const _SPURL = "StackPolicyURL";
const _SR = "StatusReason";
const _SRA = "StackRefactorActions";
const _SRAt = "StackRefactorAction";
const _SRD = "StackResourceDrifts";
const _SRDI = "StackResourceDriftInformation";
const _SRDIS = "StackResourceDriftInformationSummary";
const _SRDS = "StackResourceDriftStatus";
const _SRDSF = "StackResourceDriftStatusFilters";
const _SRDt = "StackResourceDetail";
const _SRDta = "StackResourceDrift";
const _SRE = "StaleRequestException";
const _SRI = "StackRefactorId";
const _SRIc = "ScannedResourceIdentifier";
const _SRIca = "ScannedResourceIdentifiers";
const _SRIi = "SignalResourceInput";
const _SRNFE = "StackRefactorNotFoundException";
const _SRS = "StackRefactorSummaries";
const _SRSI = "StartResourceScanInput";
const _SRSO = "StartResourceScanOutput";
const _SRSt = "StackResourceSummaries";
const _SRSta = "StackRefactorSummary";
const _SRStac = "StackResourceSummary";
const _SRStar = "StartResourceScan";
const _SRTR = "StackRefactorTagResources";
const _SRc = "ScannedResource";
const _SRca = "ScannedResources";
const _SRi = "SignalResource";
const _SRt = "StackResources";
const _SRta = "StackResource";
const _SS = "StackSet";
const _SSADTS = "StackSetAutoDeploymentTargetSummary";
const _SSADTSt = "StackSetAutoDeploymentTargetSummaries";
const _SSARN = "StackSetARN";
const _SSDDD = "StackSetDriftDetectionDetails";
const _SSF = "StackStatusFilter";
const _SSI = "StackSetId";
const _SSN = "StackSetName";
const _SSNEE = "StackSetNotEmptyException";
const _SSNFE = "StackSetNotFoundException";
const _SSO = "StackSetOperation";
const _SSOP = "StackSetOperationPreferences";
const _SSORS = "StackSetOperationResultSummary";
const _SSORSt = "StackSetOperationResultSummaries";
const _SSOS = "StackSetOperationSummary";
const _SSOSD = "StackSetOperationStatusDetails";
const _SSOSt = "StackSetOperationSummaries";
const _SSP = "SetStackPolicy";
const _SSPI = "SetStackPolicyInput";
const _SSR = "StackStatusReason";
const _SSS = "StackSetSummary";
const _SSSO = "StopStackSetOperation";
const _SSSOI = "StopStackSetOperationInput";
const _SSSOO = "StopStackSetOperationOutput";
const _SSSt = "StackSetSummaries";
const _SSt = "StackSummaries";
const _SSta = "StackStatus";
const _SStac = "StackSummary";
const _ST = "StartTime";
const _STC = "SetTypeConfiguration";
const _STCI = "SetTypeConfigurationInput";
const _STCO = "SetTypeConfigurationOutput";
const _STDV = "SetTypeDefaultVersion";
const _STDVI = "SetTypeDefaultVersionInput";
const _STDVO = "SetTypeDefaultVersionOutput";
const _STF = "ScanTypeFilter";
const _STc = "ScanType";
const _SU = "SourceUrl";
const _Sc = "Schema";
const _Sco = "Scope";
const _So = "Source";
const _St = "Stacks";
const _Sta = "Stack";
const _Su = "Summaries";
const _T = "Type";
const _TA = "TypeArn";
const _TAEE = "TokenAlreadyExistsException";
const _TB = "TemplateBody";
const _TC = "TypeConfigurations";
const _TCA = "TypeConfigurationAlias";
const _TCAy = "TypeConfigurationArn";
const _TCD = "TypeConfigurationDetails";
const _TCDL = "TypeConfigurationDetailsList";
const _TCI = "TypeConfigurationIdentifier";
const _TCIy = "TypeConfigurationIdentifiers";
const _TCNFE = "TypeConfigurationNotFoundException";
const _TCVI = "TypeConfigurationVersionId";
const _TCe = "TemplateConfiguration";
const _TCi = "TimeCreated";
const _TD = "TargetDetails";
const _TDe = "TemplateDescription";
const _TF = "TypeFilters";
const _TH = "TypeHierarchy";
const _TI = "TargetId";
const _TIM = "TimeoutInMinutes";
const _TK = "TagKey";
const _TN = "TypeName";
const _TNA = "TypeNameAlias";
const _TNFE = "TypeNotFoundException";
const _TNP = "TypeNamePrefix";
const _TP = "TemplateParameter";
const _TPe = "TemplateProgress";
const _TPem = "TemplateParameters";
const _TR = "TagResources";
const _TS = "TemplateStage";
const _TSC = "TemplateSummaryConfig";
const _TSIC = "TotalStackInstancesCount";
const _TSe = "TemplateSummary";
const _TSem = "TemplateSummaries";
const _TSy = "TypeSummaries";
const _TSyp = "TypeSummary";
const _TT = "TargetType";
const _TTI = "TestTypeInput";
const _TTN = "TargetTypeName";
const _TTO = "TestTypeOutput";
const _TTS = "TypeTestsStatus";
const _TTSD = "TypeTestsStatusDescription";
const _TTe = "TestType";
const _TURL = "TemplateURL";
const _TURTAW = "TreatUnrecognizedResourceTypesAsWarnings";
const _TV = "TagValue";
const _TVA = "TypeVersionArn";
const _TVI = "TypeVersionId";
const _TVS = "TypeVersionSummaries";
const _TVSy = "TypeVersionSummary";
const _TW = "TotalWarnings";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Target";
const _Ti = "Timestamp";
const _Ty = "Types";
const _U = "Url";
const _UGT = "UpdateGeneratedTemplate";
const _UGTI = "UpdateGeneratedTemplateInput";
const _UGTO = "UpdateGeneratedTemplateOutput";
const _UI = "UniqueId";
const _UPT = "UsePreviousTemplate";
const _UPV = "UsePreviousValue";
const _UR = "UntagResources";
const _URP = "UpdateReplacePolicy";
const _URT = "UnrecognizedResourceTypes";
const _US = "UpdateStack";
const _USI = "UpdateStackInput";
const _USII = "UpdateStackInstancesInput";
const _USIO = "UpdateStackInstancesOutput";
const _USIp = "UpdateStackInstances";
const _USO = "UpdateStackOutput";
const _USS = "UpdateStackSet";
const _USSI = "UpdateStackSetInput";
const _USSO = "UpdateStackSetOutput";
const _UTC = "UnprocessedTypeConfigurations";
const _UTP = "UpdateTerminationProtection";
const _UTPI = "UpdateTerminationProtectionInput";
const _UTPO = "UpdateTerminationProtectionOutput";
const _V = "Value";
const _VB = "VersionBump";
const _VFM = "ValidationFailureMode";
const _VI = "VersionId";
const _VN = "ValidationName";
const _VP = "ValidationPath";
const _VS = "ValidationStatus";
const _VSR = "ValidationStatusReason";
const _VT = "ValidateTemplate";
const _VTI = "ValidateTemplateInput";
const _VTO = "ValidateTemplateOutput";
const _Va = "Values";
const _Ve = "Version";
const _Vi = "Visibility";
const _W = "Warnings";
const _WD = "WarningDetail";
const _WDa = "WarningDetails";
const _WP = "WarningProperty";
const _WPa = "WarningProperties";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.cloudformation";
const n0 = "com.amazonaws.cloudformation";
const schema_1 = require("@smithy/core/schema");
const CloudFormationServiceException_1 = require("../models/CloudFormationServiceException");
const errors_1 = require("../models/errors");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.CloudFormationServiceException$ = [-3, _s, "CloudFormationServiceException", 0, [], []];
_s_registry.registerError(exports.CloudFormationServiceException$, CloudFormationServiceException_1.CloudFormationServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AlreadyExistsException$ = [-3, n0, _AEE,
    { [_aQE]: [`AlreadyExistsException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.AlreadyExistsException$, errors_1.AlreadyExistsException);
exports.CFNRegistryException$ = [-3, n0, _CFNRE,
    { [_aQE]: [`CFNRegistryException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.CFNRegistryException$, errors_1.CFNRegistryException);
exports.ChangeSetNotFoundException$ = [-3, n0, _CSNFE,
    { [_aQE]: [`ChangeSetNotFound`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ChangeSetNotFoundException$, errors_1.ChangeSetNotFoundException);
exports.ConcurrentResourcesLimitExceededException$ = [-3, n0, _CRLEE,
    { [_aQE]: [`ConcurrentResourcesLimitExceeded`, 429], [_e]: _c, [_hE]: 429 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ConcurrentResourcesLimitExceededException$, errors_1.ConcurrentResourcesLimitExceededException);
exports.CreatedButModifiedException$ = [-3, n0, _CBME,
    { [_aQE]: [`CreatedButModifiedException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.CreatedButModifiedException$, errors_1.CreatedButModifiedException);
exports.GeneratedTemplateNotFoundException$ = [-3, n0, _GTNFE,
    { [_aQE]: [`GeneratedTemplateNotFound`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.GeneratedTemplateNotFoundException$, errors_1.GeneratedTemplateNotFoundException);
exports.HookResultNotFoundException$ = [-3, n0, _HRNFE,
    { [_aQE]: [`HookResultNotFound`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.HookResultNotFoundException$, errors_1.HookResultNotFoundException);
exports.InsufficientCapabilitiesException$ = [-3, n0, _ICE,
    { [_aQE]: [`InsufficientCapabilitiesException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InsufficientCapabilitiesException$, errors_1.InsufficientCapabilitiesException);
exports.InvalidChangeSetStatusException$ = [-3, n0, _ICSSE,
    { [_aQE]: [`InvalidChangeSetStatus`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InvalidChangeSetStatusException$, errors_1.InvalidChangeSetStatusException);
exports.InvalidOperationException$ = [-3, n0, _IOE,
    { [_aQE]: [`InvalidOperationException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InvalidOperationException$, errors_1.InvalidOperationException);
exports.InvalidStateTransitionException$ = [-3, n0, _ISTE,
    { [_aQE]: [`InvalidStateTransition`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InvalidStateTransitionException$, errors_1.InvalidStateTransitionException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_aQE]: [`LimitExceededException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.NameAlreadyExistsException$ = [-3, n0, _NAEE,
    { [_aQE]: [`NameAlreadyExistsException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.NameAlreadyExistsException$, errors_1.NameAlreadyExistsException);
exports.OperationIdAlreadyExistsException$ = [-3, n0, _OIAEE,
    { [_aQE]: [`OperationIdAlreadyExistsException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.OperationIdAlreadyExistsException$, errors_1.OperationIdAlreadyExistsException);
exports.OperationInProgressException$ = [-3, n0, _OIPE,
    { [_aQE]: [`OperationInProgressException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.OperationInProgressException$, errors_1.OperationInProgressException);
exports.OperationNotFoundException$ = [-3, n0, _ONFE,
    { [_aQE]: [`OperationNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.OperationNotFoundException$, errors_1.OperationNotFoundException);
exports.OperationStatusCheckFailedException$ = [-3, n0, _OSCFE,
    { [_aQE]: [`ConditionalCheckFailed`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.OperationStatusCheckFailedException$, errors_1.OperationStatusCheckFailedException);
exports.ResourceScanInProgressException$ = [-3, n0, _RSIPE,
    { [_aQE]: [`ResourceScanInProgress`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ResourceScanInProgressException$, errors_1.ResourceScanInProgressException);
exports.ResourceScanLimitExceededException$ = [-3, n0, _RSLEE,
    { [_aQE]: [`ResourceScanLimitExceeded`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ResourceScanLimitExceededException$, errors_1.ResourceScanLimitExceededException);
exports.ResourceScanNotFoundException$ = [-3, n0, _RSNFE,
    { [_aQE]: [`ResourceScanNotFound`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ResourceScanNotFoundException$, errors_1.ResourceScanNotFoundException);
exports.StackInstanceNotFoundException$ = [-3, n0, _SINFE,
    { [_aQE]: [`StackInstanceNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StackInstanceNotFoundException$, errors_1.StackInstanceNotFoundException);
exports.StackNotFoundException$ = [-3, n0, _SNFE,
    { [_aQE]: [`StackNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StackNotFoundException$, errors_1.StackNotFoundException);
exports.StackRefactorNotFoundException$ = [-3, n0, _SRNFE,
    { [_aQE]: [`StackRefactorNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StackRefactorNotFoundException$, errors_1.StackRefactorNotFoundException);
exports.StackSetNotEmptyException$ = [-3, n0, _SSNEE,
    { [_aQE]: [`StackSetNotEmptyException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StackSetNotEmptyException$, errors_1.StackSetNotEmptyException);
exports.StackSetNotFoundException$ = [-3, n0, _SSNFE,
    { [_aQE]: [`StackSetNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StackSetNotFoundException$, errors_1.StackSetNotFoundException);
exports.StaleRequestException$ = [-3, n0, _SRE,
    { [_aQE]: [`StaleRequestException`, 409], [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(exports.StaleRequestException$, errors_1.StaleRequestException);
exports.TokenAlreadyExistsException$ = [-3, n0, _TAEE,
    { [_aQE]: [`TokenAlreadyExistsException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.TokenAlreadyExistsException$, errors_1.TokenAlreadyExistsException);
exports.TypeConfigurationNotFoundException$ = [-3, n0, _TCNFE,
    { [_aQE]: [`TypeConfigurationNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.TypeConfigurationNotFoundException$, errors_1.TypeConfigurationNotFoundException);
exports.TypeNotFoundException$ = [-3, n0, _TNFE,
    { [_aQE]: [`TypeNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.TypeNotFoundException$, errors_1.TypeNotFoundException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
exports.AccountGateResult$ = [3, n0, _AGR,
    0,
    [_S, _SR],
    [0, 0]
];
exports.AccountLimit$ = [3, n0, _AL,
    0,
    [_N, _V],
    [0, 1]
];
exports.ActivateOrganizationsAccessInput$ = [3, n0, _AOAI,
    0,
    [],
    []
];
exports.ActivateOrganizationsAccessOutput$ = [3, n0, _AOAO,
    0,
    [],
    []
];
exports.ActivateTypeInput$ = [3, n0, _ATI,
    0,
    [_T, _PTA, _PI, _TN, _TNA, _AU, _LC, _ERA, _VB, _MV],
    [0, 0, 0, 0, 0, 2, () => exports.LoggingConfig$, 0, 0, 1]
];
exports.ActivateTypeOutput$ = [3, n0, _ATO,
    0,
    [_A],
    [0]
];
exports.Annotation$ = [3, n0, _An,
    0,
    [_AN, _S, _SM, _RM, _RL, _SL],
    [0, 0, 0, 0, 0, 0]
];
exports.AutoDeployment$ = [3, n0, _AD,
    0,
    [_E, _RSOAR, _DO],
    [2, 2, 64 | 0]
];
exports.BatchDescribeTypeConfigurationsError$ = [3, n0, _BDTCE,
    0,
    [_EC, _EM, _TCI],
    [0, 0, () => exports.TypeConfigurationIdentifier$]
];
exports.BatchDescribeTypeConfigurationsInput$ = [3, n0, _BDTCI,
    0,
    [_TCIy],
    [() => TypeConfigurationIdentifiers], 1
];
exports.BatchDescribeTypeConfigurationsOutput$ = [3, n0, _BDTCO,
    0,
    [_Er, _UTC, _TC],
    [() => BatchDescribeTypeConfigurationsErrors, () => UnprocessedTypeConfigurations, () => TypeConfigurationDetailsList]
];
exports.CancelUpdateStackInput$ = [3, n0, _CUSI,
    0,
    [_SN, _CRT],
    [0, 0], 1
];
exports.Change$ = [3, n0, _C,
    0,
    [_T, _HIC, _RC],
    [0, 1, () => exports.ResourceChange$]
];
exports.ChangeSetHook$ = [3, n0, _CSH,
    0,
    [_IP, _FM, _TN, _TVI, _TCVI, _TD],
    [0, 0, 0, 0, 0, () => exports.ChangeSetHookTargetDetails$]
];
exports.ChangeSetHookResourceTargetDetails$ = [3, n0, _CSHRTD,
    0,
    [_LRI, _RT, _RA],
    [0, 0, 0]
];
exports.ChangeSetHookTargetDetails$ = [3, n0, _CSHTD,
    0,
    [_TT, _RTD],
    [0, () => exports.ChangeSetHookResourceTargetDetails$]
];
exports.ChangeSetSummary$ = [3, n0, _CSS,
    0,
    [_SI, _SN, _CSI, _CSN, _ES, _S, _SR, _CT, _D, _INS, _PCSI, _RCSI, _IER],
    [0, 0, 0, 0, 0, 0, 0, 4, 0, 2, 0, 0, 2]
];
exports.ContinueUpdateRollbackInput$ = [3, n0, _CURI,
    0,
    [_SN, _RARN, _RTS, _CRT],
    [0, 0, 64 | 0, 0], 1
];
exports.ContinueUpdateRollbackOutput$ = [3, n0, _CURO,
    0,
    [],
    []
];
exports.CreateChangeSetInput$ = [3, n0, _CCSI,
    0,
    [_SN, _CSN, _TB, _TURL, _UPT, _P, _Ca, _RTe, _RARN, _RCo, _NARN, _Ta, _CTl, _D, _CST, _RTI, _INS, _OSF, _IER, _DM],
    [0, 0, 0, 0, 2, () => _Parameters, 64 | 0, 64 | 0, 0, () => exports.RollbackConfiguration$, 64 | 0, () => Tags, 0, 0, 0, () => ResourcesToImport, 2, 0, 2, 0], 2
];
exports.CreateChangeSetOutput$ = [3, n0, _CCSO,
    0,
    [_I, _SI],
    [0, 0]
];
exports.CreateGeneratedTemplateInput$ = [3, n0, _CGTI,
    0,
    [_GTN, _R, _SN, _TCe],
    [0, () => ResourceDefinitions, 0, () => exports.TemplateConfiguration$], 1
];
exports.CreateGeneratedTemplateOutput$ = [3, n0, _CGTO,
    0,
    [_GTI],
    [0]
];
exports.CreateStackInput$ = [3, n0, _CSIr,
    0,
    [_SN, _TB, _TURL, _P, _DR, _RCo, _TIM, _NARN, _Ca, _RTe, _RARN, _OF, _SPB, _SPURL, _Ta, _CRT, _ETP, _REOC],
    [0, 0, 0, () => _Parameters, 2, () => exports.RollbackConfiguration$, 1, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, 0, () => Tags, 0, 2, 2], 1
];
exports.CreateStackInstancesInput$ = [3, n0, _CSII,
    0,
    [_SSN, _Re, _Ac, _DT, _PO, _OP, _OI, _CA],
    [0, 64 | 0, 64 | 0, () => exports.DeploymentTargets$, () => _Parameters, () => exports.StackSetOperationPreferences$, [0, 4], 0], 2
];
exports.CreateStackInstancesOutput$ = [3, n0, _CSIO,
    0,
    [_OI],
    [0]
];
exports.CreateStackOutput$ = [3, n0, _CSO,
    0,
    [_SI, _OI],
    [0, 0]
];
exports.CreateStackRefactorInput$ = [3, n0, _CSRI,
    0,
    [_SD, _D, _ESC, _RMe],
    [() => StackDefinitions, 0, 2, () => ResourceMappings], 1
];
exports.CreateStackRefactorOutput$ = [3, n0, _CSRO,
    0,
    [_SRI],
    [0], 1
];
exports.CreateStackSetInput$ = [3, n0, _CSSI,
    0,
    [_SSN, _D, _TB, _TURL, _SI, _P, _Ca, _Ta, _ARARN, _ERN, _PM, _AD, _CA, _CRT, _ME],
    [0, 0, 0, 0, 0, () => _Parameters, 64 | 0, () => Tags, 0, 0, 0, () => exports.AutoDeployment$, 0, [0, 4], () => exports.ManagedExecution$], 1
];
exports.CreateStackSetOutput$ = [3, n0, _CSSO,
    0,
    [_SSI],
    [0]
];
exports.DeactivateOrganizationsAccessInput$ = [3, n0, _DOAI,
    0,
    [],
    []
];
exports.DeactivateOrganizationsAccessOutput$ = [3, n0, _DOAO,
    0,
    [],
    []
];
exports.DeactivateTypeInput$ = [3, n0, _DTI,
    0,
    [_TN, _T, _A],
    [0, 0, 0]
];
exports.DeactivateTypeOutput$ = [3, n0, _DTO,
    0,
    [],
    []
];
exports.DeleteChangeSetInput$ = [3, n0, _DCSI,
    0,
    [_CSN, _SN],
    [0, 0], 1
];
exports.DeleteChangeSetOutput$ = [3, n0, _DCSO,
    0,
    [],
    []
];
exports.DeleteGeneratedTemplateInput$ = [3, n0, _DGTI,
    0,
    [_GTN],
    [0], 1
];
exports.DeleteStackInput$ = [3, n0, _DSI,
    0,
    [_SN, _RR, _RARN, _CRT, _DMe],
    [0, 64 | 0, 0, 0, 0], 1
];
exports.DeleteStackInstancesInput$ = [3, n0, _DSII,
    0,
    [_SSN, _Re, _RS, _Ac, _DT, _OP, _OI, _CA],
    [0, 64 | 0, 2, 64 | 0, () => exports.DeploymentTargets$, () => exports.StackSetOperationPreferences$, [0, 4], 0], 3
];
exports.DeleteStackInstancesOutput$ = [3, n0, _DSIO,
    0,
    [_OI],
    [0]
];
exports.DeleteStackSetInput$ = [3, n0, _DSSI,
    0,
    [_SSN, _CA],
    [0, 0], 1
];
exports.DeleteStackSetOutput$ = [3, n0, _DSSO,
    0,
    [],
    []
];
exports.DeploymentTargets$ = [3, n0, _DT,
    0,
    [_Ac, _AUc, _OUI, _AFT],
    [64 | 0, 0, 64 | 0, 0]
];
exports.DeregisterTypeInput$ = [3, n0, _DTIe,
    0,
    [_A, _T, _TN, _VI],
    [0, 0, 0, 0]
];
exports.DeregisterTypeOutput$ = [3, n0, _DTOe,
    0,
    [],
    []
];
exports.DescribeAccountLimitsInput$ = [3, n0, _DALI,
    0,
    [_NT],
    [0]
];
exports.DescribeAccountLimitsOutput$ = [3, n0, _DALO,
    0,
    [_ALc, _NT],
    [() => AccountLimitList, 0]
];
exports.DescribeChangeSetHooksInput$ = [3, n0, _DCSHI,
    0,
    [_CSN, _SN, _NT, _LRI],
    [0, 0, 0, 0], 1
];
exports.DescribeChangeSetHooksOutput$ = [3, n0, _DCSHO,
    0,
    [_CSI, _CSN, _H, _S, _NT, _SI, _SN],
    [0, 0, () => ChangeSetHooks, 0, 0, 0, 0]
];
exports.DescribeChangeSetInput$ = [3, n0, _DCSIe,
    0,
    [_CSN, _SN, _NT, _IPV],
    [0, 0, 0, 2], 1
];
exports.DescribeChangeSetOutput$ = [3, n0, _DCSOe,
    0,
    [_CSN, _CSI, _SI, _SN, _D, _P, _CT, _ES, _S, _SR, _SDS, _NARN, _RCo, _Ca, _Ta, _Ch, _NT, _INS, _PCSI, _RCSI, _OSF, _IER, _DM],
    [0, 0, 0, 0, 0, () => _Parameters, 4, 0, 0, 0, 0, 64 | 0, () => exports.RollbackConfiguration$, 64 | 0, () => Tags, () => Changes, 0, 2, 0, 0, 0, 2, 0]
];
exports.DescribeEventsInput$ = [3, n0, _DEI,
    0,
    [_SN, _CSN, _OI, _F, _NT],
    [0, 0, 0, () => exports.EventFilter$, 0]
];
exports.DescribeEventsOutput$ = [3, n0, _DEO,
    0,
    [_OE, _NT],
    [() => OperationEvents, 0]
];
exports.DescribeGeneratedTemplateInput$ = [3, n0, _DGTIe,
    0,
    [_GTN],
    [0], 1
];
exports.DescribeGeneratedTemplateOutput$ = [3, n0, _DGTO,
    0,
    [_GTI, _GTN, _R, _S, _SR, _CT, _LUT, _Pr, _SI, _TCe, _TW],
    [0, 0, () => ResourceDetails, 0, 0, 4, 4, () => exports.TemplateProgress$, 0, () => exports.TemplateConfiguration$, 1]
];
exports.DescribeOrganizationsAccessInput$ = [3, n0, _DOAIe,
    0,
    [_CA],
    [0]
];
exports.DescribeOrganizationsAccessOutput$ = [3, n0, _DOAOe,
    0,
    [_S],
    [0]
];
exports.DescribePublisherInput$ = [3, n0, _DPI,
    0,
    [_PI],
    [0]
];
exports.DescribePublisherOutput$ = [3, n0, _DPO,
    0,
    [_PI, _PS, _IPd, _PP],
    [0, 0, 0, 0]
];
exports.DescribeResourceScanInput$ = [3, n0, _DRSI,
    0,
    [_RSI],
    [0], 1
];
exports.DescribeResourceScanOutput$ = [3, n0, _DRSO,
    0,
    [_RSI, _S, _SR, _ST, _ET, _PC, _RTe, _RSe, _RRe, _SF],
    [0, 0, 0, 4, 4, 1, 64 | 0, 1, 1, () => ScanFilters]
];
exports.DescribeStackDriftDetectionStatusInput$ = [3, n0, _DSDDSI,
    0,
    [_SDDI],
    [0], 1
];
exports.DescribeStackDriftDetectionStatusOutput$ = [3, n0, _DSDDSO,
    0,
    [_SI, _SDDI, _DS, _Ti, _SDS, _DSR, _DSRC],
    [0, 0, 0, 4, 0, 0, 1], 4
];
exports.DescribeStackEventsInput$ = [3, n0, _DSEI,
    0,
    [_SN, _NT],
    [0, 0], 1
];
exports.DescribeStackEventsOutput$ = [3, n0, _DSEO,
    0,
    [_SE, _NT],
    [() => StackEvents, 0]
];
exports.DescribeStackInstanceInput$ = [3, n0, _DSIIe,
    0,
    [_SSN, _SIA, _SIR, _CA],
    [0, 0, 0, 0], 3
];
exports.DescribeStackInstanceOutput$ = [3, n0, _DSIOe,
    0,
    [_SIt],
    [() => exports.StackInstance$]
];
exports.DescribeStackRefactorInput$ = [3, n0, _DSRI,
    0,
    [_SRI],
    [0], 1
];
exports.DescribeStackRefactorOutput$ = [3, n0, _DSRO,
    0,
    [_D, _SRI, _SIta, _ES, _ESR, _S, _SR],
    [0, 0, 64 | 0, 0, 0, 0, 0]
];
exports.DescribeStackResourceDriftsInput$ = [3, n0, _DSRDI,
    0,
    [_SN, _SRDSF, _NT, _MR],
    [0, 64 | 0, 0, 1], 1
];
exports.DescribeStackResourceDriftsOutput$ = [3, n0, _DSRDO,
    0,
    [_SRD, _NT],
    [() => StackResourceDrifts, 0], 1
];
exports.DescribeStackResourceInput$ = [3, n0, _DSRIe,
    0,
    [_SN, _LRI],
    [0, 0], 2
];
exports.DescribeStackResourceOutput$ = [3, n0, _DSROe,
    0,
    [_SRDt],
    [() => exports.StackResourceDetail$]
];
exports.DescribeStackResourcesInput$ = [3, n0, _DSRIes,
    0,
    [_SN, _LRI, _PRI],
    [0, 0, 0]
];
exports.DescribeStackResourcesOutput$ = [3, n0, _DSROes,
    0,
    [_SRt],
    [() => StackResources]
];
exports.DescribeStackSetInput$ = [3, n0, _DSSIe,
    0,
    [_SSN, _CA],
    [0, 0], 1
];
exports.DescribeStackSetOperationInput$ = [3, n0, _DSSOI,
    0,
    [_SSN, _OI, _CA],
    [0, 0, 0], 2
];
exports.DescribeStackSetOperationOutput$ = [3, n0, _DSSOO,
    0,
    [_SSO],
    [() => exports.StackSetOperation$]
];
exports.DescribeStackSetOutput$ = [3, n0, _DSSOe,
    0,
    [_SS],
    [() => exports.StackSet$]
];
exports.DescribeStacksInput$ = [3, n0, _DSIe,
    0,
    [_SN, _NT],
    [0, 0]
];
exports.DescribeStacksOutput$ = [3, n0, _DSO,
    0,
    [_St, _NT],
    [() => Stacks, 0]
];
exports.DescribeTypeInput$ = [3, n0, _DTIes,
    0,
    [_T, _TN, _A, _VI, _PI, _PVN],
    [0, 0, 0, 0, 0, 0]
];
exports.DescribeTypeOutput$ = [3, n0, _DTOes,
    0,
    [_A, _T, _TN, _DVI, _IDV, _TTS, _TTSD, _D, _Sc, _PT, _DSe, _LC, _RAT, _ERA, _Vi, _SU, _DU, _LU, _TCi, _CS, _PI, _OTN, _OTA, _PVN, _LPV, _IA, _AU],
    [0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, () => exports.LoggingConfig$, () => RequiredActivatedTypes, 0, 0, 0, 0, 4, 4, 0, 0, 0, 0, 0, 0, 2, 2]
];
exports.DescribeTypeRegistrationInput$ = [3, n0, _DTRI,
    0,
    [_RTeg],
    [0], 1
];
exports.DescribeTypeRegistrationOutput$ = [3, n0, _DTRO,
    0,
    [_PSr, _D, _TA, _TVA],
    [0, 0, 0, 0]
];
exports.DetectStackDriftInput$ = [3, n0, _DSDI,
    0,
    [_SN, _LRIo],
    [0, 64 | 0], 1
];
exports.DetectStackDriftOutput$ = [3, n0, _DSDO,
    0,
    [_SDDI],
    [0], 1
];
exports.DetectStackResourceDriftInput$ = [3, n0, _DSRDIe,
    0,
    [_SN, _LRI],
    [0, 0], 2
];
exports.DetectStackResourceDriftOutput$ = [3, n0, _DSRDOe,
    0,
    [_SRDta],
    [() => exports.StackResourceDrift$], 1
];
exports.DetectStackSetDriftInput$ = [3, n0, _DSSDI,
    0,
    [_SSN, _OP, _OI, _CA],
    [0, () => exports.StackSetOperationPreferences$, [0, 4], 0], 1
];
exports.DetectStackSetDriftOutput$ = [3, n0, _DSSDO,
    0,
    [_OI],
    [0]
];
exports.EstimateTemplateCostInput$ = [3, n0, _ETCI,
    0,
    [_TB, _TURL, _P],
    [0, 0, () => _Parameters]
];
exports.EstimateTemplateCostOutput$ = [3, n0, _ETCO,
    0,
    [_U],
    [0]
];
exports.EventFilter$ = [3, n0, _EF,
    0,
    [_FE],
    [2]
];
exports.ExecuteChangeSetInput$ = [3, n0, _ECSI,
    0,
    [_CSN, _SN, _CRT, _DR, _REOC],
    [0, 0, 0, 2, 2], 1
];
exports.ExecuteChangeSetOutput$ = [3, n0, _ECSO,
    0,
    [],
    []
];
exports.ExecuteStackRefactorInput$ = [3, n0, _ESRI,
    0,
    [_SRI],
    [0], 1
];
exports.Export$ = [3, n0, _Ex,
    0,
    [_ESI, _N, _V],
    [0, 0, 0]
];
exports.GetGeneratedTemplateInput$ = [3, n0, _GGTI,
    0,
    [_GTN, _Fo],
    [0, 0], 1
];
exports.GetGeneratedTemplateOutput$ = [3, n0, _GGTO,
    0,
    [_S, _TB],
    [0, 0]
];
exports.GetHookResultInput$ = [3, n0, _GHRI,
    0,
    [_HRI],
    [0]
];
exports.GetHookResultOutput$ = [3, n0, _GHRO,
    0,
    [_HRI, _IP, _FM, _TN, _OTN, _TVI, _TCVI, _TA, _S, _HSR, _IAn, _Tar, _Ann],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, () => exports.HookTarget$, () => AnnotationList]
];
exports.GetStackPolicyInput$ = [3, n0, _GSPI,
    0,
    [_SN],
    [0], 1
];
exports.GetStackPolicyOutput$ = [3, n0, _GSPO,
    0,
    [_SPB],
    [0]
];
exports.GetTemplateInput$ = [3, n0, _GTIe,
    0,
    [_SN, _CSN, _TS],
    [0, 0, 0]
];
exports.GetTemplateOutput$ = [3, n0, _GTO,
    0,
    [_TB, _SA],
    [0, 64 | 0]
];
exports.GetTemplateSummaryInput$ = [3, n0, _GTSI,
    0,
    [_TB, _TURL, _SN, _SSN, _CA, _TSC],
    [0, 0, 0, 0, 0, () => exports.TemplateSummaryConfig$]
];
exports.GetTemplateSummaryOutput$ = [3, n0, _GTSO,
    0,
    [_P, _D, _Ca, _CR, _RTe, _Ve, _Me, _DTe, _RIS, _W],
    [() => ParameterDeclarations, 0, 64 | 0, 0, 64 | 0, 0, 0, 64 | 0, () => ResourceIdentifierSummaries, () => exports.Warnings$]
];
exports.HookResultSummary$ = [3, n0, _HRS,
    0,
    [_HRI, _IP, _FM, _TN, _TVI, _TCVI, _S, _HSR, _IAn, _TT, _TI, _TA, _HET],
    [0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0]
];
exports.HookTarget$ = [3, n0, _HT,
    0,
    [_TT, _TTN, _TI, _Act],
    [0, 0, 0, 0], 4
];
exports.ImportStacksToStackSetInput$ = [3, n0, _ISTSSI,
    0,
    [_SSN, _SIta, _SIU, _OUI, _OP, _OI, _CA],
    [0, 64 | 0, 0, 64 | 0, () => exports.StackSetOperationPreferences$, [0, 4], 0], 1
];
exports.ImportStacksToStackSetOutput$ = [3, n0, _ISTSSO,
    0,
    [_OI],
    [0]
];
exports.ListChangeSetsInput$ = [3, n0, _LCSI,
    0,
    [_SN, _NT],
    [0, 0], 1
];
exports.ListChangeSetsOutput$ = [3, n0, _LCSO,
    0,
    [_Su, _NT],
    [() => ChangeSetSummaries, 0]
];
exports.ListExportsInput$ = [3, n0, _LEI,
    0,
    [_NT],
    [0]
];
exports.ListExportsOutput$ = [3, n0, _LEO,
    0,
    [_Exp, _NT],
    [() => Exports, 0]
];
exports.ListGeneratedTemplatesInput$ = [3, n0, _LGTI,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListGeneratedTemplatesOutput$ = [3, n0, _LGTO,
    0,
    [_Su, _NT],
    [() => TemplateSummaries, 0]
];
exports.ListHookResultsInput$ = [3, n0, _LHRI,
    0,
    [_TT, _TI, _TA, _S, _NT],
    [0, 0, 0, 0, 0]
];
exports.ListHookResultsOutput$ = [3, n0, _LHRO,
    0,
    [_TT, _TI, _HR, _NT],
    [0, 0, () => HookResultSummaries, 0]
];
exports.ListImportsInput$ = [3, n0, _LII,
    0,
    [_EN, _NT],
    [0, 0], 1
];
exports.ListImportsOutput$ = [3, n0, _LIO,
    0,
    [_Im, _NT],
    [64 | 0, 0]
];
exports.ListResourceScanRelatedResourcesInput$ = [3, n0, _LRSRRI,
    0,
    [_RSI, _R, _NT, _MR],
    [0, () => ScannedResourceIdentifiers, 0, 1], 2
];
exports.ListResourceScanRelatedResourcesOutput$ = [3, n0, _LRSRRO,
    0,
    [_RRel, _NT],
    [() => RelatedResources, 0]
];
exports.ListResourceScanResourcesInput$ = [3, n0, _LRSRI,
    0,
    [_RSI, _RI, _RTP, _TK, _TV, _NT, _MR],
    [0, 0, 0, 0, 0, 0, 1], 1
];
exports.ListResourceScanResourcesOutput$ = [3, n0, _LRSRO,
    0,
    [_R, _NT],
    [() => ScannedResources, 0]
];
exports.ListResourceScansInput$ = [3, n0, _LRSI,
    0,
    [_NT, _MR, _STF],
    [0, 1, 0]
];
exports.ListResourceScansOutput$ = [3, n0, _LRSO,
    0,
    [_RSS, _NT],
    [() => ResourceScanSummaries, 0]
];
exports.ListStackInstanceResourceDriftsInput$ = [3, n0, _LSIRDI,
    0,
    [_SSN, _SIA, _SIR, _OI, _NT, _MR, _SIRDS, _CA],
    [0, 0, 0, 0, 0, 1, 64 | 0, 0], 4
];
exports.ListStackInstanceResourceDriftsOutput$ = [3, n0, _LSIRDO,
    0,
    [_Su, _NT],
    [() => StackInstanceResourceDriftsSummaries, 0]
];
exports.ListStackInstancesInput$ = [3, n0, _LSII,
    0,
    [_SSN, _NT, _MR, _F, _SIA, _SIR, _CA],
    [0, 0, 1, () => StackInstanceFilters, 0, 0, 0], 1
];
exports.ListStackInstancesOutput$ = [3, n0, _LSIO,
    0,
    [_Su, _NT],
    [() => StackInstanceSummaries, 0]
];
exports.ListStackRefactorActionsInput$ = [3, n0, _LSRAI,
    0,
    [_SRI, _NT, _MR],
    [0, 0, 1], 1
];
exports.ListStackRefactorActionsOutput$ = [3, n0, _LSRAO,
    0,
    [_SRA, _NT],
    [() => StackRefactorActions, 0], 1
];
exports.ListStackRefactorsInput$ = [3, n0, _LSRI,
    0,
    [_ESF, _NT, _MR],
    [64 | 0, 0, 1]
];
exports.ListStackRefactorsOutput$ = [3, n0, _LSRO,
    0,
    [_SRS, _NT],
    [() => StackRefactorSummaries, 0], 1
];
exports.ListStackResourcesInput$ = [3, n0, _LSRIi,
    0,
    [_SN, _NT],
    [0, 0], 1
];
exports.ListStackResourcesOutput$ = [3, n0, _LSROi,
    0,
    [_SRSt, _NT],
    [() => StackResourceSummaries, 0]
];
exports.ListStackSetAutoDeploymentTargetsInput$ = [3, n0, _LSSADTI,
    0,
    [_SSN, _NT, _MR, _CA],
    [0, 0, 1, 0], 1
];
exports.ListStackSetAutoDeploymentTargetsOutput$ = [3, n0, _LSSADTO,
    0,
    [_Su, _NT],
    [() => StackSetAutoDeploymentTargetSummaries, 0]
];
exports.ListStackSetOperationResultsInput$ = [3, n0, _LSSORI,
    0,
    [_SSN, _OI, _NT, _MR, _CA, _F],
    [0, 0, 0, 1, 0, () => OperationResultFilters], 2
];
exports.ListStackSetOperationResultsOutput$ = [3, n0, _LSSORO,
    0,
    [_Su, _NT],
    [() => StackSetOperationResultSummaries, 0]
];
exports.ListStackSetOperationsInput$ = [3, n0, _LSSOI,
    0,
    [_SSN, _NT, _MR, _CA],
    [0, 0, 1, 0], 1
];
exports.ListStackSetOperationsOutput$ = [3, n0, _LSSOO,
    0,
    [_Su, _NT],
    [() => StackSetOperationSummaries, 0]
];
exports.ListStackSetsInput$ = [3, n0, _LSSI,
    0,
    [_NT, _MR, _S, _CA],
    [0, 1, 0, 0]
];
exports.ListStackSetsOutput$ = [3, n0, _LSSO,
    0,
    [_Su, _NT],
    [() => StackSetSummaries, 0]
];
exports.ListStacksInput$ = [3, n0, _LSI,
    0,
    [_NT, _SSF],
    [0, 64 | 0]
];
exports.ListStacksOutput$ = [3, n0, _LSO,
    0,
    [_SSt, _NT],
    [() => StackSummaries, 0]
];
exports.ListTypeRegistrationsInput$ = [3, n0, _LTRI,
    0,
    [_T, _TN, _TA, _RSF, _MR, _NT],
    [0, 0, 0, 0, 1, 0]
];
exports.ListTypeRegistrationsOutput$ = [3, n0, _LTRO,
    0,
    [_RTL, _NT],
    [64 | 0, 0]
];
exports.ListTypesInput$ = [3, n0, _LTI,
    0,
    [_Vi, _PT, _DSe, _T, _F, _MR, _NT],
    [0, 0, 0, 0, () => exports.TypeFilters$, 1, 0]
];
exports.ListTypesOutput$ = [3, n0, _LTO,
    0,
    [_TSy, _NT],
    [() => TypeSummaries, 0]
];
exports.ListTypeVersionsInput$ = [3, n0, _LTVI,
    0,
    [_T, _TN, _A, _MR, _NT, _DSe, _PI],
    [0, 0, 0, 1, 0, 0, 0]
];
exports.ListTypeVersionsOutput$ = [3, n0, _LTVO,
    0,
    [_TVS, _NT],
    [() => TypeVersionSummaries, 0]
];
exports.LiveResourceDrift$ = [3, n0, _LRD,
    0,
    [_PV, _AV, _DDT],
    [0, 0, 4]
];
exports.LoggingConfig$ = [3, n0, _LC,
    0,
    [_LRA, _LGN],
    [0, 0], 2
];
exports.ManagedExecution$ = [3, n0, _ME,
    0,
    [_Acti],
    [2]
];
exports.ModuleInfo$ = [3, n0, _MI,
    0,
    [_TH, _LIH],
    [0, 0]
];
exports.OperationEntry$ = [3, n0, _OEp,
    0,
    [_OT, _OI],
    [0, 0]
];
exports.OperationEvent$ = [3, n0, _OEpe,
    0,
    [_EI, _SI, _OI, _OT, _OS, _ETv, _LRI, _PRI, _RT, _Ti, _ST, _ET, _RSes, _RSR, _RP, _CRT, _HTo, _HS, _HSR, _HIP, _HFM, _DSet, _VFM, _VN, _VS, _VSR, _VP],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 4, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
];
exports.OperationResultFilter$ = [3, n0, _ORF,
    0,
    [_N, _Va],
    [0, 0]
];
exports.Output$ = [3, n0, _O,
    0,
    [_OK, _OV, _D, _EN],
    [0, 0, 0, 0]
];
exports.Parameter$ = [3, n0, _Pa,
    0,
    [_PK, _PVa, _UPV, _RV],
    [0, 0, 2, 0]
];
exports.ParameterConstraints$ = [3, n0, _PCa,
    0,
    [_AVl],
    [64 | 0]
];
exports.ParameterDeclaration$ = [3, n0, _PD,
    0,
    [_PK, _DV, _PTa, _NE, _D, _PCa],
    [0, 0, 0, 2, 0, () => exports.ParameterConstraints$]
];
exports.PhysicalResourceIdContextKeyValuePair$ = [3, n0, _PRICKVP,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.PropertyDifference$ = [3, n0, _PDr,
    0,
    [_PPr, _EV, _AV, _DTi],
    [0, 0, 0, 0], 4
];
exports.PublishTypeInput$ = [3, n0, _PTI,
    0,
    [_T, _A, _TN, _PVN],
    [0, 0, 0, 0]
];
exports.PublishTypeOutput$ = [3, n0, _PTO,
    0,
    [_PTA],
    [0]
];
exports.RecordHandlerProgressInput$ = [3, n0, _RHPI,
    0,
    [_BT, _OS, _COS, _SM, _EC, _RMes, _CRT],
    [0, 0, 0, 0, 0, 0, 0], 2
];
exports.RecordHandlerProgressOutput$ = [3, n0, _RHPO,
    0,
    [],
    []
];
exports.RegisterPublisherInput$ = [3, n0, _RPI,
    0,
    [_ATAC, _CAo],
    [2, 0]
];
exports.RegisterPublisherOutput$ = [3, n0, _RPO,
    0,
    [_PI],
    [0]
];
exports.RegisterTypeInput$ = [3, n0, _RTIe,
    0,
    [_TN, _SHP, _T, _LC, _ERA, _CRT],
    [0, 0, 0, () => exports.LoggingConfig$, 0, 0], 2
];
exports.RegisterTypeOutput$ = [3, n0, _RTO,
    0,
    [_RTeg],
    [0]
];
exports.RequiredActivatedType$ = [3, n0, _RATe,
    0,
    [_TNA, _OTN, _PI, _SMV],
    [0, 0, 0, 64 | 1]
];
exports.ResourceChange$ = [3, n0, _RC,
    0,
    [_PA, _Act, _LRI, _PRI, _RT, _Rep, _Sco, _RDS, _RDIA, _De, _CSI, _MI, _BC, _AC, _PDC],
    [0, 0, 0, 0, 0, 0, 64 | 0, 0, () => ResourceDriftIgnoredAttributes, () => ResourceChangeDetails, 0, () => exports.ModuleInfo$, 0, 0, 0]
];
exports.ResourceChangeDetail$ = [3, n0, _RCD,
    0,
    [_Tar, _Ev, _CSh, _CE],
    [() => exports.ResourceTargetDefinition$, 0, 0, 0]
];
exports.ResourceDefinition$ = [3, n0, _RD,
    0,
    [_RT, _RI, _LRI],
    [0, 128 | 0, 0], 2
];
exports.ResourceDetail$ = [3, n0, _RDe,
    0,
    [_RT, _LRI, _RI, _RSes, _RSR, _W],
    [0, 0, 128 | 0, 0, 0, () => WarningDetails]
];
exports.ResourceDriftIgnoredAttribute$ = [3, n0, _RDIAe,
    0,
    [_Pat, _Rea],
    [0, 0]
];
exports.ResourceIdentifierSummary$ = [3, n0, _RISe,
    0,
    [_RT, _LRIo, _RIe],
    [0, 64 | 0, 64 | 0]
];
exports.ResourceLocation$ = [3, n0, _RLe,
    0,
    [_SN, _LRI],
    [0, 0], 2
];
exports.ResourceMapping$ = [3, n0, _RMeso,
    0,
    [_So, _Des],
    [() => exports.ResourceLocation$, () => exports.ResourceLocation$], 2
];
exports.ResourceScanSummary$ = [3, n0, _RSSe,
    0,
    [_RSI, _S, _SR, _ST, _ET, _PC, _STc],
    [0, 0, 0, 4, 4, 1, 0]
];
exports.ResourceTargetDefinition$ = [3, n0, _RTDe,
    0,
    [_At, _N, _RReq, _Pat, _BV, _AVf, _BVF, _AVF, _Dr, _ACT],
    [0, 0, 0, 0, 0, 0, 0, 0, () => exports.LiveResourceDrift$, 0]
];
exports.ResourceToImport$ = [3, n0, _RTIes,
    0,
    [_RT, _LRI, _RI],
    [0, 0, 128 | 0], 3
];
exports.RollbackConfiguration$ = [3, n0, _RCo,
    0,
    [_RTo, _MTIM],
    [() => RollbackTriggers, 1]
];
exports.RollbackStackInput$ = [3, n0, _RSIo,
    0,
    [_SN, _RARN, _CRT, _REOC],
    [0, 0, 0, 2], 1
];
exports.RollbackStackOutput$ = [3, n0, _RSO,
    0,
    [_SI, _OI],
    [0, 0]
];
exports.RollbackTrigger$ = [3, n0, _RTol,
    0,
    [_A, _T],
    [0, 0], 2
];
exports.ScanFilter$ = [3, n0, _SFc,
    0,
    [_Ty],
    [64 | 0]
];
exports.ScannedResource$ = [3, n0, _SRc,
    0,
    [_RT, _RI, _MBS],
    [0, 128 | 0, 2]
];
exports.ScannedResourceIdentifier$ = [3, n0, _SRIc,
    0,
    [_RT, _RI],
    [0, 128 | 0], 2
];
exports.SetStackPolicyInput$ = [3, n0, _SSPI,
    0,
    [_SN, _SPB, _SPURL],
    [0, 0, 0], 1
];
exports.SetTypeConfigurationInput$ = [3, n0, _STCI,
    0,
    [_Co, _TA, _CAon, _TN, _T],
    [0, 0, 0, 0, 0], 1
];
exports.SetTypeConfigurationOutput$ = [3, n0, _STCO,
    0,
    [_CAonf],
    [0]
];
exports.SetTypeDefaultVersionInput$ = [3, n0, _STDVI,
    0,
    [_A, _T, _TN, _VI],
    [0, 0, 0, 0]
];
exports.SetTypeDefaultVersionOutput$ = [3, n0, _STDVO,
    0,
    [],
    []
];
exports.SignalResourceInput$ = [3, n0, _SRIi,
    0,
    [_SN, _LRI, _UI, _S],
    [0, 0, 0, 0], 4
];
exports.Stack$ = [3, n0, _Sta,
    0,
    [_SN, _CT, _SSta, _SI, _CSI, _D, _P, _DTel, _LUT, _RCo, _SSR, _DR, _NARN, _TIM, _Ca, _Ou, _RARN, _Ta, _ETP, _PIa, _RIo, _DI, _REOC, _DMe, _DSet, _LO],
    [0, 4, 0, 0, 0, 0, () => _Parameters, 4, 4, () => exports.RollbackConfiguration$, 0, 2, 64 | 0, 1, 64 | 0, () => Outputs, 0, () => Tags, 2, 0, 0, () => exports.StackDriftInformation$, 2, 0, 0, () => LastOperations], 3
];
exports.StackDefinition$ = [3, n0, _SDt,
    0,
    [_SN, _TB, _TURL],
    [0, 0, 0]
];
exports.StackDriftInformation$ = [3, n0, _SDI,
    0,
    [_SDS, _LCT],
    [0, 4], 1
];
exports.StackDriftInformationSummary$ = [3, n0, _SDIS,
    0,
    [_SDS, _LCT],
    [0, 4], 1
];
exports.StackEvent$ = [3, n0, _SEt,
    0,
    [_SI, _EI, _SN, _Ti, _OI, _LRI, _PRI, _RT, _RSes, _RSR, _RP, _CRT, _HTo, _HS, _HSR, _HIP, _HII, _HFM, _DSet],
    [0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 4
];
exports.StackInstance$ = [3, n0, _SIt,
    0,
    [_SSI, _Reg, _Acc, _SI, _PO, _S, _SIS, _SR, _OUIr, _DSr, _LDCT, _LOI],
    [0, 0, 0, 0, () => _Parameters, 0, () => exports.StackInstanceComprehensiveStatus$, 0, 0, 0, 4, 0]
];
exports.StackInstanceComprehensiveStatus$ = [3, n0, _SICS,
    0,
    [_DSet],
    [0]
];
exports.StackInstanceFilter$ = [3, n0, _SIF,
    0,
    [_N, _Va],
    [0, 0]
];
exports.StackInstanceResourceDriftsSummary$ = [3, n0, _SIRDSt,
    0,
    [_SI, _LRI, _RT, _SRDS, _Ti, _PRI, _PRIC, _PDro],
    [0, 0, 0, 0, 4, 0, () => PhysicalResourceIdContext, () => PropertyDifferences], 5
];
exports.StackInstanceSummary$ = [3, n0, _SISt,
    0,
    [_SSI, _Reg, _Acc, _SI, _S, _SR, _SIS, _OUIr, _DSr, _LDCT, _LOI],
    [0, 0, 0, 0, 0, 0, () => exports.StackInstanceComprehensiveStatus$, 0, 0, 4, 0]
];
exports.StackRefactorAction$ = [3, n0, _SRAt,
    0,
    [_Act, _En, _PRI, _RI, _D, _Det, _DRe, _TR, _UR, _RMeso],
    [0, 0, 0, 0, 0, 0, 0, () => StackRefactorTagResources, 64 | 0, () => exports.ResourceMapping$]
];
exports.StackRefactorSummary$ = [3, n0, _SRSta,
    0,
    [_SRI, _D, _ES, _ESR, _S, _SR],
    [0, 0, 0, 0, 0, 0]
];
exports.StackResource$ = [3, n0, _SRta,
    0,
    [_LRI, _RT, _Ti, _RSes, _SN, _SI, _PRI, _RSR, _D, _DI, _MI],
    [0, 0, 4, 0, 0, 0, 0, 0, 0, () => exports.StackResourceDriftInformation$, () => exports.ModuleInfo$], 4
];
exports.StackResourceDetail$ = [3, n0, _SRDt,
    0,
    [_LRI, _RT, _LUTa, _RSes, _SN, _SI, _PRI, _RSR, _D, _Me, _DI, _MI],
    [0, 0, 4, 0, 0, 0, 0, 0, 0, 0, () => exports.StackResourceDriftInformation$, () => exports.ModuleInfo$], 4
];
exports.StackResourceDrift$ = [3, n0, _SRDta,
    0,
    [_SI, _LRI, _RT, _SRDS, _Ti, _PRI, _PRIC, _EP, _AP, _PDro, _MI, _DSRr],
    [0, 0, 0, 0, 4, 0, () => PhysicalResourceIdContext, 0, 0, () => PropertyDifferences, () => exports.ModuleInfo$, 0], 5
];
exports.StackResourceDriftInformation$ = [3, n0, _SRDI,
    0,
    [_SRDS, _LCT],
    [0, 4], 1
];
exports.StackResourceDriftInformationSummary$ = [3, n0, _SRDIS,
    0,
    [_SRDS, _LCT],
    [0, 4], 1
];
exports.StackResourceSummary$ = [3, n0, _SRStac,
    0,
    [_LRI, _RT, _LUTa, _RSes, _PRI, _RSR, _DI, _MI],
    [0, 0, 4, 0, 0, 0, () => exports.StackResourceDriftInformationSummary$, () => exports.ModuleInfo$], 4
];
exports.StackSet$ = [3, n0, _SS,
    0,
    [_SSN, _SSI, _D, _S, _TB, _P, _Ca, _Ta, _SSARN, _ARARN, _ERN, _SSDDD, _AD, _PM, _OUI, _ME, _Re],
    [0, 0, 0, 0, 0, () => _Parameters, 64 | 0, () => Tags, 0, 0, 0, () => exports.StackSetDriftDetectionDetails$, () => exports.AutoDeployment$, 0, 64 | 0, () => exports.ManagedExecution$, 64 | 0]
];
exports.StackSetAutoDeploymentTargetSummary$ = [3, n0, _SSADTS,
    0,
    [_OUIr, _Re],
    [0, 64 | 0]
];
exports.StackSetDriftDetectionDetails$ = [3, n0, _SSDDD,
    0,
    [_DSr, _DDS, _LDCT, _TSIC, _DSIC, _ISSIC, _IPSIC, _FSIC],
    [0, 0, 4, 1, 1, 1, 1, 1]
];
exports.StackSetOperation$ = [3, n0, _SSO,
    0,
    [_OI, _SSI, _Act, _S, _OP, _RS, _ARARN, _ERN, _CTr, _ETn, _DT, _SSDDD, _SR, _SDta],
    [0, 0, 0, 0, () => exports.StackSetOperationPreferences$, 2, 0, 0, 4, 4, () => exports.DeploymentTargets$, () => exports.StackSetDriftDetectionDetails$, 0, () => exports.StackSetOperationStatusDetails$]
];
exports.StackSetOperationPreferences$ = [3, n0, _SSOP,
    0,
    [_RCT, _RO, _FTC, _FTP, _MCC, _MCP, _CM],
    [0, 64 | 0, 1, 1, 1, 1, 0]
];
exports.StackSetOperationResultSummary$ = [3, n0, _SSORS,
    0,
    [_Acc, _Reg, _S, _SR, _AGR, _OUIr],
    [0, 0, 0, 0, () => exports.AccountGateResult$, 0]
];
exports.StackSetOperationStatusDetails$ = [3, n0, _SSOSD,
    0,
    [_FSIC],
    [1]
];
exports.StackSetOperationSummary$ = [3, n0, _SSOS,
    0,
    [_OI, _Act, _S, _CTr, _ETn, _SR, _SDta, _OP],
    [0, 0, 0, 4, 4, 0, () => exports.StackSetOperationStatusDetails$, () => exports.StackSetOperationPreferences$]
];
exports.StackSetSummary$ = [3, n0, _SSS,
    0,
    [_SSN, _SSI, _D, _S, _AD, _PM, _DSr, _LDCT, _ME],
    [0, 0, 0, 0, () => exports.AutoDeployment$, 0, 0, 4, () => exports.ManagedExecution$]
];
exports.StackSummary$ = [3, n0, _SStac,
    0,
    [_SN, _CT, _SSta, _SI, _TDe, _LUT, _DTel, _SSR, _PIa, _RIo, _DI, _LO],
    [0, 4, 0, 0, 0, 4, 4, 0, 0, 0, () => exports.StackDriftInformationSummary$, () => LastOperations], 3
];
exports.StartResourceScanInput$ = [3, n0, _SRSI,
    0,
    [_CRT, _SF],
    [0, () => ScanFilters]
];
exports.StartResourceScanOutput$ = [3, n0, _SRSO,
    0,
    [_RSI],
    [0]
];
exports.StopStackSetOperationInput$ = [3, n0, _SSSOI,
    0,
    [_SSN, _OI, _CA],
    [0, 0, 0], 2
];
exports.StopStackSetOperationOutput$ = [3, n0, _SSSOO,
    0,
    [],
    []
];
exports.Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.TemplateConfiguration$ = [3, n0, _TCe,
    0,
    [_DP, _URP],
    [0, 0]
];
exports.TemplateParameter$ = [3, n0, _TP,
    0,
    [_PK, _DV, _NE, _D],
    [0, 0, 2, 0]
];
exports.TemplateProgress$ = [3, n0, _TPe,
    0,
    [_RSeso, _RF, _RPe, _RPes],
    [1, 1, 1, 1]
];
exports.TemplateSummary$ = [3, n0, _TSe,
    0,
    [_GTI, _GTN, _S, _SR, _CT, _LUT, _NOR],
    [0, 0, 0, 0, 4, 4, 1]
];
exports.TemplateSummaryConfig$ = [3, n0, _TSC,
    0,
    [_TURTAW],
    [2]
];
exports.TestTypeInput$ = [3, n0, _TTI,
    0,
    [_A, _T, _TN, _VI, _LDB],
    [0, 0, 0, 0, 0]
];
exports.TestTypeOutput$ = [3, n0, _TTO,
    0,
    [_TVA],
    [0]
];
exports.TypeConfigurationDetails$ = [3, n0, _TCD,
    0,
    [_A, _Al, _Co, _LU, _TA, _TN, _IDC],
    [0, 0, 0, 4, 0, 0, 2]
];
exports.TypeConfigurationIdentifier$ = [3, n0, _TCI,
    0,
    [_TA, _TCA, _TCAy, _T, _TN],
    [0, 0, 0, 0, 0]
];
exports.TypeFilters$ = [3, n0, _TF,
    0,
    [_Cat, _PI, _TNP],
    [0, 0, 0]
];
exports.TypeSummary$ = [3, n0, _TSyp,
    0,
    [_T, _TN, _DVI, _TA, _LU, _D, _PI, _OTN, _PVN, _LPV, _PIu, _PN, _IA],
    [0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 2]
];
exports.TypeVersionSummary$ = [3, n0, _TVSy,
    0,
    [_T, _TN, _VI, _IDV, _A, _TCi, _D, _PVN],
    [0, 0, 0, 2, 0, 4, 0, 0]
];
exports.UpdateGeneratedTemplateInput$ = [3, n0, _UGTI,
    0,
    [_GTN, _NGTN, _AR, _RRem, _RAR, _TCe],
    [0, 0, () => ResourceDefinitions, 64 | 0, 2, () => exports.TemplateConfiguration$], 1
];
exports.UpdateGeneratedTemplateOutput$ = [3, n0, _UGTO,
    0,
    [_GTI],
    [0]
];
exports.UpdateStackInput$ = [3, n0, _USI,
    0,
    [_SN, _TB, _TURL, _UPT, _SPDUB, _SPDUURL, _P, _Ca, _RTe, _RARN, _RCo, _SPB, _SPURL, _NARN, _Ta, _DR, _CRT, _REOC],
    [0, 0, 0, 2, 0, 0, () => _Parameters, 64 | 0, 64 | 0, 0, () => exports.RollbackConfiguration$, 0, 0, 64 | 0, () => Tags, 2, 0, 2], 1
];
exports.UpdateStackInstancesInput$ = [3, n0, _USII,
    0,
    [_SSN, _Re, _Ac, _DT, _PO, _OP, _OI, _CA],
    [0, 64 | 0, 64 | 0, () => exports.DeploymentTargets$, () => _Parameters, () => exports.StackSetOperationPreferences$, [0, 4], 0], 2
];
exports.UpdateStackInstancesOutput$ = [3, n0, _USIO,
    0,
    [_OI],
    [0]
];
exports.UpdateStackOutput$ = [3, n0, _USO,
    0,
    [_SI, _OI],
    [0, 0]
];
exports.UpdateStackSetInput$ = [3, n0, _USSI,
    0,
    [_SSN, _D, _TB, _TURL, _UPT, _P, _Ca, _Ta, _OP, _ARARN, _ERN, _DT, _PM, _AD, _OI, _Ac, _Re, _CA, _ME],
    [0, 0, 0, 0, 2, () => _Parameters, 64 | 0, () => Tags, () => exports.StackSetOperationPreferences$, 0, 0, () => exports.DeploymentTargets$, 0, () => exports.AutoDeployment$, [0, 4], 64 | 0, 64 | 0, 0, () => exports.ManagedExecution$], 1
];
exports.UpdateStackSetOutput$ = [3, n0, _USSO,
    0,
    [_OI],
    [0]
];
exports.UpdateTerminationProtectionInput$ = [3, n0, _UTPI,
    0,
    [_ETP, _SN],
    [2, 0], 2
];
exports.UpdateTerminationProtectionOutput$ = [3, n0, _UTPO,
    0,
    [_SI],
    [0]
];
exports.ValidateTemplateInput$ = [3, n0, _VTI,
    0,
    [_TB, _TURL],
    [0, 0]
];
exports.ValidateTemplateOutput$ = [3, n0, _VTO,
    0,
    [_P, _D, _Ca, _CR, _DTe],
    [() => TemplateParameters, 0, 64 | 0, 0, 64 | 0]
];
exports.WarningDetail$ = [3, n0, _WD,
    0,
    [_T, _Pro],
    [0, () => WarningProperties]
];
exports.WarningProperty$ = [3, n0, _WP,
    0,
    [_PPr, _Req, _D],
    [0, 2, 0]
];
exports.Warnings$ = [3, n0, _W,
    0,
    [_URT],
    [64 | 0]
];
var __Unit = "unit";
var AccountLimitList = [1, n0, _ALL,
    0, () => exports.AccountLimit$
];
var AccountList = 64 | 0;
var AllowedValues = 64 | 0;
var AnnotationList = [1, n0, _ALn,
    0, () => exports.Annotation$
];
var BatchDescribeTypeConfigurationsErrors = [1, n0, _BDTCEa,
    0, () => exports.BatchDescribeTypeConfigurationsError$
];
var Capabilities = 64 | 0;
var Changes = [1, n0, _Ch,
    0, () => exports.Change$
];
var ChangeSetHooks = [1, n0, _CSHh,
    0, () => exports.ChangeSetHook$
];
var ChangeSetSummaries = [1, n0, _CSSh,
    0, () => exports.ChangeSetSummary$
];
var Exports = [1, n0, _Exp,
    0, () => exports.Export$
];
var HookResultSummaries = [1, n0, _HRSo,
    0, () => exports.HookResultSummary$
];
var Imports = 64 | 0;
var JazzLogicalResourceIds = 64 | 0;
var LastOperations = [1, n0, _LO,
    0, () => exports.OperationEntry$
];
var LogicalResourceIds = 64 | 0;
var NotificationARNs = 64 | 0;
var OperationEvents = [1, n0, _OE,
    0, () => exports.OperationEvent$
];
var OperationResultFilters = [1, n0, _ORFp,
    0, () => exports.OperationResultFilter$
];
var OrganizationalUnitIdList = 64 | 0;
var Outputs = [1, n0, _Ou,
    0, () => exports.Output$
];
var ParameterDeclarations = [1, n0, _PDa,
    0, () => exports.ParameterDeclaration$
];
var _Parameters = [1, n0, _P,
    0, () => exports.Parameter$
];
var PhysicalResourceIdContext = [1, n0, _PRIC,
    0, () => exports.PhysicalResourceIdContextKeyValuePair$
];
var PropertyDifferences = [1, n0, _PDro,
    0, () => exports.PropertyDifference$
];
var RegionList = 64 | 0;
var RegistrationTokenList = 64 | 0;
var RelatedResources = [1, n0, _RRel,
    0, () => exports.ScannedResource$
];
var RequiredActivatedTypes = [1, n0, _RAT,
    0, () => exports.RequiredActivatedType$
];
var ResourceChangeDetails = [1, n0, _RCDe,
    0, () => exports.ResourceChangeDetail$
];
var ResourceDefinitions = [1, n0, _RDes,
    0, () => exports.ResourceDefinition$
];
var ResourceDetails = [1, n0, _RDeso,
    0, () => exports.ResourceDetail$
];
var ResourceDriftIgnoredAttributes = [1, n0, _RDIA,
    0, () => exports.ResourceDriftIgnoredAttribute$
];
var ResourceIdentifiers = 64 | 0;
var ResourceIdentifierSummaries = [1, n0, _RIS,
    0, () => exports.ResourceIdentifierSummary$
];
var ResourceMappings = [1, n0, _RMe,
    0, () => exports.ResourceMapping$
];
var ResourceScanSummaries = [1, n0, _RSS,
    0, () => exports.ResourceScanSummary$
];
var ResourcesToImport = [1, n0, _RTI,
    0, () => exports.ResourceToImport$
];
var ResourcesToSkip = 64 | 0;
var ResourceTypeFilters = 64 | 0;
var ResourceTypes = 64 | 0;
var RetainResources = 64 | 0;
var RollbackTriggers = [1, n0, _RTo,
    0, () => exports.RollbackTrigger$
];
var ScanFilters = [1, n0, _SF,
    0, () => exports.ScanFilter$
];
var ScannedResourceIdentifiers = [1, n0, _SRIca,
    0, () => exports.ScannedResourceIdentifier$
];
var ScannedResources = [1, n0, _SRca,
    0, () => exports.ScannedResource$
];
var Scope = 64 | 0;
var StackDefinitions = [1, n0, _SD,
    0, () => exports.StackDefinition$
];
var StackEvents = [1, n0, _SE,
    0, () => exports.StackEvent$
];
var StackIdList = 64 | 0;
var StackIds = 64 | 0;
var StackInstanceFilters = [1, n0, _SIFt,
    0, () => exports.StackInstanceFilter$
];
var StackInstanceResourceDriftsSummaries = [1, n0, _SIRDSta,
    0, () => exports.StackInstanceResourceDriftsSummary$
];
var StackInstanceSummaries = [1, n0, _SISta,
    0, () => exports.StackInstanceSummary$
];
var StackRefactorActions = [1, n0, _SRA,
    0, () => exports.StackRefactorAction$
];
var StackRefactorExecutionStatusFilter = 64 | 0;
var StackRefactorSummaries = [1, n0, _SRS,
    0, () => exports.StackRefactorSummary$
];
var StackRefactorTagResources = [1, n0, _SRTR,
    0, () => exports.Tag$
];
var StackRefactorUntagResources = 64 | 0;
var StackResourceDrifts = [1, n0, _SRD,
    0, () => exports.StackResourceDrift$
];
var StackResourceDriftStatusFilters = 64 | 0;
var StackResources = [1, n0, _SRt,
    0, () => exports.StackResource$
];
var StackResourceSummaries = [1, n0, _SRSt,
    0, () => exports.StackResourceSummary$
];
var Stacks = [1, n0, _St,
    0, () => exports.Stack$
];
var StackSetARNList = 64 | 0;
var StackSetAutoDeploymentTargetSummaries = [1, n0, _SSADTSt,
    0, () => exports.StackSetAutoDeploymentTargetSummary$
];
var StackSetOperationResultSummaries = [1, n0, _SSORSt,
    0, () => exports.StackSetOperationResultSummary$
];
var StackSetOperationSummaries = [1, n0, _SSOSt,
    0, () => exports.StackSetOperationSummary$
];
var StackSetSummaries = [1, n0, _SSSt,
    0, () => exports.StackSetSummary$
];
var StackStatusFilter = 64 | 0;
var StackSummaries = [1, n0, _SSt,
    0, () => exports.StackSummary$
];
var StageList = 64 | 0;
var SupportedMajorVersions = 64 | 1;
var Tags = [1, n0, _Ta,
    0, () => exports.Tag$
];
var TemplateParameters = [1, n0, _TPem,
    0, () => exports.TemplateParameter$
];
var TemplateSummaries = [1, n0, _TSem,
    0, () => exports.TemplateSummary$
];
var TransformsList = 64 | 0;
var TypeConfigurationDetailsList = [1, n0, _TCDL,
    0, () => exports.TypeConfigurationDetails$
];
var TypeConfigurationIdentifiers = [1, n0, _TCIy,
    0, () => exports.TypeConfigurationIdentifier$
];
var TypeSummaries = [1, n0, _TSy,
    0, () => exports.TypeSummary$
];
var TypeVersionSummaries = [1, n0, _TVS,
    0, () => exports.TypeVersionSummary$
];
var UnprocessedTypeConfigurations = [1, n0, _UTC,
    0, () => exports.TypeConfigurationIdentifier$
];
var WarningDetails = [1, n0, _WDa,
    0, () => exports.WarningDetail$
];
var WarningProperties = [1, n0, _WPa,
    0, () => exports.WarningProperty$
];
var JazzResourceIdentifierProperties = 128 | 0;
var ResourceIdentifierProperties = 128 | 0;
exports.ActivateOrganizationsAccess$ = [9, n0, _AOA,
    0, () => exports.ActivateOrganizationsAccessInput$, () => exports.ActivateOrganizationsAccessOutput$
];
exports.ActivateType$ = [9, n0, _AT,
    2, () => exports.ActivateTypeInput$, () => exports.ActivateTypeOutput$
];
exports.BatchDescribeTypeConfigurations$ = [9, n0, _BDTC,
    0, () => exports.BatchDescribeTypeConfigurationsInput$, () => exports.BatchDescribeTypeConfigurationsOutput$
];
exports.CancelUpdateStack$ = [9, n0, _CUS,
    0, () => exports.CancelUpdateStackInput$, () => __Unit
];
exports.ContinueUpdateRollback$ = [9, n0, _CUR,
    0, () => exports.ContinueUpdateRollbackInput$, () => exports.ContinueUpdateRollbackOutput$
];
exports.CreateChangeSet$ = [9, n0, _CCS,
    0, () => exports.CreateChangeSetInput$, () => exports.CreateChangeSetOutput$
];
exports.CreateGeneratedTemplate$ = [9, n0, _CGT,
    0, () => exports.CreateGeneratedTemplateInput$, () => exports.CreateGeneratedTemplateOutput$
];
exports.CreateStack$ = [9, n0, _CSr,
    0, () => exports.CreateStackInput$, () => exports.CreateStackOutput$
];
exports.CreateStackInstances$ = [9, n0, _CSIre,
    0, () => exports.CreateStackInstancesInput$, () => exports.CreateStackInstancesOutput$
];
exports.CreateStackRefactor$ = [9, n0, _CSR,
    0, () => exports.CreateStackRefactorInput$, () => exports.CreateStackRefactorOutput$
];
exports.CreateStackSet$ = [9, n0, _CSSr,
    0, () => exports.CreateStackSetInput$, () => exports.CreateStackSetOutput$
];
exports.DeactivateOrganizationsAccess$ = [9, n0, _DOA,
    0, () => exports.DeactivateOrganizationsAccessInput$, () => exports.DeactivateOrganizationsAccessOutput$
];
exports.DeactivateType$ = [9, n0, _DTea,
    2, () => exports.DeactivateTypeInput$, () => exports.DeactivateTypeOutput$
];
exports.DeleteChangeSet$ = [9, n0, _DCS,
    0, () => exports.DeleteChangeSetInput$, () => exports.DeleteChangeSetOutput$
];
exports.DeleteGeneratedTemplate$ = [9, n0, _DGT,
    0, () => exports.DeleteGeneratedTemplateInput$, () => __Unit
];
exports.DeleteStack$ = [9, n0, _DSel,
    0, () => exports.DeleteStackInput$, () => __Unit
];
exports.DeleteStackInstances$ = [9, n0, _DSIel,
    0, () => exports.DeleteStackInstancesInput$, () => exports.DeleteStackInstancesOutput$
];
exports.DeleteStackSet$ = [9, n0, _DSS,
    0, () => exports.DeleteStackSetInput$, () => exports.DeleteStackSetOutput$
];
exports.DeregisterType$ = [9, n0, _DTer,
    2, () => exports.DeregisterTypeInput$, () => exports.DeregisterTypeOutput$
];
exports.DescribeAccountLimits$ = [9, n0, _DAL,
    0, () => exports.DescribeAccountLimitsInput$, () => exports.DescribeAccountLimitsOutput$
];
exports.DescribeChangeSet$ = [9, n0, _DCSe,
    0, () => exports.DescribeChangeSetInput$, () => exports.DescribeChangeSetOutput$
];
exports.DescribeChangeSetHooks$ = [9, n0, _DCSH,
    0, () => exports.DescribeChangeSetHooksInput$, () => exports.DescribeChangeSetHooksOutput$
];
exports.DescribeEvents$ = [9, n0, _DE,
    0, () => exports.DescribeEventsInput$, () => exports.DescribeEventsOutput$
];
exports.DescribeGeneratedTemplate$ = [9, n0, _DGTe,
    0, () => exports.DescribeGeneratedTemplateInput$, () => exports.DescribeGeneratedTemplateOutput$
];
exports.DescribeOrganizationsAccess$ = [9, n0, _DOAe,
    0, () => exports.DescribeOrganizationsAccessInput$, () => exports.DescribeOrganizationsAccessOutput$
];
exports.DescribePublisher$ = [9, n0, _DPe,
    2, () => exports.DescribePublisherInput$, () => exports.DescribePublisherOutput$
];
exports.DescribeResourceScan$ = [9, n0, _DRS,
    0, () => exports.DescribeResourceScanInput$, () => exports.DescribeResourceScanOutput$
];
exports.DescribeStackDriftDetectionStatus$ = [9, n0, _DSDDS,
    0, () => exports.DescribeStackDriftDetectionStatusInput$, () => exports.DescribeStackDriftDetectionStatusOutput$
];
exports.DescribeStackEvents$ = [9, n0, _DSE,
    0, () => exports.DescribeStackEventsInput$, () => exports.DescribeStackEventsOutput$
];
exports.DescribeStackInstance$ = [9, n0, _DSIes,
    0, () => exports.DescribeStackInstanceInput$, () => exports.DescribeStackInstanceOutput$
];
exports.DescribeStackRefactor$ = [9, n0, _DSRe,
    0, () => exports.DescribeStackRefactorInput$, () => exports.DescribeStackRefactorOutput$
];
exports.DescribeStackResource$ = [9, n0, _DSRes,
    0, () => exports.DescribeStackResourceInput$, () => exports.DescribeStackResourceOutput$
];
exports.DescribeStackResourceDrifts$ = [9, n0, _DSRD,
    0, () => exports.DescribeStackResourceDriftsInput$, () => exports.DescribeStackResourceDriftsOutput$
];
exports.DescribeStackResources$ = [9, n0, _DSResc,
    0, () => exports.DescribeStackResourcesInput$, () => exports.DescribeStackResourcesOutput$
];
exports.DescribeStacks$ = [9, n0, _DSes,
    0, () => exports.DescribeStacksInput$, () => exports.DescribeStacksOutput$
];
exports.DescribeStackSet$ = [9, n0, _DSSe,
    0, () => exports.DescribeStackSetInput$, () => exports.DescribeStackSetOutput$
];
exports.DescribeStackSetOperation$ = [9, n0, _DSSOes,
    0, () => exports.DescribeStackSetOperationInput$, () => exports.DescribeStackSetOperationOutput$
];
exports.DescribeType$ = [9, n0, _DTes,
    2, () => exports.DescribeTypeInput$, () => exports.DescribeTypeOutput$
];
exports.DescribeTypeRegistration$ = [9, n0, _DTR,
    2, () => exports.DescribeTypeRegistrationInput$, () => exports.DescribeTypeRegistrationOutput$
];
exports.DetectStackDrift$ = [9, n0, _DSD,
    0, () => exports.DetectStackDriftInput$, () => exports.DetectStackDriftOutput$
];
exports.DetectStackResourceDrift$ = [9, n0, _DSRDe,
    0, () => exports.DetectStackResourceDriftInput$, () => exports.DetectStackResourceDriftOutput$
];
exports.DetectStackSetDrift$ = [9, n0, _DSSD,
    0, () => exports.DetectStackSetDriftInput$, () => exports.DetectStackSetDriftOutput$
];
exports.EstimateTemplateCost$ = [9, n0, _ETC,
    0, () => exports.EstimateTemplateCostInput$, () => exports.EstimateTemplateCostOutput$
];
exports.ExecuteChangeSet$ = [9, n0, _ECS,
    0, () => exports.ExecuteChangeSetInput$, () => exports.ExecuteChangeSetOutput$
];
exports.ExecuteStackRefactor$ = [9, n0, _ESRx,
    0, () => exports.ExecuteStackRefactorInput$, () => __Unit
];
exports.GetGeneratedTemplate$ = [9, n0, _GGT,
    0, () => exports.GetGeneratedTemplateInput$, () => exports.GetGeneratedTemplateOutput$
];
exports.GetHookResult$ = [9, n0, _GHR,
    0, () => exports.GetHookResultInput$, () => exports.GetHookResultOutput$
];
exports.GetStackPolicy$ = [9, n0, _GSP,
    0, () => exports.GetStackPolicyInput$, () => exports.GetStackPolicyOutput$
];
exports.GetTemplate$ = [9, n0, _GT,
    0, () => exports.GetTemplateInput$, () => exports.GetTemplateOutput$
];
exports.GetTemplateSummary$ = [9, n0, _GTS,
    0, () => exports.GetTemplateSummaryInput$, () => exports.GetTemplateSummaryOutput$
];
exports.ImportStacksToStackSet$ = [9, n0, _ISTSS,
    0, () => exports.ImportStacksToStackSetInput$, () => exports.ImportStacksToStackSetOutput$
];
exports.ListChangeSets$ = [9, n0, _LCS,
    0, () => exports.ListChangeSetsInput$, () => exports.ListChangeSetsOutput$
];
exports.ListExports$ = [9, n0, _LE,
    0, () => exports.ListExportsInput$, () => exports.ListExportsOutput$
];
exports.ListGeneratedTemplates$ = [9, n0, _LGT,
    0, () => exports.ListGeneratedTemplatesInput$, () => exports.ListGeneratedTemplatesOutput$
];
exports.ListHookResults$ = [9, n0, _LHR,
    0, () => exports.ListHookResultsInput$, () => exports.ListHookResultsOutput$
];
exports.ListImports$ = [9, n0, _LI,
    0, () => exports.ListImportsInput$, () => exports.ListImportsOutput$
];
exports.ListResourceScanRelatedResources$ = [9, n0, _LRSRR,
    0, () => exports.ListResourceScanRelatedResourcesInput$, () => exports.ListResourceScanRelatedResourcesOutput$
];
exports.ListResourceScanResources$ = [9, n0, _LRSR,
    0, () => exports.ListResourceScanResourcesInput$, () => exports.ListResourceScanResourcesOutput$
];
exports.ListResourceScans$ = [9, n0, _LRS,
    0, () => exports.ListResourceScansInput$, () => exports.ListResourceScansOutput$
];
exports.ListStackInstanceResourceDrifts$ = [9, n0, _LSIRD,
    0, () => exports.ListStackInstanceResourceDriftsInput$, () => exports.ListStackInstanceResourceDriftsOutput$
];
exports.ListStackInstances$ = [9, n0, _LSIi,
    0, () => exports.ListStackInstancesInput$, () => exports.ListStackInstancesOutput$
];
exports.ListStackRefactorActions$ = [9, n0, _LSRA,
    0, () => exports.ListStackRefactorActionsInput$, () => exports.ListStackRefactorActionsOutput$
];
exports.ListStackRefactors$ = [9, n0, _LSR,
    0, () => exports.ListStackRefactorsInput$, () => exports.ListStackRefactorsOutput$
];
exports.ListStackResources$ = [9, n0, _LSRi,
    0, () => exports.ListStackResourcesInput$, () => exports.ListStackResourcesOutput$
];
exports.ListStacks$ = [9, n0, _LS,
    0, () => exports.ListStacksInput$, () => exports.ListStacksOutput$
];
exports.ListStackSetAutoDeploymentTargets$ = [9, n0, _LSSADT,
    0, () => exports.ListStackSetAutoDeploymentTargetsInput$, () => exports.ListStackSetAutoDeploymentTargetsOutput$
];
exports.ListStackSetOperationResults$ = [9, n0, _LSSOR,
    0, () => exports.ListStackSetOperationResultsInput$, () => exports.ListStackSetOperationResultsOutput$
];
exports.ListStackSetOperations$ = [9, n0, _LSSOi,
    0, () => exports.ListStackSetOperationsInput$, () => exports.ListStackSetOperationsOutput$
];
exports.ListStackSets$ = [9, n0, _LSS,
    0, () => exports.ListStackSetsInput$, () => exports.ListStackSetsOutput$
];
exports.ListTypeRegistrations$ = [9, n0, _LTR,
    2, () => exports.ListTypeRegistrationsInput$, () => exports.ListTypeRegistrationsOutput$
];
exports.ListTypes$ = [9, n0, _LT,
    2, () => exports.ListTypesInput$, () => exports.ListTypesOutput$
];
exports.ListTypeVersions$ = [9, n0, _LTV,
    2, () => exports.ListTypeVersionsInput$, () => exports.ListTypeVersionsOutput$
];
exports.PublishType$ = [9, n0, _PTu,
    2, () => exports.PublishTypeInput$, () => exports.PublishTypeOutput$
];
exports.RecordHandlerProgress$ = [9, n0, _RHP,
    2, () => exports.RecordHandlerProgressInput$, () => exports.RecordHandlerProgressOutput$
];
exports.RegisterPublisher$ = [9, n0, _RPeg,
    2, () => exports.RegisterPublisherInput$, () => exports.RegisterPublisherOutput$
];
exports.RegisterType$ = [9, n0, _RTegi,
    2, () => exports.RegisterTypeInput$, () => exports.RegisterTypeOutput$
];
exports.RollbackStack$ = [9, n0, _RSo,
    0, () => exports.RollbackStackInput$, () => exports.RollbackStackOutput$
];
exports.SetStackPolicy$ = [9, n0, _SSP,
    0, () => exports.SetStackPolicyInput$, () => __Unit
];
exports.SetTypeConfiguration$ = [9, n0, _STC,
    0, () => exports.SetTypeConfigurationInput$, () => exports.SetTypeConfigurationOutput$
];
exports.SetTypeDefaultVersion$ = [9, n0, _STDV,
    2, () => exports.SetTypeDefaultVersionInput$, () => exports.SetTypeDefaultVersionOutput$
];
exports.SignalResource$ = [9, n0, _SRi,
    0, () => exports.SignalResourceInput$, () => __Unit
];
exports.StartResourceScan$ = [9, n0, _SRStar,
    0, () => exports.StartResourceScanInput$, () => exports.StartResourceScanOutput$
];
exports.StopStackSetOperation$ = [9, n0, _SSSO,
    0, () => exports.StopStackSetOperationInput$, () => exports.StopStackSetOperationOutput$
];
exports.TestType$ = [9, n0, _TTe,
    2, () => exports.TestTypeInput$, () => exports.TestTypeOutput$
];
exports.UpdateGeneratedTemplate$ = [9, n0, _UGT,
    0, () => exports.UpdateGeneratedTemplateInput$, () => exports.UpdateGeneratedTemplateOutput$
];
exports.UpdateStack$ = [9, n0, _US,
    0, () => exports.UpdateStackInput$, () => exports.UpdateStackOutput$
];
exports.UpdateStackInstances$ = [9, n0, _USIp,
    0, () => exports.UpdateStackInstancesInput$, () => exports.UpdateStackInstancesOutput$
];
exports.UpdateStackSet$ = [9, n0, _USS,
    0, () => exports.UpdateStackSetInput$, () => exports.UpdateStackSetOutput$
];
exports.UpdateTerminationProtection$ = [9, n0, _UTP,
    0, () => exports.UpdateTerminationProtectionInput$, () => exports.UpdateTerminationProtectionOutput$
];
exports.ValidateTemplate$ = [9, n0, _VT,
    0, () => exports.ValidateTemplateInput$, () => exports.ValidateTemplateOutput$
];
