// @aws-sdk/client-sns/dist-es/schemas/schemas_0.js
const _A = "Attributes";
const _AEE = "AuthorizationErrorException";
const _AN = "ActionName";
const _ANt = "AttributeName";
const _AOU = "AuthenticateOnUnsubscribe";
const _AP = "AddPermission";
const _API = "AddPermissionInput";
const _AV = "AttributeValue";
const _AWSAI = "AWSAccountId";
const _BEINDE = "BatchEntryIdsNotDistinctException";
const _BREE = "BatchResultErrorEntry";
const _BREEL = "BatchResultErrorEntryList";
const _BRTLE = "BatchRequestTooLongException";
const _BV = "BinaryValue";
const _C = "Code";
const _CA = "CreatedAt";
const _CAE = "ConcurrentAccessException";
const _CER = "CreateEndpointResponse";
const _CIPNIOO = "CheckIfPhoneNumberIsOptedOut";
const _CIPNIOOI = "CheckIfPhoneNumberIsOptedOutInput";
const _CIPNIOOR = "CheckIfPhoneNumberIsOptedOutResponse";
const _CPA = "CreatePlatformApplication";
const _CPAI = "CreatePlatformApplicationInput";
const _CPAR = "CreatePlatformApplicationResponse";
const _CPE = "CreatePlatformEndpoint";
const _CPEI = "CreatePlatformEndpointInput";
const _CS = "ConfirmSubscription";
const _CSI = "ConfirmSubscriptionInput";
const _CSMSSPN = "CreateSMSSandboxPhoneNumber";
const _CSMSSPNI = "CreateSMSSandboxPhoneNumberInput";
const _CSMSSPNR = "CreateSMSSandboxPhoneNumberResult";
const _CSR = "ConfirmSubscriptionResponse";
const _CT = "CreateTopic";
const _CTI = "CreateTopicInput";
const _CTR = "CreateTopicResponse";
const _CUD = "CustomUserData";
const _DE = "DeleteEndpoint";
const _DEI = "DeleteEndpointInput";
const _DPA = "DeletePlatformApplication";
const _DPAI = "DeletePlatformApplicationInput";
const _DPP = "DataProtectionPolicy";
const _DSMSSPN = "DeleteSMSSandboxPhoneNumber";
const _DSMSSPNI = "DeleteSMSSandboxPhoneNumberInput";
const _DSMSSPNR = "DeleteSMSSandboxPhoneNumberResult";
const _DT = "DataType";
const _DTI = "DeleteTopicInput";
const _DTe = "DeleteTopic";
const _E = "Endpoint";
const _EA = "EndpointArn";
const _EBRE = "EmptyBatchRequestException";
const _EDE = "EndpointDisabledException";
const _En = "Endpoints";
const _F = "Failed";
const _FPLEE = "FilterPolicyLimitExceededException";
const _GDPP = "GetDataProtectionPolicy";
const _GDPPI = "GetDataProtectionPolicyInput";
const _GDPPR = "GetDataProtectionPolicyResponse";
const _GEA = "GetEndpointAttributes";
const _GEAI = "GetEndpointAttributesInput";
const _GEAR = "GetEndpointAttributesResponse";
const _GPAA = "GetPlatformApplicationAttributes";
const _GPAAI = "GetPlatformApplicationAttributesInput";
const _GPAAR = "GetPlatformApplicationAttributesResponse";
const _GSA = "GetSubscriptionAttributes";
const _GSAI = "GetSubscriptionAttributesInput";
const _GSAR = "GetSubscriptionAttributesResponse";
const _GSMSA = "GetSMSAttributes";
const _GSMSAI = "GetSMSAttributesInput";
const _GSMSAR = "GetSMSAttributesResponse";
const _GSMSSAS = "GetSMSSandboxAccountStatus";
const _GSMSSASI = "GetSMSSandboxAccountStatusInput";
const _GSMSSASR = "GetSMSSandboxAccountStatusResult";
const _GTA = "GetTopicAttributes";
const _GTAI = "GetTopicAttributesInput";
const _GTAR = "GetTopicAttributesResponse";
const _I = "Id";
const _IBEIE = "InvalidBatchEntryIdException";
const _ICC = "Iso2CountryCode";
const _IEE = "InternalErrorException";
const _IIS = "IsInSandbox";
const _IPE = "InvalidParameterException";
const _IPVE = "InvalidParameterValueException";
const _ISE = "InvalidSecurityException";
const _ISEn = "InvalidStateException";
const _K = "Key";
const _KMSADE = "KMSAccessDeniedException";
const _KMSDE = "KMSDisabledException";
const _KMSISE = "KMSInvalidStateException";
const _KMSNFE = "KMSNotFoundException";
const _KMSOIR = "KMSOptInRequired";
const _KMSTE = "KMSThrottlingException";
const _L = "Label";
const _LC = "LanguageCode";
const _LEBPA = "ListEndpointsByPlatformApplication";
const _LEBPAI = "ListEndpointsByPlatformApplicationInput";
const _LEBPAR = "ListEndpointsByPlatformApplicationResponse";
const _LOE = "ListOfEndpoints";
const _LON = "ListOriginationNumbers";
const _LONR = "ListOriginationNumbersRequest";
const _LONRi = "ListOriginationNumbersResult";
const _LOPA = "ListOfPlatformApplications";
const _LPA = "ListPlatformApplications";
const _LPAI = "ListPlatformApplicationsInput";
const _LPAR = "ListPlatformApplicationsResponse";
const _LPNOO = "ListPhoneNumbersOptedOut";
const _LPNOOI = "ListPhoneNumbersOptedOutInput";
const _LPNOOR = "ListPhoneNumbersOptedOutResponse";
const _LS = "ListSubscriptions";
const _LSBT = "ListSubscriptionsByTopic";
const _LSBTI = "ListSubscriptionsByTopicInput";
const _LSBTR = "ListSubscriptionsByTopicResponse";
const _LSI = "ListSubscriptionsInput";
const _LSMSSPN = "ListSMSSandboxPhoneNumbers";
const _LSMSSPNI = "ListSMSSandboxPhoneNumbersInput";
const _LSMSSPNR = "ListSMSSandboxPhoneNumbersResult";
const _LSR = "ListSubscriptionsResponse";
const _LT = "ListTopics";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTI = "ListTopicsInput";
const _LTR = "ListTopicsResponse";
const _M = "Message";
const _MA = "MessageAttributes";
const _MAM = "MessageAttributeMap";
const _MAV = "MessageAttributeValue";
const _MDI = "MessageDeduplicationId";
const _MGI = "MessageGroupId";
const _MI = "MessageId";
const _MR = "MaxResults";
const _MS = "MessageStructure";
const _N = "Name";
const _NC = "NumberCapabilities";
const _NFE = "NotFoundException";
const _NT = "NextToken";
const _O = "Owner";
const _OIPN = "OptInPhoneNumber";
const _OIPNI = "OptInPhoneNumberInput";
const _OIPNR = "OptInPhoneNumberResponse";
const _OOE = "OptedOutException";
const _OTP = "OneTimePassword";
const _P = "Platform";
const _PA = "PlatformApplications";
const _PAA = "PlatformApplicationArn";
const _PADE = "PlatformApplicationDisabledException";
const _PAl = "PlatformApplication";
const _PB = "PublishBatch";
const _PBI = "PublishBatchInput";
const _PBR = "PublishBatchResponse";
const _PBRE = "PublishBatchRequestEntries";
const _PBREL = "PublishBatchRequestEntryList";
const _PBRELu = "PublishBatchResultEntryList";
const _PBREu = "PublishBatchRequestEntry";
const _PBREub = "PublishBatchResultEntry";
const _PDPP = "PutDataProtectionPolicy";
const _PDPPI = "PutDataProtectionPolicyInput";
const _PI = "PublishInput";
const _PN = "PhoneNumber";
const _PNI = "PhoneNumberInformation";
const _PNIL = "PhoneNumberInformationList";
const _PNL = "PhoneNumberList";
const _PNS = "PhoneNumberString";
const _PNh = "PhoneNumbers";
const _PR = "PublishResponse";
const _Pr = "Protocol";
const _Pu = "Publish";
const _RA = "ResourceArn";
const _RLEE = "ReplayLimitExceededException";
const _RNFE = "ResourceNotFoundException";
const _RP = "RemovePermission";
const _RPI = "RemovePermissionInput";
const _RSA = "ReturnSubscriptionArn";
const _RT = "RouteType";
const _S = "Status";
const _SA = "SubscriptionArn";
const _SEA = "SetEndpointAttributes";
const _SEAI = "SetEndpointAttributesInput";
const _SF = "SenderFault";
const _SI = "SubscribeInput";
const _SL = "SubscriptionsList";
const _SLEE = "SubscriptionLimitExceededException";
const _SMSSPN = "SMSSandboxPhoneNumber";
const _SMSSPNL = "SMSSandboxPhoneNumberList";
const _SN = "SequenceNumber";
const _SPAA = "SetPlatformApplicationAttributes";
const _SPAAI = "SetPlatformApplicationAttributesInput";
const _SR = "SubscribeResponse";
const _SSA = "SetSubscriptionAttributes";
const _SSAI = "SetSubscriptionAttributesInput";
const _SSMSA = "SetSMSAttributes";
const _SSMSAI = "SetSMSAttributesInput";
const _SSMSAR = "SetSMSAttributesResponse";
const _STA = "SetTopicAttributes";
const _STAI = "SetTopicAttributesInput";
const _STE = "StaleTagException";
const _SV = "StringValue";
const _Su = "Subscriptions";
const _Sub = "Subject";
const _Subs = "Subscription";
const _Subsc = "Subscribe";
const _Suc = "Successful";
const _T = "Token";
const _TA = "TopicArn";
const _TAa = "TargetArn";
const _TE = "ThrottledException";
const _TK = "TagKeys";
const _TL = "TagList";
const _TLEE = "TagLimitExceededException";
const _TLEEo = "TopicLimitExceededException";
const _TLo = "TopicsList";
const _TMEIBRE = "TooManyEntriesInBatchRequestException";
const _TPE = "TagPolicyException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Tag = "Tag";
const _To = "Topics";
const _Top = "Topic";
const _U = "Unsubscribe";
const _UEE = "UserErrorException";
const _UI = "UnsubscribeInput";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _V = "Value";
const _VE = "ValidationException";
const _VEe = "VerificationException";
const _VSMSSPN = "VerifySMSSandboxPhoneNumber";
const _VSMSSPNI = "VerifySMSSandboxPhoneNumberInput";
const _VSMSSPNR = "VerifySMSSandboxPhoneNumberResult";
const _a = "attributes";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _iOO = "isOptedOut";
const _m = "message";
const _nT = "nextToken";
const _pN = "phoneNumber";
const _pNh = "phoneNumbers";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.sns";
const _se = "server";
const _xN = "xmlName";
const n0 = "com.amazonaws.sns";
import { TypeRegistry } from "@smithy/core/schema";
import { AuthorizationErrorException, BatchEntryIdsNotDistinctException, BatchRequestTooLongException, ConcurrentAccessException, EmptyBatchRequestException, EndpointDisabledException, FilterPolicyLimitExceededException, InternalErrorException, InvalidBatchEntryIdException, InvalidParameterException, InvalidParameterValueException, InvalidSecurityException, InvalidStateException, KMSAccessDeniedException, KMSDisabledException, KMSInvalidStateException, KMSNotFoundException, KMSOptInRequired, KMSThrottlingException, NotFoundException, OptedOutException, PlatformApplicationDisabledException, ReplayLimitExceededException, ResourceNotFoundException, StaleTagException, SubscriptionLimitExceededException, TagLimitExceededException, TagPolicyException, ThrottledException, TooManyEntriesInBatchRequestException, TopicLimitExceededException, UserErrorException, ValidationException, VerificationException, } from "../models/errors";
import { SNSServiceException } from "../models/SNSServiceException";
const _s_registry = TypeRegistry.for(_s);
export var SNSServiceException$ = [-3, _s, "SNSServiceException", 0, [], []];
_s_registry.registerError(SNSServiceException$, SNSServiceException);
const n0_registry = TypeRegistry.for(n0);
export var AuthorizationErrorException$ = [-3, n0, _AEE,
    { [_aQE]: [`AuthorizationError`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(AuthorizationErrorException$, AuthorizationErrorException);
export var BatchEntryIdsNotDistinctException$ = [-3, n0, _BEINDE,
    { [_aQE]: [`BatchEntryIdsNotDistinct`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BatchEntryIdsNotDistinctException$, BatchEntryIdsNotDistinctException);
export var BatchRequestTooLongException$ = [-3, n0, _BRTLE,
    { [_aQE]: [`BatchRequestTooLong`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BatchRequestTooLongException$, BatchRequestTooLongException);
export var ConcurrentAccessException$ = [-3, n0, _CAE,
    { [_aQE]: [`ConcurrentAccess`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(ConcurrentAccessException$, ConcurrentAccessException);
export var EmptyBatchRequestException$ = [-3, n0, _EBRE,
    { [_aQE]: [`EmptyBatchRequest`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(EmptyBatchRequestException$, EmptyBatchRequestException);
export var EndpointDisabledException$ = [-3, n0, _EDE,
    { [_aQE]: [`EndpointDisabled`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(EndpointDisabledException$, EndpointDisabledException);
export var FilterPolicyLimitExceededException$ = [-3, n0, _FPLEE,
    { [_aQE]: [`FilterPolicyLimitExceeded`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(FilterPolicyLimitExceededException$, FilterPolicyLimitExceededException);
export var InternalErrorException$ = [-3, n0, _IEE,
    { [_aQE]: [`InternalError`, 500], [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
n0_registry.registerError(InternalErrorException$, InternalErrorException);
export var InvalidBatchEntryIdException$ = [-3, n0, _IBEIE,
    { [_aQE]: [`InvalidBatchEntryId`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidBatchEntryIdException$, InvalidBatchEntryIdException);
export var InvalidParameterException$ = [-3, n0, _IPE,
    { [_aQE]: [`InvalidParameter`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidParameterException$, InvalidParameterException);
export var InvalidParameterValueException$ = [-3, n0, _IPVE,
    { [_aQE]: [`ParameterValueInvalid`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidParameterValueException$, InvalidParameterValueException);
export var InvalidSecurityException$ = [-3, n0, _ISE,
    { [_aQE]: [`InvalidSecurity`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidSecurityException$, InvalidSecurityException);
export var InvalidStateException$ = [-3, n0, _ISEn,
    { [_aQE]: [`InvalidState`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidStateException$, InvalidStateException);
export var KMSAccessDeniedException$ = [-3, n0, _KMSADE,
    { [_aQE]: [`KMSAccessDenied`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KMSAccessDeniedException$, KMSAccessDeniedException);
export var KMSDisabledException$ = [-3, n0, _KMSDE,
    { [_aQE]: [`KMSDisabled`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KMSDisabledException$, KMSDisabledException);
export var KMSInvalidStateException$ = [-3, n0, _KMSISE,
    { [_aQE]: [`KMSInvalidState`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KMSInvalidStateException$, KMSInvalidStateException);
export var KMSNotFoundException$ = [-3, n0, _KMSNFE,
    { [_aQE]: [`KMSNotFound`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KMSNotFoundException$, KMSNotFoundException);
export var KMSOptInRequired$ = [-3, n0, _KMSOIR,
    { [_aQE]: [`KMSOptInRequired`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(KMSOptInRequired$, KMSOptInRequired);
export var KMSThrottlingException$ = [-3, n0, _KMSTE,
    { [_aQE]: [`KMSThrottling`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KMSThrottlingException$, KMSThrottlingException);
export var NotFoundException$ = [-3, n0, _NFE,
    { [_aQE]: [`NotFound`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NotFoundException$, NotFoundException);
export var OptedOutException$ = [-3, n0, _OOE,
    { [_aQE]: [`OptedOut`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(OptedOutException$, OptedOutException);
export var PlatformApplicationDisabledException$ = [-3, n0, _PADE,
    { [_aQE]: [`PlatformApplicationDisabled`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(PlatformApplicationDisabledException$, PlatformApplicationDisabledException);
export var ReplayLimitExceededException$ = [-3, n0, _RLEE,
    { [_aQE]: [`ReplayLimitExceeded`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(ReplayLimitExceededException$, ReplayLimitExceededException);
export var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_aQE]: [`ResourceNotFound`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
export var StaleTagException$ = [-3, n0, _STE,
    { [_aQE]: [`StaleTag`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(StaleTagException$, StaleTagException);
export var SubscriptionLimitExceededException$ = [-3, n0, _SLEE,
    { [_aQE]: [`SubscriptionLimitExceeded`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(SubscriptionLimitExceededException$, SubscriptionLimitExceededException);
export var TagLimitExceededException$ = [-3, n0, _TLEE,
    { [_aQE]: [`TagLimitExceeded`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TagLimitExceededException$, TagLimitExceededException);
export var TagPolicyException$ = [-3, n0, _TPE,
    { [_aQE]: [`TagPolicy`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TagPolicyException$, TagPolicyException);
export var ThrottledException$ = [-3, n0, _TE,
    { [_aQE]: [`Throttled`, 429], [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
n0_registry.registerError(ThrottledException$, ThrottledException);
export var TooManyEntriesInBatchRequestException$ = [-3, n0, _TMEIBRE,
    { [_aQE]: [`TooManyEntriesInBatchRequest`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyEntriesInBatchRequestException$, TooManyEntriesInBatchRequestException);
export var TopicLimitExceededException$ = [-3, n0, _TLEEo,
    { [_aQE]: [`TopicLimitExceeded`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(TopicLimitExceededException$, TopicLimitExceededException);
export var UserErrorException$ = [-3, n0, _UEE,
    { [_aQE]: [`UserError`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(UserErrorException$, UserErrorException);
export var ValidationException$ = [-3, n0, _VE,
    { [_aQE]: [`ValidationException`, 400], [_e]: _c, [_hE]: 400 },
    [_M],
    [0], 1
];
n0_registry.registerError(ValidationException$, ValidationException);
export var VerificationException$ = [-3, n0, _VEe,
    { [_e]: _c },
    [_M, _S],
    [0, 0], 2
];
n0_registry.registerError(VerificationException$, VerificationException);
export const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var PhoneNumber = [0, n0, _PN, 8, 0];
var PhoneNumberString = [0, n0, _PNS, 8, 0];
export var AddPermissionInput$ = [3, n0, _API,
    0,
    [_TA, _L, _AWSAI, _AN],
    [0, 0, 64 | 0, 64 | 0], 4
];
export var BatchResultErrorEntry$ = [3, n0, _BREE,
    0,
    [_I, _C, _SF, _M],
    [0, 0, 2, 0], 3
];
export var CheckIfPhoneNumberIsOptedOutInput$ = [3, n0, _CIPNIOOI,
    0,
    [_pN],
    [[() => PhoneNumber, 0]], 1
];
export var CheckIfPhoneNumberIsOptedOutResponse$ = [3, n0, _CIPNIOOR,
    0,
    [_iOO],
    [2]
];
export var ConfirmSubscriptionInput$ = [3, n0, _CSI,
    0,
    [_TA, _T, _AOU],
    [0, 0, 0], 2
];
export var ConfirmSubscriptionResponse$ = [3, n0, _CSR,
    0,
    [_SA],
    [0]
];
export var CreateEndpointResponse$ = [3, n0, _CER,
    0,
    [_EA],
    [0]
];
export var CreatePlatformApplicationInput$ = [3, n0, _CPAI,
    0,
    [_N, _P, _A],
    [0, 0, 128 | 0], 3
];
export var CreatePlatformApplicationResponse$ = [3, n0, _CPAR,
    0,
    [_PAA],
    [0]
];
export var CreatePlatformEndpointInput$ = [3, n0, _CPEI,
    0,
    [_PAA, _T, _CUD, _A],
    [0, 0, 0, 128 | 0], 2
];
export var CreateSMSSandboxPhoneNumberInput$ = [3, n0, _CSMSSPNI,
    0,
    [_PN, _LC],
    [[() => PhoneNumberString, 0], 0], 1
];
export var CreateSMSSandboxPhoneNumberResult$ = [3, n0, _CSMSSPNR,
    0,
    [],
    []
];
export var CreateTopicInput$ = [3, n0, _CTI,
    0,
    [_N, _A, _Ta, _DPP],
    [0, 128 | 0, () => TagList, 0], 1
];
export var CreateTopicResponse$ = [3, n0, _CTR,
    0,
    [_TA],
    [0]
];
export var DeleteEndpointInput$ = [3, n0, _DEI,
    0,
    [_EA],
    [0], 1
];
export var DeletePlatformApplicationInput$ = [3, n0, _DPAI,
    0,
    [_PAA],
    [0], 1
];
export var DeleteSMSSandboxPhoneNumberInput$ = [3, n0, _DSMSSPNI,
    0,
    [_PN],
    [[() => PhoneNumberString, 0]], 1
];
export var DeleteSMSSandboxPhoneNumberResult$ = [3, n0, _DSMSSPNR,
    0,
    [],
    []
];
export var DeleteTopicInput$ = [3, n0, _DTI,
    0,
    [_TA],
    [0], 1
];
export var Endpoint$ = [3, n0, _E,
    0,
    [_EA, _A],
    [0, 128 | 0]
];
export var GetDataProtectionPolicyInput$ = [3, n0, _GDPPI,
    0,
    [_RA],
    [0], 1
];
export var GetDataProtectionPolicyResponse$ = [3, n0, _GDPPR,
    0,
    [_DPP],
    [0]
];
export var GetEndpointAttributesInput$ = [3, n0, _GEAI,
    0,
    [_EA],
    [0], 1
];
export var GetEndpointAttributesResponse$ = [3, n0, _GEAR,
    0,
    [_A],
    [128 | 0]
];
export var GetPlatformApplicationAttributesInput$ = [3, n0, _GPAAI,
    0,
    [_PAA],
    [0], 1
];
export var GetPlatformApplicationAttributesResponse$ = [3, n0, _GPAAR,
    0,
    [_A],
    [128 | 0]
];
export var GetSMSAttributesInput$ = [3, n0, _GSMSAI,
    0,
    [_a],
    [64 | 0]
];
export var GetSMSAttributesResponse$ = [3, n0, _GSMSAR,
    0,
    [_a],
    [128 | 0]
];
export var GetSMSSandboxAccountStatusInput$ = [3, n0, _GSMSSASI,
    0,
    [],
    []
];
export var GetSMSSandboxAccountStatusResult$ = [3, n0, _GSMSSASR,
    0,
    [_IIS],
    [2], 1
];
export var GetSubscriptionAttributesInput$ = [3, n0, _GSAI,
    0,
    [_SA],
    [0], 1
];
export var GetSubscriptionAttributesResponse$ = [3, n0, _GSAR,
    0,
    [_A],
    [128 | 0]
];
export var GetTopicAttributesInput$ = [3, n0, _GTAI,
    0,
    [_TA],
    [0], 1
];
export var GetTopicAttributesResponse$ = [3, n0, _GTAR,
    0,
    [_A],
    [128 | 0]
];
export var ListEndpointsByPlatformApplicationInput$ = [3, n0, _LEBPAI,
    0,
    [_PAA, _NT],
    [0, 0], 1
];
export var ListEndpointsByPlatformApplicationResponse$ = [3, n0, _LEBPAR,
    0,
    [_En, _NT],
    [() => ListOfEndpoints, 0]
];
export var ListOriginationNumbersRequest$ = [3, n0, _LONR,
    0,
    [_NT, _MR],
    [0, 1]
];
export var ListOriginationNumbersResult$ = [3, n0, _LONRi,
    0,
    [_NT, _PNh],
    [0, [() => PhoneNumberInformationList, 0]]
];
export var ListPhoneNumbersOptedOutInput$ = [3, n0, _LPNOOI,
    0,
    [_nT],
    [0]
];
export var ListPhoneNumbersOptedOutResponse$ = [3, n0, _LPNOOR,
    0,
    [_pNh, _nT],
    [[() => PhoneNumberList, 0], 0]
];
export var ListPlatformApplicationsInput$ = [3, n0, _LPAI,
    0,
    [_NT],
    [0]
];
export var ListPlatformApplicationsResponse$ = [3, n0, _LPAR,
    0,
    [_PA, _NT],
    [() => ListOfPlatformApplications, 0]
];
export var ListSMSSandboxPhoneNumbersInput$ = [3, n0, _LSMSSPNI,
    0,
    [_NT, _MR],
    [0, 1]
];
export var ListSMSSandboxPhoneNumbersResult$ = [3, n0, _LSMSSPNR,
    0,
    [_PNh, _NT],
    [[() => SMSSandboxPhoneNumberList, 0], 0], 1
];
export var ListSubscriptionsByTopicInput$ = [3, n0, _LSBTI,
    0,
    [_TA, _NT],
    [0, 0], 1
];
export var ListSubscriptionsByTopicResponse$ = [3, n0, _LSBTR,
    0,
    [_Su, _NT],
    [() => SubscriptionsList, 0]
];
export var ListSubscriptionsInput$ = [3, n0, _LSI,
    0,
    [_NT],
    [0]
];
export var ListSubscriptionsResponse$ = [3, n0, _LSR,
    0,
    [_Su, _NT],
    [() => SubscriptionsList, 0]
];
export var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RA],
    [0], 1
];
export var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [() => TagList]
];
export var ListTopicsInput$ = [3, n0, _LTI,
    0,
    [_NT],
    [0]
];
export var ListTopicsResponse$ = [3, n0, _LTR,
    0,
    [_To, _NT],
    [() => TopicsList, 0]
];
export var MessageAttributeValue$ = [3, n0, _MAV,
    0,
    [_DT, _SV, _BV],
    [0, 0, 21], 1
];
export var OptInPhoneNumberInput$ = [3, n0, _OIPNI,
    0,
    [_pN],
    [[() => PhoneNumber, 0]], 1
];
export var OptInPhoneNumberResponse$ = [3, n0, _OIPNR,
    0,
    [],
    []
];
export var PhoneNumberInformation$ = [3, n0, _PNI,
    0,
    [_CA, _PN, _S, _ICC, _RT, _NC],
    [4, [() => PhoneNumber, 0], 0, 0, 0, 64 | 0]
];
export var PlatformApplication$ = [3, n0, _PAl,
    0,
    [_PAA, _A],
    [0, 128 | 0]
];
export var PublishBatchInput$ = [3, n0, _PBI,
    0,
    [_TA, _PBRE],
    [0, [() => PublishBatchRequestEntryList, 0]], 2
];
export var PublishBatchRequestEntry$ = [3, n0, _PBREu,
    0,
    [_I, _M, _Sub, _MS, _MA, _MDI, _MGI],
    [0, 0, 0, 0, [() => MessageAttributeMap, 0], 0, 0], 2
];
export var PublishBatchResponse$ = [3, n0, _PBR,
    0,
    [_Suc, _F],
    [() => PublishBatchResultEntryList, () => BatchResultErrorEntryList]
];
export var PublishBatchResultEntry$ = [3, n0, _PBREub,
    0,
    [_I, _MI, _SN],
    [0, 0, 0]
];
export var PublishInput$ = [3, n0, _PI,
    0,
    [_M, _TA, _TAa, _PN, _Sub, _MS, _MA, _MDI, _MGI],
    [0, 0, 0, [() => PhoneNumber, 0], 0, 0, [() => MessageAttributeMap, 0], 0, 0], 1
];
export var PublishResponse$ = [3, n0, _PR,
    0,
    [_MI, _SN],
    [0, 0]
];
export var PutDataProtectionPolicyInput$ = [3, n0, _PDPPI,
    0,
    [_RA, _DPP],
    [0, 0], 2
];
export var RemovePermissionInput$ = [3, n0, _RPI,
    0,
    [_TA, _L],
    [0, 0], 2
];
export var SetEndpointAttributesInput$ = [3, n0, _SEAI,
    0,
    [_EA, _A],
    [0, 128 | 0], 2
];
export var SetPlatformApplicationAttributesInput$ = [3, n0, _SPAAI,
    0,
    [_PAA, _A],
    [0, 128 | 0], 2
];
export var SetSMSAttributesInput$ = [3, n0, _SSMSAI,
    0,
    [_a],
    [128 | 0], 1
];
export var SetSMSAttributesResponse$ = [3, n0, _SSMSAR,
    0,
    [],
    []
];
export var SetSubscriptionAttributesInput$ = [3, n0, _SSAI,
    0,
    [_SA, _ANt, _AV],
    [0, 0, 0], 2
];
export var SetTopicAttributesInput$ = [3, n0, _STAI,
    0,
    [_TA, _ANt, _AV],
    [0, 0, 0], 2
];
export var SMSSandboxPhoneNumber$ = [3, n0, _SMSSPN,
    0,
    [_PN, _S],
    [[() => PhoneNumberString, 0], 0]
];
export var SubscribeInput$ = [3, n0, _SI,
    0,
    [_TA, _Pr, _E, _A, _RSA],
    [0, 0, 0, 128 | 0, 2], 2
];
export var SubscribeResponse$ = [3, n0, _SR,
    0,
    [_SA],
    [0]
];
export var Subscription$ = [3, n0, _Subs,
    0,
    [_SA, _O, _Pr, _E, _TA],
    [0, 0, 0, 0, 0]
];
export var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
export var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RA, _Ta],
    [0, () => TagList], 2
];
export var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
export var Topic$ = [3, n0, _Top,
    0,
    [_TA],
    [0]
];
export var UnsubscribeInput$ = [3, n0, _UI,
    0,
    [_SA],
    [0], 1
];
export var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RA, _TK],
    [0, 64 | 0], 2
];
export var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
export var VerifySMSSandboxPhoneNumberInput$ = [3, n0, _VSMSSPNI,
    0,
    [_PN, _OTP],
    [[() => PhoneNumberString, 0], 0], 2
];
export var VerifySMSSandboxPhoneNumberResult$ = [3, n0, _VSMSSPNR,
    0,
    [],
    []
];
var __Unit = "unit";
var ActionsList = 64 | 0;
var BatchResultErrorEntryList = [1, n0, _BREEL,
    0, () => BatchResultErrorEntry$
];
var DelegatesList = 64 | 0;
var ListOfEndpoints = [1, n0, _LOE,
    0, () => Endpoint$
];
var ListOfPlatformApplications = [1, n0, _LOPA,
    0, () => PlatformApplication$
];
var ListString = 64 | 0;
var NumberCapabilityList = 64 | 0;
var PhoneNumberInformationList = [1, n0, _PNIL,
    0, [() => PhoneNumberInformation$,
        0]
];
var PhoneNumberList = [1, n0, _PNL,
    0, [() => PhoneNumber,
        0]
];
var PublishBatchRequestEntryList = [1, n0, _PBREL,
    0, [() => PublishBatchRequestEntry$,
        0]
];
var PublishBatchResultEntryList = [1, n0, _PBRELu,
    0, () => PublishBatchResultEntry$
];
var SMSSandboxPhoneNumberList = [1, n0, _SMSSPNL,
    0, [() => SMSSandboxPhoneNumber$,
        0]
];
var SubscriptionsList = [1, n0, _SL,
    0, () => Subscription$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TopicsList = [1, n0, _TLo,
    0, () => Topic$
];
var MapStringToString = 128 | 0;
var MessageAttributeMap = [2, n0, _MAM,
    0, [0,
        { [_xN]: _N }],
    [() => MessageAttributeValue$,
        { [_xN]: _V }]
];
var SubscriptionAttributesMap = 128 | 0;
var TopicAttributesMap = 128 | 0;
export var AddPermission$ = [9, n0, _AP,
    0, () => AddPermissionInput$, () => __Unit
];
export var CheckIfPhoneNumberIsOptedOut$ = [9, n0, _CIPNIOO,
    0, () => CheckIfPhoneNumberIsOptedOutInput$, () => CheckIfPhoneNumberIsOptedOutResponse$
];
export var ConfirmSubscription$ = [9, n0, _CS,
    0, () => ConfirmSubscriptionInput$, () => ConfirmSubscriptionResponse$
];
export var CreatePlatformApplication$ = [9, n0, _CPA,
    0, () => CreatePlatformApplicationInput$, () => CreatePlatformApplicationResponse$
];
export var CreatePlatformEndpoint$ = [9, n0, _CPE,
    0, () => CreatePlatformEndpointInput$, () => CreateEndpointResponse$
];
export var CreateSMSSandboxPhoneNumber$ = [9, n0, _CSMSSPN,
    0, () => CreateSMSSandboxPhoneNumberInput$, () => CreateSMSSandboxPhoneNumberResult$
];
export var CreateTopic$ = [9, n0, _CT,
    0, () => CreateTopicInput$, () => CreateTopicResponse$
];
export var DeleteEndpoint$ = [9, n0, _DE,
    0, () => DeleteEndpointInput$, () => __Unit
];
export var DeletePlatformApplication$ = [9, n0, _DPA,
    0, () => DeletePlatformApplicationInput$, () => __Unit
];
export var DeleteSMSSandboxPhoneNumber$ = [9, n0, _DSMSSPN,
    0, () => DeleteSMSSandboxPhoneNumberInput$, () => DeleteSMSSandboxPhoneNumberResult$
];
export var DeleteTopic$ = [9, n0, _DTe,
    0, () => DeleteTopicInput$, () => __Unit
];
export var GetDataProtectionPolicy$ = [9, n0, _GDPP,
    0, () => GetDataProtectionPolicyInput$, () => GetDataProtectionPolicyResponse$
];
export var GetEndpointAttributes$ = [9, n0, _GEA,
    0, () => GetEndpointAttributesInput$, () => GetEndpointAttributesResponse$
];
export var GetPlatformApplicationAttributes$ = [9, n0, _GPAA,
    0, () => GetPlatformApplicationAttributesInput$, () => GetPlatformApplicationAttributesResponse$
];
export var GetSMSAttributes$ = [9, n0, _GSMSA,
    0, () => GetSMSAttributesInput$, () => GetSMSAttributesResponse$
];
export var GetSMSSandboxAccountStatus$ = [9, n0, _GSMSSAS,
    0, () => GetSMSSandboxAccountStatusInput$, () => GetSMSSandboxAccountStatusResult$
];
export var GetSubscriptionAttributes$ = [9, n0, _GSA,
    0, () => GetSubscriptionAttributesInput$, () => GetSubscriptionAttributesResponse$
];
export var GetTopicAttributes$ = [9, n0, _GTA,
    0, () => GetTopicAttributesInput$, () => GetTopicAttributesResponse$
];
export var ListEndpointsByPlatformApplication$ = [9, n0, _LEBPA,
    0, () => ListEndpointsByPlatformApplicationInput$, () => ListEndpointsByPlatformApplicationResponse$
];
export var ListOriginationNumbers$ = [9, n0, _LON,
    0, () => ListOriginationNumbersRequest$, () => ListOriginationNumbersResult$
];
export var ListPhoneNumbersOptedOut$ = [9, n0, _LPNOO,
    0, () => ListPhoneNumbersOptedOutInput$, () => ListPhoneNumbersOptedOutResponse$
];
export var ListPlatformApplications$ = [9, n0, _LPA,
    0, () => ListPlatformApplicationsInput$, () => ListPlatformApplicationsResponse$
];
export var ListSMSSandboxPhoneNumbers$ = [9, n0, _LSMSSPN,
    0, () => ListSMSSandboxPhoneNumbersInput$, () => ListSMSSandboxPhoneNumbersResult$
];
export var ListSubscriptions$ = [9, n0, _LS,
    0, () => ListSubscriptionsInput$, () => ListSubscriptionsResponse$
];
export var ListSubscriptionsByTopic$ = [9, n0, _LSBT,
    0, () => ListSubscriptionsByTopicInput$, () => ListSubscriptionsByTopicResponse$
];
export var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
export var ListTopics$ = [9, n0, _LT,
    0, () => ListTopicsInput$, () => ListTopicsResponse$
];
export var OptInPhoneNumber$ = [9, n0, _OIPN,
    0, () => OptInPhoneNumberInput$, () => OptInPhoneNumberResponse$
];
export var Publish$ = [9, n0, _Pu,
    0, () => PublishInput$, () => PublishResponse$
];
export var PublishBatch$ = [9, n0, _PB,
    0, () => PublishBatchInput$, () => PublishBatchResponse$
];
export var PutDataProtectionPolicy$ = [9, n0, _PDPP,
    0, () => PutDataProtectionPolicyInput$, () => __Unit
];
export var RemovePermission$ = [9, n0, _RP,
    0, () => RemovePermissionInput$, () => __Unit
];
export var SetEndpointAttributes$ = [9, n0, _SEA,
    0, () => SetEndpointAttributesInput$, () => __Unit
];
export var SetPlatformApplicationAttributes$ = [9, n0, _SPAA,
    0, () => SetPlatformApplicationAttributesInput$, () => __Unit
];
export var SetSMSAttributes$ = [9, n0, _SSMSA,
    0, () => SetSMSAttributesInput$, () => SetSMSAttributesResponse$
];
export var SetSubscriptionAttributes$ = [9, n0, _SSA,
    0, () => SetSubscriptionAttributesInput$, () => __Unit
];
export var SetTopicAttributes$ = [9, n0, _STA,
    0, () => SetTopicAttributesInput$, () => __Unit
];
export var Subscribe$ = [9, n0, _Subsc,
    0, () => SubscribeInput$, () => SubscribeResponse$
];
export var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
export var Unsubscribe$ = [9, n0, _U,
    0, () => UnsubscribeInput$, () => __Unit
];
export var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
export var VerifySMSSandboxPhoneNumber$ = [9, n0, _VSMSSPN,
    0, () => VerifySMSSandboxPhoneNumberInput$, () => VerifySMSSandboxPhoneNumberResult$
];
