// @aws-sdk/client-apigatewaymanagementapi/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostToConnection$ = exports.GetConnection$ = exports.DeleteConnection$ = exports.PostToConnectionRequest$ = exports.Identity$ = exports.GetConnectionResponse$ = exports.GetConnectionRequest$ = exports.DeleteConnectionRequest$ = exports.errorTypeRegistries = exports.PayloadTooLargeException$ = exports.LimitExceededException$ = exports.GoneException$ = exports.ForbiddenException$ = exports.ApiGatewayManagementApiServiceException$ = void 0;
const _CA = "ConnectedAt";
const _CI = "ConnectionId";
const _D = "Data";
const _DC = "DeleteConnection";
const _DCR = "DeleteConnectionRequest";
const _FE = "ForbiddenException";
const _GC = "GetConnection";
const _GCR = "GetConnectionRequest";
const _GCRe = "GetConnectionResponse";
const _GE = "GoneException";
const _I = "Identity";
const _LAA = "LastActiveAt";
const _LEE = "LimitExceededException";
const _M = "Message";
const _PTC = "PostToConnection";
const _PTCR = "PostToConnectionRequest";
const _PTLE = "PayloadTooLargeException";
const _SI = "SourceIp";
const _UA = "UserAgent";
const _c = "client";
const _cA = "connectedAt";
const _e = "error";
const _h = "http";
const _hE = "httpError";
const _i = "identity";
const _jN = "jsonName";
const _lAA = "lastActiveAt";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.apigatewaymanagementapi";
const _sI = "sourceIp";
const _uA = "userAgent";
const n0 = "com.amazonaws.apigatewaymanagementapi";
const schema_1 = require("@smithy/core/schema");
const ApiGatewayManagementApiServiceException_1 = require("../models/ApiGatewayManagementApiServiceException");
const errors_1 = require("../models/errors");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.ApiGatewayManagementApiServiceException$ = [-3, _s, "ApiGatewayManagementApiServiceException", 0, [], []];
_s_registry.registerError(exports.ApiGatewayManagementApiServiceException$, ApiGatewayManagementApiServiceException_1.ApiGatewayManagementApiServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.ForbiddenException$ = [-3, n0, _FE,
    { [_e]: _c, [_hE]: 403 },
    [],
    []
];
n0_registry.registerError(exports.ForbiddenException$, errors_1.ForbiddenException);
exports.GoneException$ = [-3, n0, _GE,
    { [_e]: _c, [_hE]: 410 },
    [],
    []
];
n0_registry.registerError(exports.GoneException$, errors_1.GoneException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 429 },
    [],
    []
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.PayloadTooLargeException$ = [-3, n0, _PTLE,
    { [_e]: _c, [_hE]: 413 },
    [_M],
    [[0, { [_jN]: _m }]]
];
n0_registry.registerError(exports.PayloadTooLargeException$, errors_1.PayloadTooLargeException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
exports.DeleteConnectionRequest$ = [3, n0, _DCR,
    0,
    [_CI],
    [[0, 1]], 1
];
exports.GetConnectionRequest$ = [3, n0, _GCR,
    0,
    [_CI],
    [[0, 1]], 1
];
exports.GetConnectionResponse$ = [3, n0, _GCRe,
    0,
    [_CA, _I, _LAA],
    [[5, { [_jN]: _cA }], [() => exports.Identity$, { [_jN]: _i }], [5, { [_jN]: _lAA }]]
];
exports.Identity$ = [3, n0, _I,
    0,
    [_SI, _UA],
    [[0, { [_jN]: _sI }], [0, { [_jN]: _uA }]], 2
];
exports.PostToConnectionRequest$ = [3, n0, _PTCR,
    0,
    [_D, _CI],
    [[21, 16], [0, 1]], 2
];
var __Unit = "unit";
exports.DeleteConnection$ = [9, n0, _DC,
    { [_h]: ["DELETE", "/@connections/{ConnectionId}", 204] }, () => exports.DeleteConnectionRequest$, () => __Unit
];
exports.GetConnection$ = [9, n0, _GC,
    { [_h]: ["GET", "/@connections/{ConnectionId}", 200] }, () => exports.GetConnectionRequest$, () => exports.GetConnectionResponse$
];
exports.PostToConnection$ = [9, n0, _PTC,
    { [_h]: ["POST", "/@connections/{ConnectionId}", 200] }, () => exports.PostToConnectionRequest$, () => __Unit
];
