// @aws-sdk/client-api-gateway/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var middlewareSdkApiGateway = require('@aws-sdk/middleware-sdk-api-gateway');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var APIGatewayServiceException = require('./models/APIGatewayServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "apigateway",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class APIGatewayClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(middlewareSdkApiGateway.getAcceptHeaderPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultAPIGatewayHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class CreateApiKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateApiKey", {})
    .n("APIGatewayClient", "CreateApiKeyCommand")
    .sc(schemas_0.CreateApiKey$)
    .build() {
}

class CreateAuthorizerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateAuthorizer", {})
    .n("APIGatewayClient", "CreateAuthorizerCommand")
    .sc(schemas_0.CreateAuthorizer$)
    .build() {
}

class CreateBasePathMappingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateBasePathMapping", {})
    .n("APIGatewayClient", "CreateBasePathMappingCommand")
    .sc(schemas_0.CreateBasePathMapping$)
    .build() {
}

class CreateDeploymentCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDeployment", {})
    .n("APIGatewayClient", "CreateDeploymentCommand")
    .sc(schemas_0.CreateDeployment$)
    .build() {
}

class CreateDocumentationPartCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDocumentationPart", {})
    .n("APIGatewayClient", "CreateDocumentationPartCommand")
    .sc(schemas_0.CreateDocumentationPart$)
    .build() {
}

class CreateDocumentationVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDocumentationVersion", {})
    .n("APIGatewayClient", "CreateDocumentationVersionCommand")
    .sc(schemas_0.CreateDocumentationVersion$)
    .build() {
}

class CreateDomainNameAccessAssociationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "CreateDomainNameAccessAssociationCommand")
    .sc(schemas_0.CreateDomainNameAccessAssociation$)
    .build() {
}

class CreateDomainNameCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDomainName", {})
    .n("APIGatewayClient", "CreateDomainNameCommand")
    .sc(schemas_0.CreateDomainName$)
    .build() {
}

class CreateModelCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateModel", {})
    .n("APIGatewayClient", "CreateModelCommand")
    .sc(schemas_0.CreateModel$)
    .build() {
}

class CreateRequestValidatorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateRequestValidator", {})
    .n("APIGatewayClient", "CreateRequestValidatorCommand")
    .sc(schemas_0.CreateRequestValidator$)
    .build() {
}

class CreateResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateResource", {})
    .n("APIGatewayClient", "CreateResourceCommand")
    .sc(schemas_0.CreateResource$)
    .build() {
}

class CreateRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateRestApi", {})
    .n("APIGatewayClient", "CreateRestApiCommand")
    .sc(schemas_0.CreateRestApi$)
    .build() {
}

class CreateStageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateStage", {})
    .n("APIGatewayClient", "CreateStageCommand")
    .sc(schemas_0.CreateStage$)
    .build() {
}

class CreateUsagePlanCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateUsagePlan", {})
    .n("APIGatewayClient", "CreateUsagePlanCommand")
    .sc(schemas_0.CreateUsagePlan$)
    .build() {
}

class CreateUsagePlanKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateUsagePlanKey", {})
    .n("APIGatewayClient", "CreateUsagePlanKeyCommand")
    .sc(schemas_0.CreateUsagePlanKey$)
    .build() {
}

class CreateVpcLinkCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateVpcLink", {})
    .n("APIGatewayClient", "CreateVpcLinkCommand")
    .sc(schemas_0.CreateVpcLink$)
    .build() {
}

class DeleteApiKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteApiKey", {})
    .n("APIGatewayClient", "DeleteApiKeyCommand")
    .sc(schemas_0.DeleteApiKey$)
    .build() {
}

class DeleteAuthorizerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteAuthorizer", {})
    .n("APIGatewayClient", "DeleteAuthorizerCommand")
    .sc(schemas_0.DeleteAuthorizer$)
    .build() {
}

class DeleteBasePathMappingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteBasePathMapping", {})
    .n("APIGatewayClient", "DeleteBasePathMappingCommand")
    .sc(schemas_0.DeleteBasePathMapping$)
    .build() {
}

class DeleteClientCertificateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteClientCertificate", {})
    .n("APIGatewayClient", "DeleteClientCertificateCommand")
    .sc(schemas_0.DeleteClientCertificate$)
    .build() {
}

class DeleteDeploymentCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDeployment", {})
    .n("APIGatewayClient", "DeleteDeploymentCommand")
    .sc(schemas_0.DeleteDeployment$)
    .build() {
}

class DeleteDocumentationPartCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDocumentationPart", {})
    .n("APIGatewayClient", "DeleteDocumentationPartCommand")
    .sc(schemas_0.DeleteDocumentationPart$)
    .build() {
}

class DeleteDocumentationVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDocumentationVersion", {})
    .n("APIGatewayClient", "DeleteDocumentationVersionCommand")
    .sc(schemas_0.DeleteDocumentationVersion$)
    .build() {
}

class DeleteDomainNameAccessAssociationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "DeleteDomainNameAccessAssociationCommand")
    .sc(schemas_0.DeleteDomainNameAccessAssociation$)
    .build() {
}

class DeleteDomainNameCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDomainName", {})
    .n("APIGatewayClient", "DeleteDomainNameCommand")
    .sc(schemas_0.DeleteDomainName$)
    .build() {
}

class DeleteGatewayResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteGatewayResponse", {})
    .n("APIGatewayClient", "DeleteGatewayResponseCommand")
    .sc(schemas_0.DeleteGatewayResponse$)
    .build() {
}

class DeleteIntegrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteIntegration", {})
    .n("APIGatewayClient", "DeleteIntegrationCommand")
    .sc(schemas_0.DeleteIntegration$)
    .build() {
}

class DeleteIntegrationResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteIntegrationResponse", {})
    .n("APIGatewayClient", "DeleteIntegrationResponseCommand")
    .sc(schemas_0.DeleteIntegrationResponse$)
    .build() {
}

class DeleteMethodCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteMethod", {})
    .n("APIGatewayClient", "DeleteMethodCommand")
    .sc(schemas_0.DeleteMethod$)
    .build() {
}

class DeleteMethodResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteMethodResponse", {})
    .n("APIGatewayClient", "DeleteMethodResponseCommand")
    .sc(schemas_0.DeleteMethodResponse$)
    .build() {
}

class DeleteModelCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteModel", {})
    .n("APIGatewayClient", "DeleteModelCommand")
    .sc(schemas_0.DeleteModel$)
    .build() {
}

class DeleteRequestValidatorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteRequestValidator", {})
    .n("APIGatewayClient", "DeleteRequestValidatorCommand")
    .sc(schemas_0.DeleteRequestValidator$)
    .build() {
}

class DeleteResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteResource", {})
    .n("APIGatewayClient", "DeleteResourceCommand")
    .sc(schemas_0.DeleteResource$)
    .build() {
}

class DeleteRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteRestApi", {})
    .n("APIGatewayClient", "DeleteRestApiCommand")
    .sc(schemas_0.DeleteRestApi$)
    .build() {
}

class DeleteStageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteStage", {})
    .n("APIGatewayClient", "DeleteStageCommand")
    .sc(schemas_0.DeleteStage$)
    .build() {
}

class DeleteUsagePlanCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteUsagePlan", {})
    .n("APIGatewayClient", "DeleteUsagePlanCommand")
    .sc(schemas_0.DeleteUsagePlan$)
    .build() {
}

class DeleteUsagePlanKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteUsagePlanKey", {})
    .n("APIGatewayClient", "DeleteUsagePlanKeyCommand")
    .sc(schemas_0.DeleteUsagePlanKey$)
    .build() {
}

class DeleteVpcLinkCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteVpcLink", {})
    .n("APIGatewayClient", "DeleteVpcLinkCommand")
    .sc(schemas_0.DeleteVpcLink$)
    .build() {
}

class FlushStageAuthorizersCacheCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "FlushStageAuthorizersCache", {})
    .n("APIGatewayClient", "FlushStageAuthorizersCacheCommand")
    .sc(schemas_0.FlushStageAuthorizersCache$)
    .build() {
}

class FlushStageCacheCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "FlushStageCache", {})
    .n("APIGatewayClient", "FlushStageCacheCommand")
    .sc(schemas_0.FlushStageCache$)
    .build() {
}

class GenerateClientCertificateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GenerateClientCertificate", {})
    .n("APIGatewayClient", "GenerateClientCertificateCommand")
    .sc(schemas_0.GenerateClientCertificate$)
    .build() {
}

class GetAccountCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAccount", {})
    .n("APIGatewayClient", "GetAccountCommand")
    .sc(schemas_0.GetAccount$)
    .build() {
}

class GetApiKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetApiKey", {})
    .n("APIGatewayClient", "GetApiKeyCommand")
    .sc(schemas_0.GetApiKey$)
    .build() {
}

class GetApiKeysCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetApiKeys", {})
    .n("APIGatewayClient", "GetApiKeysCommand")
    .sc(schemas_0.GetApiKeys$)
    .build() {
}

class GetAuthorizerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAuthorizer", {})
    .n("APIGatewayClient", "GetAuthorizerCommand")
    .sc(schemas_0.GetAuthorizer$)
    .build() {
}

class GetAuthorizersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAuthorizers", {})
    .n("APIGatewayClient", "GetAuthorizersCommand")
    .sc(schemas_0.GetAuthorizers$)
    .build() {
}

class GetBasePathMappingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetBasePathMapping", {})
    .n("APIGatewayClient", "GetBasePathMappingCommand")
    .sc(schemas_0.GetBasePathMapping$)
    .build() {
}

class GetBasePathMappingsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetBasePathMappings", {})
    .n("APIGatewayClient", "GetBasePathMappingsCommand")
    .sc(schemas_0.GetBasePathMappings$)
    .build() {
}

class GetClientCertificateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetClientCertificate", {})
    .n("APIGatewayClient", "GetClientCertificateCommand")
    .sc(schemas_0.GetClientCertificate$)
    .build() {
}

class GetClientCertificatesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetClientCertificates", {})
    .n("APIGatewayClient", "GetClientCertificatesCommand")
    .sc(schemas_0.GetClientCertificates$)
    .build() {
}

class GetDeploymentCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDeployment", {})
    .n("APIGatewayClient", "GetDeploymentCommand")
    .sc(schemas_0.GetDeployment$)
    .build() {
}

class GetDeploymentsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDeployments", {})
    .n("APIGatewayClient", "GetDeploymentsCommand")
    .sc(schemas_0.GetDeployments$)
    .build() {
}

class GetDocumentationPartCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationPart", {})
    .n("APIGatewayClient", "GetDocumentationPartCommand")
    .sc(schemas_0.GetDocumentationPart$)
    .build() {
}

class GetDocumentationPartsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationParts", {})
    .n("APIGatewayClient", "GetDocumentationPartsCommand")
    .sc(schemas_0.GetDocumentationParts$)
    .build() {
}

class GetDocumentationVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationVersion", {})
    .n("APIGatewayClient", "GetDocumentationVersionCommand")
    .sc(schemas_0.GetDocumentationVersion$)
    .build() {
}

class GetDocumentationVersionsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationVersions", {})
    .n("APIGatewayClient", "GetDocumentationVersionsCommand")
    .sc(schemas_0.GetDocumentationVersions$)
    .build() {
}

class GetDomainNameAccessAssociationsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainNameAccessAssociations", {})
    .n("APIGatewayClient", "GetDomainNameAccessAssociationsCommand")
    .sc(schemas_0.GetDomainNameAccessAssociations$)
    .build() {
}

class GetDomainNameCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainName", {})
    .n("APIGatewayClient", "GetDomainNameCommand")
    .sc(schemas_0.GetDomainName$)
    .build() {
}

class GetDomainNamesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainNames", {})
    .n("APIGatewayClient", "GetDomainNamesCommand")
    .sc(schemas_0.GetDomainNames$)
    .build() {
}

class GetExportCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetExport", {})
    .n("APIGatewayClient", "GetExportCommand")
    .sc(schemas_0.GetExport$)
    .build() {
}

class GetGatewayResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetGatewayResponse", {})
    .n("APIGatewayClient", "GetGatewayResponseCommand")
    .sc(schemas_0.GetGatewayResponse$)
    .build() {
}

class GetGatewayResponsesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetGatewayResponses", {})
    .n("APIGatewayClient", "GetGatewayResponsesCommand")
    .sc(schemas_0.GetGatewayResponses$)
    .build() {
}

class GetIntegrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetIntegration", {})
    .n("APIGatewayClient", "GetIntegrationCommand")
    .sc(schemas_0.GetIntegration$)
    .build() {
}

class GetIntegrationResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetIntegrationResponse", {})
    .n("APIGatewayClient", "GetIntegrationResponseCommand")
    .sc(schemas_0.GetIntegrationResponse$)
    .build() {
}

class GetMethodCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetMethod", {})
    .n("APIGatewayClient", "GetMethodCommand")
    .sc(schemas_0.GetMethod$)
    .build() {
}

class GetMethodResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetMethodResponse", {})
    .n("APIGatewayClient", "GetMethodResponseCommand")
    .sc(schemas_0.GetMethodResponse$)
    .build() {
}

class GetModelCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModel", {})
    .n("APIGatewayClient", "GetModelCommand")
    .sc(schemas_0.GetModel$)
    .build() {
}

class GetModelsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModels", {})
    .n("APIGatewayClient", "GetModelsCommand")
    .sc(schemas_0.GetModels$)
    .build() {
}

class GetModelTemplateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModelTemplate", {})
    .n("APIGatewayClient", "GetModelTemplateCommand")
    .sc(schemas_0.GetModelTemplate$)
    .build() {
}

class GetRequestValidatorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRequestValidator", {})
    .n("APIGatewayClient", "GetRequestValidatorCommand")
    .sc(schemas_0.GetRequestValidator$)
    .build() {
}

class GetRequestValidatorsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRequestValidators", {})
    .n("APIGatewayClient", "GetRequestValidatorsCommand")
    .sc(schemas_0.GetRequestValidators$)
    .build() {
}

class GetResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetResource", {})
    .n("APIGatewayClient", "GetResourceCommand")
    .sc(schemas_0.GetResource$)
    .build() {
}

class GetResourcesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetResources", {})
    .n("APIGatewayClient", "GetResourcesCommand")
    .sc(schemas_0.GetResources$)
    .build() {
}

class GetRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRestApi", {})
    .n("APIGatewayClient", "GetRestApiCommand")
    .sc(schemas_0.GetRestApi$)
    .build() {
}

class GetRestApisCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRestApis", {})
    .n("APIGatewayClient", "GetRestApisCommand")
    .sc(schemas_0.GetRestApis$)
    .build() {
}

class GetSdkCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdk", {})
    .n("APIGatewayClient", "GetSdkCommand")
    .sc(schemas_0.GetSdk$)
    .build() {
}

class GetSdkTypeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdkType", {})
    .n("APIGatewayClient", "GetSdkTypeCommand")
    .sc(schemas_0.GetSdkType$)
    .build() {
}

class GetSdkTypesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdkTypes", {})
    .n("APIGatewayClient", "GetSdkTypesCommand")
    .sc(schemas_0.GetSdkTypes$)
    .build() {
}

class GetStageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetStage", {})
    .n("APIGatewayClient", "GetStageCommand")
    .sc(schemas_0.GetStage$)
    .build() {
}

class GetStagesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetStages", {})
    .n("APIGatewayClient", "GetStagesCommand")
    .sc(schemas_0.GetStages$)
    .build() {
}

class GetTagsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetTags", {})
    .n("APIGatewayClient", "GetTagsCommand")
    .sc(schemas_0.GetTags$)
    .build() {
}

class GetUsageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsage", {})
    .n("APIGatewayClient", "GetUsageCommand")
    .sc(schemas_0.GetUsage$)
    .build() {
}

class GetUsagePlanCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlan", {})
    .n("APIGatewayClient", "GetUsagePlanCommand")
    .sc(schemas_0.GetUsagePlan$)
    .build() {
}

class GetUsagePlanKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlanKey", {})
    .n("APIGatewayClient", "GetUsagePlanKeyCommand")
    .sc(schemas_0.GetUsagePlanKey$)
    .build() {
}

class GetUsagePlanKeysCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlanKeys", {})
    .n("APIGatewayClient", "GetUsagePlanKeysCommand")
    .sc(schemas_0.GetUsagePlanKeys$)
    .build() {
}

class GetUsagePlansCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlans", {})
    .n("APIGatewayClient", "GetUsagePlansCommand")
    .sc(schemas_0.GetUsagePlans$)
    .build() {
}

class GetVpcLinkCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetVpcLink", {})
    .n("APIGatewayClient", "GetVpcLinkCommand")
    .sc(schemas_0.GetVpcLink$)
    .build() {
}

class GetVpcLinksCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetVpcLinks", {})
    .n("APIGatewayClient", "GetVpcLinksCommand")
    .sc(schemas_0.GetVpcLinks$)
    .build() {
}

class ImportApiKeysCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportApiKeys", {})
    .n("APIGatewayClient", "ImportApiKeysCommand")
    .sc(schemas_0.ImportApiKeys$)
    .build() {
}

class ImportDocumentationPartsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportDocumentationParts", {})
    .n("APIGatewayClient", "ImportDocumentationPartsCommand")
    .sc(schemas_0.ImportDocumentationParts$)
    .build() {
}

class ImportRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportRestApi", {})
    .n("APIGatewayClient", "ImportRestApiCommand")
    .sc(schemas_0.ImportRestApi$)
    .build() {
}

class PutGatewayResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutGatewayResponse", {})
    .n("APIGatewayClient", "PutGatewayResponseCommand")
    .sc(schemas_0.PutGatewayResponse$)
    .build() {
}

class PutIntegrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutIntegration", {})
    .n("APIGatewayClient", "PutIntegrationCommand")
    .sc(schemas_0.PutIntegration$)
    .build() {
}

class PutIntegrationResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutIntegrationResponse", {})
    .n("APIGatewayClient", "PutIntegrationResponseCommand")
    .sc(schemas_0.PutIntegrationResponse$)
    .build() {
}

class PutMethodCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutMethod", {})
    .n("APIGatewayClient", "PutMethodCommand")
    .sc(schemas_0.PutMethod$)
    .build() {
}

class PutMethodResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutMethodResponse", {})
    .n("APIGatewayClient", "PutMethodResponseCommand")
    .sc(schemas_0.PutMethodResponse$)
    .build() {
}

class PutRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutRestApi", {})
    .n("APIGatewayClient", "PutRestApiCommand")
    .sc(schemas_0.PutRestApi$)
    .build() {
}

class RejectDomainNameAccessAssociationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "RejectDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "RejectDomainNameAccessAssociationCommand")
    .sc(schemas_0.RejectDomainNameAccessAssociation$)
    .build() {
}

class TagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TagResource", {})
    .n("APIGatewayClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class TestInvokeAuthorizerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TestInvokeAuthorizer", {})
    .n("APIGatewayClient", "TestInvokeAuthorizerCommand")
    .sc(schemas_0.TestInvokeAuthorizer$)
    .build() {
}

class TestInvokeMethodCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TestInvokeMethod", {})
    .n("APIGatewayClient", "TestInvokeMethodCommand")
    .sc(schemas_0.TestInvokeMethod$)
    .build() {
}

class UntagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UntagResource", {})
    .n("APIGatewayClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateAccountCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateAccount", {})
    .n("APIGatewayClient", "UpdateAccountCommand")
    .sc(schemas_0.UpdateAccount$)
    .build() {
}

class UpdateApiKeyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateApiKey", {})
    .n("APIGatewayClient", "UpdateApiKeyCommand")
    .sc(schemas_0.UpdateApiKey$)
    .build() {
}

class UpdateAuthorizerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateAuthorizer", {})
    .n("APIGatewayClient", "UpdateAuthorizerCommand")
    .sc(schemas_0.UpdateAuthorizer$)
    .build() {
}

class UpdateBasePathMappingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateBasePathMapping", {})
    .n("APIGatewayClient", "UpdateBasePathMappingCommand")
    .sc(schemas_0.UpdateBasePathMapping$)
    .build() {
}

class UpdateClientCertificateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateClientCertificate", {})
    .n("APIGatewayClient", "UpdateClientCertificateCommand")
    .sc(schemas_0.UpdateClientCertificate$)
    .build() {
}

class UpdateDeploymentCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDeployment", {})
    .n("APIGatewayClient", "UpdateDeploymentCommand")
    .sc(schemas_0.UpdateDeployment$)
    .build() {
}

class UpdateDocumentationPartCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDocumentationPart", {})
    .n("APIGatewayClient", "UpdateDocumentationPartCommand")
    .sc(schemas_0.UpdateDocumentationPart$)
    .build() {
}

class UpdateDocumentationVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDocumentationVersion", {})
    .n("APIGatewayClient", "UpdateDocumentationVersionCommand")
    .sc(schemas_0.UpdateDocumentationVersion$)
    .build() {
}

class UpdateDomainNameCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDomainName", {})
    .n("APIGatewayClient", "UpdateDomainNameCommand")
    .sc(schemas_0.UpdateDomainName$)
    .build() {
}

class UpdateGatewayResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateGatewayResponse", {})
    .n("APIGatewayClient", "UpdateGatewayResponseCommand")
    .sc(schemas_0.UpdateGatewayResponse$)
    .build() {
}

class UpdateIntegrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateIntegration", {})
    .n("APIGatewayClient", "UpdateIntegrationCommand")
    .sc(schemas_0.UpdateIntegration$)
    .build() {
}

class UpdateIntegrationResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateIntegrationResponse", {})
    .n("APIGatewayClient", "UpdateIntegrationResponseCommand")
    .sc(schemas_0.UpdateIntegrationResponse$)
    .build() {
}

class UpdateMethodCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateMethod", {})
    .n("APIGatewayClient", "UpdateMethodCommand")
    .sc(schemas_0.UpdateMethod$)
    .build() {
}

class UpdateMethodResponseCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateMethodResponse", {})
    .n("APIGatewayClient", "UpdateMethodResponseCommand")
    .sc(schemas_0.UpdateMethodResponse$)
    .build() {
}

class UpdateModelCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateModel", {})
    .n("APIGatewayClient", "UpdateModelCommand")
    .sc(schemas_0.UpdateModel$)
    .build() {
}

class UpdateRequestValidatorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateRequestValidator", {})
    .n("APIGatewayClient", "UpdateRequestValidatorCommand")
    .sc(schemas_0.UpdateRequestValidator$)
    .build() {
}

class UpdateResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateResource", {})
    .n("APIGatewayClient", "UpdateResourceCommand")
    .sc(schemas_0.UpdateResource$)
    .build() {
}

class UpdateRestApiCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateRestApi", {})
    .n("APIGatewayClient", "UpdateRestApiCommand")
    .sc(schemas_0.UpdateRestApi$)
    .build() {
}

class UpdateStageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateStage", {})
    .n("APIGatewayClient", "UpdateStageCommand")
    .sc(schemas_0.UpdateStage$)
    .build() {
}

class UpdateUsageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateUsage", {})
    .n("APIGatewayClient", "UpdateUsageCommand")
    .sc(schemas_0.UpdateUsage$)
    .build() {
}

class UpdateUsagePlanCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateUsagePlan", {})
    .n("APIGatewayClient", "UpdateUsagePlanCommand")
    .sc(schemas_0.UpdateUsagePlan$)
    .build() {
}

class UpdateVpcLinkCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateVpcLink", {})
    .n("APIGatewayClient", "UpdateVpcLinkCommand")
    .sc(schemas_0.UpdateVpcLink$)
    .build() {
}

const paginateGetApiKeys = core.createPaginator(APIGatewayClient, GetApiKeysCommand, "position", "position", "limit");

const paginateGetBasePathMappings = core.createPaginator(APIGatewayClient, GetBasePathMappingsCommand, "position", "position", "limit");

const paginateGetClientCertificates = core.createPaginator(APIGatewayClient, GetClientCertificatesCommand, "position", "position", "limit");

const paginateGetDeployments = core.createPaginator(APIGatewayClient, GetDeploymentsCommand, "position", "position", "limit");

const paginateGetDomainNames = core.createPaginator(APIGatewayClient, GetDomainNamesCommand, "position", "position", "limit");

const paginateGetModels = core.createPaginator(APIGatewayClient, GetModelsCommand, "position", "position", "limit");

const paginateGetResources = core.createPaginator(APIGatewayClient, GetResourcesCommand, "position", "position", "limit");

const paginateGetRestApis = core.createPaginator(APIGatewayClient, GetRestApisCommand, "position", "position", "limit");

const paginateGetUsage = core.createPaginator(APIGatewayClient, GetUsageCommand, "position", "position", "limit");

const paginateGetUsagePlanKeys = core.createPaginator(APIGatewayClient, GetUsagePlanKeysCommand, "position", "position", "limit");

const paginateGetUsagePlans = core.createPaginator(APIGatewayClient, GetUsagePlansCommand, "position", "position", "limit");

const paginateGetVpcLinks = core.createPaginator(APIGatewayClient, GetVpcLinksCommand, "position", "position", "limit");

const commands = {
    CreateApiKeyCommand,
    CreateAuthorizerCommand,
    CreateBasePathMappingCommand,
    CreateDeploymentCommand,
    CreateDocumentationPartCommand,
    CreateDocumentationVersionCommand,
    CreateDomainNameCommand,
    CreateDomainNameAccessAssociationCommand,
    CreateModelCommand,
    CreateRequestValidatorCommand,
    CreateResourceCommand,
    CreateRestApiCommand,
    CreateStageCommand,
    CreateUsagePlanCommand,
    CreateUsagePlanKeyCommand,
    CreateVpcLinkCommand,
    DeleteApiKeyCommand,
    DeleteAuthorizerCommand,
    DeleteBasePathMappingCommand,
    DeleteClientCertificateCommand,
    DeleteDeploymentCommand,
    DeleteDocumentationPartCommand,
    DeleteDocumentationVersionCommand,
    DeleteDomainNameCommand,
    DeleteDomainNameAccessAssociationCommand,
    DeleteGatewayResponseCommand,
    DeleteIntegrationCommand,
    DeleteIntegrationResponseCommand,
    DeleteMethodCommand,
    DeleteMethodResponseCommand,
    DeleteModelCommand,
    DeleteRequestValidatorCommand,
    DeleteResourceCommand,
    DeleteRestApiCommand,
    DeleteStageCommand,
    DeleteUsagePlanCommand,
    DeleteUsagePlanKeyCommand,
    DeleteVpcLinkCommand,
    FlushStageAuthorizersCacheCommand,
    FlushStageCacheCommand,
    GenerateClientCertificateCommand,
    GetAccountCommand,
    GetApiKeyCommand,
    GetApiKeysCommand,
    GetAuthorizerCommand,
    GetAuthorizersCommand,
    GetBasePathMappingCommand,
    GetBasePathMappingsCommand,
    GetClientCertificateCommand,
    GetClientCertificatesCommand,
    GetDeploymentCommand,
    GetDeploymentsCommand,
    GetDocumentationPartCommand,
    GetDocumentationPartsCommand,
    GetDocumentationVersionCommand,
    GetDocumentationVersionsCommand,
    GetDomainNameCommand,
    GetDomainNameAccessAssociationsCommand,
    GetDomainNamesCommand,
    GetExportCommand,
    GetGatewayResponseCommand,
    GetGatewayResponsesCommand,
    GetIntegrationCommand,
    GetIntegrationResponseCommand,
    GetMethodCommand,
    GetMethodResponseCommand,
    GetModelCommand,
    GetModelsCommand,
    GetModelTemplateCommand,
    GetRequestValidatorCommand,
    GetRequestValidatorsCommand,
    GetResourceCommand,
    GetResourcesCommand,
    GetRestApiCommand,
    GetRestApisCommand,
    GetSdkCommand,
    GetSdkTypeCommand,
    GetSdkTypesCommand,
    GetStageCommand,
    GetStagesCommand,
    GetTagsCommand,
    GetUsageCommand,
    GetUsagePlanCommand,
    GetUsagePlanKeyCommand,
    GetUsagePlanKeysCommand,
    GetUsagePlansCommand,
    GetVpcLinkCommand,
    GetVpcLinksCommand,
    ImportApiKeysCommand,
    ImportDocumentationPartsCommand,
    ImportRestApiCommand,
    PutGatewayResponseCommand,
    PutIntegrationCommand,
    PutIntegrationResponseCommand,
    PutMethodCommand,
    PutMethodResponseCommand,
    PutRestApiCommand,
    RejectDomainNameAccessAssociationCommand,
    TagResourceCommand,
    TestInvokeAuthorizerCommand,
    TestInvokeMethodCommand,
    UntagResourceCommand,
    UpdateAccountCommand,
    UpdateApiKeyCommand,
    UpdateAuthorizerCommand,
    UpdateBasePathMappingCommand,
    UpdateClientCertificateCommand,
    UpdateDeploymentCommand,
    UpdateDocumentationPartCommand,
    UpdateDocumentationVersionCommand,
    UpdateDomainNameCommand,
    UpdateGatewayResponseCommand,
    UpdateIntegrationCommand,
    UpdateIntegrationResponseCommand,
    UpdateMethodCommand,
    UpdateMethodResponseCommand,
    UpdateModelCommand,
    UpdateRequestValidatorCommand,
    UpdateResourceCommand,
    UpdateRestApiCommand,
    UpdateStageCommand,
    UpdateUsageCommand,
    UpdateUsagePlanCommand,
    UpdateVpcLinkCommand,
};
const paginators = {
    paginateGetApiKeys,
    paginateGetBasePathMappings,
    paginateGetClientCertificates,
    paginateGetDeployments,
    paginateGetDomainNames,
    paginateGetModels,
    paginateGetResources,
    paginateGetRestApis,
    paginateGetUsage,
    paginateGetUsagePlanKeys,
    paginateGetUsagePlans,
    paginateGetVpcLinks,
};
class APIGateway extends APIGatewayClient {
}
client.createAggregatedClient(commands, APIGateway, { paginators });

const AccessAssociationSourceType = {
    VPCE: "VPCE",
};
const ApiKeysFormat = {
    csv: "csv",
};
const ApiKeySourceType = {
    AUTHORIZER: "AUTHORIZER",
    HEADER: "HEADER",
};
const ApiStatus = {
    AVAILABLE: "AVAILABLE",
    FAILED: "FAILED",
    PENDING: "PENDING",
    UPDATING: "UPDATING",
};
const AuthorizerType = {
    COGNITO_USER_POOLS: "COGNITO_USER_POOLS",
    REQUEST: "REQUEST",
    TOKEN: "TOKEN",
};
const CacheClusterSize = {
    SIZE_0_POINT_5_GB: "0.5",
    SIZE_118_GB: "118",
    SIZE_13_POINT_5_GB: "13.5",
    SIZE_1_POINT_6_GB: "1.6",
    SIZE_237_GB: "237",
    SIZE_28_POINT_4_GB: "28.4",
    SIZE_58_POINT_2_GB: "58.2",
    SIZE_6_POINT_1_GB: "6.1",
};
const DocumentationPartType = {
    API: "API",
    AUTHORIZER: "AUTHORIZER",
    METHOD: "METHOD",
    MODEL: "MODEL",
    PATH_PARAMETER: "PATH_PARAMETER",
    QUERY_PARAMETER: "QUERY_PARAMETER",
    REQUEST_BODY: "REQUEST_BODY",
    REQUEST_HEADER: "REQUEST_HEADER",
    RESOURCE: "RESOURCE",
    RESPONSE: "RESPONSE",
    RESPONSE_BODY: "RESPONSE_BODY",
    RESPONSE_HEADER: "RESPONSE_HEADER",
};
const EndpointAccessMode = {
    BASIC: "BASIC",
    STRICT: "STRICT",
};
const IpAddressType = {
    dualstack: "dualstack",
    ipv4: "ipv4",
};
const EndpointType = {
    EDGE: "EDGE",
    PRIVATE: "PRIVATE",
    REGIONAL: "REGIONAL",
};
const RoutingMode = {
    BASE_PATH_MAPPING_ONLY: "BASE_PATH_MAPPING_ONLY",
    ROUTING_RULE_ONLY: "ROUTING_RULE_ONLY",
    ROUTING_RULE_THEN_BASE_PATH_MAPPING: "ROUTING_RULE_THEN_BASE_PATH_MAPPING",
};
const SecurityPolicy = {
    SecurityPolicy_TLS12_2018_EDGE: "SecurityPolicy_TLS12_2018_EDGE",
    SecurityPolicy_TLS12_PFS_2025_EDGE: "SecurityPolicy_TLS12_PFS_2025_EDGE",
    SecurityPolicy_TLS13_1_2_2021_06: "SecurityPolicy_TLS13_1_2_2021_06",
    SecurityPolicy_TLS13_1_2_FIPS_PFS_PQ_2025_09: "SecurityPolicy_TLS13_1_2_FIPS_PFS_PQ_2025_09",
    SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09: "SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09",
    SecurityPolicy_TLS13_1_2_PFS_PQ_2025_09: "SecurityPolicy_TLS13_1_2_PFS_PQ_2025_09",
    SecurityPolicy_TLS13_1_2_PQ_2025_09: "SecurityPolicy_TLS13_1_2_PQ_2025_09",
    SecurityPolicy_TLS13_1_3_2025_09: "SecurityPolicy_TLS13_1_3_2025_09",
    SecurityPolicy_TLS13_1_3_FIPS_2025_09: "SecurityPolicy_TLS13_1_3_FIPS_2025_09",
    SecurityPolicy_TLS13_2025_EDGE: "SecurityPolicy_TLS13_2025_EDGE",
    TLS_1_0: "TLS_1_0",
    TLS_1_2: "TLS_1_2",
};
const DomainNameStatus = {
    AVAILABLE: "AVAILABLE",
    FAILED: "FAILED",
    PENDING: "PENDING",
    PENDING_CERTIFICATE_REIMPORT: "PENDING_CERTIFICATE_REIMPORT",
    PENDING_OWNERSHIP_VERIFICATION: "PENDING_OWNERSHIP_VERIFICATION",
    UPDATING: "UPDATING",
};
const ConnectionType = {
    INTERNET: "INTERNET",
    VPC_LINK: "VPC_LINK",
};
const ContentHandlingStrategy = {
    CONVERT_TO_BINARY: "CONVERT_TO_BINARY",
    CONVERT_TO_TEXT: "CONVERT_TO_TEXT",
};
const ResponseTransferMode = {
    BUFFERED: "BUFFERED",
    STREAM: "STREAM",
};
const IntegrationType = {
    AWS: "AWS",
    AWS_PROXY: "AWS_PROXY",
    HTTP: "HTTP",
    HTTP_PROXY: "HTTP_PROXY",
    MOCK: "MOCK",
};
const CacheClusterStatus = {
    AVAILABLE: "AVAILABLE",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    FLUSH_IN_PROGRESS: "FLUSH_IN_PROGRESS",
    NOT_AVAILABLE: "NOT_AVAILABLE",
};
const UnauthorizedCacheControlHeaderStrategy = {
    FAIL_WITH_403: "FAIL_WITH_403",
    SUCCEED_WITHOUT_RESPONSE_HEADER: "SUCCEED_WITHOUT_RESPONSE_HEADER",
    SUCCEED_WITH_RESPONSE_HEADER: "SUCCEED_WITH_RESPONSE_HEADER",
};
const QuotaPeriodType = {
    DAY: "DAY",
    MONTH: "MONTH",
    WEEK: "WEEK",
};
const VpcLinkStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
    FAILED: "FAILED",
    PENDING: "PENDING",
};
const GatewayResponseType = {
    ACCESS_DENIED: "ACCESS_DENIED",
    API_CONFIGURATION_ERROR: "API_CONFIGURATION_ERROR",
    AUTHORIZER_CONFIGURATION_ERROR: "AUTHORIZER_CONFIGURATION_ERROR",
    AUTHORIZER_FAILURE: "AUTHORIZER_FAILURE",
    BAD_REQUEST_BODY: "BAD_REQUEST_BODY",
    BAD_REQUEST_PARAMETERS: "BAD_REQUEST_PARAMETERS",
    DEFAULT_4XX: "DEFAULT_4XX",
    DEFAULT_5XX: "DEFAULT_5XX",
    EXPIRED_TOKEN: "EXPIRED_TOKEN",
    INTEGRATION_FAILURE: "INTEGRATION_FAILURE",
    INTEGRATION_TIMEOUT: "INTEGRATION_TIMEOUT",
    INVALID_API_KEY: "INVALID_API_KEY",
    INVALID_SIGNATURE: "INVALID_SIGNATURE",
    MISSING_AUTHENTICATION_TOKEN: "MISSING_AUTHENTICATION_TOKEN",
    QUOTA_EXCEEDED: "QUOTA_EXCEEDED",
    REQUEST_TOO_LARGE: "REQUEST_TOO_LARGE",
    RESOURCE_NOT_FOUND: "RESOURCE_NOT_FOUND",
    THROTTLED: "THROTTLED",
    UNAUTHORIZED: "UNAUTHORIZED",
    UNSUPPORTED_MEDIA_TYPE: "UNSUPPORTED_MEDIA_TYPE",
    WAF_FILTERED: "WAF_FILTERED",
};
const LocationStatusType = {
    DOCUMENTED: "DOCUMENTED",
    UNDOCUMENTED: "UNDOCUMENTED",
};
const ResourceOwner = {
    OTHER_ACCOUNTS: "OTHER_ACCOUNTS",
    SELF: "SELF",
};
const PutMode = {
    Merge: "merge",
    Overwrite: "overwrite",
};
const Op = {
    add: "add",
    copy: "copy",
    move: "move",
    remove: "remove",
    replace: "replace",
    test: "test",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.APIGatewayServiceException = APIGatewayServiceException.APIGatewayServiceException;
exports.APIGateway = APIGateway;
exports.APIGatewayClient = APIGatewayClient;
exports.AccessAssociationSourceType = AccessAssociationSourceType;
exports.ApiKeySourceType = ApiKeySourceType;
exports.ApiKeysFormat = ApiKeysFormat;
exports.ApiStatus = ApiStatus;
exports.AuthorizerType = AuthorizerType;
exports.CacheClusterSize = CacheClusterSize;
exports.CacheClusterStatus = CacheClusterStatus;
exports.ConnectionType = ConnectionType;
exports.ContentHandlingStrategy = ContentHandlingStrategy;
exports.CreateApiKeyCommand = CreateApiKeyCommand;
exports.CreateAuthorizerCommand = CreateAuthorizerCommand;
exports.CreateBasePathMappingCommand = CreateBasePathMappingCommand;
exports.CreateDeploymentCommand = CreateDeploymentCommand;
exports.CreateDocumentationPartCommand = CreateDocumentationPartCommand;
exports.CreateDocumentationVersionCommand = CreateDocumentationVersionCommand;
exports.CreateDomainNameAccessAssociationCommand = CreateDomainNameAccessAssociationCommand;
exports.CreateDomainNameCommand = CreateDomainNameCommand;
exports.CreateModelCommand = CreateModelCommand;
exports.CreateRequestValidatorCommand = CreateRequestValidatorCommand;
exports.CreateResourceCommand = CreateResourceCommand;
exports.CreateRestApiCommand = CreateRestApiCommand;
exports.CreateStageCommand = CreateStageCommand;
exports.CreateUsagePlanCommand = CreateUsagePlanCommand;
exports.CreateUsagePlanKeyCommand = CreateUsagePlanKeyCommand;
exports.CreateVpcLinkCommand = CreateVpcLinkCommand;
exports.DeleteApiKeyCommand = DeleteApiKeyCommand;
exports.DeleteAuthorizerCommand = DeleteAuthorizerCommand;
exports.DeleteBasePathMappingCommand = DeleteBasePathMappingCommand;
exports.DeleteClientCertificateCommand = DeleteClientCertificateCommand;
exports.DeleteDeploymentCommand = DeleteDeploymentCommand;
exports.DeleteDocumentationPartCommand = DeleteDocumentationPartCommand;
exports.DeleteDocumentationVersionCommand = DeleteDocumentationVersionCommand;
exports.DeleteDomainNameAccessAssociationCommand = DeleteDomainNameAccessAssociationCommand;
exports.DeleteDomainNameCommand = DeleteDomainNameCommand;
exports.DeleteGatewayResponseCommand = DeleteGatewayResponseCommand;
exports.DeleteIntegrationCommand = DeleteIntegrationCommand;
exports.DeleteIntegrationResponseCommand = DeleteIntegrationResponseCommand;
exports.DeleteMethodCommand = DeleteMethodCommand;
exports.DeleteMethodResponseCommand = DeleteMethodResponseCommand;
exports.DeleteModelCommand = DeleteModelCommand;
exports.DeleteRequestValidatorCommand = DeleteRequestValidatorCommand;
exports.DeleteResourceCommand = DeleteResourceCommand;
exports.DeleteRestApiCommand = DeleteRestApiCommand;
exports.DeleteStageCommand = DeleteStageCommand;
exports.DeleteUsagePlanCommand = DeleteUsagePlanCommand;
exports.DeleteUsagePlanKeyCommand = DeleteUsagePlanKeyCommand;
exports.DeleteVpcLinkCommand = DeleteVpcLinkCommand;
exports.DocumentationPartType = DocumentationPartType;
exports.DomainNameStatus = DomainNameStatus;
exports.EndpointAccessMode = EndpointAccessMode;
exports.EndpointType = EndpointType;
exports.FlushStageAuthorizersCacheCommand = FlushStageAuthorizersCacheCommand;
exports.FlushStageCacheCommand = FlushStageCacheCommand;
exports.GatewayResponseType = GatewayResponseType;
exports.GenerateClientCertificateCommand = GenerateClientCertificateCommand;
exports.GetAccountCommand = GetAccountCommand;
exports.GetApiKeyCommand = GetApiKeyCommand;
exports.GetApiKeysCommand = GetApiKeysCommand;
exports.GetAuthorizerCommand = GetAuthorizerCommand;
exports.GetAuthorizersCommand = GetAuthorizersCommand;
exports.GetBasePathMappingCommand = GetBasePathMappingCommand;
exports.GetBasePathMappingsCommand = GetBasePathMappingsCommand;
exports.GetClientCertificateCommand = GetClientCertificateCommand;
exports.GetClientCertificatesCommand = GetClientCertificatesCommand;
exports.GetDeploymentCommand = GetDeploymentCommand;
exports.GetDeploymentsCommand = GetDeploymentsCommand;
exports.GetDocumentationPartCommand = GetDocumentationPartCommand;
exports.GetDocumentationPartsCommand = GetDocumentationPartsCommand;
exports.GetDocumentationVersionCommand = GetDocumentationVersionCommand;
exports.GetDocumentationVersionsCommand = GetDocumentationVersionsCommand;
exports.GetDomainNameAccessAssociationsCommand = GetDomainNameAccessAssociationsCommand;
exports.GetDomainNameCommand = GetDomainNameCommand;
exports.GetDomainNamesCommand = GetDomainNamesCommand;
exports.GetExportCommand = GetExportCommand;
exports.GetGatewayResponseCommand = GetGatewayResponseCommand;
exports.GetGatewayResponsesCommand = GetGatewayResponsesCommand;
exports.GetIntegrationCommand = GetIntegrationCommand;
exports.GetIntegrationResponseCommand = GetIntegrationResponseCommand;
exports.GetMethodCommand = GetMethodCommand;
exports.GetMethodResponseCommand = GetMethodResponseCommand;
exports.GetModelCommand = GetModelCommand;
exports.GetModelTemplateCommand = GetModelTemplateCommand;
exports.GetModelsCommand = GetModelsCommand;
exports.GetRequestValidatorCommand = GetRequestValidatorCommand;
exports.GetRequestValidatorsCommand = GetRequestValidatorsCommand;
exports.GetResourceCommand = GetResourceCommand;
exports.GetResourcesCommand = GetResourcesCommand;
exports.GetRestApiCommand = GetRestApiCommand;
exports.GetRestApisCommand = GetRestApisCommand;
exports.GetSdkCommand = GetSdkCommand;
exports.GetSdkTypeCommand = GetSdkTypeCommand;
exports.GetSdkTypesCommand = GetSdkTypesCommand;
exports.GetStageCommand = GetStageCommand;
exports.GetStagesCommand = GetStagesCommand;
exports.GetTagsCommand = GetTagsCommand;
exports.GetUsageCommand = GetUsageCommand;
exports.GetUsagePlanCommand = GetUsagePlanCommand;
exports.GetUsagePlanKeyCommand = GetUsagePlanKeyCommand;
exports.GetUsagePlanKeysCommand = GetUsagePlanKeysCommand;
exports.GetUsagePlansCommand = GetUsagePlansCommand;
exports.GetVpcLinkCommand = GetVpcLinkCommand;
exports.GetVpcLinksCommand = GetVpcLinksCommand;
exports.ImportApiKeysCommand = ImportApiKeysCommand;
exports.ImportDocumentationPartsCommand = ImportDocumentationPartsCommand;
exports.ImportRestApiCommand = ImportRestApiCommand;
exports.IntegrationType = IntegrationType;
exports.IpAddressType = IpAddressType;
exports.LocationStatusType = LocationStatusType;
exports.Op = Op;
exports.PutGatewayResponseCommand = PutGatewayResponseCommand;
exports.PutIntegrationCommand = PutIntegrationCommand;
exports.PutIntegrationResponseCommand = PutIntegrationResponseCommand;
exports.PutMethodCommand = PutMethodCommand;
exports.PutMethodResponseCommand = PutMethodResponseCommand;
exports.PutMode = PutMode;
exports.PutRestApiCommand = PutRestApiCommand;
exports.QuotaPeriodType = QuotaPeriodType;
exports.RejectDomainNameAccessAssociationCommand = RejectDomainNameAccessAssociationCommand;
exports.ResourceOwner = ResourceOwner;
exports.ResponseTransferMode = ResponseTransferMode;
exports.RoutingMode = RoutingMode;
exports.SecurityPolicy = SecurityPolicy;
exports.TagResourceCommand = TagResourceCommand;
exports.TestInvokeAuthorizerCommand = TestInvokeAuthorizerCommand;
exports.TestInvokeMethodCommand = TestInvokeMethodCommand;
exports.UnauthorizedCacheControlHeaderStrategy = UnauthorizedCacheControlHeaderStrategy;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateAccountCommand = UpdateAccountCommand;
exports.UpdateApiKeyCommand = UpdateApiKeyCommand;
exports.UpdateAuthorizerCommand = UpdateAuthorizerCommand;
exports.UpdateBasePathMappingCommand = UpdateBasePathMappingCommand;
exports.UpdateClientCertificateCommand = UpdateClientCertificateCommand;
exports.UpdateDeploymentCommand = UpdateDeploymentCommand;
exports.UpdateDocumentationPartCommand = UpdateDocumentationPartCommand;
exports.UpdateDocumentationVersionCommand = UpdateDocumentationVersionCommand;
exports.UpdateDomainNameCommand = UpdateDomainNameCommand;
exports.UpdateGatewayResponseCommand = UpdateGatewayResponseCommand;
exports.UpdateIntegrationCommand = UpdateIntegrationCommand;
exports.UpdateIntegrationResponseCommand = UpdateIntegrationResponseCommand;
exports.UpdateMethodCommand = UpdateMethodCommand;
exports.UpdateMethodResponseCommand = UpdateMethodResponseCommand;
exports.UpdateModelCommand = UpdateModelCommand;
exports.UpdateRequestValidatorCommand = UpdateRequestValidatorCommand;
exports.UpdateResourceCommand = UpdateResourceCommand;
exports.UpdateRestApiCommand = UpdateRestApiCommand;
exports.UpdateStageCommand = UpdateStageCommand;
exports.UpdateUsageCommand = UpdateUsageCommand;
exports.UpdateUsagePlanCommand = UpdateUsagePlanCommand;
exports.UpdateVpcLinkCommand = UpdateVpcLinkCommand;
exports.VpcLinkStatus = VpcLinkStatus;
exports.paginateGetApiKeys = paginateGetApiKeys;
exports.paginateGetBasePathMappings = paginateGetBasePathMappings;
exports.paginateGetClientCertificates = paginateGetClientCertificates;
exports.paginateGetDeployments = paginateGetDeployments;
exports.paginateGetDomainNames = paginateGetDomainNames;
exports.paginateGetModels = paginateGetModels;
exports.paginateGetResources = paginateGetResources;
exports.paginateGetRestApis = paginateGetRestApis;
exports.paginateGetUsage = paginateGetUsage;
exports.paginateGetUsagePlanKeys = paginateGetUsagePlanKeys;
exports.paginateGetUsagePlans = paginateGetUsagePlans;
exports.paginateGetVpcLinks = paginateGetVpcLinks;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
