// @aws-sdk/client-cognito-identity-provider/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var CognitoIdentityProviderServiceException = require('./models/CognitoIdentityProviderServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "cognito-idp",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class CognitoIdentityProviderClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultCognitoIdentityProviderHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AddCustomAttributesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AddCustomAttributes", {})
    .n("CognitoIdentityProviderClient", "AddCustomAttributesCommand")
    .sc(schemas_0.AddCustomAttributes$)
    .build() {
}

class AddUserPoolClientSecretCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AddUserPoolClientSecret", {})
    .n("CognitoIdentityProviderClient", "AddUserPoolClientSecretCommand")
    .sc(schemas_0.AddUserPoolClientSecret$)
    .build() {
}

class AdminAddUserToGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminAddUserToGroup", {})
    .n("CognitoIdentityProviderClient", "AdminAddUserToGroupCommand")
    .sc(schemas_0.AdminAddUserToGroup$)
    .build() {
}

class AdminConfirmSignUpCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminConfirmSignUp", {})
    .n("CognitoIdentityProviderClient", "AdminConfirmSignUpCommand")
    .sc(schemas_0.AdminConfirmSignUp$)
    .build() {
}

class AdminCreateUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminCreateUser", {})
    .n("CognitoIdentityProviderClient", "AdminCreateUserCommand")
    .sc(schemas_0.AdminCreateUser$)
    .build() {
}

class AdminDeleteUserAttributesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminDeleteUserAttributes", {})
    .n("CognitoIdentityProviderClient", "AdminDeleteUserAttributesCommand")
    .sc(schemas_0.AdminDeleteUserAttributes$)
    .build() {
}

class AdminDeleteUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminDeleteUser", {})
    .n("CognitoIdentityProviderClient", "AdminDeleteUserCommand")
    .sc(schemas_0.AdminDeleteUser$)
    .build() {
}

class AdminDisableProviderForUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminDisableProviderForUser", {})
    .n("CognitoIdentityProviderClient", "AdminDisableProviderForUserCommand")
    .sc(schemas_0.AdminDisableProviderForUser$)
    .build() {
}

class AdminDisableUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminDisableUser", {})
    .n("CognitoIdentityProviderClient", "AdminDisableUserCommand")
    .sc(schemas_0.AdminDisableUser$)
    .build() {
}

class AdminEnableUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminEnableUser", {})
    .n("CognitoIdentityProviderClient", "AdminEnableUserCommand")
    .sc(schemas_0.AdminEnableUser$)
    .build() {
}

class AdminForgetDeviceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminForgetDevice", {})
    .n("CognitoIdentityProviderClient", "AdminForgetDeviceCommand")
    .sc(schemas_0.AdminForgetDevice$)
    .build() {
}

class AdminGetDeviceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminGetDevice", {})
    .n("CognitoIdentityProviderClient", "AdminGetDeviceCommand")
    .sc(schemas_0.AdminGetDevice$)
    .build() {
}

class AdminGetUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminGetUser", {})
    .n("CognitoIdentityProviderClient", "AdminGetUserCommand")
    .sc(schemas_0.AdminGetUser$)
    .build() {
}

class AdminInitiateAuthCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminInitiateAuth", {})
    .n("CognitoIdentityProviderClient", "AdminInitiateAuthCommand")
    .sc(schemas_0.AdminInitiateAuth$)
    .build() {
}

class AdminLinkProviderForUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminLinkProviderForUser", {})
    .n("CognitoIdentityProviderClient", "AdminLinkProviderForUserCommand")
    .sc(schemas_0.AdminLinkProviderForUser$)
    .build() {
}

class AdminListDevicesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminListDevices", {})
    .n("CognitoIdentityProviderClient", "AdminListDevicesCommand")
    .sc(schemas_0.AdminListDevices$)
    .build() {
}

class AdminListGroupsForUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminListGroupsForUser", {})
    .n("CognitoIdentityProviderClient", "AdminListGroupsForUserCommand")
    .sc(schemas_0.AdminListGroupsForUser$)
    .build() {
}

class AdminListUserAuthEventsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminListUserAuthEvents", {})
    .n("CognitoIdentityProviderClient", "AdminListUserAuthEventsCommand")
    .sc(schemas_0.AdminListUserAuthEvents$)
    .build() {
}

class AdminRemoveUserFromGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminRemoveUserFromGroup", {})
    .n("CognitoIdentityProviderClient", "AdminRemoveUserFromGroupCommand")
    .sc(schemas_0.AdminRemoveUserFromGroup$)
    .build() {
}

class AdminResetUserPasswordCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminResetUserPassword", {})
    .n("CognitoIdentityProviderClient", "AdminResetUserPasswordCommand")
    .sc(schemas_0.AdminResetUserPassword$)
    .build() {
}

class AdminRespondToAuthChallengeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminRespondToAuthChallenge", {})
    .n("CognitoIdentityProviderClient", "AdminRespondToAuthChallengeCommand")
    .sc(schemas_0.AdminRespondToAuthChallenge$)
    .build() {
}

class AdminSetUserMFAPreferenceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminSetUserMFAPreference", {})
    .n("CognitoIdentityProviderClient", "AdminSetUserMFAPreferenceCommand")
    .sc(schemas_0.AdminSetUserMFAPreference$)
    .build() {
}

class AdminSetUserPasswordCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminSetUserPassword", {})
    .n("CognitoIdentityProviderClient", "AdminSetUserPasswordCommand")
    .sc(schemas_0.AdminSetUserPassword$)
    .build() {
}

class AdminSetUserSettingsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminSetUserSettings", {})
    .n("CognitoIdentityProviderClient", "AdminSetUserSettingsCommand")
    .sc(schemas_0.AdminSetUserSettings$)
    .build() {
}

class AdminUpdateAuthEventFeedbackCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminUpdateAuthEventFeedback", {})
    .n("CognitoIdentityProviderClient", "AdminUpdateAuthEventFeedbackCommand")
    .sc(schemas_0.AdminUpdateAuthEventFeedback$)
    .build() {
}

class AdminUpdateDeviceStatusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminUpdateDeviceStatus", {})
    .n("CognitoIdentityProviderClient", "AdminUpdateDeviceStatusCommand")
    .sc(schemas_0.AdminUpdateDeviceStatus$)
    .build() {
}

class AdminUpdateUserAttributesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminUpdateUserAttributes", {})
    .n("CognitoIdentityProviderClient", "AdminUpdateUserAttributesCommand")
    .sc(schemas_0.AdminUpdateUserAttributes$)
    .build() {
}

class AdminUserGlobalSignOutCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AdminUserGlobalSignOut", {})
    .n("CognitoIdentityProviderClient", "AdminUserGlobalSignOutCommand")
    .sc(schemas_0.AdminUserGlobalSignOut$)
    .build() {
}

class AssociateSoftwareTokenCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "AssociateSoftwareToken", {})
    .n("CognitoIdentityProviderClient", "AssociateSoftwareTokenCommand")
    .sc(schemas_0.AssociateSoftwareToken$)
    .build() {
}

class ChangePasswordCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ChangePassword", {})
    .n("CognitoIdentityProviderClient", "ChangePasswordCommand")
    .sc(schemas_0.ChangePassword$)
    .build() {
}

class CompleteWebAuthnRegistrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CompleteWebAuthnRegistration", {})
    .n("CognitoIdentityProviderClient", "CompleteWebAuthnRegistrationCommand")
    .sc(schemas_0.CompleteWebAuthnRegistration$)
    .build() {
}

class ConfirmDeviceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ConfirmDevice", {})
    .n("CognitoIdentityProviderClient", "ConfirmDeviceCommand")
    .sc(schemas_0.ConfirmDevice$)
    .build() {
}

class ConfirmForgotPasswordCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ConfirmForgotPassword", {})
    .n("CognitoIdentityProviderClient", "ConfirmForgotPasswordCommand")
    .sc(schemas_0.ConfirmForgotPassword$)
    .build() {
}

class ConfirmSignUpCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ConfirmSignUp", {})
    .n("CognitoIdentityProviderClient", "ConfirmSignUpCommand")
    .sc(schemas_0.ConfirmSignUp$)
    .build() {
}

class CreateGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateGroup", {})
    .n("CognitoIdentityProviderClient", "CreateGroupCommand")
    .sc(schemas_0.CreateGroup$)
    .build() {
}

class CreateIdentityProviderCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateIdentityProvider", {})
    .n("CognitoIdentityProviderClient", "CreateIdentityProviderCommand")
    .sc(schemas_0.CreateIdentityProvider$)
    .build() {
}

class CreateManagedLoginBrandingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateManagedLoginBranding", {})
    .n("CognitoIdentityProviderClient", "CreateManagedLoginBrandingCommand")
    .sc(schemas_0.CreateManagedLoginBranding$)
    .build() {
}

class CreateResourceServerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateResourceServer", {})
    .n("CognitoIdentityProviderClient", "CreateResourceServerCommand")
    .sc(schemas_0.CreateResourceServer$)
    .build() {
}

class CreateTermsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateTerms", {})
    .n("CognitoIdentityProviderClient", "CreateTermsCommand")
    .sc(schemas_0.CreateTerms$)
    .build() {
}

class CreateUserImportJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateUserImportJob", {})
    .n("CognitoIdentityProviderClient", "CreateUserImportJobCommand")
    .sc(schemas_0.CreateUserImportJob$)
    .build() {
}

class CreateUserPoolClientCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateUserPoolClient", {})
    .n("CognitoIdentityProviderClient", "CreateUserPoolClientCommand")
    .sc(schemas_0.CreateUserPoolClient$)
    .build() {
}

class CreateUserPoolCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateUserPool", {})
    .n("CognitoIdentityProviderClient", "CreateUserPoolCommand")
    .sc(schemas_0.CreateUserPool$)
    .build() {
}

class CreateUserPoolDomainCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateUserPoolDomain", {})
    .n("CognitoIdentityProviderClient", "CreateUserPoolDomainCommand")
    .sc(schemas_0.CreateUserPoolDomain$)
    .build() {
}

class CreateUserPoolReplicaCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "CreateUserPoolReplica", {})
    .n("CognitoIdentityProviderClient", "CreateUserPoolReplicaCommand")
    .sc(schemas_0.CreateUserPoolReplica$)
    .build() {
}

class DeleteGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteGroup", {})
    .n("CognitoIdentityProviderClient", "DeleteGroupCommand")
    .sc(schemas_0.DeleteGroup$)
    .build() {
}

class DeleteIdentityProviderCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteIdentityProvider", {})
    .n("CognitoIdentityProviderClient", "DeleteIdentityProviderCommand")
    .sc(schemas_0.DeleteIdentityProvider$)
    .build() {
}

class DeleteManagedLoginBrandingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteManagedLoginBranding", {})
    .n("CognitoIdentityProviderClient", "DeleteManagedLoginBrandingCommand")
    .sc(schemas_0.DeleteManagedLoginBranding$)
    .build() {
}

class DeleteResourceServerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteResourceServer", {})
    .n("CognitoIdentityProviderClient", "DeleteResourceServerCommand")
    .sc(schemas_0.DeleteResourceServer$)
    .build() {
}

class DeleteTermsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteTerms", {})
    .n("CognitoIdentityProviderClient", "DeleteTermsCommand")
    .sc(schemas_0.DeleteTerms$)
    .build() {
}

class DeleteUserAttributesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserAttributes", {})
    .n("CognitoIdentityProviderClient", "DeleteUserAttributesCommand")
    .sc(schemas_0.DeleteUserAttributes$)
    .build() {
}

class DeleteUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUser", {})
    .n("CognitoIdentityProviderClient", "DeleteUserCommand")
    .sc(schemas_0.DeleteUser$)
    .build() {
}

class DeleteUserPoolClientCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserPoolClient", {})
    .n("CognitoIdentityProviderClient", "DeleteUserPoolClientCommand")
    .sc(schemas_0.DeleteUserPoolClient$)
    .build() {
}

class DeleteUserPoolClientSecretCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserPoolClientSecret", {})
    .n("CognitoIdentityProviderClient", "DeleteUserPoolClientSecretCommand")
    .sc(schemas_0.DeleteUserPoolClientSecret$)
    .build() {
}

class DeleteUserPoolCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserPool", {})
    .n("CognitoIdentityProviderClient", "DeleteUserPoolCommand")
    .sc(schemas_0.DeleteUserPool$)
    .build() {
}

class DeleteUserPoolDomainCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserPoolDomain", {})
    .n("CognitoIdentityProviderClient", "DeleteUserPoolDomainCommand")
    .sc(schemas_0.DeleteUserPoolDomain$)
    .build() {
}

class DeleteUserPoolReplicaCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteUserPoolReplica", {})
    .n("CognitoIdentityProviderClient", "DeleteUserPoolReplicaCommand")
    .sc(schemas_0.DeleteUserPoolReplica$)
    .build() {
}

class DeleteWebAuthnCredentialCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DeleteWebAuthnCredential", {})
    .n("CognitoIdentityProviderClient", "DeleteWebAuthnCredentialCommand")
    .sc(schemas_0.DeleteWebAuthnCredential$)
    .build() {
}

class DescribeIdentityProviderCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeIdentityProvider", {})
    .n("CognitoIdentityProviderClient", "DescribeIdentityProviderCommand")
    .sc(schemas_0.DescribeIdentityProvider$)
    .build() {
}

class DescribeManagedLoginBrandingByClientCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeManagedLoginBrandingByClient", {})
    .n("CognitoIdentityProviderClient", "DescribeManagedLoginBrandingByClientCommand")
    .sc(schemas_0.DescribeManagedLoginBrandingByClient$)
    .build() {
}

class DescribeManagedLoginBrandingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeManagedLoginBranding", {})
    .n("CognitoIdentityProviderClient", "DescribeManagedLoginBrandingCommand")
    .sc(schemas_0.DescribeManagedLoginBranding$)
    .build() {
}

class DescribeResourceServerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeResourceServer", {})
    .n("CognitoIdentityProviderClient", "DescribeResourceServerCommand")
    .sc(schemas_0.DescribeResourceServer$)
    .build() {
}

class DescribeRiskConfigurationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeRiskConfiguration", {})
    .n("CognitoIdentityProviderClient", "DescribeRiskConfigurationCommand")
    .sc(schemas_0.DescribeRiskConfiguration$)
    .build() {
}

class DescribeTermsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeTerms", {})
    .n("CognitoIdentityProviderClient", "DescribeTermsCommand")
    .sc(schemas_0.DescribeTerms$)
    .build() {
}

class DescribeUserImportJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeUserImportJob", {})
    .n("CognitoIdentityProviderClient", "DescribeUserImportJobCommand")
    .sc(schemas_0.DescribeUserImportJob$)
    .build() {
}

class DescribeUserPoolClientCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeUserPoolClient", {})
    .n("CognitoIdentityProviderClient", "DescribeUserPoolClientCommand")
    .sc(schemas_0.DescribeUserPoolClient$)
    .build() {
}

class DescribeUserPoolCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeUserPool", {})
    .n("CognitoIdentityProviderClient", "DescribeUserPoolCommand")
    .sc(schemas_0.DescribeUserPool$)
    .build() {
}

class DescribeUserPoolDomainCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "DescribeUserPoolDomain", {})
    .n("CognitoIdentityProviderClient", "DescribeUserPoolDomainCommand")
    .sc(schemas_0.DescribeUserPoolDomain$)
    .build() {
}

class ForgetDeviceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ForgetDevice", {})
    .n("CognitoIdentityProviderClient", "ForgetDeviceCommand")
    .sc(schemas_0.ForgetDevice$)
    .build() {
}

class ForgotPasswordCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ForgotPassword", {})
    .n("CognitoIdentityProviderClient", "ForgotPasswordCommand")
    .sc(schemas_0.ForgotPassword$)
    .build() {
}

class GetCSVHeaderCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetCSVHeader", {})
    .n("CognitoIdentityProviderClient", "GetCSVHeaderCommand")
    .sc(schemas_0.GetCSVHeader$)
    .build() {
}

class GetDeviceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetDevice", {})
    .n("CognitoIdentityProviderClient", "GetDeviceCommand")
    .sc(schemas_0.GetDevice$)
    .build() {
}

class GetGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetGroup", {})
    .n("CognitoIdentityProviderClient", "GetGroupCommand")
    .sc(schemas_0.GetGroup$)
    .build() {
}

class GetIdentityProviderByIdentifierCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetIdentityProviderByIdentifier", {})
    .n("CognitoIdentityProviderClient", "GetIdentityProviderByIdentifierCommand")
    .sc(schemas_0.GetIdentityProviderByIdentifier$)
    .build() {
}

class GetLogDeliveryConfigurationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetLogDeliveryConfiguration", {})
    .n("CognitoIdentityProviderClient", "GetLogDeliveryConfigurationCommand")
    .sc(schemas_0.GetLogDeliveryConfiguration$)
    .build() {
}

class GetSigningCertificateCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetSigningCertificate", {})
    .n("CognitoIdentityProviderClient", "GetSigningCertificateCommand")
    .sc(schemas_0.GetSigningCertificate$)
    .build() {
}

class GetTokensFromRefreshTokenCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetTokensFromRefreshToken", {})
    .n("CognitoIdentityProviderClient", "GetTokensFromRefreshTokenCommand")
    .sc(schemas_0.GetTokensFromRefreshToken$)
    .build() {
}

class GetUICustomizationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetUICustomization", {})
    .n("CognitoIdentityProviderClient", "GetUICustomizationCommand")
    .sc(schemas_0.GetUICustomization$)
    .build() {
}

class GetUserAttributeVerificationCodeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetUserAttributeVerificationCode", {})
    .n("CognitoIdentityProviderClient", "GetUserAttributeVerificationCodeCommand")
    .sc(schemas_0.GetUserAttributeVerificationCode$)
    .build() {
}

class GetUserAuthFactorsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetUserAuthFactors", {})
    .n("CognitoIdentityProviderClient", "GetUserAuthFactorsCommand")
    .sc(schemas_0.GetUserAuthFactors$)
    .build() {
}

class GetUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetUser", {})
    .n("CognitoIdentityProviderClient", "GetUserCommand")
    .sc(schemas_0.GetUser$)
    .build() {
}

class GetUserPoolMfaConfigCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GetUserPoolMfaConfig", {})
    .n("CognitoIdentityProviderClient", "GetUserPoolMfaConfigCommand")
    .sc(schemas_0.GetUserPoolMfaConfig$)
    .build() {
}

class GlobalSignOutCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "GlobalSignOut", {})
    .n("CognitoIdentityProviderClient", "GlobalSignOutCommand")
    .sc(schemas_0.GlobalSignOut$)
    .build() {
}

class InitiateAuthCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "InitiateAuth", {})
    .n("CognitoIdentityProviderClient", "InitiateAuthCommand")
    .sc(schemas_0.InitiateAuth$)
    .build() {
}

class ListDevicesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListDevices", {})
    .n("CognitoIdentityProviderClient", "ListDevicesCommand")
    .sc(schemas_0.ListDevices$)
    .build() {
}

class ListGroupsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListGroups", {})
    .n("CognitoIdentityProviderClient", "ListGroupsCommand")
    .sc(schemas_0.ListGroups$)
    .build() {
}

class ListIdentityProvidersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListIdentityProviders", {})
    .n("CognitoIdentityProviderClient", "ListIdentityProvidersCommand")
    .sc(schemas_0.ListIdentityProviders$)
    .build() {
}

class ListResourceServersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListResourceServers", {})
    .n("CognitoIdentityProviderClient", "ListResourceServersCommand")
    .sc(schemas_0.ListResourceServers$)
    .build() {
}

class ListTagsForResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListTagsForResource", {})
    .n("CognitoIdentityProviderClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTermsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListTerms", {})
    .n("CognitoIdentityProviderClient", "ListTermsCommand")
    .sc(schemas_0.ListTerms$)
    .build() {
}

class ListUserImportJobsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUserImportJobs", {})
    .n("CognitoIdentityProviderClient", "ListUserImportJobsCommand")
    .sc(schemas_0.ListUserImportJobs$)
    .build() {
}

class ListUserPoolClientsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUserPoolClients", {})
    .n("CognitoIdentityProviderClient", "ListUserPoolClientsCommand")
    .sc(schemas_0.ListUserPoolClients$)
    .build() {
}

class ListUserPoolClientSecretsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUserPoolClientSecrets", {})
    .n("CognitoIdentityProviderClient", "ListUserPoolClientSecretsCommand")
    .sc(schemas_0.ListUserPoolClientSecrets$)
    .build() {
}

class ListUserPoolReplicasCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUserPoolReplicas", {})
    .n("CognitoIdentityProviderClient", "ListUserPoolReplicasCommand")
    .sc(schemas_0.ListUserPoolReplicas$)
    .build() {
}

class ListUserPoolsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUserPools", {})
    .n("CognitoIdentityProviderClient", "ListUserPoolsCommand")
    .sc(schemas_0.ListUserPools$)
    .build() {
}

class ListUsersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUsers", {})
    .n("CognitoIdentityProviderClient", "ListUsersCommand")
    .sc(schemas_0.ListUsers$)
    .build() {
}

class ListUsersInGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListUsersInGroup", {})
    .n("CognitoIdentityProviderClient", "ListUsersInGroupCommand")
    .sc(schemas_0.ListUsersInGroup$)
    .build() {
}

class ListWebAuthnCredentialsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ListWebAuthnCredentials", {})
    .n("CognitoIdentityProviderClient", "ListWebAuthnCredentialsCommand")
    .sc(schemas_0.ListWebAuthnCredentials$)
    .build() {
}

class ResendConfirmationCodeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "ResendConfirmationCode", {})
    .n("CognitoIdentityProviderClient", "ResendConfirmationCodeCommand")
    .sc(schemas_0.ResendConfirmationCode$)
    .build() {
}

class RespondToAuthChallengeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "RespondToAuthChallenge", {})
    .n("CognitoIdentityProviderClient", "RespondToAuthChallengeCommand")
    .sc(schemas_0.RespondToAuthChallenge$)
    .build() {
}

class RevokeTokenCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "RevokeToken", {})
    .n("CognitoIdentityProviderClient", "RevokeTokenCommand")
    .sc(schemas_0.RevokeToken$)
    .build() {
}

class SetLogDeliveryConfigurationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetLogDeliveryConfiguration", {})
    .n("CognitoIdentityProviderClient", "SetLogDeliveryConfigurationCommand")
    .sc(schemas_0.SetLogDeliveryConfiguration$)
    .build() {
}

class SetRiskConfigurationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetRiskConfiguration", {})
    .n("CognitoIdentityProviderClient", "SetRiskConfigurationCommand")
    .sc(schemas_0.SetRiskConfiguration$)
    .build() {
}

class SetUICustomizationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetUICustomization", {})
    .n("CognitoIdentityProviderClient", "SetUICustomizationCommand")
    .sc(schemas_0.SetUICustomization$)
    .build() {
}

class SetUserMFAPreferenceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetUserMFAPreference", {})
    .n("CognitoIdentityProviderClient", "SetUserMFAPreferenceCommand")
    .sc(schemas_0.SetUserMFAPreference$)
    .build() {
}

class SetUserPoolMfaConfigCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetUserPoolMfaConfig", {})
    .n("CognitoIdentityProviderClient", "SetUserPoolMfaConfigCommand")
    .sc(schemas_0.SetUserPoolMfaConfig$)
    .build() {
}

class SetUserSettingsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SetUserSettings", {})
    .n("CognitoIdentityProviderClient", "SetUserSettingsCommand")
    .sc(schemas_0.SetUserSettings$)
    .build() {
}

class SignUpCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "SignUp", {})
    .n("CognitoIdentityProviderClient", "SignUpCommand")
    .sc(schemas_0.SignUp$)
    .build() {
}

class StartUserImportJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "StartUserImportJob", {})
    .n("CognitoIdentityProviderClient", "StartUserImportJobCommand")
    .sc(schemas_0.StartUserImportJob$)
    .build() {
}

class StartWebAuthnRegistrationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "StartWebAuthnRegistration", {})
    .n("CognitoIdentityProviderClient", "StartWebAuthnRegistrationCommand")
    .sc(schemas_0.StartWebAuthnRegistration$)
    .build() {
}

class StopUserImportJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "StopUserImportJob", {})
    .n("CognitoIdentityProviderClient", "StopUserImportJobCommand")
    .sc(schemas_0.StopUserImportJob$)
    .build() {
}

class TagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "TagResource", {})
    .n("CognitoIdentityProviderClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UntagResource", {})
    .n("CognitoIdentityProviderClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateAuthEventFeedbackCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateAuthEventFeedback", {})
    .n("CognitoIdentityProviderClient", "UpdateAuthEventFeedbackCommand")
    .sc(schemas_0.UpdateAuthEventFeedback$)
    .build() {
}

class UpdateDeviceStatusCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateDeviceStatus", {})
    .n("CognitoIdentityProviderClient", "UpdateDeviceStatusCommand")
    .sc(schemas_0.UpdateDeviceStatus$)
    .build() {
}

class UpdateGroupCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateGroup", {})
    .n("CognitoIdentityProviderClient", "UpdateGroupCommand")
    .sc(schemas_0.UpdateGroup$)
    .build() {
}

class UpdateIdentityProviderCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateIdentityProvider", {})
    .n("CognitoIdentityProviderClient", "UpdateIdentityProviderCommand")
    .sc(schemas_0.UpdateIdentityProvider$)
    .build() {
}

class UpdateManagedLoginBrandingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateManagedLoginBranding", {})
    .n("CognitoIdentityProviderClient", "UpdateManagedLoginBrandingCommand")
    .sc(schemas_0.UpdateManagedLoginBranding$)
    .build() {
}

class UpdateResourceServerCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateResourceServer", {})
    .n("CognitoIdentityProviderClient", "UpdateResourceServerCommand")
    .sc(schemas_0.UpdateResourceServer$)
    .build() {
}

class UpdateTermsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateTerms", {})
    .n("CognitoIdentityProviderClient", "UpdateTermsCommand")
    .sc(schemas_0.UpdateTerms$)
    .build() {
}

class UpdateUserAttributesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateUserAttributes", {})
    .n("CognitoIdentityProviderClient", "UpdateUserAttributesCommand")
    .sc(schemas_0.UpdateUserAttributes$)
    .build() {
}

class UpdateUserPoolClientCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateUserPoolClient", {})
    .n("CognitoIdentityProviderClient", "UpdateUserPoolClientCommand")
    .sc(schemas_0.UpdateUserPoolClient$)
    .build() {
}

class UpdateUserPoolCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateUserPool", {})
    .n("CognitoIdentityProviderClient", "UpdateUserPoolCommand")
    .sc(schemas_0.UpdateUserPool$)
    .build() {
}

class UpdateUserPoolDomainCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateUserPoolDomain", {})
    .n("CognitoIdentityProviderClient", "UpdateUserPoolDomainCommand")
    .sc(schemas_0.UpdateUserPoolDomain$)
    .build() {
}

class UpdateUserPoolReplicaCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "UpdateUserPoolReplica", {})
    .n("CognitoIdentityProviderClient", "UpdateUserPoolReplicaCommand")
    .sc(schemas_0.UpdateUserPoolReplica$)
    .build() {
}

class VerifySoftwareTokenCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "VerifySoftwareToken", {})
    .n("CognitoIdentityProviderClient", "VerifySoftwareTokenCommand")
    .sc(schemas_0.VerifySoftwareToken$)
    .build() {
}

class VerifyUserAttributeCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSCognitoIdentityProviderService", "VerifyUserAttribute", {})
    .n("CognitoIdentityProviderClient", "VerifyUserAttributeCommand")
    .sc(schemas_0.VerifyUserAttribute$)
    .build() {
}

const paginateAdminListGroupsForUser = core.createPaginator(CognitoIdentityProviderClient, AdminListGroupsForUserCommand, "NextToken", "NextToken", "Limit");

const paginateAdminListUserAuthEvents = core.createPaginator(CognitoIdentityProviderClient, AdminListUserAuthEventsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListGroups = core.createPaginator(CognitoIdentityProviderClient, ListGroupsCommand, "NextToken", "NextToken", "Limit");

const paginateListIdentityProviders = core.createPaginator(CognitoIdentityProviderClient, ListIdentityProvidersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListResourceServers = core.createPaginator(CognitoIdentityProviderClient, ListResourceServersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListUserPoolClients = core.createPaginator(CognitoIdentityProviderClient, ListUserPoolClientsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListUserPools = core.createPaginator(CognitoIdentityProviderClient, ListUserPoolsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListUsersInGroup = core.createPaginator(CognitoIdentityProviderClient, ListUsersInGroupCommand, "NextToken", "NextToken", "Limit");

const paginateListUsers = core.createPaginator(CognitoIdentityProviderClient, ListUsersCommand, "PaginationToken", "PaginationToken", "Limit");

const commands = {
    AddCustomAttributesCommand,
    AddUserPoolClientSecretCommand,
    AdminAddUserToGroupCommand,
    AdminConfirmSignUpCommand,
    AdminCreateUserCommand,
    AdminDeleteUserCommand,
    AdminDeleteUserAttributesCommand,
    AdminDisableProviderForUserCommand,
    AdminDisableUserCommand,
    AdminEnableUserCommand,
    AdminForgetDeviceCommand,
    AdminGetDeviceCommand,
    AdminGetUserCommand,
    AdminInitiateAuthCommand,
    AdminLinkProviderForUserCommand,
    AdminListDevicesCommand,
    AdminListGroupsForUserCommand,
    AdminListUserAuthEventsCommand,
    AdminRemoveUserFromGroupCommand,
    AdminResetUserPasswordCommand,
    AdminRespondToAuthChallengeCommand,
    AdminSetUserMFAPreferenceCommand,
    AdminSetUserPasswordCommand,
    AdminSetUserSettingsCommand,
    AdminUpdateAuthEventFeedbackCommand,
    AdminUpdateDeviceStatusCommand,
    AdminUpdateUserAttributesCommand,
    AdminUserGlobalSignOutCommand,
    AssociateSoftwareTokenCommand,
    ChangePasswordCommand,
    CompleteWebAuthnRegistrationCommand,
    ConfirmDeviceCommand,
    ConfirmForgotPasswordCommand,
    ConfirmSignUpCommand,
    CreateGroupCommand,
    CreateIdentityProviderCommand,
    CreateManagedLoginBrandingCommand,
    CreateResourceServerCommand,
    CreateTermsCommand,
    CreateUserImportJobCommand,
    CreateUserPoolCommand,
    CreateUserPoolClientCommand,
    CreateUserPoolDomainCommand,
    CreateUserPoolReplicaCommand,
    DeleteGroupCommand,
    DeleteIdentityProviderCommand,
    DeleteManagedLoginBrandingCommand,
    DeleteResourceServerCommand,
    DeleteTermsCommand,
    DeleteUserCommand,
    DeleteUserAttributesCommand,
    DeleteUserPoolCommand,
    DeleteUserPoolClientCommand,
    DeleteUserPoolClientSecretCommand,
    DeleteUserPoolDomainCommand,
    DeleteUserPoolReplicaCommand,
    DeleteWebAuthnCredentialCommand,
    DescribeIdentityProviderCommand,
    DescribeManagedLoginBrandingCommand,
    DescribeManagedLoginBrandingByClientCommand,
    DescribeResourceServerCommand,
    DescribeRiskConfigurationCommand,
    DescribeTermsCommand,
    DescribeUserImportJobCommand,
    DescribeUserPoolCommand,
    DescribeUserPoolClientCommand,
    DescribeUserPoolDomainCommand,
    ForgetDeviceCommand,
    ForgotPasswordCommand,
    GetCSVHeaderCommand,
    GetDeviceCommand,
    GetGroupCommand,
    GetIdentityProviderByIdentifierCommand,
    GetLogDeliveryConfigurationCommand,
    GetSigningCertificateCommand,
    GetTokensFromRefreshTokenCommand,
    GetUICustomizationCommand,
    GetUserCommand,
    GetUserAttributeVerificationCodeCommand,
    GetUserAuthFactorsCommand,
    GetUserPoolMfaConfigCommand,
    GlobalSignOutCommand,
    InitiateAuthCommand,
    ListDevicesCommand,
    ListGroupsCommand,
    ListIdentityProvidersCommand,
    ListResourceServersCommand,
    ListTagsForResourceCommand,
    ListTermsCommand,
    ListUserImportJobsCommand,
    ListUserPoolClientsCommand,
    ListUserPoolClientSecretsCommand,
    ListUserPoolReplicasCommand,
    ListUserPoolsCommand,
    ListUsersCommand,
    ListUsersInGroupCommand,
    ListWebAuthnCredentialsCommand,
    ResendConfirmationCodeCommand,
    RespondToAuthChallengeCommand,
    RevokeTokenCommand,
    SetLogDeliveryConfigurationCommand,
    SetRiskConfigurationCommand,
    SetUICustomizationCommand,
    SetUserMFAPreferenceCommand,
    SetUserPoolMfaConfigCommand,
    SetUserSettingsCommand,
    SignUpCommand,
    StartUserImportJobCommand,
    StartWebAuthnRegistrationCommand,
    StopUserImportJobCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAuthEventFeedbackCommand,
    UpdateDeviceStatusCommand,
    UpdateGroupCommand,
    UpdateIdentityProviderCommand,
    UpdateManagedLoginBrandingCommand,
    UpdateResourceServerCommand,
    UpdateTermsCommand,
    UpdateUserAttributesCommand,
    UpdateUserPoolCommand,
    UpdateUserPoolClientCommand,
    UpdateUserPoolDomainCommand,
    UpdateUserPoolReplicaCommand,
    VerifySoftwareTokenCommand,
    VerifyUserAttributeCommand,
};
const paginators = {
    paginateAdminListGroupsForUser,
    paginateAdminListUserAuthEvents,
    paginateListGroups,
    paginateListIdentityProviders,
    paginateListResourceServers,
    paginateListUserPoolClients,
    paginateListUserPools,
    paginateListUsers,
    paginateListUsersInGroup,
};
class CognitoIdentityProvider extends CognitoIdentityProviderClient {
}
client.createAggregatedClient(commands, CognitoIdentityProvider, { paginators });

const RecoveryOptionNameType = {
    ADMIN_ONLY: "admin_only",
    VERIFIED_EMAIL: "verified_email",
    VERIFIED_PHONE_NUMBER: "verified_phone_number",
};
const AccountTakeoverEventActionType = {
    BLOCK: "BLOCK",
    MFA_IF_CONFIGURED: "MFA_IF_CONFIGURED",
    MFA_REQUIRED: "MFA_REQUIRED",
    NO_ACTION: "NO_ACTION",
};
const AttributeDataType = {
    BOOLEAN: "Boolean",
    DATETIME: "DateTime",
    NUMBER: "Number",
    STRING: "String",
};
const DeliveryMediumType = {
    EMAIL: "EMAIL",
    SMS: "SMS",
};
const MessageActionType = {
    RESEND: "RESEND",
    SUPPRESS: "SUPPRESS",
};
const UserStatusType = {
    ARCHIVED: "ARCHIVED",
    COMPROMISED: "COMPROMISED",
    CONFIRMED: "CONFIRMED",
    EXTERNAL_PROVIDER: "EXTERNAL_PROVIDER",
    FORCE_CHANGE_PASSWORD: "FORCE_CHANGE_PASSWORD",
    RESET_REQUIRED: "RESET_REQUIRED",
    UNCONFIRMED: "UNCONFIRMED",
    UNKNOWN: "UNKNOWN",
};
const AuthFlowType = {
    ADMIN_NO_SRP_AUTH: "ADMIN_NO_SRP_AUTH",
    ADMIN_USER_PASSWORD_AUTH: "ADMIN_USER_PASSWORD_AUTH",
    CUSTOM_AUTH: "CUSTOM_AUTH",
    REFRESH_TOKEN: "REFRESH_TOKEN",
    REFRESH_TOKEN_AUTH: "REFRESH_TOKEN_AUTH",
    USER_AUTH: "USER_AUTH",
    USER_PASSWORD_AUTH: "USER_PASSWORD_AUTH",
    USER_SRP_AUTH: "USER_SRP_AUTH",
};
const ChallengeNameType = {
    ADMIN_NO_SRP_AUTH: "ADMIN_NO_SRP_AUTH",
    CUSTOM_CHALLENGE: "CUSTOM_CHALLENGE",
    DEVICE_PASSWORD_VERIFIER: "DEVICE_PASSWORD_VERIFIER",
    DEVICE_SRP_AUTH: "DEVICE_SRP_AUTH",
    EMAIL_OTP: "EMAIL_OTP",
    MFA_SETUP: "MFA_SETUP",
    NEW_PASSWORD_REQUIRED: "NEW_PASSWORD_REQUIRED",
    PASSWORD: "PASSWORD",
    PASSWORD_SRP: "PASSWORD_SRP",
    PASSWORD_VERIFIER: "PASSWORD_VERIFIER",
    SELECT_CHALLENGE: "SELECT_CHALLENGE",
    SELECT_MFA_TYPE: "SELECT_MFA_TYPE",
    SMS_MFA: "SMS_MFA",
    SMS_OTP: "SMS_OTP",
    SOFTWARE_TOKEN_MFA: "SOFTWARE_TOKEN_MFA",
    WEB_AUTHN: "WEB_AUTHN",
};
const ChallengeName = {
    Mfa: "Mfa",
    Password: "Password",
};
const ChallengeResponse = {
    Failure: "Failure",
    Success: "Success",
};
const FeedbackValueType = {
    INVALID: "Invalid",
    VALID: "Valid",
};
const EventResponseType = {
    Fail: "Fail",
    InProgress: "InProgress",
    Pass: "Pass",
};
const RiskDecisionType = {
    AccountTakeover: "AccountTakeover",
    Block: "Block",
    NoRisk: "NoRisk",
};
const RiskLevelType = {
    High: "High",
    Low: "Low",
    Medium: "Medium",
};
const EventType = {
    ForgotPassword: "ForgotPassword",
    PasswordChange: "PasswordChange",
    ResendCode: "ResendCode",
    SignIn: "SignIn",
    SignUp: "SignUp",
};
const DeviceRememberedStatusType = {
    NOT_REMEMBERED: "not_remembered",
    REMEMBERED: "remembered",
};
const AdvancedSecurityEnabledModeType = {
    AUDIT: "AUDIT",
    ENFORCED: "ENFORCED",
};
const AdvancedSecurityModeType = {
    AUDIT: "AUDIT",
    ENFORCED: "ENFORCED",
    OFF: "OFF",
};
const AliasAttributeType = {
    EMAIL: "email",
    PHONE_NUMBER: "phone_number",
    PREFERRED_USERNAME: "preferred_username",
};
const AuthFactorType = {
    EMAIL_OTP: "EMAIL_OTP",
    PASSWORD: "PASSWORD",
    SMS_OTP: "SMS_OTP",
    WEB_AUTHN: "WEB_AUTHN",
};
const AssetCategoryType = {
    AUTH_APP_GRAPHIC: "AUTH_APP_GRAPHIC",
    EMAIL_GRAPHIC: "EMAIL_GRAPHIC",
    FAVICON_ICO: "FAVICON_ICO",
    FAVICON_SVG: "FAVICON_SVG",
    FORM_BACKGROUND: "FORM_BACKGROUND",
    FORM_LOGO: "FORM_LOGO",
    IDP_BUTTON_ICON: "IDP_BUTTON_ICON",
    PAGE_BACKGROUND: "PAGE_BACKGROUND",
    PAGE_FOOTER_BACKGROUND: "PAGE_FOOTER_BACKGROUND",
    PAGE_FOOTER_LOGO: "PAGE_FOOTER_LOGO",
    PAGE_HEADER_BACKGROUND: "PAGE_HEADER_BACKGROUND",
    PAGE_HEADER_LOGO: "PAGE_HEADER_LOGO",
    PASSKEY_GRAPHIC: "PASSKEY_GRAPHIC",
    PASSWORD_GRAPHIC: "PASSWORD_GRAPHIC",
    SMS_GRAPHIC: "SMS_GRAPHIC",
};
const AssetExtensionType = {
    ICO: "ICO",
    JPEG: "JPEG",
    PNG: "PNG",
    SVG: "SVG",
    WEBP: "WEBP",
};
const ColorSchemeModeType = {
    DARK: "DARK",
    DYNAMIC: "DYNAMIC",
    LIGHT: "LIGHT",
};
const VerifiedAttributeType = {
    EMAIL: "email",
    PHONE_NUMBER: "phone_number",
};
const IdentityProviderTypeType = {
    Facebook: "Facebook",
    Google: "Google",
    LoginWithAmazon: "LoginWithAmazon",
    OIDC: "OIDC",
    SAML: "SAML",
    SignInWithApple: "SignInWithApple",
};
const TermsEnforcementType = {
    NONE: "NONE",
};
const TermsSourceType = {
    LINK: "LINK",
};
const UserImportJobStatusType = {
    Created: "Created",
    Expired: "Expired",
    Failed: "Failed",
    InProgress: "InProgress",
    Pending: "Pending",
    Stopped: "Stopped",
    Stopping: "Stopping",
    Succeeded: "Succeeded",
};
const DeletionProtectionType = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const EmailSendingAccountType = {
    COGNITO_DEFAULT: "COGNITO_DEFAULT",
    DEVELOPER: "DEVELOPER",
};
const IssuerType = {
    ORIGINAL: "ORIGINAL",
    UPDATED: "UPDATED",
};
const EncryptionKeyType = {
    AWS_OWNED_KEY: "AWS_OWNED_KEY",
    CUSTOMER_MANAGED_KEY: "CUSTOMER_MANAGED_KEY",
};
const CustomEmailSenderLambdaVersionType = {
    V1_0: "V1_0",
};
const CustomSMSSenderLambdaVersionType = {
    V1_0: "V1_0",
};
const InboundFederationLambdaVersionType = {
    V1_0: "V1_0",
};
const PreTokenGenerationLambdaVersionType = {
    V1_0: "V1_0",
    V2_0: "V2_0",
    V3_0: "V3_0",
};
const UserPoolMfaType = {
    OFF: "OFF",
    ON: "ON",
    OPTIONAL: "OPTIONAL",
};
const UsernameAttributeType = {
    EMAIL: "email",
    PHONE_NUMBER: "phone_number",
};
const UserPoolTierType = {
    ESSENTIALS: "ESSENTIALS",
    LITE: "LITE",
    PLUS: "PLUS",
};
const DefaultEmailOptionType = {
    CONFIRM_WITH_CODE: "CONFIRM_WITH_CODE",
    CONFIRM_WITH_LINK: "CONFIRM_WITH_LINK",
};
const StatusType = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
const OAuthFlowType = {
    client_credentials: "client_credentials",
    code: "code",
    implicit: "implicit",
};
const ExplicitAuthFlowsType = {
    ADMIN_NO_SRP_AUTH: "ADMIN_NO_SRP_AUTH",
    ALLOW_ADMIN_USER_PASSWORD_AUTH: "ALLOW_ADMIN_USER_PASSWORD_AUTH",
    ALLOW_CUSTOM_AUTH: "ALLOW_CUSTOM_AUTH",
    ALLOW_REFRESH_TOKEN_AUTH: "ALLOW_REFRESH_TOKEN_AUTH",
    ALLOW_USER_AUTH: "ALLOW_USER_AUTH",
    ALLOW_USER_PASSWORD_AUTH: "ALLOW_USER_PASSWORD_AUTH",
    ALLOW_USER_SRP_AUTH: "ALLOW_USER_SRP_AUTH",
    CUSTOM_AUTH_FLOW_ONLY: "CUSTOM_AUTH_FLOW_ONLY",
    USER_PASSWORD_AUTH: "USER_PASSWORD_AUTH",
};
const PreventUserExistenceErrorTypes = {
    ENABLED: "ENABLED",
    LEGACY: "LEGACY",
};
const FeatureType = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const TimeUnitsType = {
    DAYS: "days",
    HOURS: "hours",
    MINUTES: "minutes",
    SECONDS: "seconds",
};
const ReplicaRoleType = {
    PRIMARY: "PRIMARY",
    SECONDARY: "SECONDARY",
};
const ReplicaStatusType = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    INACTIVE: "INACTIVE",
};
const CompromisedCredentialsEventActionType = {
    BLOCK: "BLOCK",
    NO_ACTION: "NO_ACTION",
};
const EventFilterType = {
    PASSWORD_CHANGE: "PASSWORD_CHANGE",
    SIGN_IN: "SIGN_IN",
    SIGN_UP: "SIGN_UP",
};
const DomainStatusType = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};
const EventSourceName = {
    USER_AUTH_EVENTS: "userAuthEvents",
    USER_NOTIFICATION: "userNotification",
};
const LogLevel = {
    ERROR: "ERROR",
    INFO: "INFO",
};
const WebAuthnFactorConfigurationType = {
    MULTI_FACTOR_WITH_USER_VERIFICATION: "MULTI_FACTOR_WITH_USER_VERIFICATION",
    SINGLE_FACTOR: "SINGLE_FACTOR",
};
const UserVerificationType = {
    PREFERRED: "preferred",
    REQUIRED: "required",
};
const UpdateReplicaStatusType = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const VerifySoftwareTokenResponseType = {
    ERROR: "ERROR",
    SUCCESS: "SUCCESS",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.CognitoIdentityProviderServiceException = CognitoIdentityProviderServiceException.CognitoIdentityProviderServiceException;
exports.AccountTakeoverEventActionType = AccountTakeoverEventActionType;
exports.AddCustomAttributesCommand = AddCustomAttributesCommand;
exports.AddUserPoolClientSecretCommand = AddUserPoolClientSecretCommand;
exports.AdminAddUserToGroupCommand = AdminAddUserToGroupCommand;
exports.AdminConfirmSignUpCommand = AdminConfirmSignUpCommand;
exports.AdminCreateUserCommand = AdminCreateUserCommand;
exports.AdminDeleteUserAttributesCommand = AdminDeleteUserAttributesCommand;
exports.AdminDeleteUserCommand = AdminDeleteUserCommand;
exports.AdminDisableProviderForUserCommand = AdminDisableProviderForUserCommand;
exports.AdminDisableUserCommand = AdminDisableUserCommand;
exports.AdminEnableUserCommand = AdminEnableUserCommand;
exports.AdminForgetDeviceCommand = AdminForgetDeviceCommand;
exports.AdminGetDeviceCommand = AdminGetDeviceCommand;
exports.AdminGetUserCommand = AdminGetUserCommand;
exports.AdminInitiateAuthCommand = AdminInitiateAuthCommand;
exports.AdminLinkProviderForUserCommand = AdminLinkProviderForUserCommand;
exports.AdminListDevicesCommand = AdminListDevicesCommand;
exports.AdminListGroupsForUserCommand = AdminListGroupsForUserCommand;
exports.AdminListUserAuthEventsCommand = AdminListUserAuthEventsCommand;
exports.AdminRemoveUserFromGroupCommand = AdminRemoveUserFromGroupCommand;
exports.AdminResetUserPasswordCommand = AdminResetUserPasswordCommand;
exports.AdminRespondToAuthChallengeCommand = AdminRespondToAuthChallengeCommand;
exports.AdminSetUserMFAPreferenceCommand = AdminSetUserMFAPreferenceCommand;
exports.AdminSetUserPasswordCommand = AdminSetUserPasswordCommand;
exports.AdminSetUserSettingsCommand = AdminSetUserSettingsCommand;
exports.AdminUpdateAuthEventFeedbackCommand = AdminUpdateAuthEventFeedbackCommand;
exports.AdminUpdateDeviceStatusCommand = AdminUpdateDeviceStatusCommand;
exports.AdminUpdateUserAttributesCommand = AdminUpdateUserAttributesCommand;
exports.AdminUserGlobalSignOutCommand = AdminUserGlobalSignOutCommand;
exports.AdvancedSecurityEnabledModeType = AdvancedSecurityEnabledModeType;
exports.AdvancedSecurityModeType = AdvancedSecurityModeType;
exports.AliasAttributeType = AliasAttributeType;
exports.AssetCategoryType = AssetCategoryType;
exports.AssetExtensionType = AssetExtensionType;
exports.AssociateSoftwareTokenCommand = AssociateSoftwareTokenCommand;
exports.AttributeDataType = AttributeDataType;
exports.AuthFactorType = AuthFactorType;
exports.AuthFlowType = AuthFlowType;
exports.ChallengeName = ChallengeName;
exports.ChallengeNameType = ChallengeNameType;
exports.ChallengeResponse = ChallengeResponse;
exports.ChangePasswordCommand = ChangePasswordCommand;
exports.CognitoIdentityProvider = CognitoIdentityProvider;
exports.CognitoIdentityProviderClient = CognitoIdentityProviderClient;
exports.ColorSchemeModeType = ColorSchemeModeType;
exports.CompleteWebAuthnRegistrationCommand = CompleteWebAuthnRegistrationCommand;
exports.CompromisedCredentialsEventActionType = CompromisedCredentialsEventActionType;
exports.ConfirmDeviceCommand = ConfirmDeviceCommand;
exports.ConfirmForgotPasswordCommand = ConfirmForgotPasswordCommand;
exports.ConfirmSignUpCommand = ConfirmSignUpCommand;
exports.CreateGroupCommand = CreateGroupCommand;
exports.CreateIdentityProviderCommand = CreateIdentityProviderCommand;
exports.CreateManagedLoginBrandingCommand = CreateManagedLoginBrandingCommand;
exports.CreateResourceServerCommand = CreateResourceServerCommand;
exports.CreateTermsCommand = CreateTermsCommand;
exports.CreateUserImportJobCommand = CreateUserImportJobCommand;
exports.CreateUserPoolClientCommand = CreateUserPoolClientCommand;
exports.CreateUserPoolCommand = CreateUserPoolCommand;
exports.CreateUserPoolDomainCommand = CreateUserPoolDomainCommand;
exports.CreateUserPoolReplicaCommand = CreateUserPoolReplicaCommand;
exports.CustomEmailSenderLambdaVersionType = CustomEmailSenderLambdaVersionType;
exports.CustomSMSSenderLambdaVersionType = CustomSMSSenderLambdaVersionType;
exports.DefaultEmailOptionType = DefaultEmailOptionType;
exports.DeleteGroupCommand = DeleteGroupCommand;
exports.DeleteIdentityProviderCommand = DeleteIdentityProviderCommand;
exports.DeleteManagedLoginBrandingCommand = DeleteManagedLoginBrandingCommand;
exports.DeleteResourceServerCommand = DeleteResourceServerCommand;
exports.DeleteTermsCommand = DeleteTermsCommand;
exports.DeleteUserAttributesCommand = DeleteUserAttributesCommand;
exports.DeleteUserCommand = DeleteUserCommand;
exports.DeleteUserPoolClientCommand = DeleteUserPoolClientCommand;
exports.DeleteUserPoolClientSecretCommand = DeleteUserPoolClientSecretCommand;
exports.DeleteUserPoolCommand = DeleteUserPoolCommand;
exports.DeleteUserPoolDomainCommand = DeleteUserPoolDomainCommand;
exports.DeleteUserPoolReplicaCommand = DeleteUserPoolReplicaCommand;
exports.DeleteWebAuthnCredentialCommand = DeleteWebAuthnCredentialCommand;
exports.DeletionProtectionType = DeletionProtectionType;
exports.DeliveryMediumType = DeliveryMediumType;
exports.DescribeIdentityProviderCommand = DescribeIdentityProviderCommand;
exports.DescribeManagedLoginBrandingByClientCommand = DescribeManagedLoginBrandingByClientCommand;
exports.DescribeManagedLoginBrandingCommand = DescribeManagedLoginBrandingCommand;
exports.DescribeResourceServerCommand = DescribeResourceServerCommand;
exports.DescribeRiskConfigurationCommand = DescribeRiskConfigurationCommand;
exports.DescribeTermsCommand = DescribeTermsCommand;
exports.DescribeUserImportJobCommand = DescribeUserImportJobCommand;
exports.DescribeUserPoolClientCommand = DescribeUserPoolClientCommand;
exports.DescribeUserPoolCommand = DescribeUserPoolCommand;
exports.DescribeUserPoolDomainCommand = DescribeUserPoolDomainCommand;
exports.DeviceRememberedStatusType = DeviceRememberedStatusType;
exports.DomainStatusType = DomainStatusType;
exports.EmailSendingAccountType = EmailSendingAccountType;
exports.EncryptionKeyType = EncryptionKeyType;
exports.EventFilterType = EventFilterType;
exports.EventResponseType = EventResponseType;
exports.EventSourceName = EventSourceName;
exports.EventType = EventType;
exports.ExplicitAuthFlowsType = ExplicitAuthFlowsType;
exports.FeatureType = FeatureType;
exports.FeedbackValueType = FeedbackValueType;
exports.ForgetDeviceCommand = ForgetDeviceCommand;
exports.ForgotPasswordCommand = ForgotPasswordCommand;
exports.GetCSVHeaderCommand = GetCSVHeaderCommand;
exports.GetDeviceCommand = GetDeviceCommand;
exports.GetGroupCommand = GetGroupCommand;
exports.GetIdentityProviderByIdentifierCommand = GetIdentityProviderByIdentifierCommand;
exports.GetLogDeliveryConfigurationCommand = GetLogDeliveryConfigurationCommand;
exports.GetSigningCertificateCommand = GetSigningCertificateCommand;
exports.GetTokensFromRefreshTokenCommand = GetTokensFromRefreshTokenCommand;
exports.GetUICustomizationCommand = GetUICustomizationCommand;
exports.GetUserAttributeVerificationCodeCommand = GetUserAttributeVerificationCodeCommand;
exports.GetUserAuthFactorsCommand = GetUserAuthFactorsCommand;
exports.GetUserCommand = GetUserCommand;
exports.GetUserPoolMfaConfigCommand = GetUserPoolMfaConfigCommand;
exports.GlobalSignOutCommand = GlobalSignOutCommand;
exports.IdentityProviderTypeType = IdentityProviderTypeType;
exports.InboundFederationLambdaVersionType = InboundFederationLambdaVersionType;
exports.InitiateAuthCommand = InitiateAuthCommand;
exports.IssuerType = IssuerType;
exports.ListDevicesCommand = ListDevicesCommand;
exports.ListGroupsCommand = ListGroupsCommand;
exports.ListIdentityProvidersCommand = ListIdentityProvidersCommand;
exports.ListResourceServersCommand = ListResourceServersCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTermsCommand = ListTermsCommand;
exports.ListUserImportJobsCommand = ListUserImportJobsCommand;
exports.ListUserPoolClientSecretsCommand = ListUserPoolClientSecretsCommand;
exports.ListUserPoolClientsCommand = ListUserPoolClientsCommand;
exports.ListUserPoolReplicasCommand = ListUserPoolReplicasCommand;
exports.ListUserPoolsCommand = ListUserPoolsCommand;
exports.ListUsersCommand = ListUsersCommand;
exports.ListUsersInGroupCommand = ListUsersInGroupCommand;
exports.ListWebAuthnCredentialsCommand = ListWebAuthnCredentialsCommand;
exports.LogLevel = LogLevel;
exports.MessageActionType = MessageActionType;
exports.OAuthFlowType = OAuthFlowType;
exports.PreTokenGenerationLambdaVersionType = PreTokenGenerationLambdaVersionType;
exports.PreventUserExistenceErrorTypes = PreventUserExistenceErrorTypes;
exports.RecoveryOptionNameType = RecoveryOptionNameType;
exports.ReplicaRoleType = ReplicaRoleType;
exports.ReplicaStatusType = ReplicaStatusType;
exports.ResendConfirmationCodeCommand = ResendConfirmationCodeCommand;
exports.RespondToAuthChallengeCommand = RespondToAuthChallengeCommand;
exports.RevokeTokenCommand = RevokeTokenCommand;
exports.RiskDecisionType = RiskDecisionType;
exports.RiskLevelType = RiskLevelType;
exports.SetLogDeliveryConfigurationCommand = SetLogDeliveryConfigurationCommand;
exports.SetRiskConfigurationCommand = SetRiskConfigurationCommand;
exports.SetUICustomizationCommand = SetUICustomizationCommand;
exports.SetUserMFAPreferenceCommand = SetUserMFAPreferenceCommand;
exports.SetUserPoolMfaConfigCommand = SetUserPoolMfaConfigCommand;
exports.SetUserSettingsCommand = SetUserSettingsCommand;
exports.SignUpCommand = SignUpCommand;
exports.StartUserImportJobCommand = StartUserImportJobCommand;
exports.StartWebAuthnRegistrationCommand = StartWebAuthnRegistrationCommand;
exports.StatusType = StatusType;
exports.StopUserImportJobCommand = StopUserImportJobCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TermsEnforcementType = TermsEnforcementType;
exports.TermsSourceType = TermsSourceType;
exports.TimeUnitsType = TimeUnitsType;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateAuthEventFeedbackCommand = UpdateAuthEventFeedbackCommand;
exports.UpdateDeviceStatusCommand = UpdateDeviceStatusCommand;
exports.UpdateGroupCommand = UpdateGroupCommand;
exports.UpdateIdentityProviderCommand = UpdateIdentityProviderCommand;
exports.UpdateManagedLoginBrandingCommand = UpdateManagedLoginBrandingCommand;
exports.UpdateReplicaStatusType = UpdateReplicaStatusType;
exports.UpdateResourceServerCommand = UpdateResourceServerCommand;
exports.UpdateTermsCommand = UpdateTermsCommand;
exports.UpdateUserAttributesCommand = UpdateUserAttributesCommand;
exports.UpdateUserPoolClientCommand = UpdateUserPoolClientCommand;
exports.UpdateUserPoolCommand = UpdateUserPoolCommand;
exports.UpdateUserPoolDomainCommand = UpdateUserPoolDomainCommand;
exports.UpdateUserPoolReplicaCommand = UpdateUserPoolReplicaCommand;
exports.UserImportJobStatusType = UserImportJobStatusType;
exports.UserPoolMfaType = UserPoolMfaType;
exports.UserPoolTierType = UserPoolTierType;
exports.UserStatusType = UserStatusType;
exports.UserVerificationType = UserVerificationType;
exports.UsernameAttributeType = UsernameAttributeType;
exports.VerifiedAttributeType = VerifiedAttributeType;
exports.VerifySoftwareTokenCommand = VerifySoftwareTokenCommand;
exports.VerifySoftwareTokenResponseType = VerifySoftwareTokenResponseType;
exports.VerifyUserAttributeCommand = VerifyUserAttributeCommand;
exports.WebAuthnFactorConfigurationType = WebAuthnFactorConfigurationType;
exports.paginateAdminListGroupsForUser = paginateAdminListGroupsForUser;
exports.paginateAdminListUserAuthEvents = paginateAdminListUserAuthEvents;
exports.paginateListGroups = paginateListGroups;
exports.paginateListIdentityProviders = paginateListIdentityProviders;
exports.paginateListResourceServers = paginateListResourceServers;
exports.paginateListUserPoolClients = paginateListUserPoolClients;
exports.paginateListUserPools = paginateListUserPools;
exports.paginateListUsers = paginateListUsers;
exports.paginateListUsersInGroup = paginateListUsersInGroup;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
