// @aws-sdk/client-cloudformation/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceScanLimitExceededException = exports.OperationStatusCheckFailedException = exports.InvalidStateTransitionException = exports.ResourceScanInProgressException = exports.StackNotFoundException = exports.HookResultNotFoundException = exports.StackRefactorNotFoundException = exports.StackInstanceNotFoundException = exports.ResourceScanNotFoundException = exports.StackSetNotEmptyException = exports.GeneratedTemplateNotFoundException = exports.InvalidChangeSetStatusException = exports.NameAlreadyExistsException = exports.CreatedButModifiedException = exports.StaleRequestException = exports.StackSetNotFoundException = exports.OperationInProgressException = exports.OperationIdAlreadyExistsException = exports.ConcurrentResourcesLimitExceededException = exports.LimitExceededException = exports.InsufficientCapabilitiesException = exports.ChangeSetNotFoundException = exports.TokenAlreadyExistsException = exports.TypeConfigurationNotFoundException = exports.AlreadyExistsException = exports.TypeNotFoundException = exports.CFNRegistryException = exports.OperationNotFoundException = exports.InvalidOperationException = void 0;
const CloudFormationServiceException_1 = require("./CloudFormationServiceException");
class InvalidOperationException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "InvalidOperationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidOperationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOperationException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOperationException = InvalidOperationException;
class OperationNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "OperationNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.OperationNotFoundException = OperationNotFoundException;
class CFNRegistryException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "CFNRegistryException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CFNRegistryException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CFNRegistryException.prototype);
        this.Message = opts.Message;
    }
}
exports.CFNRegistryException = CFNRegistryException;
class TypeNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "TypeNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TypeNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TypeNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.TypeNotFoundException = TypeNotFoundException;
class AlreadyExistsException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "AlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.AlreadyExistsException = AlreadyExistsException;
class TypeConfigurationNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "TypeConfigurationNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TypeConfigurationNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TypeConfigurationNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.TypeConfigurationNotFoundException = TypeConfigurationNotFoundException;
class TokenAlreadyExistsException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "TokenAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TokenAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TokenAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.TokenAlreadyExistsException = TokenAlreadyExistsException;
class ChangeSetNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "ChangeSetNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ChangeSetNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ChangeSetNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.ChangeSetNotFoundException = ChangeSetNotFoundException;
class InsufficientCapabilitiesException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "InsufficientCapabilitiesException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InsufficientCapabilitiesException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InsufficientCapabilitiesException.prototype);
        this.Message = opts.Message;
    }
}
exports.InsufficientCapabilitiesException = InsufficientCapabilitiesException;
class LimitExceededException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "LimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.LimitExceededException = LimitExceededException;
class ConcurrentResourcesLimitExceededException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "ConcurrentResourcesLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConcurrentResourcesLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentResourcesLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConcurrentResourcesLimitExceededException = ConcurrentResourcesLimitExceededException;
class OperationIdAlreadyExistsException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "OperationIdAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationIdAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationIdAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.OperationIdAlreadyExistsException = OperationIdAlreadyExistsException;
class OperationInProgressException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "OperationInProgressException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationInProgressException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationInProgressException.prototype);
        this.Message = opts.Message;
    }
}
exports.OperationInProgressException = OperationInProgressException;
class StackSetNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StackSetNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StackSetNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StackSetNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.StackSetNotFoundException = StackSetNotFoundException;
class StaleRequestException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StaleRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StaleRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StaleRequestException.prototype);
        this.Message = opts.Message;
    }
}
exports.StaleRequestException = StaleRequestException;
class CreatedButModifiedException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "CreatedButModifiedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CreatedButModifiedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CreatedButModifiedException.prototype);
        this.Message = opts.Message;
    }
}
exports.CreatedButModifiedException = CreatedButModifiedException;
class NameAlreadyExistsException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "NameAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NameAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NameAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.NameAlreadyExistsException = NameAlreadyExistsException;
class InvalidChangeSetStatusException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "InvalidChangeSetStatusException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidChangeSetStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidChangeSetStatusException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidChangeSetStatusException = InvalidChangeSetStatusException;
class GeneratedTemplateNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "GeneratedTemplateNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "GeneratedTemplateNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GeneratedTemplateNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.GeneratedTemplateNotFoundException = GeneratedTemplateNotFoundException;
class StackSetNotEmptyException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StackSetNotEmptyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StackSetNotEmptyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StackSetNotEmptyException.prototype);
        this.Message = opts.Message;
    }
}
exports.StackSetNotEmptyException = StackSetNotEmptyException;
class ResourceScanNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "ResourceScanNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceScanNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceScanNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceScanNotFoundException = ResourceScanNotFoundException;
class StackInstanceNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StackInstanceNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StackInstanceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StackInstanceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.StackInstanceNotFoundException = StackInstanceNotFoundException;
class StackRefactorNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StackRefactorNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StackRefactorNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StackRefactorNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.StackRefactorNotFoundException = StackRefactorNotFoundException;
class HookResultNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "HookResultNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "HookResultNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HookResultNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.HookResultNotFoundException = HookResultNotFoundException;
class StackNotFoundException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "StackNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "StackNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StackNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.StackNotFoundException = StackNotFoundException;
class ResourceScanInProgressException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "ResourceScanInProgressException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceScanInProgressException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceScanInProgressException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceScanInProgressException = ResourceScanInProgressException;
class InvalidStateTransitionException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "InvalidStateTransitionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidStateTransitionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidStateTransitionException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidStateTransitionException = InvalidStateTransitionException;
class OperationStatusCheckFailedException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "OperationStatusCheckFailedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationStatusCheckFailedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationStatusCheckFailedException.prototype);
        this.Message = opts.Message;
    }
}
exports.OperationStatusCheckFailedException = OperationStatusCheckFailedException;
class ResourceScanLimitExceededException extends CloudFormationServiceException_1.CloudFormationServiceException {
    name = "ResourceScanLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceScanLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceScanLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceScanLimitExceededException = ResourceScanLimitExceededException;
