// @aws-sdk/client-kinesis/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListShardsInput$ = exports.IncreaseStreamRetentionPeriodInput$ = exports.HashKeyRange$ = exports.GetShardIteratorOutput$ = exports.GetShardIteratorInput$ = exports.GetResourcePolicyOutput$ = exports.GetResourcePolicyInput$ = exports.GetRecordsOutput$ = exports.GetRecordsInput$ = exports.EnhancedMonitoringOutput$ = exports.EnhancedMetrics$ = exports.EnableEnhancedMonitoringInput$ = exports.DisableEnhancedMonitoringInput$ = exports.DescribeStreamSummaryOutput$ = exports.DescribeStreamSummaryInput$ = exports.DescribeStreamOutput$ = exports.DescribeStreamInput$ = exports.DescribeStreamConsumerOutput$ = exports.DescribeStreamConsumerInput$ = exports.DescribeLimitsOutput$ = exports.DescribeLimitsInput$ = exports.DescribeAccountSettingsOutput$ = exports.DescribeAccountSettingsInput$ = exports.DeregisterStreamConsumerInput$ = exports.DeleteStreamInput$ = exports.DeleteResourcePolicyInput$ = exports.DecreaseStreamRetentionPeriodInput$ = exports.CreateStreamInput$ = exports.ConsumerDescription$ = exports.Consumer$ = exports.ChildShard$ = exports.AddTagsToStreamInput$ = exports.errorTypeRegistries = exports.ValidationException$ = exports.ResourceNotFoundException$ = exports.ResourceInUseException$ = exports.ProvisionedThroughputExceededException$ = exports.LimitExceededException$ = exports.KMSThrottlingException$ = exports.KMSOptInRequired$ = exports.KMSNotFoundException$ = exports.KMSInvalidStateException$ = exports.KMSDisabledException$ = exports.KMSAccessDeniedException$ = exports.InvalidArgumentException$ = exports.InternalFailureException$ = exports.ExpiredNextTokenException$ = exports.ExpiredIteratorException$ = exports.AccessDeniedException$ = exports.KinesisServiceException$ = void 0;
exports.SubscribeToShardEventStream$ = exports.WarmThroughputObject$ = exports.UpdateStreamWarmThroughputOutput$ = exports.UpdateStreamWarmThroughputInput$ = exports.UpdateStreamModeInput$ = exports.UpdateShardCountOutput$ = exports.UpdateShardCountInput$ = exports.UpdateMaxRecordSizeInput$ = exports.UpdateAccountSettingsOutput$ = exports.UpdateAccountSettingsInput$ = exports.UntagResourceInput$ = exports.TagResourceInput$ = exports.Tag$ = exports.SubscribeToShardOutput$ = exports.SubscribeToShardInput$ = exports.SubscribeToShardEvent$ = exports.StreamSummary$ = exports.StreamModeDetails$ = exports.StreamDescriptionSummary$ = exports.StreamDescription$ = exports.StopStreamEncryptionInput$ = exports.StartStreamEncryptionInput$ = exports.StartingPosition$ = exports.SplitShardInput$ = exports.ShardFilter$ = exports.Shard$ = exports.SequenceNumberRange$ = exports.RemoveTagsFromStreamInput$ = exports.RegisterStreamConsumerOutput$ = exports.RegisterStreamConsumerInput$ = exports._Record$ = exports.PutResourcePolicyInput$ = exports.PutRecordsResultEntry$ = exports.PutRecordsRequestEntry$ = exports.PutRecordsOutput$ = exports.PutRecordsInput$ = exports.PutRecordOutput$ = exports.PutRecordInput$ = exports.MinimumThroughputBillingCommitmentOutput$ = exports.MinimumThroughputBillingCommitmentInput$ = exports.MergeShardsInput$ = exports.ListTagsForStreamOutput$ = exports.ListTagsForStreamInput$ = exports.ListTagsForResourceOutput$ = exports.ListTagsForResourceInput$ = exports.ListStreamsOutput$ = exports.ListStreamsInput$ = exports.ListStreamConsumersOutput$ = exports.ListStreamConsumersInput$ = exports.ListShardsOutput$ = void 0;
exports.UpdateStreamWarmThroughput$ = exports.UpdateStreamMode$ = exports.UpdateShardCount$ = exports.UpdateMaxRecordSize$ = exports.UpdateAccountSettings$ = exports.UntagResource$ = exports.TagResource$ = exports.SubscribeToShard$ = exports.StopStreamEncryption$ = exports.StartStreamEncryption$ = exports.SplitShard$ = exports.RemoveTagsFromStream$ = exports.RegisterStreamConsumer$ = exports.PutResourcePolicy$ = exports.PutRecords$ = exports.PutRecord$ = exports.MergeShards$ = exports.ListTagsForStream$ = exports.ListTagsForResource$ = exports.ListStreams$ = exports.ListStreamConsumers$ = exports.ListShards$ = exports.IncreaseStreamRetentionPeriod$ = exports.GetShardIterator$ = exports.GetResourcePolicy$ = exports.GetRecords$ = exports.EnableEnhancedMonitoring$ = exports.DisableEnhancedMonitoring$ = exports.DescribeStreamSummary$ = exports.DescribeStreamConsumer$ = exports.DescribeStream$ = exports.DescribeLimits$ = exports.DescribeAccountSettings$ = exports.DeregisterStreamConsumer$ = exports.DeleteStream$ = exports.DeleteResourcePolicy$ = exports.DecreaseStreamRetentionPeriod$ = exports.CreateStream$ = exports.AddTagsToStream$ = void 0;
const _AAT = "ApproximateArrivalTimestamp";
const _ADE = "AccessDeniedException";
const _APSI = "AdjacentParentShardId";
const _ASTM = "AdjacentShardToMerge";
const _ATTS = "AddTagsToStream";
const _ATTSI = "AddTagsToStreamInput";
const _C = "Consumer";
const _CARN = "ConsumerARN";
const _CC = "ConsumerCount";
const _CCT = "ConsumerCreationTimestamp";
const _CD = "ConsumerDescription";
const _CL = "ConsumerList";
const _CMB = "CurrentMiBps";
const _CN = "ConsumerName";
const _CS = "ChildShard";
const _CSC = "CurrentShardCount";
const _CSI = "CreateStreamInput";
const _CSL = "ChildShardList";
const _CSLM = "CurrentShardLevelMetrics";
const _CSN = "ContinuationSequenceNumber";
const _CSh = "ChildShards";
const _CSo = "ConsumerStatus";
const _CSr = "CreateStream";
const _Co = "Consumers";
const _D = "Data";
const _DAS = "DescribeAccountSettings";
const _DASI = "DescribeAccountSettingsInput";
const _DASO = "DescribeAccountSettingsOutput";
const _DEM = "DisableEnhancedMonitoring";
const _DEMI = "DisableEnhancedMonitoringInput";
const _DL = "DescribeLimits";
const _DLI = "DescribeLimitsInput";
const _DLO = "DescribeLimitsOutput";
const _DRP = "DeleteResourcePolicy";
const _DRPI = "DeleteResourcePolicyInput";
const _DS = "DeleteStream";
const _DSC = "DeregisterStreamConsumer";
const _DSCI = "DeregisterStreamConsumerInput";
const _DSCIe = "DescribeStreamConsumerInput";
const _DSCO = "DescribeStreamConsumerOutput";
const _DSCe = "DescribeStreamConsumer";
const _DSI = "DeleteStreamInput";
const _DSIe = "DescribeStreamInput";
const _DSLM = "DesiredShardLevelMetrics";
const _DSO = "DescribeStreamOutput";
const _DSRP = "DecreaseStreamRetentionPeriod";
const _DSRPI = "DecreaseStreamRetentionPeriodInput";
const _DSS = "DescribeStreamSummary";
const _DSSI = "DescribeStreamSummaryInput";
const _DSSO = "DescribeStreamSummaryOutput";
const _DSe = "DescribeStream";
const _EA = "EndedAt";
const _EAEA = "EarliestAllowedEndAt";
const _EC = "ErrorCode";
const _ECD = "EnforceConsumerDeletion";
const _EEM = "EnableEnhancedMonitoring";
const _EEMI = "EnableEnhancedMonitoringInput";
const _EHK = "EndingHashKey";
const _EHKx = "ExplicitHashKey";
const _EIE = "ExpiredIteratorException";
const _EM = "EnhancedMetrics";
const _EML = "EnhancedMonitoringList";
const _EMO = "EnhancedMonitoringOutput";
const _EMn = "EnhancedMonitoring";
const _EMr = "ErrorMessage";
const _ENTE = "ExpiredNextTokenException";
const _ES = "EventStream";
const _ESN = "EndingSequenceNumber";
const _ESSI = "ExclusiveStartShardId";
const _ESSN = "ExclusiveStartStreamName";
const _ESTK = "ExclusiveStartTagKey";
const _ET = "EncryptionType";
const _FRC = "FailedRecordCount";
const _GR = "GetRecords";
const _GRI = "GetRecordsInput";
const _GRO = "GetRecordsOutput";
const _GRP = "GetResourcePolicy";
const _GRPI = "GetResourcePolicyInput";
const _GRPO = "GetResourcePolicyOutput";
const _GSI = "GetShardIterator";
const _GSII = "GetShardIteratorInput";
const _GSIO = "GetShardIteratorOutput";
const _HKR = "HashKeyRange";
const _HMS = "HasMoreStreams";
const _HMSa = "HasMoreShards";
const _HMT = "HasMoreTags";
const _IAE = "InvalidArgumentException";
const _IFE = "InternalFailureException";
const _ISRP = "IncreaseStreamRetentionPeriod";
const _ISRPI = "IncreaseStreamRetentionPeriodInput";
const _K = "Key";
const _KI = "KeyId";
const _KMSADE = "KMSAccessDeniedException";
const _KMSDE = "KMSDisabledException";
const _KMSISE = "KMSInvalidStateException";
const _KMSNFE = "KMSNotFoundException";
const _KMSOIR = "KMSOptInRequired";
const _KMSTE = "KMSThrottlingException";
const _L = "Limit";
const _LEE = "LimitExceededException";
const _LS = "ListShards";
const _LSC = "ListStreamConsumers";
const _LSCI = "ListStreamConsumersInput";
const _LSCO = "ListStreamConsumersOutput";
const _LSI = "ListShardsInput";
const _LSIi = "ListStreamsInput";
const _LSO = "ListShardsOutput";
const _LSOi = "ListStreamsOutput";
const _LSi = "ListStreams";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _LTFS = "ListTagsForStream";
const _LTFSI = "ListTagsForStreamInput";
const _LTFSO = "ListTagsForStreamOutput";
const _MBL = "MillisBehindLatest";
const _MR = "MaxResults";
const _MRSIKB = "MaxRecordSizeInKiB";
const _MS = "MergeShards";
const _MSI = "MergeShardsInput";
const _MTBC = "MinimumThroughputBillingCommitment";
const _MTBCI = "MinimumThroughputBillingCommitmentInput";
const _MTBCO = "MinimumThroughputBillingCommitmentOutput";
const _NSHK = "NewStartingHashKey";
const _NSI = "NextShardIterator";
const _NT = "NextToken";
const _ODSC = "OnDemandStreamCount";
const _ODSCL = "OnDemandStreamCountLimit";
const _OSC = "OpenShardCount";
const _P = "Policy";
const _PK = "PartitionKey";
const _PR = "PutRecord";
const _PRI = "PutRecordInput";
const _PRIu = "PutRecordsInput";
const _PRO = "PutRecordOutput";
const _PROu = "PutRecordsOutput";
const _PRP = "PutResourcePolicy";
const _PRPI = "PutResourcePolicyInput";
const _PRRE = "PutRecordsRequestEntry";
const _PRREL = "PutRecordsRequestEntryList";
const _PRRELu = "PutRecordsResultEntryList";
const _PRREu = "PutRecordsResultEntry";
const _PRu = "PutRecords";
const _PS = "ParentShards";
const _PSI = "ParentShardId";
const _PTEE = "ProvisionedThroughputExceededException";
const _R = "Records";
const _RARN = "ResourceARN";
const _RIUE = "ResourceInUseException";
const _RL = "RecordList";
const _RNFE = "ResourceNotFoundException";
const _RPH = "RetentionPeriodHours";
const _RSC = "RegisterStreamConsumer";
const _RSCI = "RegisterStreamConsumerInput";
const _RSCO = "RegisterStreamConsumerOutput";
const _RTFS = "RemoveTagsFromStream";
const _RTFSI = "RemoveTagsFromStreamInput";
const _Re = "Record";
const _S = "Shards";
const _SA = "StartedAt";
const _SARN = "StreamARN";
const _SC = "ShardCount";
const _SCT = "StreamCreationTimestamp";
const _SD = "StreamDescription";
const _SDS = "StreamDescriptionSummary";
const _SF = "ShardFilter";
const _SHK = "StartingHashKey";
const _SI = "StreamId";
const _SIT = "ShardIteratorType";
const _SIh = "ShardId";
const _SIha = "ShardIterator";
const _SL = "ShardLimit";
const _SLM = "ShardLevelMetrics";
const _SLh = "ShardList";
const _SM = "StreamMode";
const _SMD = "StreamModeDetails";
const _SN = "StreamName";
const _SNFO = "SequenceNumberForOrdering";
const _SNR = "SequenceNumberRange";
const _SNe = "SequenceNumber";
const _SNt = "StreamNames";
const _SP = "StartingPosition";
const _SS = "StreamSummaries";
const _SSE = "StartStreamEncryption";
const _SSEI = "StartStreamEncryptionInput";
const _SSEIt = "StopStreamEncryptionInput";
const _SSEt = "StopStreamEncryption";
const _SSI = "SplitShardInput";
const _SSL = "StreamSummaryList";
const _SSN = "StartingSequenceNumber";
const _SSp = "SplitShard";
const _SSt = "StreamStatus";
const _SStr = "StreamSummary";
const _ST = "ScalingType";
const _STM = "ShardToMerge";
const _STS = "ShardToSplit";
const _STSE = "SubscribeToShardEvent";
const _STSES = "SubscribeToShardEventStream";
const _STSI = "SubscribeToShardInput";
const _STSO = "SubscribeToShardOutput";
const _STSu = "SubscribeToShard";
const _Sh = "Shard";
const _St = "Status";
const _T = "Tags";
const _TK = "TagKeys";
const _TL = "TagList";
const _TMB = "TargetMiBps";
const _TR = "TagResource";
const _TRI = "TagResourceInput";
const _TSC = "TargetShardCount";
const _Ta = "Tag";
const _Ti = "Timestamp";
const _Ty = "Type";
const _UAS = "UpdateAccountSettings";
const _UASI = "UpdateAccountSettingsInput";
const _UASO = "UpdateAccountSettingsOutput";
const _UMRS = "UpdateMaxRecordSize";
const _UMRSI = "UpdateMaxRecordSizeInput";
const _UR = "UntagResource";
const _URI = "UntagResourceInput";
const _USC = "UpdateShardCount";
const _USCI = "UpdateShardCountInput";
const _USCO = "UpdateShardCountOutput";
const _USM = "UpdateStreamMode";
const _USMI = "UpdateStreamModeInput";
const _USWT = "UpdateStreamWarmThroughput";
const _USWTI = "UpdateStreamWarmThroughputInput";
const _USWTO = "UpdateStreamWarmThroughputOutput";
const _V = "Value";
const _VE = "ValidationException";
const _WT = "WarmThroughput";
const _WTMB = "WarmThroughputMiBps";
const _WTO = "WarmThroughputObject";
const _c = "client";
const _e = "error";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.kinesis";
const _se = "server";
const _st = "streaming";
const n0 = "com.amazonaws.kinesis";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const KinesisServiceException_1 = require("../models/KinesisServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.KinesisServiceException$ = [-3, _s, "KinesisServiceException", 0, [], []];
_s_registry.registerError(exports.KinesisServiceException$, KinesisServiceException_1.KinesisServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.AccessDeniedException$, errors_1.AccessDeniedException);
exports.ExpiredIteratorException$ = [-3, n0, _EIE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ExpiredIteratorException$, errors_1.ExpiredIteratorException);
exports.ExpiredNextTokenException$ = [-3, n0, _ENTE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ExpiredNextTokenException$, errors_1.ExpiredNextTokenException);
exports.InternalFailureException$ = [-3, n0, _IFE,
    { [_e]: _se },
    [_m],
    [0]
];
n0_registry.registerError(exports.InternalFailureException$, errors_1.InternalFailureException);
exports.InvalidArgumentException$ = [-3, n0, _IAE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidArgumentException$, errors_1.InvalidArgumentException);
exports.KMSAccessDeniedException$ = [-3, n0, _KMSADE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSAccessDeniedException$, errors_1.KMSAccessDeniedException);
exports.KMSDisabledException$ = [-3, n0, _KMSDE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSDisabledException$, errors_1.KMSDisabledException);
exports.KMSInvalidStateException$ = [-3, n0, _KMSISE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSInvalidStateException$, errors_1.KMSInvalidStateException);
exports.KMSNotFoundException$ = [-3, n0, _KMSNFE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSNotFoundException$, errors_1.KMSNotFoundException);
exports.KMSOptInRequired$ = [-3, n0, _KMSOIR,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSOptInRequired$, errors_1.KMSOptInRequired);
exports.KMSThrottlingException$ = [-3, n0, _KMSTE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.KMSThrottlingException$, errors_1.KMSThrottlingException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.ProvisionedThroughputExceededException$ = [-3, n0, _PTEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ProvisionedThroughputExceededException$, errors_1.ProvisionedThroughputExceededException);
exports.ResourceInUseException$ = [-3, n0, _RIUE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceInUseException$, errors_1.ResourceInUseException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.ValidationException$ = [-3, n0, _VE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ValidationException$, errors_1.ValidationException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
exports.AddTagsToStreamInput$ = [3, n0, _ATTSI,
    0,
    [_T, _SN, _SARN, _SI],
    [128 | 0, 0, 0, 0], 1
];
exports.ChildShard$ = [3, n0, _CS,
    0,
    [_SIh, _PS, _HKR],
    [0, 64 | 0, () => exports.HashKeyRange$], 3
];
exports.Consumer$ = [3, n0, _C,
    0,
    [_CN, _CARN, _CSo, _CCT],
    [0, 0, 0, 4], 4
];
exports.ConsumerDescription$ = [3, n0, _CD,
    0,
    [_CN, _CARN, _CSo, _CCT, _SARN],
    [0, 0, 0, 4, 0], 5
];
exports.CreateStreamInput$ = [3, n0, _CSI,
    0,
    [_SN, _SC, _SMD, _T, _WTMB, _MRSIKB],
    [0, 1, () => exports.StreamModeDetails$, 128 | 0, 1, 1], 1
];
exports.DecreaseStreamRetentionPeriodInput$ = [3, n0, _DSRPI,
    0,
    [_RPH, _SN, _SARN, _SI],
    [1, 0, 0, 0], 1
];
exports.DeleteResourcePolicyInput$ = [3, n0, _DRPI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.DeleteStreamInput$ = [3, n0, _DSI,
    0,
    [_SN, _ECD, _SARN, _SI],
    [0, 2, 0, 0]
];
exports.DeregisterStreamConsumerInput$ = [3, n0, _DSCI,
    0,
    [_SARN, _CN, _CARN, _SI],
    [0, 0, 0, 0]
];
exports.DescribeAccountSettingsInput$ = [3, n0, _DASI,
    0,
    [],
    []
];
exports.DescribeAccountSettingsOutput$ = [3, n0, _DASO,
    0,
    [_MTBC],
    [() => exports.MinimumThroughputBillingCommitmentOutput$]
];
exports.DescribeLimitsInput$ = [3, n0, _DLI,
    0,
    [],
    []
];
exports.DescribeLimitsOutput$ = [3, n0, _DLO,
    0,
    [_SL, _OSC, _ODSC, _ODSCL],
    [1, 1, 1, 1], 4
];
exports.DescribeStreamConsumerInput$ = [3, n0, _DSCIe,
    0,
    [_SARN, _CN, _CARN, _SI],
    [0, 0, 0, 0]
];
exports.DescribeStreamConsumerOutput$ = [3, n0, _DSCO,
    0,
    [_CD],
    [() => exports.ConsumerDescription$], 1
];
exports.DescribeStreamInput$ = [3, n0, _DSIe,
    0,
    [_SN, _L, _ESSI, _SARN, _SI],
    [0, 1, 0, 0, 0]
];
exports.DescribeStreamOutput$ = [3, n0, _DSO,
    0,
    [_SD],
    [() => exports.StreamDescription$], 1
];
exports.DescribeStreamSummaryInput$ = [3, n0, _DSSI,
    0,
    [_SN, _SARN, _SI],
    [0, 0, 0]
];
exports.DescribeStreamSummaryOutput$ = [3, n0, _DSSO,
    0,
    [_SDS],
    [() => exports.StreamDescriptionSummary$], 1
];
exports.DisableEnhancedMonitoringInput$ = [3, n0, _DEMI,
    0,
    [_SLM, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.EnableEnhancedMonitoringInput$ = [3, n0, _EEMI,
    0,
    [_SLM, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.EnhancedMetrics$ = [3, n0, _EM,
    0,
    [_SLM],
    [64 | 0]
];
exports.EnhancedMonitoringOutput$ = [3, n0, _EMO,
    0,
    [_SN, _CSLM, _DSLM, _SARN],
    [0, 64 | 0, 64 | 0, 0]
];
exports.GetRecordsInput$ = [3, n0, _GRI,
    0,
    [_SIha, _L, _SARN, _SI],
    [0, 1, 0, 0], 1
];
exports.GetRecordsOutput$ = [3, n0, _GRO,
    0,
    [_R, _NSI, _MBL, _CSh],
    [() => RecordList, 0, 1, () => ChildShardList], 1
];
exports.GetResourcePolicyInput$ = [3, n0, _GRPI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.GetResourcePolicyOutput$ = [3, n0, _GRPO,
    0,
    [_P],
    [0], 1
];
exports.GetShardIteratorInput$ = [3, n0, _GSII,
    0,
    [_SIh, _SIT, _SN, _SSN, _Ti, _SARN, _SI],
    [0, 0, 0, 0, 4, 0, 0], 2
];
exports.GetShardIteratorOutput$ = [3, n0, _GSIO,
    0,
    [_SIha],
    [0]
];
exports.HashKeyRange$ = [3, n0, _HKR,
    0,
    [_SHK, _EHK],
    [0, 0], 2
];
exports.IncreaseStreamRetentionPeriodInput$ = [3, n0, _ISRPI,
    0,
    [_RPH, _SN, _SARN, _SI],
    [1, 0, 0, 0], 1
];
exports.ListShardsInput$ = [3, n0, _LSI,
    0,
    [_SN, _NT, _ESSI, _MR, _SCT, _SF, _SARN, _SI],
    [0, 0, 0, 1, 4, () => exports.ShardFilter$, 0, 0]
];
exports.ListShardsOutput$ = [3, n0, _LSO,
    0,
    [_S, _NT],
    [() => ShardList, 0]
];
exports.ListStreamConsumersInput$ = [3, n0, _LSCI,
    0,
    [_SARN, _NT, _MR, _SCT, _SI],
    [0, 0, 1, 4, 0], 1
];
exports.ListStreamConsumersOutput$ = [3, n0, _LSCO,
    0,
    [_Co, _NT],
    [() => ConsumerList, 0]
];
exports.ListStreamsInput$ = [3, n0, _LSIi,
    0,
    [_L, _ESSN, _NT],
    [1, 0, 0]
];
exports.ListStreamsOutput$ = [3, n0, _LSOi,
    0,
    [_SNt, _HMS, _NT, _SS],
    [64 | 0, 2, 0, () => StreamSummaryList], 2
];
exports.ListTagsForResourceInput$ = [3, n0, _LTFRI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.ListTagsForResourceOutput$ = [3, n0, _LTFRO,
    0,
    [_T],
    [() => TagList]
];
exports.ListTagsForStreamInput$ = [3, n0, _LTFSI,
    0,
    [_SN, _ESTK, _L, _SARN, _SI],
    [0, 0, 1, 0, 0]
];
exports.ListTagsForStreamOutput$ = [3, n0, _LTFSO,
    0,
    [_T, _HMT],
    [() => TagList, 2], 2
];
exports.MergeShardsInput$ = [3, n0, _MSI,
    0,
    [_STM, _ASTM, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.MinimumThroughputBillingCommitmentInput$ = [3, n0, _MTBCI,
    0,
    [_St],
    [0], 1
];
exports.MinimumThroughputBillingCommitmentOutput$ = [3, n0, _MTBCO,
    0,
    [_St, _SA, _EA, _EAEA],
    [0, 4, 4, 4], 1
];
exports.PutRecordInput$ = [3, n0, _PRI,
    0,
    [_D, _PK, _SN, _EHKx, _SNFO, _SARN, _SI],
    [21, 0, 0, 0, 0, 0, 0], 2
];
exports.PutRecordOutput$ = [3, n0, _PRO,
    0,
    [_SIh, _SNe, _ET],
    [0, 0, 0], 2
];
exports.PutRecordsInput$ = [3, n0, _PRIu,
    0,
    [_R, _SN, _SARN, _SI],
    [() => PutRecordsRequestEntryList, 0, 0, 0], 1
];
exports.PutRecordsOutput$ = [3, n0, _PROu,
    0,
    [_R, _FRC, _ET],
    [() => PutRecordsResultEntryList, 1, 0], 1
];
exports.PutRecordsRequestEntry$ = [3, n0, _PRRE,
    0,
    [_D, _PK, _EHKx],
    [21, 0, 0], 2
];
exports.PutRecordsResultEntry$ = [3, n0, _PRREu,
    0,
    [_SNe, _SIh, _EC, _EMr],
    [0, 0, 0, 0]
];
exports.PutResourcePolicyInput$ = [3, n0, _PRPI,
    0,
    [_RARN, _P, _SI],
    [0, 0, 0], 2
];
exports._Record$ = [3, n0, _Re,
    0,
    [_SNe, _D, _PK, _AAT, _ET],
    [0, 21, 0, 4, 0], 3
];
exports.RegisterStreamConsumerInput$ = [3, n0, _RSCI,
    0,
    [_SARN, _CN, _SI, _T],
    [0, 0, 0, 128 | 0], 2
];
exports.RegisterStreamConsumerOutput$ = [3, n0, _RSCO,
    0,
    [_C],
    [() => exports.Consumer$], 1
];
exports.RemoveTagsFromStreamInput$ = [3, n0, _RTFSI,
    0,
    [_TK, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.SequenceNumberRange$ = [3, n0, _SNR,
    0,
    [_SSN, _ESN],
    [0, 0], 1
];
exports.Shard$ = [3, n0, _Sh,
    0,
    [_SIh, _HKR, _SNR, _PSI, _APSI],
    [0, () => exports.HashKeyRange$, () => exports.SequenceNumberRange$, 0, 0], 3
];
exports.ShardFilter$ = [3, n0, _SF,
    0,
    [_Ty, _SIh, _Ti],
    [0, 0, 4], 1
];
exports.SplitShardInput$ = [3, n0, _SSI,
    0,
    [_STS, _NSHK, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.StartingPosition$ = [3, n0, _SP,
    0,
    [_Ty, _SNe, _Ti],
    [0, 0, 4], 1
];
exports.StartStreamEncryptionInput$ = [3, n0, _SSEI,
    0,
    [_ET, _KI, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.StopStreamEncryptionInput$ = [3, n0, _SSEIt,
    0,
    [_ET, _KI, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.StreamDescription$ = [3, n0, _SD,
    0,
    [_SN, _SARN, _SSt, _S, _HMSa, _RPH, _SCT, _EMn, _SMD, _ET, _KI],
    [0, 0, 0, () => ShardList, 2, 1, 4, () => EnhancedMonitoringList, () => exports.StreamModeDetails$, 0, 0], 8
];
exports.StreamDescriptionSummary$ = [3, n0, _SDS,
    0,
    [_SN, _SARN, _SSt, _RPH, _SCT, _EMn, _OSC, _SI, _SMD, _ET, _KI, _CC, _WT, _MRSIKB],
    [0, 0, 0, 1, 4, () => EnhancedMonitoringList, 1, 0, () => exports.StreamModeDetails$, 0, 0, 1, () => exports.WarmThroughputObject$, 1], 7
];
exports.StreamModeDetails$ = [3, n0, _SMD,
    0,
    [_SM],
    [0], 1
];
exports.StreamSummary$ = [3, n0, _SStr,
    0,
    [_SN, _SARN, _SSt, _SMD, _SCT],
    [0, 0, 0, () => exports.StreamModeDetails$, 4], 3
];
exports.SubscribeToShardEvent$ = [3, n0, _STSE,
    0,
    [_R, _CSN, _MBL, _CSh],
    [() => RecordList, 0, 1, () => ChildShardList], 3
];
exports.SubscribeToShardInput$ = [3, n0, _STSI,
    0,
    [_CARN, _SIh, _SP, _SI],
    [0, 0, () => exports.StartingPosition$, 0], 3
];
exports.SubscribeToShardOutput$ = [3, n0, _STSO,
    0,
    [_ES],
    [[() => exports.SubscribeToShardEventStream$, 0]], 1
];
exports.Tag$ = [3, n0, _Ta,
    0,
    [_K, _V],
    [0, 0], 1
];
exports.TagResourceInput$ = [3, n0, _TRI,
    0,
    [_T, _RARN, _SI],
    [128 | 0, 0, 0], 2
];
exports.UntagResourceInput$ = [3, n0, _URI,
    0,
    [_TK, _RARN, _SI],
    [64 | 0, 0, 0], 2
];
exports.UpdateAccountSettingsInput$ = [3, n0, _UASI,
    0,
    [_MTBC],
    [() => exports.MinimumThroughputBillingCommitmentInput$], 1
];
exports.UpdateAccountSettingsOutput$ = [3, n0, _UASO,
    0,
    [_MTBC],
    [() => exports.MinimumThroughputBillingCommitmentOutput$]
];
exports.UpdateMaxRecordSizeInput$ = [3, n0, _UMRSI,
    0,
    [_MRSIKB, _SARN, _SI],
    [1, 0, 0], 1
];
exports.UpdateShardCountInput$ = [3, n0, _USCI,
    0,
    [_TSC, _ST, _SN, _SARN, _SI],
    [1, 0, 0, 0, 0], 2
];
exports.UpdateShardCountOutput$ = [3, n0, _USCO,
    0,
    [_SN, _CSC, _TSC, _SARN],
    [0, 1, 1, 0]
];
exports.UpdateStreamModeInput$ = [3, n0, _USMI,
    0,
    [_SARN, _SMD, _SI, _WTMB],
    [0, () => exports.StreamModeDetails$, 0, 1], 2
];
exports.UpdateStreamWarmThroughputInput$ = [3, n0, _USWTI,
    0,
    [_WTMB, _SARN, _SN, _SI],
    [1, 0, 0, 0], 1
];
exports.UpdateStreamWarmThroughputOutput$ = [3, n0, _USWTO,
    0,
    [_SARN, _SN, _WT],
    [0, 0, () => exports.WarmThroughputObject$]
];
exports.WarmThroughputObject$ = [3, n0, _WTO,
    0,
    [_TMB, _CMB],
    [1, 1]
];
var __Unit = "unit";
var ChildShardList = [1, n0, _CSL,
    0, () => exports.ChildShard$
];
var ConsumerList = [1, n0, _CL,
    0, () => exports.Consumer$
];
var EnhancedMonitoringList = [1, n0, _EML,
    0, () => exports.EnhancedMetrics$
];
var MetricsNameList = 64 | 0;
var PutRecordsRequestEntryList = [1, n0, _PRREL,
    0, () => exports.PutRecordsRequestEntry$
];
var PutRecordsResultEntryList = [1, n0, _PRRELu,
    0, () => exports.PutRecordsResultEntry$
];
var RecordList = [1, n0, _RL,
    0, () => exports._Record$
];
var ShardIdList = 64 | 0;
var ShardList = [1, n0, _SLh,
    0, () => exports.Shard$
];
var StreamNameList = 64 | 0;
var StreamSummaryList = [1, n0, _SSL,
    0, () => exports.StreamSummary$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => exports.Tag$
];
var TagMap = 128 | 0;
exports.SubscribeToShardEventStream$ = [4, n0, _STSES,
    { [_st]: 1 },
    [_STSE, _RNFE, _RIUE, _KMSDE, _KMSISE, _KMSADE, _KMSNFE, _KMSOIR, _KMSTE, _IFE],
    [() => exports.SubscribeToShardEvent$, [() => exports.ResourceNotFoundException$, 0], [() => exports.ResourceInUseException$, 0], [() => exports.KMSDisabledException$, 0], [() => exports.KMSInvalidStateException$, 0], [() => exports.KMSAccessDeniedException$, 0], [() => exports.KMSNotFoundException$, 0], [() => exports.KMSOptInRequired$, 0], [() => exports.KMSThrottlingException$, 0], [() => exports.InternalFailureException$, 0]]
];
exports.AddTagsToStream$ = [9, n0, _ATTS,
    0, () => exports.AddTagsToStreamInput$, () => __Unit
];
exports.CreateStream$ = [9, n0, _CSr,
    0, () => exports.CreateStreamInput$, () => __Unit
];
exports.DecreaseStreamRetentionPeriod$ = [9, n0, _DSRP,
    0, () => exports.DecreaseStreamRetentionPeriodInput$, () => __Unit
];
exports.DeleteResourcePolicy$ = [9, n0, _DRP,
    0, () => exports.DeleteResourcePolicyInput$, () => __Unit
];
exports.DeleteStream$ = [9, n0, _DS,
    0, () => exports.DeleteStreamInput$, () => __Unit
];
exports.DeregisterStreamConsumer$ = [9, n0, _DSC,
    0, () => exports.DeregisterStreamConsumerInput$, () => __Unit
];
exports.DescribeAccountSettings$ = [9, n0, _DAS,
    0, () => exports.DescribeAccountSettingsInput$, () => exports.DescribeAccountSettingsOutput$
];
exports.DescribeLimits$ = [9, n0, _DL,
    0, () => exports.DescribeLimitsInput$, () => exports.DescribeLimitsOutput$
];
exports.DescribeStream$ = [9, n0, _DSe,
    0, () => exports.DescribeStreamInput$, () => exports.DescribeStreamOutput$
];
exports.DescribeStreamConsumer$ = [9, n0, _DSCe,
    0, () => exports.DescribeStreamConsumerInput$, () => exports.DescribeStreamConsumerOutput$
];
exports.DescribeStreamSummary$ = [9, n0, _DSS,
    0, () => exports.DescribeStreamSummaryInput$, () => exports.DescribeStreamSummaryOutput$
];
exports.DisableEnhancedMonitoring$ = [9, n0, _DEM,
    0, () => exports.DisableEnhancedMonitoringInput$, () => exports.EnhancedMonitoringOutput$
];
exports.EnableEnhancedMonitoring$ = [9, n0, _EEM,
    0, () => exports.EnableEnhancedMonitoringInput$, () => exports.EnhancedMonitoringOutput$
];
exports.GetRecords$ = [9, n0, _GR,
    0, () => exports.GetRecordsInput$, () => exports.GetRecordsOutput$
];
exports.GetResourcePolicy$ = [9, n0, _GRP,
    0, () => exports.GetResourcePolicyInput$, () => exports.GetResourcePolicyOutput$
];
exports.GetShardIterator$ = [9, n0, _GSI,
    0, () => exports.GetShardIteratorInput$, () => exports.GetShardIteratorOutput$
];
exports.IncreaseStreamRetentionPeriod$ = [9, n0, _ISRP,
    0, () => exports.IncreaseStreamRetentionPeriodInput$, () => __Unit
];
exports.ListShards$ = [9, n0, _LS,
    0, () => exports.ListShardsInput$, () => exports.ListShardsOutput$
];
exports.ListStreamConsumers$ = [9, n0, _LSC,
    0, () => exports.ListStreamConsumersInput$, () => exports.ListStreamConsumersOutput$
];
exports.ListStreams$ = [9, n0, _LSi,
    0, () => exports.ListStreamsInput$, () => exports.ListStreamsOutput$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => exports.ListTagsForResourceInput$, () => exports.ListTagsForResourceOutput$
];
exports.ListTagsForStream$ = [9, n0, _LTFS,
    0, () => exports.ListTagsForStreamInput$, () => exports.ListTagsForStreamOutput$
];
exports.MergeShards$ = [9, n0, _MS,
    0, () => exports.MergeShardsInput$, () => __Unit
];
exports.PutRecord$ = [9, n0, _PR,
    0, () => exports.PutRecordInput$, () => exports.PutRecordOutput$
];
exports.PutRecords$ = [9, n0, _PRu,
    0, () => exports.PutRecordsInput$, () => exports.PutRecordsOutput$
];
exports.PutResourcePolicy$ = [9, n0, _PRP,
    0, () => exports.PutResourcePolicyInput$, () => __Unit
];
exports.RegisterStreamConsumer$ = [9, n0, _RSC,
    0, () => exports.RegisterStreamConsumerInput$, () => exports.RegisterStreamConsumerOutput$
];
exports.RemoveTagsFromStream$ = [9, n0, _RTFS,
    0, () => exports.RemoveTagsFromStreamInput$, () => __Unit
];
exports.SplitShard$ = [9, n0, _SSp,
    0, () => exports.SplitShardInput$, () => __Unit
];
exports.StartStreamEncryption$ = [9, n0, _SSE,
    0, () => exports.StartStreamEncryptionInput$, () => __Unit
];
exports.StopStreamEncryption$ = [9, n0, _SSEt,
    0, () => exports.StopStreamEncryptionInput$, () => __Unit
];
exports.SubscribeToShard$ = [9, n0, _STSu,
    0, () => exports.SubscribeToShardInput$, () => exports.SubscribeToShardOutput$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => exports.TagResourceInput$, () => __Unit
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => exports.UntagResourceInput$, () => __Unit
];
exports.UpdateAccountSettings$ = [9, n0, _UAS,
    0, () => exports.UpdateAccountSettingsInput$, () => exports.UpdateAccountSettingsOutput$
];
exports.UpdateMaxRecordSize$ = [9, n0, _UMRS,
    0, () => exports.UpdateMaxRecordSizeInput$, () => __Unit
];
exports.UpdateShardCount$ = [9, n0, _USC,
    0, () => exports.UpdateShardCountInput$, () => exports.UpdateShardCountOutput$
];
exports.UpdateStreamMode$ = [9, n0, _USM,
    0, () => exports.UpdateStreamModeInput$, () => __Unit
];
exports.UpdateStreamWarmThroughput$ = [9, n0, _USWT,
    0, () => exports.UpdateStreamWarmThroughputInput$, () => exports.UpdateStreamWarmThroughputOutput$
];
