// @aws-sdk/client-eventbridge/dist-es/schemas/schemas_0.js
const _A = "Archive";
const _AA = "ArchiveArn";
const _AD = "ApiDestination";
const _ADA = "ApiDestinationArn";
const _ADE = "AccessDeniedException";
const _ADRL = "ApiDestinationResponseList";
const _ADS = "ApiDestinationState";
const _ADp = "ApiDestinations";
const _AE = "AuthorizationEndpoint";
const _AES = "ActivateEventSource";
const _AESR = "ActivateEventSourceRequest";
const _AHPS = "AuthHeaderParametersSensitive";
const _AKAP = "ApiKeyAuthParameters";
const _AKN = "ApiKeyName";
const _AKV = "ApiKeyValue";
const _AN = "ArchiveName";
const _AP = "ArrayProperties";
const _API = "AssignPublicIp";
const _APu = "AuthParameters";
const _ARL = "ArchiveResponseList";
const _ASP = "AppSyncParameters";
const _AT = "AuthorizationType";
const _AVC = "AwsVpcConfiguration";
const _Ac = "Account";
const _Act = "Action";
const _Ar = "Arn";
const _Arc = "Archives";
const _At = "Attempts";
const _BAP = "BatchArrayProperties";
const _BAPa = "BasicAuthParameters";
const _BP = "BatchParameters";
const _BPo = "BodyParameters";
const _BRS = "BatchRetryStrategy";
const _C = "Condition";
const _CA = "ConnectionArn";
const _CAD = "CreateApiDestination";
const _CADR = "CreateApiDestinationRequest";
const _CADRr = "CreateApiDestinationResponse";
const _CAKARP = "ConnectionApiKeyAuthResponseParameters";
const _CAR = "CreateArchiveRequest";
const _CARP = "ConnectionAuthResponseParameters";
const _CARr = "CreateArchiveResponse";
const _CAr = "CreateArchive";
const _CB = "CreatedBy";
const _CBARP = "ConnectionBasicAuthResponseParameters";
const _CBP = "ConnectionBodyParameter";
const _CBPL = "ConnectionBodyParametersList";
const _CC = "CreateConnection";
const _CCAKARP = "CreateConnectionApiKeyAuthRequestParameters";
const _CCARP = "CreateConnectionAuthRequestParameters";
const _CCBARP = "CreateConnectionBasicAuthRequestParameters";
const _CCOACRP = "CreateConnectionOAuthClientRequestParameters";
const _CCOARP = "CreateConnectionOAuthRequestParameters";
const _CCR = "CreateConnectionRequest";
const _CCRr = "CreateConnectionResponse";
const _CE = "CreateEndpoint";
const _CEB = "CreateEventBus";
const _CEBR = "CreateEventBusRequest";
const _CEBRr = "CreateEventBusResponse";
const _CER = "CreateEndpointRequest";
const _CERr = "CreateEndpointResponse";
const _CHP = "ConnectionHeaderParameter";
const _CHPL = "ConnectionHeaderParametersList";
const _CHPo = "ConnectionHttpParameters";
const _CID = "ClientID";
const _CME = "ConcurrentModificationException";
const _COACRP = "ConnectionOAuthClientResponseParameters";
const _COARP = "ConnectionOAuthResponseParameters";
const _CP = "ConnectivityParameters";
const _CPES = "CreatePartnerEventSource";
const _CPESR = "CreatePartnerEventSourceRequest";
const _CPESRr = "CreatePartnerEventSourceResponse";
const _CPS = "CapacityProviderStrategy";
const _CPSI = "CapacityProviderStrategyItem";
const _CPl = "ClientParameters";
const _CQSP = "ConnectionQueryStringParameter";
const _CQSPL = "ConnectionQueryStringParametersList";
const _CR = "CancelReplay";
const _CRCA = "ConnectivityResourceConfigurationArn";
const _CRL = "ConnectionResponseList";
const _CRP = "ConnectivityResourceParameters";
const _CRR = "CancelReplayRequest";
const _CRRa = "CancelReplayResponse";
const _CS = "ConnectionState";
const _CSl = "ClientSecret";
const _CT = "CreationTime";
const _Co = "Connection";
const _Con = "Connections";
const _D = "Description";
const _DA = "DeleteArchive";
const _DAD = "DeleteApiDestination";
const _DADR = "DeleteApiDestinationRequest";
const _DADRe = "DeleteApiDestinationResponse";
const _DADRes = "DescribeApiDestinationRequest";
const _DADResc = "DescribeApiDestinationResponse";
const _DADe = "DescribeApiDestination";
const _DAR = "DeleteArchiveRequest";
const _DARe = "DeleteArchiveResponse";
const _DARes = "DescribeArchiveRequest";
const _DAResc = "DescribeArchiveResponse";
const _DAe = "DescribeArchive";
const _DC = "DeauthorizeConnection";
const _DCCP = "DescribeConnectionConnectivityParameters";
const _DCR = "DeauthorizeConnectionRequest";
const _DCRP = "DescribeConnectionResourceParameters";
const _DCRe = "DeauthorizeConnectionResponse";
const _DCRel = "DeleteConnectionRequest";
const _DCRele = "DeleteConnectionResponse";
const _DCRes = "DescribeConnectionRequest";
const _DCResc = "DescribeConnectionResponse";
const _DCe = "DeleteConnection";
const _DCes = "DescribeConnection";
const _DE = "DeleteEndpoint";
const _DEB = "DeleteEventBus";
const _DEBR = "DeleteEventBusRequest";
const _DEBRe = "DescribeEventBusRequest";
const _DEBRes = "DescribeEventBusResponse";
const _DEBe = "DescribeEventBus";
const _DER = "DeleteEndpointRequest";
const _DERe = "DeleteEndpointResponse";
const _DERes = "DescribeEndpointRequest";
const _DEResc = "DescribeEndpointResponse";
const _DES = "DeactivateEventSource";
const _DESR = "DeactivateEventSourceRequest";
const _DESRe = "DescribeEventSourceRequest";
const _DESRes = "DescribeEventSourceResponse";
const _DESe = "DescribeEventSource";
const _DEe = "DescribeEndpoint";
const _DLC = "DeadLetterConfig";
const _DPES = "DeletePartnerEventSource";
const _DPESR = "DeletePartnerEventSourceRequest";
const _DPESRe = "DescribePartnerEventSourceRequest";
const _DPESRes = "DescribePartnerEventSourceResponse";
const _DPESe = "DescribePartnerEventSource";
const _DR = "DeleteRule";
const _DRR = "DeleteRuleRequest";
const _DRRe = "DescribeReplayRequest";
const _DRRes = "DescribeReplayResponse";
const _DRResc = "DescribeRuleRequest";
const _DRRescr = "DescribeRuleResponse";
const _DRRi = "DisableRuleRequest";
const _DRe = "DescribeReplay";
const _DRes = "DescribeRule";
const _DRi = "DisableRule";
const _DT = "DetailType";
const _DU = "DbUser";
const _Da = "Database";
const _De = "Destination";
const _Det = "Detail";
const _E = "Endpoint";
const _EB = "EventBuses";
const _EBA = "EventBusArn";
const _EBL = "EventBusList";
const _EBN = "EventBusName";
const _EBv = "EventBus";
const _EC = "EventCount";
const _ECr = "ErrorCode";
const _EEB = "EndpointEventBus";
const _EEBL = "EndpointEventBusList";
const _EEC = "EnableExecuteCommand";
const _EECSMT = "EnableECSManagedTags";
const _EET = "EventEndTime";
const _EI = "EndpointId";
const _EIv = "EventId";
const _EL = "EndpointList";
const _ELRT = "EventLastReplayedTime";
const _EM = "ErrorMessage";
const _EP = "EventPattern";
const _EPc = "EcsParameters";
const _ER = "EnableRule";
const _ERR = "EnableRuleRequest";
const _ES = "EventSource";
const _ESA = "EventSourceArn";
const _ESL = "EventSourceList";
const _ESN = "EventSourceName";
const _EST = "EventStartTime";
const _ESv = "EventSources";
const _ET = "ExpirationTime";
const _EU = "EndpointUrl";
const _En = "Endpoints";
const _Ent = "Entries";
const _Ev = "Event";
const _F = "Force";
const _FA = "FilterArns";
const _FC = "FailoverConfig";
const _FE = "FailedEntries";
const _FEC = "FailedEntryCount";
const _G = "Group";
const _GQLO = "GraphQLOperation";
const _HC = "HealthCheck";
const _HM = "HttpMethod";
const _HP = "HeaderParameters";
const _HPt = "HttpParameters";
const _HR = "HomeRegion";
const _HVS = "HeaderValueSensitive";
const _I = "Ids";
const _ICP = "InvocationConnectivityParameters";
const _ID = "IncludeDetail";
const _IE = "InternalException";
const _IEPE = "InvalidEventPatternException";
const _IEn = "InvocationEndpoint";
const _IHP = "InvocationHttpParameters";
const _IP = "InputPath";
const _IPM = "InputPathsMap";
const _IRLPS = "InvocationRateLimitPerSecond";
const _ISE = "IllegalStatusException";
const _ISEn = "InvalidStateException";
const _IT = "InputTransformer";
const _ITn = "InputTemplate";
const _IVS = "IsValueSecret";
const _Id = "Id";
const _In = "Input";
const _JD = "JobDefinition";
const _JN = "JobName";
const _K = "Key";
const _KKI = "KmsKeyIdentifier";
const _KP = "KinesisParameters";
const _L = "Limit";
const _LA = "ListArchives";
const _LAD = "ListApiDestinations";
const _LADR = "ListApiDestinationsRequest";
const _LADRi = "ListApiDestinationsResponse";
const _LAR = "ListArchivesRequest";
const _LARi = "ListArchivesResponse";
const _LAT = "LastAuthorizedTime";
const _LC = "LogConfig";
const _LCR = "ListConnectionsRequest";
const _LCRi = "ListConnectionsResponse";
const _LCi = "ListConnections";
const _LE = "ListEndpoints";
const _LEB = "ListEventBuses";
const _LEBR = "ListEventBusesRequest";
const _LEBRi = "ListEventBusesResponse";
const _LEE = "LimitExceededException";
const _LER = "ListEndpointsRequest";
const _LERi = "ListEndpointsResponse";
const _LES = "ListEventSources";
const _LESR = "ListEventSourcesRequest";
const _LESRi = "ListEventSourcesResponse";
const _LMT = "LastModifiedTime";
const _LPES = "ListPartnerEventSources";
const _LPESA = "ListPartnerEventSourceAccounts";
const _LPESAR = "ListPartnerEventSourceAccountsRequest";
const _LPESARi = "ListPartnerEventSourceAccountsResponse";
const _LPESR = "ListPartnerEventSourcesRequest";
const _LPESRi = "ListPartnerEventSourcesResponse";
const _LR = "ListReplays";
const _LRNBT = "ListRuleNamesByTarget";
const _LRNBTR = "ListRuleNamesByTargetRequest";
const _LRNBTRi = "ListRuleNamesByTargetResponse";
const _LRR = "ListReplaysRequest";
const _LRRi = "ListReplaysResponse";
const _LRRis = "ListRulesRequest";
const _LRRist = "ListRulesResponse";
const _LRi = "ListRules";
const _LT = "LaunchType";
const _LTBR = "ListTargetsByRule";
const _LTBRR = "ListTargetsByRuleRequest";
const _LTBRRi = "ListTargetsByRuleResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _Le = "Level";
const _MB = "ManagedBy";
const _MEAIS = "MaximumEventAgeInSeconds";
const _MGI = "MessageGroupId";
const _MR = "MaxResults";
const _MRA = "MaximumRetryAttempts";
const _MRE = "ManagedRuleException";
const _N = "Name";
const _NC = "NetworkConfiguration";
const _NP = "NamePrefix";
const _NT = "NextToken";
const _OAHP = "OAuthHttpParameters";
const _OAP = "OAuthParameters";
const _ODE = "OperationDisabledException";
const _P = "Password";
const _PC = "PlacementConstraints";
const _PCl = "PlacementConstraint";
const _PE = "PutEvents";
const _PER = "PutEventsRequest";
const _PERE = "PutEventsRequestEntry";
const _PEREL = "PutEventsRequestEntryList";
const _PERELu = "PutEventsResultEntryList";
const _PEREu = "PutEventsResultEntry";
const _PERu = "PutEventsResponse";
const _PES = "PartnerEventSources";
const _PESA = "PartnerEventSourceAccounts";
const _PESAL = "PartnerEventSourceAccountList";
const _PESAa = "PartnerEventSourceAccount";
const _PESL = "PartnerEventSourceList";
const _PESa = "PartnerEventSource";
const _PKP = "PartitionKeyPath";
const _PLEE = "PolicyLengthExceededException";
const _PP = "PutPermission";
const _PPE = "PutPartnerEvents";
const _PPER = "PutPartnerEventsRequest";
const _PPERE = "PutPartnerEventsRequestEntry";
const _PPEREL = "PutPartnerEventsRequestEntryList";
const _PPERELu = "PutPartnerEventsResultEntryList";
const _PPEREu = "PutPartnerEventsResultEntry";
const _PPERu = "PutPartnerEventsResponse";
const _PPL = "PipelineParameterList";
const _PPR = "PutPermissionRequest";
const _PPV = "PathParameterValues";
const _PR = "PutRule";
const _PRR = "PutRuleRequest";
const _PRRu = "PutRuleResponse";
const _PS = "PlacementStrategy";
const _PSl = "PlacementStrategies";
const _PT = "PropagateTags";
const _PTR = "PutTargetsRequest";
const _PTRE = "PutTargetsResultEntry";
const _PTREL = "PutTargetsResultEntryList";
const _PTRu = "PutTargetsResponse";
const _PTu = "PutTargets";
const _PV = "PlatformVersion";
const _Po = "Policy";
const _Pr = "Primary";
const _Pri = "Principal";
const _QSP = "QueryStringParameters";
const _QSVS = "QueryStringValueSensitive";
const _R = "Replays";
const _RA = "ReplayArn";
const _RAA = "ResourceAssociationArn";
const _RAEE = "ResourceAlreadyExistsException";
const _RAP = "RemoveAllPermissions";
const _RARN = "ResourceARN";
const _RAo = "RoleArn";
const _RAu = "RuleArn";
const _RC = "RoutingConfig";
const _RCA = "ResourceConfigurationArn";
const _RCP = "RunCommandParameters";
const _RCT = "RunCommandTargets";
const _RCTu = "RunCommandTarget";
const _RCe = "ReplicationConfig";
const _RD = "RetentionDays";
const _RDP = "RedshiftDataParameters";
const _RDe = "ReplayDestination";
const _RET = "ReplayEndTime";
const _RI = "ReferenceId";
const _RL = "ReplayList";
const _RN = "ReplayName";
const _RNFE = "ResourceNotFoundException";
const _RNu = "RuleNames";
const _RP = "ResourceParameters";
const _RPR = "RemovePermissionRequest";
const _RPe = "RetryPolicy";
const _RPem = "RemovePermission";
const _RRL = "RuleResponseList";
const _RS = "RetryStrategy";
const _RST = "ReplayStartTime";
const _RT = "RemoveTargets";
const _RTR = "RemoveTargetsRequest";
const _RTRE = "RemoveTargetsResultEntry";
const _RTREL = "RemoveTargetsResultEntryList";
const _RTRe = "RemoveTargetsResponse";
const _Re = "Resources";
const _Rep = "Replay";
const _Res = "Result";
const _Ro = "Route";
const _Ru = "Rules";
const _Rul = "Rule";
const _S = "Sql";
const _SA = "SecretArn";
const _SB = "SizeBytes";
const _SE = "ScheduleExpression";
const _SG = "SecurityGroups";
const _SI = "StatementId";
const _SMA = "SecretManagerArn";
const _SMPP = "SageMakerPipelineParameter";
const _SMPPL = "SageMakerPipelineParameterList";
const _SMPPa = "SageMakerPipelineParameters";
const _SN = "StatementName";
const _SP = "SqsParameters";
const _SR = "StateReason";
const _SRR = "StartReplayRequest";
const _SRRt = "StartReplayResponse";
const _SRt = "StartReplay";
const _SS = "SensitiveString";
const _Se = "Secondary";
const _Si = "Size";
const _So = "Source";
const _Sq = "Sqls";
const _St = "State";
const _Su = "Subnets";
const _T = "Type";
const _TA = "TargetArn";
const _TC = "TaskCount";
const _TDA = "TaskDefinitionArn";
const _TE = "ThrottlingException";
const _TEP = "TestEventPattern";
const _TEPR = "TestEventPatternRequest";
const _TEPRe = "TestEventPatternResponse";
const _TH = "TraceHeader";
const _TI = "TargetId";
const _TK = "TagKeys";
const _TL = "TagList";
const _TLa = "TargetList";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Targets";
const _Targ = "Target";
const _Ti = "Time";
const _U = "Username";
const _UA = "UpdateArchive";
const _UAD = "UpdateApiDestination";
const _UADR = "UpdateApiDestinationRequest";
const _UADRp = "UpdateApiDestinationResponse";
const _UAR = "UpdateArchiveRequest";
const _UARp = "UpdateArchiveResponse";
const _UC = "UpdateConnection";
const _UCAKARP = "UpdateConnectionApiKeyAuthRequestParameters";
const _UCARP = "UpdateConnectionAuthRequestParameters";
const _UCBARP = "UpdateConnectionBasicAuthRequestParameters";
const _UCOACRP = "UpdateConnectionOAuthClientRequestParameters";
const _UCOARP = "UpdateConnectionOAuthRequestParameters";
const _UCR = "UpdateConnectionRequest";
const _UCRp = "UpdateConnectionResponse";
const _UE = "UpdateEndpoint";
const _UEB = "UpdateEventBus";
const _UEBR = "UpdateEventBusRequest";
const _UEBRp = "UpdateEventBusResponse";
const _UER = "UpdateEndpointRequest";
const _UERp = "UpdateEndpointResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _V = "Value";
const _Va = "Values";
const _WE = "WithEvent";
const _aC = "awsvpcConfiguration";
const _b = "base";
const _c = "client";
const _cP = "capacityProvider";
const _e = "error";
const _ex = "expression";
const _f = "field";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.eventbridge";
const _se = "server";
const _t = "type";
const _w = "weight";
const n0 = "com.amazonaws.eventbridge";
import { TypeRegistry } from "@smithy/core/schema";
import { AccessDeniedException, ConcurrentModificationException, IllegalStatusException, InternalException, InvalidEventPatternException, InvalidStateException, LimitExceededException, ManagedRuleException, OperationDisabledException, PolicyLengthExceededException, ResourceAlreadyExistsException, ResourceNotFoundException, ThrottlingException, } from "../models/errors";
import { EventBridgeServiceException } from "../models/EventBridgeServiceException";
const _s_registry = TypeRegistry.for(_s);
export var EventBridgeServiceException$ = [-3, _s, "EventBridgeServiceException", 0, [], []];
_s_registry.registerError(EventBridgeServiceException$, EventBridgeServiceException);
const n0_registry = TypeRegistry.for(n0);
export var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
export var ConcurrentModificationException$ = [-3, n0, _CME,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ConcurrentModificationException$, ConcurrentModificationException);
export var IllegalStatusException$ = [-3, n0, _ISE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(IllegalStatusException$, IllegalStatusException);
export var InternalException$ = [-3, n0, _IE,
    { [_e]: _se },
    [_m],
    [0]
];
n0_registry.registerError(InternalException$, InternalException);
export var InvalidEventPatternException$ = [-3, n0, _IEPE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidEventPatternException$, InvalidEventPatternException);
export var InvalidStateException$ = [-3, n0, _ISEn,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidStateException$, InvalidStateException);
export var LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(LimitExceededException$, LimitExceededException);
export var ManagedRuleException$ = [-3, n0, _MRE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ManagedRuleException$, ManagedRuleException);
export var OperationDisabledException$ = [-3, n0, _ODE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(OperationDisabledException$, OperationDisabledException);
export var PolicyLengthExceededException$ = [-3, n0, _PLEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(PolicyLengthExceededException$, PolicyLengthExceededException);
export var ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ResourceAlreadyExistsException$, ResourceAlreadyExistsException);
export var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
export var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ThrottlingException$, ThrottlingException);
export const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AuthHeaderParametersSensitive = [0, n0, _AHPS, 8, 0];
var GraphQLOperation = [0, n0, _GQLO, 8, 0];
var HeaderValueSensitive = [0, n0, _HVS, 8, 0];
var QueryStringValueSensitive = [0, n0, _QSVS, 8, 0];
var SensitiveString = [0, n0, _SS, 8, 0];
var Sql = [0, n0, _S, 8, 0];
export var ActivateEventSourceRequest$ = [3, n0, _AESR,
    0,
    [_N],
    [0], 1
];
export var ApiDestination$ = [3, n0, _AD,
    0,
    [_ADA, _N, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 1, 4, 4]
];
export var AppSyncParameters$ = [3, n0, _ASP,
    0,
    [_GQLO],
    [[() => GraphQLOperation, 0]]
];
export var Archive$ = [3, n0, _A,
    0,
    [_AN, _ESA, _St, _SR, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 1, 1, 1, 4]
];
export var AwsVpcConfiguration$ = [3, n0, _AVC,
    0,
    [_Su, _SG, _API],
    [64 | 0, 64 | 0, 0], 1
];
export var BatchArrayProperties$ = [3, n0, _BAP,
    0,
    [_Si],
    [1]
];
export var BatchParameters$ = [3, n0, _BP,
    0,
    [_JD, _JN, _AP, _RS],
    [0, 0, () => BatchArrayProperties$, () => BatchRetryStrategy$], 2
];
export var BatchRetryStrategy$ = [3, n0, _BRS,
    0,
    [_At],
    [1]
];
export var CancelReplayRequest$ = [3, n0, _CRR,
    0,
    [_RN],
    [0], 1
];
export var CancelReplayResponse$ = [3, n0, _CRRa,
    0,
    [_RA, _St, _SR],
    [0, 0, 0]
];
export var CapacityProviderStrategyItem$ = [3, n0, _CPSI,
    0,
    [_cP, _w, _b],
    [0, 1, 1], 1
];
export var Condition$ = [3, n0, _C,
    0,
    [_T, _K, _V],
    [0, 0, 0], 3
];
export var Connection$ = [3, n0, _Co,
    0,
    [_CA, _N, _CS, _SR, _AT, _CT, _LMT, _LAT],
    [0, 0, 0, 0, 0, 4, 4, 4]
];
export var ConnectionApiKeyAuthResponseParameters$ = [3, n0, _CAKARP,
    0,
    [_AKN],
    [0]
];
export var ConnectionAuthResponseParameters$ = [3, n0, _CARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [() => ConnectionBasicAuthResponseParameters$, [() => ConnectionOAuthResponseParameters$, 0], () => ConnectionApiKeyAuthResponseParameters$, [() => ConnectionHttpParameters$, 0], () => DescribeConnectionConnectivityParameters$]
];
export var ConnectionBasicAuthResponseParameters$ = [3, n0, _CBARP,
    0,
    [_U],
    [0]
];
export var ConnectionBodyParameter$ = [3, n0, _CBP,
    0,
    [_K, _V, _IVS],
    [0, [() => SensitiveString, 0], 2]
];
export var ConnectionHeaderParameter$ = [3, n0, _CHP,
    0,
    [_K, _V, _IVS],
    [0, [() => HeaderValueSensitive, 0], 2]
];
export var ConnectionHttpParameters$ = [3, n0, _CHPo,
    0,
    [_HP, _QSP, _BPo],
    [[() => ConnectionHeaderParametersList, 0], [() => ConnectionQueryStringParametersList, 0], [() => ConnectionBodyParametersList, 0]]
];
export var ConnectionOAuthClientResponseParameters$ = [3, n0, _COACRP,
    0,
    [_CID],
    [0]
];
export var ConnectionOAuthResponseParameters$ = [3, n0, _COARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [() => ConnectionOAuthClientResponseParameters$, 0, 0, [() => ConnectionHttpParameters$, 0]]
];
export var ConnectionQueryStringParameter$ = [3, n0, _CQSP,
    0,
    [_K, _V, _IVS],
    [0, [() => QueryStringValueSensitive, 0], 2]
];
export var ConnectivityResourceConfigurationArn$ = [3, n0, _CRCA,
    0,
    [_RCA],
    [0], 1
];
export var ConnectivityResourceParameters$ = [3, n0, _CRP,
    0,
    [_RP],
    [() => ConnectivityResourceConfigurationArn$], 1
];
export var CreateApiDestinationRequest$ = [3, n0, _CADR,
    0,
    [_N, _CA, _IEn, _HM, _D, _IRLPS],
    [0, 0, 0, 0, 0, 1], 4
];
export var CreateApiDestinationResponse$ = [3, n0, _CADRr,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
export var CreateArchiveRequest$ = [3, n0, _CAR,
    0,
    [_AN, _ESA, _D, _EP, _RD, _KKI],
    [0, 0, 0, 0, 1, 0], 2
];
export var CreateArchiveResponse$ = [3, n0, _CARr,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
export var CreateConnectionApiKeyAuthRequestParameters$ = [3, n0, _CCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
export var CreateConnectionAuthRequestParameters$ = [3, n0, _CCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => CreateConnectionBasicAuthRequestParameters$, 0], [() => CreateConnectionOAuthRequestParameters$, 0], [() => CreateConnectionApiKeyAuthRequestParameters$, 0], [() => ConnectionHttpParameters$, 0], () => ConnectivityResourceParameters$]
];
export var CreateConnectionBasicAuthRequestParameters$ = [3, n0, _CCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
export var CreateConnectionOAuthClientRequestParameters$ = [3, n0, _CCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
export var CreateConnectionOAuthRequestParameters$ = [3, n0, _CCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => CreateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => ConnectionHttpParameters$, 0]], 3
];
export var CreateConnectionRequest$ = [3, n0, _CCR,
    0,
    [_N, _AT, _APu, _D, _ICP, _KKI],
    [0, 0, [() => CreateConnectionAuthRequestParameters$, 0], 0, () => ConnectivityResourceParameters$, 0], 3
];
export var CreateConnectionResponse$ = [3, n0, _CCRr,
    0,
    [_CA, _CS, _CT, _LMT],
    [0, 0, 4, 4]
];
export var CreateEndpointRequest$ = [3, n0, _CER,
    0,
    [_N, _RC, _EB, _D, _RCe, _RAo],
    [0, () => RoutingConfig$, () => EndpointEventBusList, 0, () => ReplicationConfig$, 0], 3
];
export var CreateEndpointResponse$ = [3, n0, _CERr,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _St],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0]
];
export var CreateEventBusRequest$ = [3, n0, _CEBR,
    0,
    [_N, _ESN, _D, _KKI, _DLC, _LC, _Ta],
    [0, 0, 0, 0, () => DeadLetterConfig$, () => LogConfig$, () => TagList], 1
];
export var CreateEventBusResponse$ = [3, n0, _CEBRr,
    0,
    [_EBA, _D, _KKI, _DLC, _LC],
    [0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
export var CreatePartnerEventSourceRequest$ = [3, n0, _CPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
export var CreatePartnerEventSourceResponse$ = [3, n0, _CPESRr,
    0,
    [_ESA],
    [0]
];
export var DeactivateEventSourceRequest$ = [3, n0, _DESR,
    0,
    [_N],
    [0], 1
];
export var DeadLetterConfig$ = [3, n0, _DLC,
    0,
    [_Ar],
    [0]
];
export var DeauthorizeConnectionRequest$ = [3, n0, _DCR,
    0,
    [_N],
    [0], 1
];
export var DeauthorizeConnectionResponse$ = [3, n0, _DCRe,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
export var DeleteApiDestinationRequest$ = [3, n0, _DADR,
    0,
    [_N],
    [0], 1
];
export var DeleteApiDestinationResponse$ = [3, n0, _DADRe,
    0,
    [],
    []
];
export var DeleteArchiveRequest$ = [3, n0, _DAR,
    0,
    [_AN],
    [0], 1
];
export var DeleteArchiveResponse$ = [3, n0, _DARe,
    0,
    [],
    []
];
export var DeleteConnectionRequest$ = [3, n0, _DCRel,
    0,
    [_N],
    [0], 1
];
export var DeleteConnectionResponse$ = [3, n0, _DCRele,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
export var DeleteEndpointRequest$ = [3, n0, _DER,
    0,
    [_N],
    [0], 1
];
export var DeleteEndpointResponse$ = [3, n0, _DERe,
    0,
    [],
    []
];
export var DeleteEventBusRequest$ = [3, n0, _DEBR,
    0,
    [_N],
    [0], 1
];
export var DeletePartnerEventSourceRequest$ = [3, n0, _DPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
export var DeleteRuleRequest$ = [3, n0, _DRR,
    0,
    [_N, _EBN, _F],
    [0, 0, 2], 1
];
export var DescribeApiDestinationRequest$ = [3, n0, _DADRes,
    0,
    [_N],
    [0], 1
];
export var DescribeApiDestinationResponse$ = [3, n0, _DADResc,
    0,
    [_ADA, _N, _D, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 0, 1, 4, 4]
];
export var DescribeArchiveRequest$ = [3, n0, _DARes,
    0,
    [_AN],
    [0], 1
];
export var DescribeArchiveResponse$ = [3, n0, _DAResc,
    0,
    [_AA, _AN, _ESA, _D, _EP, _St, _SR, _KKI, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 4]
];
export var DescribeConnectionConnectivityParameters$ = [3, n0, _DCCP,
    0,
    [_RP],
    [() => DescribeConnectionResourceParameters$], 1
];
export var DescribeConnectionRequest$ = [3, n0, _DCRes,
    0,
    [_N],
    [0], 1
];
export var DescribeConnectionResourceParameters$ = [3, n0, _DCRP,
    0,
    [_RCA, _RAA],
    [0, 0], 2
];
export var DescribeConnectionResponse$ = [3, n0, _DCResc,
    0,
    [_CA, _N, _D, _ICP, _CS, _SR, _AT, _SA, _KKI, _APu, _CT, _LMT, _LAT],
    [0, 0, 0, () => DescribeConnectionConnectivityParameters$, 0, 0, 0, 0, 0, [() => ConnectionAuthResponseParameters$, 0], 4, 4, 4]
];
export var DescribeEndpointRequest$ = [3, n0, _DERes,
    0,
    [_N, _HR],
    [0, 0], 1
];
export var DescribeEndpointResponse$ = [3, n0, _DEResc,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
export var DescribeEventBusRequest$ = [3, n0, _DEBRe,
    0,
    [_N],
    [0]
];
export var DescribeEventBusResponse$ = [3, n0, _DEBRes,
    0,
    [_N, _Ar, _D, _KKI, _DLC, _Po, _LC, _CT, _LMT],
    [0, 0, 0, 0, () => DeadLetterConfig$, 0, () => LogConfig$, 4, 4]
];
export var DescribeEventSourceRequest$ = [3, n0, _DESRe,
    0,
    [_N],
    [0], 1
];
export var DescribeEventSourceResponse$ = [3, n0, _DESRes,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
export var DescribePartnerEventSourceRequest$ = [3, n0, _DPESRe,
    0,
    [_N],
    [0], 1
];
export var DescribePartnerEventSourceResponse$ = [3, n0, _DPESRes,
    0,
    [_Ar, _N],
    [0, 0]
];
export var DescribeReplayRequest$ = [3, n0, _DRRe,
    0,
    [_RN],
    [0], 1
];
export var DescribeReplayResponse$ = [3, n0, _DRRes,
    0,
    [_RN, _RA, _D, _St, _SR, _ESA, _De, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 0, 0, () => ReplayDestination$, 4, 4, 4, 4, 4]
];
export var DescribeRuleRequest$ = [3, n0, _DRResc,
    0,
    [_N, _EBN],
    [0, 0], 1
];
export var DescribeRuleResponse$ = [3, n0, _DRRescr,
    0,
    [_N, _Ar, _EP, _SE, _St, _D, _RAo, _MB, _EBN, _CB],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
];
export var DisableRuleRequest$ = [3, n0, _DRRi,
    0,
    [_N, _EBN],
    [0, 0], 1
];
export var EcsParameters$ = [3, n0, _EPc,
    0,
    [_TDA, _TC, _LT, _NC, _PV, _G, _CPS, _EECSMT, _EEC, _PC, _PS, _PT, _RI, _Ta],
    [0, 1, 0, () => NetworkConfiguration$, 0, 0, () => CapacityProviderStrategy, 2, 2, () => PlacementConstraints, () => PlacementStrategies, 0, 0, () => TagList], 1
];
export var EnableRuleRequest$ = [3, n0, _ERR,
    0,
    [_N, _EBN],
    [0, 0], 1
];
export var Endpoint$ = [3, n0, _E,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
export var EndpointEventBus$ = [3, n0, _EEB,
    0,
    [_EBA],
    [0], 1
];
export var EventBus$ = [3, n0, _EBv,
    0,
    [_N, _Ar, _D, _Po, _CT, _LMT],
    [0, 0, 0, 0, 4, 4]
];
export var EventSource$ = [3, n0, _ES,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
export var FailoverConfig$ = [3, n0, _FC,
    0,
    [_Pr, _Se],
    [() => Primary$, () => Secondary$], 2
];
export var HttpParameters$ = [3, n0, _HPt,
    0,
    [_PPV, _HP, _QSP],
    [64 | 0, 128 | 0, 128 | 0]
];
export var InputTransformer$ = [3, n0, _IT,
    0,
    [_ITn, _IPM],
    [0, 128 | 0], 1
];
export var KinesisParameters$ = [3, n0, _KP,
    0,
    [_PKP],
    [0], 1
];
export var ListApiDestinationsRequest$ = [3, n0, _LADR,
    0,
    [_NP, _CA, _NT, _L],
    [0, 0, 0, 1]
];
export var ListApiDestinationsResponse$ = [3, n0, _LADRi,
    0,
    [_ADp, _NT],
    [() => ApiDestinationResponseList, 0]
];
export var ListArchivesRequest$ = [3, n0, _LAR,
    0,
    [_NP, _ESA, _St, _NT, _L],
    [0, 0, 0, 0, 1]
];
export var ListArchivesResponse$ = [3, n0, _LARi,
    0,
    [_Arc, _NT],
    [() => ArchiveResponseList, 0]
];
export var ListConnectionsRequest$ = [3, n0, _LCR,
    0,
    [_NP, _CS, _NT, _L],
    [0, 0, 0, 1]
];
export var ListConnectionsResponse$ = [3, n0, _LCRi,
    0,
    [_Con, _NT],
    [() => ConnectionResponseList, 0]
];
export var ListEndpointsRequest$ = [3, n0, _LER,
    0,
    [_NP, _HR, _NT, _MR],
    [0, 0, 0, 1]
];
export var ListEndpointsResponse$ = [3, n0, _LERi,
    0,
    [_En, _NT],
    [() => EndpointList, 0]
];
export var ListEventBusesRequest$ = [3, n0, _LEBR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
export var ListEventBusesResponse$ = [3, n0, _LEBRi,
    0,
    [_EB, _NT],
    [() => EventBusList, 0]
];
export var ListEventSourcesRequest$ = [3, n0, _LESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
export var ListEventSourcesResponse$ = [3, n0, _LESRi,
    0,
    [_ESv, _NT],
    [() => EventSourceList, 0]
];
export var ListPartnerEventSourceAccountsRequest$ = [3, n0, _LPESAR,
    0,
    [_ESN, _NT, _L],
    [0, 0, 1], 1
];
export var ListPartnerEventSourceAccountsResponse$ = [3, n0, _LPESARi,
    0,
    [_PESA, _NT],
    [() => PartnerEventSourceAccountList, 0]
];
export var ListPartnerEventSourcesRequest$ = [3, n0, _LPESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1], 1
];
export var ListPartnerEventSourcesResponse$ = [3, n0, _LPESRi,
    0,
    [_PES, _NT],
    [() => PartnerEventSourceList, 0]
];
export var ListReplaysRequest$ = [3, n0, _LRR,
    0,
    [_NP, _St, _ESA, _NT, _L],
    [0, 0, 0, 0, 1]
];
export var ListReplaysResponse$ = [3, n0, _LRRi,
    0,
    [_R, _NT],
    [() => ReplayList, 0]
];
export var ListRuleNamesByTargetRequest$ = [3, n0, _LRNBTR,
    0,
    [_TA, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
export var ListRuleNamesByTargetResponse$ = [3, n0, _LRNBTRi,
    0,
    [_RNu, _NT],
    [64 | 0, 0]
];
export var ListRulesRequest$ = [3, n0, _LRRis,
    0,
    [_NP, _EBN, _NT, _L],
    [0, 0, 0, 1]
];
export var ListRulesResponse$ = [3, n0, _LRRist,
    0,
    [_Ru, _NT],
    [() => RuleResponseList, 0]
];
export var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
export var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [() => TagList]
];
export var ListTargetsByRuleRequest$ = [3, n0, _LTBRR,
    0,
    [_Rul, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
export var ListTargetsByRuleResponse$ = [3, n0, _LTBRRi,
    0,
    [_Tar, _NT],
    [[() => TargetList, 0], 0]
];
export var LogConfig$ = [3, n0, _LC,
    0,
    [_ID, _Le],
    [0, 0]
];
export var NetworkConfiguration$ = [3, n0, _NC,
    0,
    [_aC],
    [() => AwsVpcConfiguration$]
];
export var PartnerEventSource$ = [3, n0, _PESa,
    0,
    [_Ar, _N],
    [0, 0]
];
export var PartnerEventSourceAccount$ = [3, n0, _PESAa,
    0,
    [_Ac, _CT, _ET, _St],
    [0, 4, 4, 0]
];
export var PlacementConstraint$ = [3, n0, _PCl,
    0,
    [_t, _ex],
    [0, 0]
];
export var PlacementStrategy$ = [3, n0, _PS,
    0,
    [_t, _f],
    [0, 0]
];
export var Primary$ = [3, n0, _Pr,
    0,
    [_HC],
    [0], 1
];
export var PutEventsRequest$ = [3, n0, _PER,
    0,
    [_Ent, _EI],
    [() => PutEventsRequestEntryList, 0], 1
];
export var PutEventsRequestEntry$ = [3, n0, _PERE,
    0,
    [_Ti, _So, _Re, _DT, _Det, _EBN, _TH],
    [4, 0, 64 | 0, 0, 0, 0, 0]
];
export var PutEventsResponse$ = [3, n0, _PERu,
    0,
    [_FEC, _Ent],
    [1, () => PutEventsResultEntryList]
];
export var PutEventsResultEntry$ = [3, n0, _PEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
export var PutPartnerEventsRequest$ = [3, n0, _PPER,
    0,
    [_Ent],
    [() => PutPartnerEventsRequestEntryList], 1
];
export var PutPartnerEventsRequestEntry$ = [3, n0, _PPERE,
    0,
    [_Ti, _So, _Re, _DT, _Det],
    [4, 0, 64 | 0, 0, 0]
];
export var PutPartnerEventsResponse$ = [3, n0, _PPERu,
    0,
    [_FEC, _Ent],
    [1, () => PutPartnerEventsResultEntryList]
];
export var PutPartnerEventsResultEntry$ = [3, n0, _PPEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
export var PutPermissionRequest$ = [3, n0, _PPR,
    0,
    [_EBN, _Act, _Pri, _SI, _C, _Po],
    [0, 0, 0, 0, () => Condition$, 0]
];
export var PutRuleRequest$ = [3, n0, _PRR,
    0,
    [_N, _SE, _EP, _St, _D, _RAo, _Ta, _EBN],
    [0, 0, 0, 0, 0, 0, () => TagList, 0], 1
];
export var PutRuleResponse$ = [3, n0, _PRRu,
    0,
    [_RAu],
    [0]
];
export var PutTargetsRequest$ = [3, n0, _PTR,
    0,
    [_Rul, _Tar, _EBN],
    [0, [() => TargetList, 0], 0], 2
];
export var PutTargetsResponse$ = [3, n0, _PTRu,
    0,
    [_FEC, _FE],
    [1, () => PutTargetsResultEntryList]
];
export var PutTargetsResultEntry$ = [3, n0, _PTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
export var RedshiftDataParameters$ = [3, n0, _RDP,
    0,
    [_Da, _SMA, _DU, _S, _SN, _WE, _Sq],
    [0, 0, 0, [() => Sql, 0], 0, 2, [() => Sqls, 0]], 1
];
export var RemovePermissionRequest$ = [3, n0, _RPR,
    0,
    [_SI, _RAP, _EBN],
    [0, 2, 0]
];
export var RemoveTargetsRequest$ = [3, n0, _RTR,
    0,
    [_Rul, _I, _EBN, _F],
    [0, 64 | 0, 0, 2], 2
];
export var RemoveTargetsResponse$ = [3, n0, _RTRe,
    0,
    [_FEC, _FE],
    [1, () => RemoveTargetsResultEntryList]
];
export var RemoveTargetsResultEntry$ = [3, n0, _RTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
export var Replay$ = [3, n0, _Rep,
    0,
    [_RN, _ESA, _St, _SR, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 4, 4, 4, 4, 4]
];
export var ReplayDestination$ = [3, n0, _RDe,
    0,
    [_Ar, _FA],
    [0, 64 | 0], 1
];
export var ReplicationConfig$ = [3, n0, _RCe,
    0,
    [_St],
    [0]
];
export var RetryPolicy$ = [3, n0, _RPe,
    0,
    [_MRA, _MEAIS],
    [1, 1]
];
export var RoutingConfig$ = [3, n0, _RC,
    0,
    [_FC],
    [() => FailoverConfig$], 1
];
export var Rule$ = [3, n0, _Rul,
    0,
    [_N, _Ar, _EP, _St, _D, _SE, _RAo, _MB, _EBN],
    [0, 0, 0, 0, 0, 0, 0, 0, 0]
];
export var RunCommandParameters$ = [3, n0, _RCP,
    0,
    [_RCT],
    [() => RunCommandTargets], 1
];
export var RunCommandTarget$ = [3, n0, _RCTu,
    0,
    [_K, _Va],
    [0, 64 | 0], 2
];
export var SageMakerPipelineParameter$ = [3, n0, _SMPP,
    0,
    [_N, _V],
    [0, 0], 2
];
export var SageMakerPipelineParameters$ = [3, n0, _SMPPa,
    0,
    [_PPL],
    [() => SageMakerPipelineParameterList]
];
export var Secondary$ = [3, n0, _Se,
    0,
    [_Ro],
    [0], 1
];
export var SqsParameters$ = [3, n0, _SP,
    0,
    [_MGI],
    [0]
];
export var StartReplayRequest$ = [3, n0, _SRR,
    0,
    [_RN, _ESA, _EST, _EET, _De, _D],
    [0, 0, 4, 4, () => ReplayDestination$, 0], 5
];
export var StartReplayResponse$ = [3, n0, _SRRt,
    0,
    [_RA, _St, _SR, _RST],
    [0, 0, 0, 4]
];
export var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
export var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _Ta],
    [0, () => TagList], 2
];
export var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
export var Target$ = [3, n0, _Targ,
    0,
    [_Id, _Ar, _RAo, _In, _IP, _IT, _KP, _RCP, _EPc, _BP, _SP, _HPt, _RDP, _SMPPa, _DLC, _RPe, _ASP],
    [0, 0, 0, 0, 0, () => InputTransformer$, () => KinesisParameters$, () => RunCommandParameters$, () => EcsParameters$, () => BatchParameters$, () => SqsParameters$, () => HttpParameters$, [() => RedshiftDataParameters$, 0], () => SageMakerPipelineParameters$, () => DeadLetterConfig$, () => RetryPolicy$, [() => AppSyncParameters$, 0]], 2
];
export var TestEventPatternRequest$ = [3, n0, _TEPR,
    0,
    [_EP, _Ev],
    [0, 0], 2
];
export var TestEventPatternResponse$ = [3, n0, _TEPRe,
    0,
    [_Res],
    [2]
];
export var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
export var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
export var UpdateApiDestinationRequest$ = [3, n0, _UADR,
    0,
    [_N, _D, _CA, _IEn, _HM, _IRLPS],
    [0, 0, 0, 0, 0, 1], 1
];
export var UpdateApiDestinationResponse$ = [3, n0, _UADRp,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
export var UpdateArchiveRequest$ = [3, n0, _UAR,
    0,
    [_AN, _D, _EP, _RD, _KKI],
    [0, 0, 0, 1, 0], 1
];
export var UpdateArchiveResponse$ = [3, n0, _UARp,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
export var UpdateConnectionApiKeyAuthRequestParameters$ = [3, n0, _UCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
export var UpdateConnectionAuthRequestParameters$ = [3, n0, _UCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => UpdateConnectionBasicAuthRequestParameters$, 0], [() => UpdateConnectionOAuthRequestParameters$, 0], [() => UpdateConnectionApiKeyAuthRequestParameters$, 0], [() => ConnectionHttpParameters$, 0], () => ConnectivityResourceParameters$]
];
export var UpdateConnectionBasicAuthRequestParameters$ = [3, n0, _UCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
export var UpdateConnectionOAuthClientRequestParameters$ = [3, n0, _UCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
export var UpdateConnectionOAuthRequestParameters$ = [3, n0, _UCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => UpdateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => ConnectionHttpParameters$, 0]]
];
export var UpdateConnectionRequest$ = [3, n0, _UCR,
    0,
    [_N, _D, _AT, _APu, _ICP, _KKI],
    [0, 0, 0, [() => UpdateConnectionAuthRequestParameters$, 0], () => ConnectivityResourceParameters$, 0], 1
];
export var UpdateConnectionResponse$ = [3, n0, _UCRp,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
export var UpdateEndpointRequest$ = [3, n0, _UER,
    0,
    [_N, _D, _RC, _RCe, _EB, _RAo],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0], 1
];
export var UpdateEndpointResponse$ = [3, n0, _UERp,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0]
];
export var UpdateEventBusRequest$ = [3, n0, _UEBR,
    0,
    [_N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
export var UpdateEventBusResponse$ = [3, n0, _UEBRp,
    0,
    [_Ar, _N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
var __Unit = "unit";
var ApiDestinationResponseList = [1, n0, _ADRL,
    0, () => ApiDestination$
];
var ArchiveResponseList = [1, n0, _ARL,
    0, () => Archive$
];
var CapacityProviderStrategy = [1, n0, _CPS,
    0, () => CapacityProviderStrategyItem$
];
var ConnectionBodyParametersList = [1, n0, _CBPL,
    0, [() => ConnectionBodyParameter$,
        0]
];
var ConnectionHeaderParametersList = [1, n0, _CHPL,
    0, [() => ConnectionHeaderParameter$,
        0]
];
var ConnectionQueryStringParametersList = [1, n0, _CQSPL,
    0, [() => ConnectionQueryStringParameter$,
        0]
];
var ConnectionResponseList = [1, n0, _CRL,
    0, () => Connection$
];
var EndpointEventBusList = [1, n0, _EEBL,
    0, () => EndpointEventBus$
];
var EndpointList = [1, n0, _EL,
    0, () => Endpoint$
];
var EventBusList = [1, n0, _EBL,
    0, () => EventBus$
];
var EventResourceList = 64 | 0;
var EventSourceList = [1, n0, _ESL,
    0, () => EventSource$
];
var PartnerEventSourceAccountList = [1, n0, _PESAL,
    0, () => PartnerEventSourceAccount$
];
var PartnerEventSourceList = [1, n0, _PESL,
    0, () => PartnerEventSource$
];
var PathParameterList = 64 | 0;
var PlacementConstraints = [1, n0, _PC,
    0, () => PlacementConstraint$
];
var PlacementStrategies = [1, n0, _PSl,
    0, () => PlacementStrategy$
];
var PutEventsRequestEntryList = [1, n0, _PEREL,
    0, () => PutEventsRequestEntry$
];
var PutEventsResultEntryList = [1, n0, _PERELu,
    0, () => PutEventsResultEntry$
];
var PutPartnerEventsRequestEntryList = [1, n0, _PPEREL,
    0, () => PutPartnerEventsRequestEntry$
];
var PutPartnerEventsResultEntryList = [1, n0, _PPERELu,
    0, () => PutPartnerEventsResultEntry$
];
var PutTargetsResultEntryList = [1, n0, _PTREL,
    0, () => PutTargetsResultEntry$
];
var RemoveTargetsResultEntryList = [1, n0, _RTREL,
    0, () => RemoveTargetsResultEntry$
];
var ReplayDestinationFilters = 64 | 0;
var ReplayList = [1, n0, _RL,
    0, () => Replay$
];
var RuleNameList = 64 | 0;
var RuleResponseList = [1, n0, _RRL,
    0, () => Rule$
];
var RunCommandTargets = [1, n0, _RCT,
    0, () => RunCommandTarget$
];
var RunCommandTargetValues = 64 | 0;
var SageMakerPipelineParameterList = [1, n0, _SMPPL,
    0, () => SageMakerPipelineParameter$
];
var Sqls = [1, n0, _Sq,
    8, [() => Sql,
        0]
];
var StringList = 64 | 0;
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TargetIdList = 64 | 0;
var TargetList = [1, n0, _TLa,
    0, [() => Target$,
        0]
];
var HeaderParametersMap = 128 | 0;
var QueryStringParametersMap = 128 | 0;
var TransformerPaths = 128 | 0;
export var ActivateEventSource$ = [9, n0, _AES,
    0, () => ActivateEventSourceRequest$, () => __Unit
];
export var CancelReplay$ = [9, n0, _CR,
    0, () => CancelReplayRequest$, () => CancelReplayResponse$
];
export var CreateApiDestination$ = [9, n0, _CAD,
    0, () => CreateApiDestinationRequest$, () => CreateApiDestinationResponse$
];
export var CreateArchive$ = [9, n0, _CAr,
    0, () => CreateArchiveRequest$, () => CreateArchiveResponse$
];
export var CreateConnection$ = [9, n0, _CC,
    0, () => CreateConnectionRequest$, () => CreateConnectionResponse$
];
export var CreateEndpoint$ = [9, n0, _CE,
    0, () => CreateEndpointRequest$, () => CreateEndpointResponse$
];
export var CreateEventBus$ = [9, n0, _CEB,
    0, () => CreateEventBusRequest$, () => CreateEventBusResponse$
];
export var CreatePartnerEventSource$ = [9, n0, _CPES,
    0, () => CreatePartnerEventSourceRequest$, () => CreatePartnerEventSourceResponse$
];
export var DeactivateEventSource$ = [9, n0, _DES,
    0, () => DeactivateEventSourceRequest$, () => __Unit
];
export var DeauthorizeConnection$ = [9, n0, _DC,
    0, () => DeauthorizeConnectionRequest$, () => DeauthorizeConnectionResponse$
];
export var DeleteApiDestination$ = [9, n0, _DAD,
    0, () => DeleteApiDestinationRequest$, () => DeleteApiDestinationResponse$
];
export var DeleteArchive$ = [9, n0, _DA,
    0, () => DeleteArchiveRequest$, () => DeleteArchiveResponse$
];
export var DeleteConnection$ = [9, n0, _DCe,
    0, () => DeleteConnectionRequest$, () => DeleteConnectionResponse$
];
export var DeleteEndpoint$ = [9, n0, _DE,
    0, () => DeleteEndpointRequest$, () => DeleteEndpointResponse$
];
export var DeleteEventBus$ = [9, n0, _DEB,
    0, () => DeleteEventBusRequest$, () => __Unit
];
export var DeletePartnerEventSource$ = [9, n0, _DPES,
    0, () => DeletePartnerEventSourceRequest$, () => __Unit
];
export var DeleteRule$ = [9, n0, _DR,
    0, () => DeleteRuleRequest$, () => __Unit
];
export var DescribeApiDestination$ = [9, n0, _DADe,
    0, () => DescribeApiDestinationRequest$, () => DescribeApiDestinationResponse$
];
export var DescribeArchive$ = [9, n0, _DAe,
    0, () => DescribeArchiveRequest$, () => DescribeArchiveResponse$
];
export var DescribeConnection$ = [9, n0, _DCes,
    0, () => DescribeConnectionRequest$, () => DescribeConnectionResponse$
];
export var DescribeEndpoint$ = [9, n0, _DEe,
    0, () => DescribeEndpointRequest$, () => DescribeEndpointResponse$
];
export var DescribeEventBus$ = [9, n0, _DEBe,
    0, () => DescribeEventBusRequest$, () => DescribeEventBusResponse$
];
export var DescribeEventSource$ = [9, n0, _DESe,
    0, () => DescribeEventSourceRequest$, () => DescribeEventSourceResponse$
];
export var DescribePartnerEventSource$ = [9, n0, _DPESe,
    0, () => DescribePartnerEventSourceRequest$, () => DescribePartnerEventSourceResponse$
];
export var DescribeReplay$ = [9, n0, _DRe,
    0, () => DescribeReplayRequest$, () => DescribeReplayResponse$
];
export var DescribeRule$ = [9, n0, _DRes,
    0, () => DescribeRuleRequest$, () => DescribeRuleResponse$
];
export var DisableRule$ = [9, n0, _DRi,
    0, () => DisableRuleRequest$, () => __Unit
];
export var EnableRule$ = [9, n0, _ER,
    0, () => EnableRuleRequest$, () => __Unit
];
export var ListApiDestinations$ = [9, n0, _LAD,
    0, () => ListApiDestinationsRequest$, () => ListApiDestinationsResponse$
];
export var ListArchives$ = [9, n0, _LA,
    0, () => ListArchivesRequest$, () => ListArchivesResponse$
];
export var ListConnections$ = [9, n0, _LCi,
    0, () => ListConnectionsRequest$, () => ListConnectionsResponse$
];
export var ListEndpoints$ = [9, n0, _LE,
    0, () => ListEndpointsRequest$, () => ListEndpointsResponse$
];
export var ListEventBuses$ = [9, n0, _LEB,
    0, () => ListEventBusesRequest$, () => ListEventBusesResponse$
];
export var ListEventSources$ = [9, n0, _LES,
    0, () => ListEventSourcesRequest$, () => ListEventSourcesResponse$
];
export var ListPartnerEventSourceAccounts$ = [9, n0, _LPESA,
    0, () => ListPartnerEventSourceAccountsRequest$, () => ListPartnerEventSourceAccountsResponse$
];
export var ListPartnerEventSources$ = [9, n0, _LPES,
    0, () => ListPartnerEventSourcesRequest$, () => ListPartnerEventSourcesResponse$
];
export var ListReplays$ = [9, n0, _LR,
    0, () => ListReplaysRequest$, () => ListReplaysResponse$
];
export var ListRuleNamesByTarget$ = [9, n0, _LRNBT,
    0, () => ListRuleNamesByTargetRequest$, () => ListRuleNamesByTargetResponse$
];
export var ListRules$ = [9, n0, _LRi,
    0, () => ListRulesRequest$, () => ListRulesResponse$
];
export var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
export var ListTargetsByRule$ = [9, n0, _LTBR,
    0, () => ListTargetsByRuleRequest$, () => ListTargetsByRuleResponse$
];
export var PutEvents$ = [9, n0, _PE,
    0, () => PutEventsRequest$, () => PutEventsResponse$
];
export var PutPartnerEvents$ = [9, n0, _PPE,
    0, () => PutPartnerEventsRequest$, () => PutPartnerEventsResponse$
];
export var PutPermission$ = [9, n0, _PP,
    0, () => PutPermissionRequest$, () => __Unit
];
export var PutRule$ = [9, n0, _PR,
    0, () => PutRuleRequest$, () => PutRuleResponse$
];
export var PutTargets$ = [9, n0, _PTu,
    0, () => PutTargetsRequest$, () => PutTargetsResponse$
];
export var RemovePermission$ = [9, n0, _RPem,
    0, () => RemovePermissionRequest$, () => __Unit
];
export var RemoveTargets$ = [9, n0, _RT,
    0, () => RemoveTargetsRequest$, () => RemoveTargetsResponse$
];
export var StartReplay$ = [9, n0, _SRt,
    0, () => StartReplayRequest$, () => StartReplayResponse$
];
export var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
export var TestEventPattern$ = [9, n0, _TEP,
    0, () => TestEventPatternRequest$, () => TestEventPatternResponse$
];
export var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
export var UpdateApiDestination$ = [9, n0, _UAD,
    0, () => UpdateApiDestinationRequest$, () => UpdateApiDestinationResponse$
];
export var UpdateArchive$ = [9, n0, _UA,
    0, () => UpdateArchiveRequest$, () => UpdateArchiveResponse$
];
export var UpdateConnection$ = [9, n0, _UC,
    0, () => UpdateConnectionRequest$, () => UpdateConnectionResponse$
];
export var UpdateEndpoint$ = [9, n0, _UE,
    0, () => UpdateEndpointRequest$, () => UpdateEndpointResponse$
];
export var UpdateEventBus$ = [9, n0, _UEB,
    0, () => UpdateEventBusRequest$, () => UpdateEventBusResponse$
];
