// @aws-sdk/client-eventbridge/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateConnectionResponse$ = exports.CreateConnectionRequest$ = exports.CreateConnectionOAuthRequestParameters$ = exports.CreateConnectionOAuthClientRequestParameters$ = exports.CreateConnectionBasicAuthRequestParameters$ = exports.CreateConnectionAuthRequestParameters$ = exports.CreateConnectionApiKeyAuthRequestParameters$ = exports.CreateArchiveResponse$ = exports.CreateArchiveRequest$ = exports.CreateApiDestinationResponse$ = exports.CreateApiDestinationRequest$ = exports.ConnectivityResourceParameters$ = exports.ConnectivityResourceConfigurationArn$ = exports.ConnectionQueryStringParameter$ = exports.ConnectionOAuthResponseParameters$ = exports.ConnectionOAuthClientResponseParameters$ = exports.ConnectionHttpParameters$ = exports.ConnectionHeaderParameter$ = exports.ConnectionBodyParameter$ = exports.ConnectionBasicAuthResponseParameters$ = exports.ConnectionAuthResponseParameters$ = exports.ConnectionApiKeyAuthResponseParameters$ = exports.Connection$ = exports.Condition$ = exports.CapacityProviderStrategyItem$ = exports.CancelReplayResponse$ = exports.CancelReplayRequest$ = exports.BatchRetryStrategy$ = exports.BatchParameters$ = exports.BatchArrayProperties$ = exports.AwsVpcConfiguration$ = exports.Archive$ = exports.AppSyncParameters$ = exports.ApiDestination$ = exports.ActivateEventSourceRequest$ = exports.errorTypeRegistries = exports.ThrottlingException$ = exports.ResourceNotFoundException$ = exports.ResourceAlreadyExistsException$ = exports.PolicyLengthExceededException$ = exports.OperationDisabledException$ = exports.ManagedRuleException$ = exports.LimitExceededException$ = exports.InvalidStateException$ = exports.InvalidEventPatternException$ = exports.InternalException$ = exports.IllegalStatusException$ = exports.ConcurrentModificationException$ = exports.AccessDeniedException$ = exports.EventBridgeServiceException$ = void 0;
exports.HttpParameters$ = exports.FailoverConfig$ = exports.EventSource$ = exports.EventBus$ = exports.EndpointEventBus$ = exports.Endpoint$ = exports.EnableRuleRequest$ = exports.EcsParameters$ = exports.DisableRuleRequest$ = exports.DescribeRuleResponse$ = exports.DescribeRuleRequest$ = exports.DescribeReplayResponse$ = exports.DescribeReplayRequest$ = exports.DescribePartnerEventSourceResponse$ = exports.DescribePartnerEventSourceRequest$ = exports.DescribeEventSourceResponse$ = exports.DescribeEventSourceRequest$ = exports.DescribeEventBusResponse$ = exports.DescribeEventBusRequest$ = exports.DescribeEndpointResponse$ = exports.DescribeEndpointRequest$ = exports.DescribeConnectionResponse$ = exports.DescribeConnectionResourceParameters$ = exports.DescribeConnectionRequest$ = exports.DescribeConnectionConnectivityParameters$ = exports.DescribeArchiveResponse$ = exports.DescribeArchiveRequest$ = exports.DescribeApiDestinationResponse$ = exports.DescribeApiDestinationRequest$ = exports.DeleteRuleRequest$ = exports.DeletePartnerEventSourceRequest$ = exports.DeleteEventBusRequest$ = exports.DeleteEndpointResponse$ = exports.DeleteEndpointRequest$ = exports.DeleteConnectionResponse$ = exports.DeleteConnectionRequest$ = exports.DeleteArchiveResponse$ = exports.DeleteArchiveRequest$ = exports.DeleteApiDestinationResponse$ = exports.DeleteApiDestinationRequest$ = exports.DeauthorizeConnectionResponse$ = exports.DeauthorizeConnectionRequest$ = exports.DeadLetterConfig$ = exports.DeactivateEventSourceRequest$ = exports.CreatePartnerEventSourceResponse$ = exports.CreatePartnerEventSourceRequest$ = exports.CreateEventBusResponse$ = exports.CreateEventBusRequest$ = exports.CreateEndpointResponse$ = exports.CreateEndpointRequest$ = void 0;
exports.RedshiftDataParameters$ = exports.PutTargetsResultEntry$ = exports.PutTargetsResponse$ = exports.PutTargetsRequest$ = exports.PutRuleResponse$ = exports.PutRuleRequest$ = exports.PutPermissionRequest$ = exports.PutPartnerEventsResultEntry$ = exports.PutPartnerEventsResponse$ = exports.PutPartnerEventsRequestEntry$ = exports.PutPartnerEventsRequest$ = exports.PutEventsResultEntry$ = exports.PutEventsResponse$ = exports.PutEventsRequestEntry$ = exports.PutEventsRequest$ = exports.Primary$ = exports.PlacementStrategy$ = exports.PlacementConstraint$ = exports.PartnerEventSourceAccount$ = exports.PartnerEventSource$ = exports.NetworkConfiguration$ = exports.LogConfig$ = exports.ListTargetsByRuleResponse$ = exports.ListTargetsByRuleRequest$ = exports.ListTagsForResourceResponse$ = exports.ListTagsForResourceRequest$ = exports.ListRulesResponse$ = exports.ListRulesRequest$ = exports.ListRuleNamesByTargetResponse$ = exports.ListRuleNamesByTargetRequest$ = exports.ListReplaysResponse$ = exports.ListReplaysRequest$ = exports.ListPartnerEventSourcesResponse$ = exports.ListPartnerEventSourcesRequest$ = exports.ListPartnerEventSourceAccountsResponse$ = exports.ListPartnerEventSourceAccountsRequest$ = exports.ListEventSourcesResponse$ = exports.ListEventSourcesRequest$ = exports.ListEventBusesResponse$ = exports.ListEventBusesRequest$ = exports.ListEndpointsResponse$ = exports.ListEndpointsRequest$ = exports.ListConnectionsResponse$ = exports.ListConnectionsRequest$ = exports.ListArchivesResponse$ = exports.ListArchivesRequest$ = exports.ListApiDestinationsResponse$ = exports.ListApiDestinationsRequest$ = exports.KinesisParameters$ = exports.InputTransformer$ = void 0;
exports.DeactivateEventSource$ = exports.CreatePartnerEventSource$ = exports.CreateEventBus$ = exports.CreateEndpoint$ = exports.CreateConnection$ = exports.CreateArchive$ = exports.CreateApiDestination$ = exports.CancelReplay$ = exports.ActivateEventSource$ = exports.UpdateEventBusResponse$ = exports.UpdateEventBusRequest$ = exports.UpdateEndpointResponse$ = exports.UpdateEndpointRequest$ = exports.UpdateConnectionResponse$ = exports.UpdateConnectionRequest$ = exports.UpdateConnectionOAuthRequestParameters$ = exports.UpdateConnectionOAuthClientRequestParameters$ = exports.UpdateConnectionBasicAuthRequestParameters$ = exports.UpdateConnectionAuthRequestParameters$ = exports.UpdateConnectionApiKeyAuthRequestParameters$ = exports.UpdateArchiveResponse$ = exports.UpdateArchiveRequest$ = exports.UpdateApiDestinationResponse$ = exports.UpdateApiDestinationRequest$ = exports.UntagResourceResponse$ = exports.UntagResourceRequest$ = exports.TestEventPatternResponse$ = exports.TestEventPatternRequest$ = exports.Target$ = exports.TagResourceResponse$ = exports.TagResourceRequest$ = exports.Tag$ = exports.StartReplayResponse$ = exports.StartReplayRequest$ = exports.SqsParameters$ = exports.Secondary$ = exports.SageMakerPipelineParameters$ = exports.SageMakerPipelineParameter$ = exports.RunCommandTarget$ = exports.RunCommandParameters$ = exports.Rule$ = exports.RoutingConfig$ = exports.RetryPolicy$ = exports.ReplicationConfig$ = exports.ReplayDestination$ = exports.Replay$ = exports.RemoveTargetsResultEntry$ = exports.RemoveTargetsResponse$ = exports.RemoveTargetsRequest$ = exports.RemovePermissionRequest$ = void 0;
exports.UpdateEventBus$ = exports.UpdateEndpoint$ = exports.UpdateConnection$ = exports.UpdateArchive$ = exports.UpdateApiDestination$ = exports.UntagResource$ = exports.TestEventPattern$ = exports.TagResource$ = exports.StartReplay$ = exports.RemoveTargets$ = exports.RemovePermission$ = exports.PutTargets$ = exports.PutRule$ = exports.PutPermission$ = exports.PutPartnerEvents$ = exports.PutEvents$ = exports.ListTargetsByRule$ = exports.ListTagsForResource$ = exports.ListRules$ = exports.ListRuleNamesByTarget$ = exports.ListReplays$ = exports.ListPartnerEventSources$ = exports.ListPartnerEventSourceAccounts$ = exports.ListEventSources$ = exports.ListEventBuses$ = exports.ListEndpoints$ = exports.ListConnections$ = exports.ListArchives$ = exports.ListApiDestinations$ = exports.EnableRule$ = exports.DisableRule$ = exports.DescribeRule$ = exports.DescribeReplay$ = exports.DescribePartnerEventSource$ = exports.DescribeEventSource$ = exports.DescribeEventBus$ = exports.DescribeEndpoint$ = exports.DescribeConnection$ = exports.DescribeArchive$ = exports.DescribeApiDestination$ = exports.DeleteRule$ = exports.DeletePartnerEventSource$ = exports.DeleteEventBus$ = exports.DeleteEndpoint$ = exports.DeleteConnection$ = exports.DeleteArchive$ = exports.DeleteApiDestination$ = exports.DeauthorizeConnection$ = void 0;
const _A = "Archive";
const _AA = "ArchiveArn";
const _AD = "ApiDestination";
const _ADA = "ApiDestinationArn";
const _ADE = "AccessDeniedException";
const _ADRL = "ApiDestinationResponseList";
const _ADS = "ApiDestinationState";
const _ADp = "ApiDestinations";
const _AE = "AuthorizationEndpoint";
const _AES = "ActivateEventSource";
const _AESR = "ActivateEventSourceRequest";
const _AHPS = "AuthHeaderParametersSensitive";
const _AKAP = "ApiKeyAuthParameters";
const _AKN = "ApiKeyName";
const _AKV = "ApiKeyValue";
const _AN = "ArchiveName";
const _AP = "ArrayProperties";
const _API = "AssignPublicIp";
const _APu = "AuthParameters";
const _ARL = "ArchiveResponseList";
const _ASP = "AppSyncParameters";
const _AT = "AuthorizationType";
const _AVC = "AwsVpcConfiguration";
const _Ac = "Account";
const _Act = "Action";
const _Ar = "Arn";
const _Arc = "Archives";
const _At = "Attempts";
const _BAP = "BatchArrayProperties";
const _BAPa = "BasicAuthParameters";
const _BP = "BatchParameters";
const _BPo = "BodyParameters";
const _BRS = "BatchRetryStrategy";
const _C = "Condition";
const _CA = "ConnectionArn";
const _CAD = "CreateApiDestination";
const _CADR = "CreateApiDestinationRequest";
const _CADRr = "CreateApiDestinationResponse";
const _CAKARP = "ConnectionApiKeyAuthResponseParameters";
const _CAR = "CreateArchiveRequest";
const _CARP = "ConnectionAuthResponseParameters";
const _CARr = "CreateArchiveResponse";
const _CAr = "CreateArchive";
const _CB = "CreatedBy";
const _CBARP = "ConnectionBasicAuthResponseParameters";
const _CBP = "ConnectionBodyParameter";
const _CBPL = "ConnectionBodyParametersList";
const _CC = "CreateConnection";
const _CCAKARP = "CreateConnectionApiKeyAuthRequestParameters";
const _CCARP = "CreateConnectionAuthRequestParameters";
const _CCBARP = "CreateConnectionBasicAuthRequestParameters";
const _CCOACRP = "CreateConnectionOAuthClientRequestParameters";
const _CCOARP = "CreateConnectionOAuthRequestParameters";
const _CCR = "CreateConnectionRequest";
const _CCRr = "CreateConnectionResponse";
const _CE = "CreateEndpoint";
const _CEB = "CreateEventBus";
const _CEBR = "CreateEventBusRequest";
const _CEBRr = "CreateEventBusResponse";
const _CER = "CreateEndpointRequest";
const _CERr = "CreateEndpointResponse";
const _CHP = "ConnectionHeaderParameter";
const _CHPL = "ConnectionHeaderParametersList";
const _CHPo = "ConnectionHttpParameters";
const _CID = "ClientID";
const _CME = "ConcurrentModificationException";
const _COACRP = "ConnectionOAuthClientResponseParameters";
const _COARP = "ConnectionOAuthResponseParameters";
const _CP = "ConnectivityParameters";
const _CPES = "CreatePartnerEventSource";
const _CPESR = "CreatePartnerEventSourceRequest";
const _CPESRr = "CreatePartnerEventSourceResponse";
const _CPS = "CapacityProviderStrategy";
const _CPSI = "CapacityProviderStrategyItem";
const _CPl = "ClientParameters";
const _CQSP = "ConnectionQueryStringParameter";
const _CQSPL = "ConnectionQueryStringParametersList";
const _CR = "CancelReplay";
const _CRCA = "ConnectivityResourceConfigurationArn";
const _CRL = "ConnectionResponseList";
const _CRP = "ConnectivityResourceParameters";
const _CRR = "CancelReplayRequest";
const _CRRa = "CancelReplayResponse";
const _CS = "ConnectionState";
const _CSl = "ClientSecret";
const _CT = "CreationTime";
const _Co = "Connection";
const _Con = "Connections";
const _D = "Description";
const _DA = "DeleteArchive";
const _DAD = "DeleteApiDestination";
const _DADR = "DeleteApiDestinationRequest";
const _DADRe = "DeleteApiDestinationResponse";
const _DADRes = "DescribeApiDestinationRequest";
const _DADResc = "DescribeApiDestinationResponse";
const _DADe = "DescribeApiDestination";
const _DAR = "DeleteArchiveRequest";
const _DARe = "DeleteArchiveResponse";
const _DARes = "DescribeArchiveRequest";
const _DAResc = "DescribeArchiveResponse";
const _DAe = "DescribeArchive";
const _DC = "DeauthorizeConnection";
const _DCCP = "DescribeConnectionConnectivityParameters";
const _DCR = "DeauthorizeConnectionRequest";
const _DCRP = "DescribeConnectionResourceParameters";
const _DCRe = "DeauthorizeConnectionResponse";
const _DCRel = "DeleteConnectionRequest";
const _DCRele = "DeleteConnectionResponse";
const _DCRes = "DescribeConnectionRequest";
const _DCResc = "DescribeConnectionResponse";
const _DCe = "DeleteConnection";
const _DCes = "DescribeConnection";
const _DE = "DeleteEndpoint";
const _DEB = "DeleteEventBus";
const _DEBR = "DeleteEventBusRequest";
const _DEBRe = "DescribeEventBusRequest";
const _DEBRes = "DescribeEventBusResponse";
const _DEBe = "DescribeEventBus";
const _DER = "DeleteEndpointRequest";
const _DERe = "DeleteEndpointResponse";
const _DERes = "DescribeEndpointRequest";
const _DEResc = "DescribeEndpointResponse";
const _DES = "DeactivateEventSource";
const _DESR = "DeactivateEventSourceRequest";
const _DESRe = "DescribeEventSourceRequest";
const _DESRes = "DescribeEventSourceResponse";
const _DESe = "DescribeEventSource";
const _DEe = "DescribeEndpoint";
const _DLC = "DeadLetterConfig";
const _DPES = "DeletePartnerEventSource";
const _DPESR = "DeletePartnerEventSourceRequest";
const _DPESRe = "DescribePartnerEventSourceRequest";
const _DPESRes = "DescribePartnerEventSourceResponse";
const _DPESe = "DescribePartnerEventSource";
const _DR = "DeleteRule";
const _DRR = "DeleteRuleRequest";
const _DRRe = "DescribeReplayRequest";
const _DRRes = "DescribeReplayResponse";
const _DRResc = "DescribeRuleRequest";
const _DRRescr = "DescribeRuleResponse";
const _DRRi = "DisableRuleRequest";
const _DRe = "DescribeReplay";
const _DRes = "DescribeRule";
const _DRi = "DisableRule";
const _DT = "DetailType";
const _DU = "DbUser";
const _Da = "Database";
const _De = "Destination";
const _Det = "Detail";
const _E = "Endpoint";
const _EB = "EventBuses";
const _EBA = "EventBusArn";
const _EBL = "EventBusList";
const _EBN = "EventBusName";
const _EBv = "EventBus";
const _EC = "EventCount";
const _ECr = "ErrorCode";
const _EEB = "EndpointEventBus";
const _EEBL = "EndpointEventBusList";
const _EEC = "EnableExecuteCommand";
const _EECSMT = "EnableECSManagedTags";
const _EET = "EventEndTime";
const _EI = "EndpointId";
const _EIv = "EventId";
const _EL = "EndpointList";
const _ELRT = "EventLastReplayedTime";
const _EM = "ErrorMessage";
const _EP = "EventPattern";
const _EPc = "EcsParameters";
const _ER = "EnableRule";
const _ERR = "EnableRuleRequest";
const _ES = "EventSource";
const _ESA = "EventSourceArn";
const _ESL = "EventSourceList";
const _ESN = "EventSourceName";
const _EST = "EventStartTime";
const _ESv = "EventSources";
const _ET = "ExpirationTime";
const _EU = "EndpointUrl";
const _En = "Endpoints";
const _Ent = "Entries";
const _Ev = "Event";
const _F = "Force";
const _FA = "FilterArns";
const _FC = "FailoverConfig";
const _FE = "FailedEntries";
const _FEC = "FailedEntryCount";
const _G = "Group";
const _GQLO = "GraphQLOperation";
const _HC = "HealthCheck";
const _HM = "HttpMethod";
const _HP = "HeaderParameters";
const _HPt = "HttpParameters";
const _HR = "HomeRegion";
const _HVS = "HeaderValueSensitive";
const _I = "Ids";
const _ICP = "InvocationConnectivityParameters";
const _ID = "IncludeDetail";
const _IE = "InternalException";
const _IEPE = "InvalidEventPatternException";
const _IEn = "InvocationEndpoint";
const _IHP = "InvocationHttpParameters";
const _IP = "InputPath";
const _IPM = "InputPathsMap";
const _IRLPS = "InvocationRateLimitPerSecond";
const _ISE = "IllegalStatusException";
const _ISEn = "InvalidStateException";
const _IT = "InputTransformer";
const _ITn = "InputTemplate";
const _IVS = "IsValueSecret";
const _Id = "Id";
const _In = "Input";
const _JD = "JobDefinition";
const _JN = "JobName";
const _K = "Key";
const _KKI = "KmsKeyIdentifier";
const _KP = "KinesisParameters";
const _L = "Limit";
const _LA = "ListArchives";
const _LAD = "ListApiDestinations";
const _LADR = "ListApiDestinationsRequest";
const _LADRi = "ListApiDestinationsResponse";
const _LAR = "ListArchivesRequest";
const _LARi = "ListArchivesResponse";
const _LAT = "LastAuthorizedTime";
const _LC = "LogConfig";
const _LCR = "ListConnectionsRequest";
const _LCRi = "ListConnectionsResponse";
const _LCi = "ListConnections";
const _LE = "ListEndpoints";
const _LEB = "ListEventBuses";
const _LEBR = "ListEventBusesRequest";
const _LEBRi = "ListEventBusesResponse";
const _LEE = "LimitExceededException";
const _LER = "ListEndpointsRequest";
const _LERi = "ListEndpointsResponse";
const _LES = "ListEventSources";
const _LESR = "ListEventSourcesRequest";
const _LESRi = "ListEventSourcesResponse";
const _LMT = "LastModifiedTime";
const _LPES = "ListPartnerEventSources";
const _LPESA = "ListPartnerEventSourceAccounts";
const _LPESAR = "ListPartnerEventSourceAccountsRequest";
const _LPESARi = "ListPartnerEventSourceAccountsResponse";
const _LPESR = "ListPartnerEventSourcesRequest";
const _LPESRi = "ListPartnerEventSourcesResponse";
const _LR = "ListReplays";
const _LRNBT = "ListRuleNamesByTarget";
const _LRNBTR = "ListRuleNamesByTargetRequest";
const _LRNBTRi = "ListRuleNamesByTargetResponse";
const _LRR = "ListReplaysRequest";
const _LRRi = "ListReplaysResponse";
const _LRRis = "ListRulesRequest";
const _LRRist = "ListRulesResponse";
const _LRi = "ListRules";
const _LT = "LaunchType";
const _LTBR = "ListTargetsByRule";
const _LTBRR = "ListTargetsByRuleRequest";
const _LTBRRi = "ListTargetsByRuleResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _Le = "Level";
const _MB = "ManagedBy";
const _MEAIS = "MaximumEventAgeInSeconds";
const _MGI = "MessageGroupId";
const _MR = "MaxResults";
const _MRA = "MaximumRetryAttempts";
const _MRE = "ManagedRuleException";
const _N = "Name";
const _NC = "NetworkConfiguration";
const _NP = "NamePrefix";
const _NT = "NextToken";
const _OAHP = "OAuthHttpParameters";
const _OAP = "OAuthParameters";
const _ODE = "OperationDisabledException";
const _P = "Password";
const _PC = "PlacementConstraints";
const _PCl = "PlacementConstraint";
const _PE = "PutEvents";
const _PER = "PutEventsRequest";
const _PERE = "PutEventsRequestEntry";
const _PEREL = "PutEventsRequestEntryList";
const _PERELu = "PutEventsResultEntryList";
const _PEREu = "PutEventsResultEntry";
const _PERu = "PutEventsResponse";
const _PES = "PartnerEventSources";
const _PESA = "PartnerEventSourceAccounts";
const _PESAL = "PartnerEventSourceAccountList";
const _PESAa = "PartnerEventSourceAccount";
const _PESL = "PartnerEventSourceList";
const _PESa = "PartnerEventSource";
const _PKP = "PartitionKeyPath";
const _PLEE = "PolicyLengthExceededException";
const _PP = "PutPermission";
const _PPE = "PutPartnerEvents";
const _PPER = "PutPartnerEventsRequest";
const _PPERE = "PutPartnerEventsRequestEntry";
const _PPEREL = "PutPartnerEventsRequestEntryList";
const _PPERELu = "PutPartnerEventsResultEntryList";
const _PPEREu = "PutPartnerEventsResultEntry";
const _PPERu = "PutPartnerEventsResponse";
const _PPL = "PipelineParameterList";
const _PPR = "PutPermissionRequest";
const _PPV = "PathParameterValues";
const _PR = "PutRule";
const _PRR = "PutRuleRequest";
const _PRRu = "PutRuleResponse";
const _PS = "PlacementStrategy";
const _PSl = "PlacementStrategies";
const _PT = "PropagateTags";
const _PTR = "PutTargetsRequest";
const _PTRE = "PutTargetsResultEntry";
const _PTREL = "PutTargetsResultEntryList";
const _PTRu = "PutTargetsResponse";
const _PTu = "PutTargets";
const _PV = "PlatformVersion";
const _Po = "Policy";
const _Pr = "Primary";
const _Pri = "Principal";
const _QSP = "QueryStringParameters";
const _QSVS = "QueryStringValueSensitive";
const _R = "Replays";
const _RA = "ReplayArn";
const _RAA = "ResourceAssociationArn";
const _RAEE = "ResourceAlreadyExistsException";
const _RAP = "RemoveAllPermissions";
const _RARN = "ResourceARN";
const _RAo = "RoleArn";
const _RAu = "RuleArn";
const _RC = "RoutingConfig";
const _RCA = "ResourceConfigurationArn";
const _RCP = "RunCommandParameters";
const _RCT = "RunCommandTargets";
const _RCTu = "RunCommandTarget";
const _RCe = "ReplicationConfig";
const _RD = "RetentionDays";
const _RDP = "RedshiftDataParameters";
const _RDe = "ReplayDestination";
const _RET = "ReplayEndTime";
const _RI = "ReferenceId";
const _RL = "ReplayList";
const _RN = "ReplayName";
const _RNFE = "ResourceNotFoundException";
const _RNu = "RuleNames";
const _RP = "ResourceParameters";
const _RPR = "RemovePermissionRequest";
const _RPe = "RetryPolicy";
const _RPem = "RemovePermission";
const _RRL = "RuleResponseList";
const _RS = "RetryStrategy";
const _RST = "ReplayStartTime";
const _RT = "RemoveTargets";
const _RTR = "RemoveTargetsRequest";
const _RTRE = "RemoveTargetsResultEntry";
const _RTREL = "RemoveTargetsResultEntryList";
const _RTRe = "RemoveTargetsResponse";
const _Re = "Resources";
const _Rep = "Replay";
const _Res = "Result";
const _Ro = "Route";
const _Ru = "Rules";
const _Rul = "Rule";
const _S = "Sql";
const _SA = "SecretArn";
const _SB = "SizeBytes";
const _SE = "ScheduleExpression";
const _SG = "SecurityGroups";
const _SI = "StatementId";
const _SMA = "SecretManagerArn";
const _SMPP = "SageMakerPipelineParameter";
const _SMPPL = "SageMakerPipelineParameterList";
const _SMPPa = "SageMakerPipelineParameters";
const _SN = "StatementName";
const _SP = "SqsParameters";
const _SR = "StateReason";
const _SRR = "StartReplayRequest";
const _SRRt = "StartReplayResponse";
const _SRt = "StartReplay";
const _SS = "SensitiveString";
const _Se = "Secondary";
const _Si = "Size";
const _So = "Source";
const _Sq = "Sqls";
const _St = "State";
const _Su = "Subnets";
const _T = "Type";
const _TA = "TargetArn";
const _TC = "TaskCount";
const _TDA = "TaskDefinitionArn";
const _TE = "ThrottlingException";
const _TEP = "TestEventPattern";
const _TEPR = "TestEventPatternRequest";
const _TEPRe = "TestEventPatternResponse";
const _TH = "TraceHeader";
const _TI = "TargetId";
const _TK = "TagKeys";
const _TL = "TagList";
const _TLa = "TargetList";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Targets";
const _Targ = "Target";
const _Ti = "Time";
const _U = "Username";
const _UA = "UpdateArchive";
const _UAD = "UpdateApiDestination";
const _UADR = "UpdateApiDestinationRequest";
const _UADRp = "UpdateApiDestinationResponse";
const _UAR = "UpdateArchiveRequest";
const _UARp = "UpdateArchiveResponse";
const _UC = "UpdateConnection";
const _UCAKARP = "UpdateConnectionApiKeyAuthRequestParameters";
const _UCARP = "UpdateConnectionAuthRequestParameters";
const _UCBARP = "UpdateConnectionBasicAuthRequestParameters";
const _UCOACRP = "UpdateConnectionOAuthClientRequestParameters";
const _UCOARP = "UpdateConnectionOAuthRequestParameters";
const _UCR = "UpdateConnectionRequest";
const _UCRp = "UpdateConnectionResponse";
const _UE = "UpdateEndpoint";
const _UEB = "UpdateEventBus";
const _UEBR = "UpdateEventBusRequest";
const _UEBRp = "UpdateEventBusResponse";
const _UER = "UpdateEndpointRequest";
const _UERp = "UpdateEndpointResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _V = "Value";
const _Va = "Values";
const _WE = "WithEvent";
const _aC = "awsvpcConfiguration";
const _b = "base";
const _c = "client";
const _cP = "capacityProvider";
const _e = "error";
const _ex = "expression";
const _f = "field";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.eventbridge";
const _se = "server";
const _t = "type";
const _w = "weight";
const n0 = "com.amazonaws.eventbridge";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const EventBridgeServiceException_1 = require("../models/EventBridgeServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.EventBridgeServiceException$ = [-3, _s, "EventBridgeServiceException", 0, [], []];
_s_registry.registerError(exports.EventBridgeServiceException$, EventBridgeServiceException_1.EventBridgeServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.AccessDeniedException$, errors_1.AccessDeniedException);
exports.ConcurrentModificationException$ = [-3, n0, _CME,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ConcurrentModificationException$, errors_1.ConcurrentModificationException);
exports.IllegalStatusException$ = [-3, n0, _ISE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.IllegalStatusException$, errors_1.IllegalStatusException);
exports.InternalException$ = [-3, n0, _IE,
    { [_e]: _se },
    [_m],
    [0]
];
n0_registry.registerError(exports.InternalException$, errors_1.InternalException);
exports.InvalidEventPatternException$ = [-3, n0, _IEPE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidEventPatternException$, errors_1.InvalidEventPatternException);
exports.InvalidStateException$ = [-3, n0, _ISEn,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidStateException$, errors_1.InvalidStateException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.ManagedRuleException$ = [-3, n0, _MRE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ManagedRuleException$, errors_1.ManagedRuleException);
exports.OperationDisabledException$ = [-3, n0, _ODE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.OperationDisabledException$, errors_1.OperationDisabledException);
exports.PolicyLengthExceededException$ = [-3, n0, _PLEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.PolicyLengthExceededException$, errors_1.PolicyLengthExceededException);
exports.ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceAlreadyExistsException$, errors_1.ResourceAlreadyExistsException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(exports.ThrottlingException$, errors_1.ThrottlingException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AuthHeaderParametersSensitive = [0, n0, _AHPS, 8, 0];
var GraphQLOperation = [0, n0, _GQLO, 8, 0];
var HeaderValueSensitive = [0, n0, _HVS, 8, 0];
var QueryStringValueSensitive = [0, n0, _QSVS, 8, 0];
var SensitiveString = [0, n0, _SS, 8, 0];
var Sql = [0, n0, _S, 8, 0];
exports.ActivateEventSourceRequest$ = [3, n0, _AESR,
    0,
    [_N],
    [0], 1
];
exports.ApiDestination$ = [3, n0, _AD,
    0,
    [_ADA, _N, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 1, 4, 4]
];
exports.AppSyncParameters$ = [3, n0, _ASP,
    0,
    [_GQLO],
    [[() => GraphQLOperation, 0]]
];
exports.Archive$ = [3, n0, _A,
    0,
    [_AN, _ESA, _St, _SR, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 1, 1, 1, 4]
];
exports.AwsVpcConfiguration$ = [3, n0, _AVC,
    0,
    [_Su, _SG, _API],
    [64 | 0, 64 | 0, 0], 1
];
exports.BatchArrayProperties$ = [3, n0, _BAP,
    0,
    [_Si],
    [1]
];
exports.BatchParameters$ = [3, n0, _BP,
    0,
    [_JD, _JN, _AP, _RS],
    [0, 0, () => exports.BatchArrayProperties$, () => exports.BatchRetryStrategy$], 2
];
exports.BatchRetryStrategy$ = [3, n0, _BRS,
    0,
    [_At],
    [1]
];
exports.CancelReplayRequest$ = [3, n0, _CRR,
    0,
    [_RN],
    [0], 1
];
exports.CancelReplayResponse$ = [3, n0, _CRRa,
    0,
    [_RA, _St, _SR],
    [0, 0, 0]
];
exports.CapacityProviderStrategyItem$ = [3, n0, _CPSI,
    0,
    [_cP, _w, _b],
    [0, 1, 1], 1
];
exports.Condition$ = [3, n0, _C,
    0,
    [_T, _K, _V],
    [0, 0, 0], 3
];
exports.Connection$ = [3, n0, _Co,
    0,
    [_CA, _N, _CS, _SR, _AT, _CT, _LMT, _LAT],
    [0, 0, 0, 0, 0, 4, 4, 4]
];
exports.ConnectionApiKeyAuthResponseParameters$ = [3, n0, _CAKARP,
    0,
    [_AKN],
    [0]
];
exports.ConnectionAuthResponseParameters$ = [3, n0, _CARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [() => exports.ConnectionBasicAuthResponseParameters$, [() => exports.ConnectionOAuthResponseParameters$, 0], () => exports.ConnectionApiKeyAuthResponseParameters$, [() => exports.ConnectionHttpParameters$, 0], () => exports.DescribeConnectionConnectivityParameters$]
];
exports.ConnectionBasicAuthResponseParameters$ = [3, n0, _CBARP,
    0,
    [_U],
    [0]
];
exports.ConnectionBodyParameter$ = [3, n0, _CBP,
    0,
    [_K, _V, _IVS],
    [0, [() => SensitiveString, 0], 2]
];
exports.ConnectionHeaderParameter$ = [3, n0, _CHP,
    0,
    [_K, _V, _IVS],
    [0, [() => HeaderValueSensitive, 0], 2]
];
exports.ConnectionHttpParameters$ = [3, n0, _CHPo,
    0,
    [_HP, _QSP, _BPo],
    [[() => ConnectionHeaderParametersList, 0], [() => ConnectionQueryStringParametersList, 0], [() => ConnectionBodyParametersList, 0]]
];
exports.ConnectionOAuthClientResponseParameters$ = [3, n0, _COACRP,
    0,
    [_CID],
    [0]
];
exports.ConnectionOAuthResponseParameters$ = [3, n0, _COARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [() => exports.ConnectionOAuthClientResponseParameters$, 0, 0, [() => exports.ConnectionHttpParameters$, 0]]
];
exports.ConnectionQueryStringParameter$ = [3, n0, _CQSP,
    0,
    [_K, _V, _IVS],
    [0, [() => QueryStringValueSensitive, 0], 2]
];
exports.ConnectivityResourceConfigurationArn$ = [3, n0, _CRCA,
    0,
    [_RCA],
    [0], 1
];
exports.ConnectivityResourceParameters$ = [3, n0, _CRP,
    0,
    [_RP],
    [() => exports.ConnectivityResourceConfigurationArn$], 1
];
exports.CreateApiDestinationRequest$ = [3, n0, _CADR,
    0,
    [_N, _CA, _IEn, _HM, _D, _IRLPS],
    [0, 0, 0, 0, 0, 1], 4
];
exports.CreateApiDestinationResponse$ = [3, n0, _CADRr,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
exports.CreateArchiveRequest$ = [3, n0, _CAR,
    0,
    [_AN, _ESA, _D, _EP, _RD, _KKI],
    [0, 0, 0, 0, 1, 0], 2
];
exports.CreateArchiveResponse$ = [3, n0, _CARr,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
exports.CreateConnectionApiKeyAuthRequestParameters$ = [3, n0, _CCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
exports.CreateConnectionAuthRequestParameters$ = [3, n0, _CCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => exports.CreateConnectionBasicAuthRequestParameters$, 0], [() => exports.CreateConnectionOAuthRequestParameters$, 0], [() => exports.CreateConnectionApiKeyAuthRequestParameters$, 0], [() => exports.ConnectionHttpParameters$, 0], () => exports.ConnectivityResourceParameters$]
];
exports.CreateConnectionBasicAuthRequestParameters$ = [3, n0, _CCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
exports.CreateConnectionOAuthClientRequestParameters$ = [3, n0, _CCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
exports.CreateConnectionOAuthRequestParameters$ = [3, n0, _CCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => exports.CreateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => exports.ConnectionHttpParameters$, 0]], 3
];
exports.CreateConnectionRequest$ = [3, n0, _CCR,
    0,
    [_N, _AT, _APu, _D, _ICP, _KKI],
    [0, 0, [() => exports.CreateConnectionAuthRequestParameters$, 0], 0, () => exports.ConnectivityResourceParameters$, 0], 3
];
exports.CreateConnectionResponse$ = [3, n0, _CCRr,
    0,
    [_CA, _CS, _CT, _LMT],
    [0, 0, 4, 4]
];
exports.CreateEndpointRequest$ = [3, n0, _CER,
    0,
    [_N, _RC, _EB, _D, _RCe, _RAo],
    [0, () => exports.RoutingConfig$, () => EndpointEventBusList, 0, () => exports.ReplicationConfig$, 0], 3
];
exports.CreateEndpointResponse$ = [3, n0, _CERr,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _St],
    [0, 0, () => exports.RoutingConfig$, () => exports.ReplicationConfig$, () => EndpointEventBusList, 0, 0]
];
exports.CreateEventBusRequest$ = [3, n0, _CEBR,
    0,
    [_N, _ESN, _D, _KKI, _DLC, _LC, _Ta],
    [0, 0, 0, 0, () => exports.DeadLetterConfig$, () => exports.LogConfig$, () => TagList], 1
];
exports.CreateEventBusResponse$ = [3, n0, _CEBRr,
    0,
    [_EBA, _D, _KKI, _DLC, _LC],
    [0, 0, 0, () => exports.DeadLetterConfig$, () => exports.LogConfig$]
];
exports.CreatePartnerEventSourceRequest$ = [3, n0, _CPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
exports.CreatePartnerEventSourceResponse$ = [3, n0, _CPESRr,
    0,
    [_ESA],
    [0]
];
exports.DeactivateEventSourceRequest$ = [3, n0, _DESR,
    0,
    [_N],
    [0], 1
];
exports.DeadLetterConfig$ = [3, n0, _DLC,
    0,
    [_Ar],
    [0]
];
exports.DeauthorizeConnectionRequest$ = [3, n0, _DCR,
    0,
    [_N],
    [0], 1
];
exports.DeauthorizeConnectionResponse$ = [3, n0, _DCRe,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
exports.DeleteApiDestinationRequest$ = [3, n0, _DADR,
    0,
    [_N],
    [0], 1
];
exports.DeleteApiDestinationResponse$ = [3, n0, _DADRe,
    0,
    [],
    []
];
exports.DeleteArchiveRequest$ = [3, n0, _DAR,
    0,
    [_AN],
    [0], 1
];
exports.DeleteArchiveResponse$ = [3, n0, _DARe,
    0,
    [],
    []
];
exports.DeleteConnectionRequest$ = [3, n0, _DCRel,
    0,
    [_N],
    [0], 1
];
exports.DeleteConnectionResponse$ = [3, n0, _DCRele,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
exports.DeleteEndpointRequest$ = [3, n0, _DER,
    0,
    [_N],
    [0], 1
];
exports.DeleteEndpointResponse$ = [3, n0, _DERe,
    0,
    [],
    []
];
exports.DeleteEventBusRequest$ = [3, n0, _DEBR,
    0,
    [_N],
    [0], 1
];
exports.DeletePartnerEventSourceRequest$ = [3, n0, _DPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
exports.DeleteRuleRequest$ = [3, n0, _DRR,
    0,
    [_N, _EBN, _F],
    [0, 0, 2], 1
];
exports.DescribeApiDestinationRequest$ = [3, n0, _DADRes,
    0,
    [_N],
    [0], 1
];
exports.DescribeApiDestinationResponse$ = [3, n0, _DADResc,
    0,
    [_ADA, _N, _D, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 0, 1, 4, 4]
];
exports.DescribeArchiveRequest$ = [3, n0, _DARes,
    0,
    [_AN],
    [0], 1
];
exports.DescribeArchiveResponse$ = [3, n0, _DAResc,
    0,
    [_AA, _AN, _ESA, _D, _EP, _St, _SR, _KKI, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 4]
];
exports.DescribeConnectionConnectivityParameters$ = [3, n0, _DCCP,
    0,
    [_RP],
    [() => exports.DescribeConnectionResourceParameters$], 1
];
exports.DescribeConnectionRequest$ = [3, n0, _DCRes,
    0,
    [_N],
    [0], 1
];
exports.DescribeConnectionResourceParameters$ = [3, n0, _DCRP,
    0,
    [_RCA, _RAA],
    [0, 0], 2
];
exports.DescribeConnectionResponse$ = [3, n0, _DCResc,
    0,
    [_CA, _N, _D, _ICP, _CS, _SR, _AT, _SA, _KKI, _APu, _CT, _LMT, _LAT],
    [0, 0, 0, () => exports.DescribeConnectionConnectivityParameters$, 0, 0, 0, 0, 0, [() => exports.ConnectionAuthResponseParameters$, 0], 4, 4, 4]
];
exports.DescribeEndpointRequest$ = [3, n0, _DERes,
    0,
    [_N, _HR],
    [0, 0], 1
];
exports.DescribeEndpointResponse$ = [3, n0, _DEResc,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => exports.RoutingConfig$, () => exports.ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
exports.DescribeEventBusRequest$ = [3, n0, _DEBRe,
    0,
    [_N],
    [0]
];
exports.DescribeEventBusResponse$ = [3, n0, _DEBRes,
    0,
    [_N, _Ar, _D, _KKI, _DLC, _Po, _LC, _CT, _LMT],
    [0, 0, 0, 0, () => exports.DeadLetterConfig$, 0, () => exports.LogConfig$, 4, 4]
];
exports.DescribeEventSourceRequest$ = [3, n0, _DESRe,
    0,
    [_N],
    [0], 1
];
exports.DescribeEventSourceResponse$ = [3, n0, _DESRes,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
exports.DescribePartnerEventSourceRequest$ = [3, n0, _DPESRe,
    0,
    [_N],
    [0], 1
];
exports.DescribePartnerEventSourceResponse$ = [3, n0, _DPESRes,
    0,
    [_Ar, _N],
    [0, 0]
];
exports.DescribeReplayRequest$ = [3, n0, _DRRe,
    0,
    [_RN],
    [0], 1
];
exports.DescribeReplayResponse$ = [3, n0, _DRRes,
    0,
    [_RN, _RA, _D, _St, _SR, _ESA, _De, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 0, 0, () => exports.ReplayDestination$, 4, 4, 4, 4, 4]
];
exports.DescribeRuleRequest$ = [3, n0, _DRResc,
    0,
    [_N, _EBN],
    [0, 0], 1
];
exports.DescribeRuleResponse$ = [3, n0, _DRRescr,
    0,
    [_N, _Ar, _EP, _SE, _St, _D, _RAo, _MB, _EBN, _CB],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
];
exports.DisableRuleRequest$ = [3, n0, _DRRi,
    0,
    [_N, _EBN],
    [0, 0], 1
];
exports.EcsParameters$ = [3, n0, _EPc,
    0,
    [_TDA, _TC, _LT, _NC, _PV, _G, _CPS, _EECSMT, _EEC, _PC, _PS, _PT, _RI, _Ta],
    [0, 1, 0, () => exports.NetworkConfiguration$, 0, 0, () => CapacityProviderStrategy, 2, 2, () => PlacementConstraints, () => PlacementStrategies, 0, 0, () => TagList], 1
];
exports.EnableRuleRequest$ = [3, n0, _ERR,
    0,
    [_N, _EBN],
    [0, 0], 1
];
exports.Endpoint$ = [3, n0, _E,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => exports.RoutingConfig$, () => exports.ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
exports.EndpointEventBus$ = [3, n0, _EEB,
    0,
    [_EBA],
    [0], 1
];
exports.EventBus$ = [3, n0, _EBv,
    0,
    [_N, _Ar, _D, _Po, _CT, _LMT],
    [0, 0, 0, 0, 4, 4]
];
exports.EventSource$ = [3, n0, _ES,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
exports.FailoverConfig$ = [3, n0, _FC,
    0,
    [_Pr, _Se],
    [() => exports.Primary$, () => exports.Secondary$], 2
];
exports.HttpParameters$ = [3, n0, _HPt,
    0,
    [_PPV, _HP, _QSP],
    [64 | 0, 128 | 0, 128 | 0]
];
exports.InputTransformer$ = [3, n0, _IT,
    0,
    [_ITn, _IPM],
    [0, 128 | 0], 1
];
exports.KinesisParameters$ = [3, n0, _KP,
    0,
    [_PKP],
    [0], 1
];
exports.ListApiDestinationsRequest$ = [3, n0, _LADR,
    0,
    [_NP, _CA, _NT, _L],
    [0, 0, 0, 1]
];
exports.ListApiDestinationsResponse$ = [3, n0, _LADRi,
    0,
    [_ADp, _NT],
    [() => ApiDestinationResponseList, 0]
];
exports.ListArchivesRequest$ = [3, n0, _LAR,
    0,
    [_NP, _ESA, _St, _NT, _L],
    [0, 0, 0, 0, 1]
];
exports.ListArchivesResponse$ = [3, n0, _LARi,
    0,
    [_Arc, _NT],
    [() => ArchiveResponseList, 0]
];
exports.ListConnectionsRequest$ = [3, n0, _LCR,
    0,
    [_NP, _CS, _NT, _L],
    [0, 0, 0, 1]
];
exports.ListConnectionsResponse$ = [3, n0, _LCRi,
    0,
    [_Con, _NT],
    [() => ConnectionResponseList, 0]
];
exports.ListEndpointsRequest$ = [3, n0, _LER,
    0,
    [_NP, _HR, _NT, _MR],
    [0, 0, 0, 1]
];
exports.ListEndpointsResponse$ = [3, n0, _LERi,
    0,
    [_En, _NT],
    [() => EndpointList, 0]
];
exports.ListEventBusesRequest$ = [3, n0, _LEBR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
exports.ListEventBusesResponse$ = [3, n0, _LEBRi,
    0,
    [_EB, _NT],
    [() => EventBusList, 0]
];
exports.ListEventSourcesRequest$ = [3, n0, _LESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
exports.ListEventSourcesResponse$ = [3, n0, _LESRi,
    0,
    [_ESv, _NT],
    [() => EventSourceList, 0]
];
exports.ListPartnerEventSourceAccountsRequest$ = [3, n0, _LPESAR,
    0,
    [_ESN, _NT, _L],
    [0, 0, 1], 1
];
exports.ListPartnerEventSourceAccountsResponse$ = [3, n0, _LPESARi,
    0,
    [_PESA, _NT],
    [() => PartnerEventSourceAccountList, 0]
];
exports.ListPartnerEventSourcesRequest$ = [3, n0, _LPESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1], 1
];
exports.ListPartnerEventSourcesResponse$ = [3, n0, _LPESRi,
    0,
    [_PES, _NT],
    [() => PartnerEventSourceList, 0]
];
exports.ListReplaysRequest$ = [3, n0, _LRR,
    0,
    [_NP, _St, _ESA, _NT, _L],
    [0, 0, 0, 0, 1]
];
exports.ListReplaysResponse$ = [3, n0, _LRRi,
    0,
    [_R, _NT],
    [() => ReplayList, 0]
];
exports.ListRuleNamesByTargetRequest$ = [3, n0, _LRNBTR,
    0,
    [_TA, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
exports.ListRuleNamesByTargetResponse$ = [3, n0, _LRNBTRi,
    0,
    [_RNu, _NT],
    [64 | 0, 0]
];
exports.ListRulesRequest$ = [3, n0, _LRRis,
    0,
    [_NP, _EBN, _NT, _L],
    [0, 0, 0, 1]
];
exports.ListRulesResponse$ = [3, n0, _LRRist,
    0,
    [_Ru, _NT],
    [() => RuleResponseList, 0]
];
exports.ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
exports.ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [() => TagList]
];
exports.ListTargetsByRuleRequest$ = [3, n0, _LTBRR,
    0,
    [_Rul, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
exports.ListTargetsByRuleResponse$ = [3, n0, _LTBRRi,
    0,
    [_Tar, _NT],
    [[() => TargetList, 0], 0]
];
exports.LogConfig$ = [3, n0, _LC,
    0,
    [_ID, _Le],
    [0, 0]
];
exports.NetworkConfiguration$ = [3, n0, _NC,
    0,
    [_aC],
    [() => exports.AwsVpcConfiguration$]
];
exports.PartnerEventSource$ = [3, n0, _PESa,
    0,
    [_Ar, _N],
    [0, 0]
];
exports.PartnerEventSourceAccount$ = [3, n0, _PESAa,
    0,
    [_Ac, _CT, _ET, _St],
    [0, 4, 4, 0]
];
exports.PlacementConstraint$ = [3, n0, _PCl,
    0,
    [_t, _ex],
    [0, 0]
];
exports.PlacementStrategy$ = [3, n0, _PS,
    0,
    [_t, _f],
    [0, 0]
];
exports.Primary$ = [3, n0, _Pr,
    0,
    [_HC],
    [0], 1
];
exports.PutEventsRequest$ = [3, n0, _PER,
    0,
    [_Ent, _EI],
    [() => PutEventsRequestEntryList, 0], 1
];
exports.PutEventsRequestEntry$ = [3, n0, _PERE,
    0,
    [_Ti, _So, _Re, _DT, _Det, _EBN, _TH],
    [4, 0, 64 | 0, 0, 0, 0, 0]
];
exports.PutEventsResponse$ = [3, n0, _PERu,
    0,
    [_FEC, _Ent],
    [1, () => PutEventsResultEntryList]
];
exports.PutEventsResultEntry$ = [3, n0, _PEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
exports.PutPartnerEventsRequest$ = [3, n0, _PPER,
    0,
    [_Ent],
    [() => PutPartnerEventsRequestEntryList], 1
];
exports.PutPartnerEventsRequestEntry$ = [3, n0, _PPERE,
    0,
    [_Ti, _So, _Re, _DT, _Det],
    [4, 0, 64 | 0, 0, 0]
];
exports.PutPartnerEventsResponse$ = [3, n0, _PPERu,
    0,
    [_FEC, _Ent],
    [1, () => PutPartnerEventsResultEntryList]
];
exports.PutPartnerEventsResultEntry$ = [3, n0, _PPEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
exports.PutPermissionRequest$ = [3, n0, _PPR,
    0,
    [_EBN, _Act, _Pri, _SI, _C, _Po],
    [0, 0, 0, 0, () => exports.Condition$, 0]
];
exports.PutRuleRequest$ = [3, n0, _PRR,
    0,
    [_N, _SE, _EP, _St, _D, _RAo, _Ta, _EBN],
    [0, 0, 0, 0, 0, 0, () => TagList, 0], 1
];
exports.PutRuleResponse$ = [3, n0, _PRRu,
    0,
    [_RAu],
    [0]
];
exports.PutTargetsRequest$ = [3, n0, _PTR,
    0,
    [_Rul, _Tar, _EBN],
    [0, [() => TargetList, 0], 0], 2
];
exports.PutTargetsResponse$ = [3, n0, _PTRu,
    0,
    [_FEC, _FE],
    [1, () => PutTargetsResultEntryList]
];
exports.PutTargetsResultEntry$ = [3, n0, _PTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
exports.RedshiftDataParameters$ = [3, n0, _RDP,
    0,
    [_Da, _SMA, _DU, _S, _SN, _WE, _Sq],
    [0, 0, 0, [() => Sql, 0], 0, 2, [() => Sqls, 0]], 1
];
exports.RemovePermissionRequest$ = [3, n0, _RPR,
    0,
    [_SI, _RAP, _EBN],
    [0, 2, 0]
];
exports.RemoveTargetsRequest$ = [3, n0, _RTR,
    0,
    [_Rul, _I, _EBN, _F],
    [0, 64 | 0, 0, 2], 2
];
exports.RemoveTargetsResponse$ = [3, n0, _RTRe,
    0,
    [_FEC, _FE],
    [1, () => RemoveTargetsResultEntryList]
];
exports.RemoveTargetsResultEntry$ = [3, n0, _RTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
exports.Replay$ = [3, n0, _Rep,
    0,
    [_RN, _ESA, _St, _SR, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 4, 4, 4, 4, 4]
];
exports.ReplayDestination$ = [3, n0, _RDe,
    0,
    [_Ar, _FA],
    [0, 64 | 0], 1
];
exports.ReplicationConfig$ = [3, n0, _RCe,
    0,
    [_St],
    [0]
];
exports.RetryPolicy$ = [3, n0, _RPe,
    0,
    [_MRA, _MEAIS],
    [1, 1]
];
exports.RoutingConfig$ = [3, n0, _RC,
    0,
    [_FC],
    [() => exports.FailoverConfig$], 1
];
exports.Rule$ = [3, n0, _Rul,
    0,
    [_N, _Ar, _EP, _St, _D, _SE, _RAo, _MB, _EBN],
    [0, 0, 0, 0, 0, 0, 0, 0, 0]
];
exports.RunCommandParameters$ = [3, n0, _RCP,
    0,
    [_RCT],
    [() => RunCommandTargets], 1
];
exports.RunCommandTarget$ = [3, n0, _RCTu,
    0,
    [_K, _Va],
    [0, 64 | 0], 2
];
exports.SageMakerPipelineParameter$ = [3, n0, _SMPP,
    0,
    [_N, _V],
    [0, 0], 2
];
exports.SageMakerPipelineParameters$ = [3, n0, _SMPPa,
    0,
    [_PPL],
    [() => SageMakerPipelineParameterList]
];
exports.Secondary$ = [3, n0, _Se,
    0,
    [_Ro],
    [0], 1
];
exports.SqsParameters$ = [3, n0, _SP,
    0,
    [_MGI],
    [0]
];
exports.StartReplayRequest$ = [3, n0, _SRR,
    0,
    [_RN, _ESA, _EST, _EET, _De, _D],
    [0, 0, 4, 4, () => exports.ReplayDestination$, 0], 5
];
exports.StartReplayResponse$ = [3, n0, _SRRt,
    0,
    [_RA, _St, _SR, _RST],
    [0, 0, 0, 4]
];
exports.Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _Ta],
    [0, () => TagList], 2
];
exports.TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.Target$ = [3, n0, _Targ,
    0,
    [_Id, _Ar, _RAo, _In, _IP, _IT, _KP, _RCP, _EPc, _BP, _SP, _HPt, _RDP, _SMPPa, _DLC, _RPe, _ASP],
    [0, 0, 0, 0, 0, () => exports.InputTransformer$, () => exports.KinesisParameters$, () => exports.RunCommandParameters$, () => exports.EcsParameters$, () => exports.BatchParameters$, () => exports.SqsParameters$, () => exports.HttpParameters$, [() => exports.RedshiftDataParameters$, 0], () => exports.SageMakerPipelineParameters$, () => exports.DeadLetterConfig$, () => exports.RetryPolicy$, [() => exports.AppSyncParameters$, 0]], 2
];
exports.TestEventPatternRequest$ = [3, n0, _TEPR,
    0,
    [_EP, _Ev],
    [0, 0], 2
];
exports.TestEventPatternResponse$ = [3, n0, _TEPRe,
    0,
    [_Res],
    [2]
];
exports.UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UpdateApiDestinationRequest$ = [3, n0, _UADR,
    0,
    [_N, _D, _CA, _IEn, _HM, _IRLPS],
    [0, 0, 0, 0, 0, 1], 1
];
exports.UpdateApiDestinationResponse$ = [3, n0, _UADRp,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
exports.UpdateArchiveRequest$ = [3, n0, _UAR,
    0,
    [_AN, _D, _EP, _RD, _KKI],
    [0, 0, 0, 1, 0], 1
];
exports.UpdateArchiveResponse$ = [3, n0, _UARp,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
exports.UpdateConnectionApiKeyAuthRequestParameters$ = [3, n0, _UCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
exports.UpdateConnectionAuthRequestParameters$ = [3, n0, _UCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => exports.UpdateConnectionBasicAuthRequestParameters$, 0], [() => exports.UpdateConnectionOAuthRequestParameters$, 0], [() => exports.UpdateConnectionApiKeyAuthRequestParameters$, 0], [() => exports.ConnectionHttpParameters$, 0], () => exports.ConnectivityResourceParameters$]
];
exports.UpdateConnectionBasicAuthRequestParameters$ = [3, n0, _UCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
exports.UpdateConnectionOAuthClientRequestParameters$ = [3, n0, _UCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
exports.UpdateConnectionOAuthRequestParameters$ = [3, n0, _UCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => exports.UpdateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => exports.ConnectionHttpParameters$, 0]]
];
exports.UpdateConnectionRequest$ = [3, n0, _UCR,
    0,
    [_N, _D, _AT, _APu, _ICP, _KKI],
    [0, 0, 0, [() => exports.UpdateConnectionAuthRequestParameters$, 0], () => exports.ConnectivityResourceParameters$, 0], 1
];
exports.UpdateConnectionResponse$ = [3, n0, _UCRp,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
exports.UpdateEndpointRequest$ = [3, n0, _UER,
    0,
    [_N, _D, _RC, _RCe, _EB, _RAo],
    [0, 0, () => exports.RoutingConfig$, () => exports.ReplicationConfig$, () => EndpointEventBusList, 0], 1
];
exports.UpdateEndpointResponse$ = [3, n0, _UERp,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St],
    [0, 0, () => exports.RoutingConfig$, () => exports.ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0]
];
exports.UpdateEventBusRequest$ = [3, n0, _UEBR,
    0,
    [_N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, () => exports.DeadLetterConfig$, () => exports.LogConfig$]
];
exports.UpdateEventBusResponse$ = [3, n0, _UEBRp,
    0,
    [_Ar, _N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, 0, () => exports.DeadLetterConfig$, () => exports.LogConfig$]
];
var __Unit = "unit";
var ApiDestinationResponseList = [1, n0, _ADRL,
    0, () => exports.ApiDestination$
];
var ArchiveResponseList = [1, n0, _ARL,
    0, () => exports.Archive$
];
var CapacityProviderStrategy = [1, n0, _CPS,
    0, () => exports.CapacityProviderStrategyItem$
];
var ConnectionBodyParametersList = [1, n0, _CBPL,
    0, [() => exports.ConnectionBodyParameter$,
        0]
];
var ConnectionHeaderParametersList = [1, n0, _CHPL,
    0, [() => exports.ConnectionHeaderParameter$,
        0]
];
var ConnectionQueryStringParametersList = [1, n0, _CQSPL,
    0, [() => exports.ConnectionQueryStringParameter$,
        0]
];
var ConnectionResponseList = [1, n0, _CRL,
    0, () => exports.Connection$
];
var EndpointEventBusList = [1, n0, _EEBL,
    0, () => exports.EndpointEventBus$
];
var EndpointList = [1, n0, _EL,
    0, () => exports.Endpoint$
];
var EventBusList = [1, n0, _EBL,
    0, () => exports.EventBus$
];
var EventResourceList = 64 | 0;
var EventSourceList = [1, n0, _ESL,
    0, () => exports.EventSource$
];
var PartnerEventSourceAccountList = [1, n0, _PESAL,
    0, () => exports.PartnerEventSourceAccount$
];
var PartnerEventSourceList = [1, n0, _PESL,
    0, () => exports.PartnerEventSource$
];
var PathParameterList = 64 | 0;
var PlacementConstraints = [1, n0, _PC,
    0, () => exports.PlacementConstraint$
];
var PlacementStrategies = [1, n0, _PSl,
    0, () => exports.PlacementStrategy$
];
var PutEventsRequestEntryList = [1, n0, _PEREL,
    0, () => exports.PutEventsRequestEntry$
];
var PutEventsResultEntryList = [1, n0, _PERELu,
    0, () => exports.PutEventsResultEntry$
];
var PutPartnerEventsRequestEntryList = [1, n0, _PPEREL,
    0, () => exports.PutPartnerEventsRequestEntry$
];
var PutPartnerEventsResultEntryList = [1, n0, _PPERELu,
    0, () => exports.PutPartnerEventsResultEntry$
];
var PutTargetsResultEntryList = [1, n0, _PTREL,
    0, () => exports.PutTargetsResultEntry$
];
var RemoveTargetsResultEntryList = [1, n0, _RTREL,
    0, () => exports.RemoveTargetsResultEntry$
];
var ReplayDestinationFilters = 64 | 0;
var ReplayList = [1, n0, _RL,
    0, () => exports.Replay$
];
var RuleNameList = 64 | 0;
var RuleResponseList = [1, n0, _RRL,
    0, () => exports.Rule$
];
var RunCommandTargets = [1, n0, _RCT,
    0, () => exports.RunCommandTarget$
];
var RunCommandTargetValues = 64 | 0;
var SageMakerPipelineParameterList = [1, n0, _SMPPL,
    0, () => exports.SageMakerPipelineParameter$
];
var Sqls = [1, n0, _Sq,
    8, [() => Sql,
        0]
];
var StringList = 64 | 0;
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => exports.Tag$
];
var TargetIdList = 64 | 0;
var TargetList = [1, n0, _TLa,
    0, [() => exports.Target$,
        0]
];
var HeaderParametersMap = 128 | 0;
var QueryStringParametersMap = 128 | 0;
var TransformerPaths = 128 | 0;
exports.ActivateEventSource$ = [9, n0, _AES,
    0, () => exports.ActivateEventSourceRequest$, () => __Unit
];
exports.CancelReplay$ = [9, n0, _CR,
    0, () => exports.CancelReplayRequest$, () => exports.CancelReplayResponse$
];
exports.CreateApiDestination$ = [9, n0, _CAD,
    0, () => exports.CreateApiDestinationRequest$, () => exports.CreateApiDestinationResponse$
];
exports.CreateArchive$ = [9, n0, _CAr,
    0, () => exports.CreateArchiveRequest$, () => exports.CreateArchiveResponse$
];
exports.CreateConnection$ = [9, n0, _CC,
    0, () => exports.CreateConnectionRequest$, () => exports.CreateConnectionResponse$
];
exports.CreateEndpoint$ = [9, n0, _CE,
    0, () => exports.CreateEndpointRequest$, () => exports.CreateEndpointResponse$
];
exports.CreateEventBus$ = [9, n0, _CEB,
    0, () => exports.CreateEventBusRequest$, () => exports.CreateEventBusResponse$
];
exports.CreatePartnerEventSource$ = [9, n0, _CPES,
    0, () => exports.CreatePartnerEventSourceRequest$, () => exports.CreatePartnerEventSourceResponse$
];
exports.DeactivateEventSource$ = [9, n0, _DES,
    0, () => exports.DeactivateEventSourceRequest$, () => __Unit
];
exports.DeauthorizeConnection$ = [9, n0, _DC,
    0, () => exports.DeauthorizeConnectionRequest$, () => exports.DeauthorizeConnectionResponse$
];
exports.DeleteApiDestination$ = [9, n0, _DAD,
    0, () => exports.DeleteApiDestinationRequest$, () => exports.DeleteApiDestinationResponse$
];
exports.DeleteArchive$ = [9, n0, _DA,
    0, () => exports.DeleteArchiveRequest$, () => exports.DeleteArchiveResponse$
];
exports.DeleteConnection$ = [9, n0, _DCe,
    0, () => exports.DeleteConnectionRequest$, () => exports.DeleteConnectionResponse$
];
exports.DeleteEndpoint$ = [9, n0, _DE,
    0, () => exports.DeleteEndpointRequest$, () => exports.DeleteEndpointResponse$
];
exports.DeleteEventBus$ = [9, n0, _DEB,
    0, () => exports.DeleteEventBusRequest$, () => __Unit
];
exports.DeletePartnerEventSource$ = [9, n0, _DPES,
    0, () => exports.DeletePartnerEventSourceRequest$, () => __Unit
];
exports.DeleteRule$ = [9, n0, _DR,
    0, () => exports.DeleteRuleRequest$, () => __Unit
];
exports.DescribeApiDestination$ = [9, n0, _DADe,
    0, () => exports.DescribeApiDestinationRequest$, () => exports.DescribeApiDestinationResponse$
];
exports.DescribeArchive$ = [9, n0, _DAe,
    0, () => exports.DescribeArchiveRequest$, () => exports.DescribeArchiveResponse$
];
exports.DescribeConnection$ = [9, n0, _DCes,
    0, () => exports.DescribeConnectionRequest$, () => exports.DescribeConnectionResponse$
];
exports.DescribeEndpoint$ = [9, n0, _DEe,
    0, () => exports.DescribeEndpointRequest$, () => exports.DescribeEndpointResponse$
];
exports.DescribeEventBus$ = [9, n0, _DEBe,
    0, () => exports.DescribeEventBusRequest$, () => exports.DescribeEventBusResponse$
];
exports.DescribeEventSource$ = [9, n0, _DESe,
    0, () => exports.DescribeEventSourceRequest$, () => exports.DescribeEventSourceResponse$
];
exports.DescribePartnerEventSource$ = [9, n0, _DPESe,
    0, () => exports.DescribePartnerEventSourceRequest$, () => exports.DescribePartnerEventSourceResponse$
];
exports.DescribeReplay$ = [9, n0, _DRe,
    0, () => exports.DescribeReplayRequest$, () => exports.DescribeReplayResponse$
];
exports.DescribeRule$ = [9, n0, _DRes,
    0, () => exports.DescribeRuleRequest$, () => exports.DescribeRuleResponse$
];
exports.DisableRule$ = [9, n0, _DRi,
    0, () => exports.DisableRuleRequest$, () => __Unit
];
exports.EnableRule$ = [9, n0, _ER,
    0, () => exports.EnableRuleRequest$, () => __Unit
];
exports.ListApiDestinations$ = [9, n0, _LAD,
    0, () => exports.ListApiDestinationsRequest$, () => exports.ListApiDestinationsResponse$
];
exports.ListArchives$ = [9, n0, _LA,
    0, () => exports.ListArchivesRequest$, () => exports.ListArchivesResponse$
];
exports.ListConnections$ = [9, n0, _LCi,
    0, () => exports.ListConnectionsRequest$, () => exports.ListConnectionsResponse$
];
exports.ListEndpoints$ = [9, n0, _LE,
    0, () => exports.ListEndpointsRequest$, () => exports.ListEndpointsResponse$
];
exports.ListEventBuses$ = [9, n0, _LEB,
    0, () => exports.ListEventBusesRequest$, () => exports.ListEventBusesResponse$
];
exports.ListEventSources$ = [9, n0, _LES,
    0, () => exports.ListEventSourcesRequest$, () => exports.ListEventSourcesResponse$
];
exports.ListPartnerEventSourceAccounts$ = [9, n0, _LPESA,
    0, () => exports.ListPartnerEventSourceAccountsRequest$, () => exports.ListPartnerEventSourceAccountsResponse$
];
exports.ListPartnerEventSources$ = [9, n0, _LPES,
    0, () => exports.ListPartnerEventSourcesRequest$, () => exports.ListPartnerEventSourcesResponse$
];
exports.ListReplays$ = [9, n0, _LR,
    0, () => exports.ListReplaysRequest$, () => exports.ListReplaysResponse$
];
exports.ListRuleNamesByTarget$ = [9, n0, _LRNBT,
    0, () => exports.ListRuleNamesByTargetRequest$, () => exports.ListRuleNamesByTargetResponse$
];
exports.ListRules$ = [9, n0, _LRi,
    0, () => exports.ListRulesRequest$, () => exports.ListRulesResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => exports.ListTagsForResourceRequest$, () => exports.ListTagsForResourceResponse$
];
exports.ListTargetsByRule$ = [9, n0, _LTBR,
    0, () => exports.ListTargetsByRuleRequest$, () => exports.ListTargetsByRuleResponse$
];
exports.PutEvents$ = [9, n0, _PE,
    0, () => exports.PutEventsRequest$, () => exports.PutEventsResponse$
];
exports.PutPartnerEvents$ = [9, n0, _PPE,
    0, () => exports.PutPartnerEventsRequest$, () => exports.PutPartnerEventsResponse$
];
exports.PutPermission$ = [9, n0, _PP,
    0, () => exports.PutPermissionRequest$, () => __Unit
];
exports.PutRule$ = [9, n0, _PR,
    0, () => exports.PutRuleRequest$, () => exports.PutRuleResponse$
];
exports.PutTargets$ = [9, n0, _PTu,
    0, () => exports.PutTargetsRequest$, () => exports.PutTargetsResponse$
];
exports.RemovePermission$ = [9, n0, _RPem,
    0, () => exports.RemovePermissionRequest$, () => __Unit
];
exports.RemoveTargets$ = [9, n0, _RT,
    0, () => exports.RemoveTargetsRequest$, () => exports.RemoveTargetsResponse$
];
exports.StartReplay$ = [9, n0, _SRt,
    0, () => exports.StartReplayRequest$, () => exports.StartReplayResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => exports.TagResourceRequest$, () => exports.TagResourceResponse$
];
exports.TestEventPattern$ = [9, n0, _TEP,
    0, () => exports.TestEventPatternRequest$, () => exports.TestEventPatternResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => exports.UntagResourceRequest$, () => exports.UntagResourceResponse$
];
exports.UpdateApiDestination$ = [9, n0, _UAD,
    0, () => exports.UpdateApiDestinationRequest$, () => exports.UpdateApiDestinationResponse$
];
exports.UpdateArchive$ = [9, n0, _UA,
    0, () => exports.UpdateArchiveRequest$, () => exports.UpdateArchiveResponse$
];
exports.UpdateConnection$ = [9, n0, _UC,
    0, () => exports.UpdateConnectionRequest$, () => exports.UpdateConnectionResponse$
];
exports.UpdateEndpoint$ = [9, n0, _UE,
    0, () => exports.UpdateEndpointRequest$, () => exports.UpdateEndpointResponse$
];
exports.UpdateEventBus$ = [9, n0, _UEB,
    0, () => exports.UpdateEventBusRequest$, () => exports.UpdateEventBusResponse$
];
